{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## HOMER for SOX10 KD library"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Active vs inactive regions"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "__Format files for Homer__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "terminate called after throwing an instance of 'std::invalid_argument'\n",
      "  what():  stoll\n",
      "Aborted\n",
      "terminate called after throwing an instance of 'std::invalid_argument'\n",
      "  what():  stoll\n",
      "Aborted\n"
     ]
    }
   ],
   "source": [
    "# Convert the tiles to bed format\n",
    "cd /staging/leuven/stg_00002/lcb/lcb_projects/CSE/SOX10-KD_library/analysis/HOMER/data\n",
    "\n",
    "for file in *-human-names.txt; do \\\n",
    " cat $file | tr \"@@\" \"\\t\"  | tr \":\" \"\\t\" | tr \"-\" \"\\t\" | awk '{print $1 \"\\t\" $2 \"\\t\" $3 \"\\t\" $1 \":\" $2 \"-\" $3 \"@@\" $4 }' > ${file%%.*}.bed ; done\n",
    "\n",
    "# Remove entries that are mutated and the shuffled tiles\n",
    "for file in *-human-names.bed; do \\\n",
    " cat $file | grep -v \"Neg-Control\" > ${file%%.*}.noNC.bed ; done\n",
    "\n",
    "# Convert bed to fasta\n",
    "module load BEDTools\n",
    "for file in *.noNC.bed; do \\\n",
    "bedtools getfasta \\\n",
    "-fi /staging/leuven/stg_00002/lcb/resources/human/hg19/Homo_sapiens_assembly19_sorted.fa \\\n",
    "-bed $file -name -fo ${file%%.*}.noNC.fa ; done\n",
    "\n",
    "rm *.bed"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "__Run HOMER__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Selected Options:\n",
      "\tInput file = active-human-names.noNC.fa\n",
      "\tPromoter Set = fasta\n",
      "\tOutput Directory = results/\n",
      "\tUsing repeat masked sequences\n",
      "\tWill not run homer for de novo motifs\n",
      "\tMotif length set at 6, 8, 10, 11, 12, 17, *** might want to hit ctrl+C to quit if it takes too long!\n",
      "*** If you run out of memmory try reducing the number background sequences\n",
      "18, *** might want to hit ctrl+C to quit if it takes too long!\n",
      "*** If you run out of memmory try reducing the number background sequences\n",
      "\n",
      "\tWill use FASTA files for motif finding\n",
      "\t\tTarget Sequences = active-human-names.noNC.fa\n",
      "\t\tBackground Sequences = inactive-human-names.noNC.fa\n",
      "\tUsing custom gene IDs for GO analysis\n",
      "\tParsing FASTA format files...\n",
      "\tFound 329 sequences\n",
      "\tFound 3055 sequences\n",
      "\n",
      "\tProgress: Step4 - removing redundant promoters\n",
      "\n",
      "\tProgress: Step5 - adjusting background sequences for GC/CpG content...\n",
      "\n",
      "\tSequences processed:\n",
      "\t\tAuto detected maximum sequence length of 190 bp\n",
      "\t\t3384 total\n",
      "\n",
      "\tFrequency Bins: 0.2 0.25 0.3 0.35 0.4 0.45 0.5 0.6 0.7 0.8\n",
      "\tFreq\tBin\tCount\n",
      "\t0.35\t3\t17\n",
      "\t0.4\t4\t224\n",
      "\t0.45\t5\t751\n",
      "\t0.5\t6\t876\n",
      "\t0.6\t7\t1150\n",
      "\t0.7\t8\t330\n",
      "\t0.8\t9\t32\n",
      "\t10\t10\t4\n",
      "\tBin\t# Targets\t# Background\tBackground Weight\n",
      "\t3\t5\t12\t3.869\n",
      "\t4\t25\t199\t1.167\n",
      "\t5\t73\t678\t1.000\n",
      "\t6\t83\t793\t0.972\n",
      "\t7\t96\t1054\t0.846\n",
      "\t8\t36\t294\t1.137\n",
      "\t9\t9\t23\t3.634\n",
      "\t10\t2\t2\t9.286\n",
      "\n",
      "\tNormalizing lower order oligos using homer2\n",
      "\n",
      "\tReading input files...\n",
      "\t3384 total sequences read\n",
      "\tAutonormalization: 1-mers (4 total)\n",
      "\t\tA\t25.00%\t25.01%\t1.000\n",
      "\t\tC\t25.00%\t24.99%\t1.000\n",
      "\t\tG\t25.00%\t24.99%\t1.000\n",
      "\t\tT\t25.00%\t25.01%\t1.000\n",
      "\tAutonormalization: 2-mers (16 total)\n",
      "\t\tAA\t7.20%\t6.86%\t1.050\n",
      "\t\tCA\t7.93%\t8.05%\t0.985\n",
      "\t\tGA\t6.36%\t6.30%\t1.008\n",
      "\t\tTA\t3.53%\t3.80%\t0.928\n",
      "\t\tAC\t5.11%\t5.10%\t1.001\n",
      "\t\tCC\t7.29%\t7.24%\t1.007\n",
      "\t\tGC\t6.22%\t6.33%\t0.983\n",
      "\t\tTC\t6.36%\t6.30%\t1.008\n",
      "\t\tAG\t7.83%\t7.94%\t0.987\n",
      "\t\tCG\t1.94%\t1.76%\t1.103\n",
      "\t\tGG\t7.29%\t7.24%\t1.007\n",
      "\t\tTG\t7.93%\t8.05%\t0.985\n",
      "\t\tAT\t4.87%\t5.12%\t0.952\n",
      "\t\tCT\t7.83%\t7.94%\t0.987\n",
      "\t\tGT\t5.11%\t5.10%\t1.001\n",
      "\t\tTT\t7.20%\t6.86%\t1.050\n",
      "\tAutonormalization: 3-mers (64 total)\n",
      "\tNormalization weights can be found in file: results//seq.autonorm.tsv\n",
      "\tConverging on autonormalization solution:\n",
      "\t...............................................................................\n",
      "\tFinal normalization:\tAutonormalization: 1-mers (4 total)\n",
      "\t\tA\t25.00%\t24.99%\t1.001\n",
      "\t\tC\t25.00%\t25.01%\t0.999\n",
      "\t\tG\t25.00%\t25.01%\t0.999\n",
      "\t\tT\t25.00%\t24.99%\t1.001\n",
      "\tAutonormalization: 2-mers (16 total)\n",
      "\t\tAA\t7.20%\t7.10%\t1.014\n",
      "\t\tCA\t7.93%\t7.95%\t0.997\n",
      "\t\tGA\t6.36%\t6.34%\t1.003\n",
      "\t\tTA\t3.53%\t3.60%\t0.980\n",
      "\t\tAC\t5.11%\t5.11%\t1.001\n",
      "\t\tCC\t7.29%\t7.28%\t1.002\n",
      "\t\tGC\t6.22%\t6.28%\t0.991\n",
      "\t\tTC\t6.36%\t6.34%\t1.003\n",
      "\t\tAG\t7.83%\t7.87%\t0.996\n",
      "\t\tCG\t1.94%\t1.91%\t1.014\n",
      "\t\tGG\t7.29%\t7.28%\t1.002\n",
      "\t\tTG\t7.93%\t7.95%\t0.997\n",
      "\t\tAT\t4.87%\t4.92%\t0.990\n",
      "\t\tCT\t7.83%\t7.87%\t0.996\n",
      "\t\tGT\t5.11%\t5.11%\t1.001\n",
      "\t\tTT\t7.20%\t7.10%\t1.014\n",
      "\tAutonormalization: 3-mers (64 total)\n",
      "\n",
      "\tProgress: Step6 - Gene Ontology Enrichment Analysis\n",
      "\tSkipping...\n",
      "\n",
      "\tProgress: Step7 - Known motif enrichment\n",
      "\n",
      "\tReading input files...\n",
      "\t3384 total sequences read\n",
      "\t994 motifs loaded\n",
      "\tCache length = 11180\n",
      "\tUsing hypergeometric scoring\n",
      "\tChecking enrichment of 994 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tPreparing HTML output with sequence logos...\n",
      "\t\t1 of 994 (1e-28) VRN1(ABI3VP1)/col-VRN1-DAP-Seq(GSE60143)/Homer\n",
      "\t\t2 of 994 (1e-19) RLR1?/SacCer-Promoters/Homer\n",
      "\t\t3 of 994 (1e-16) REM19(REM)/colamp-REM19-DAP-Seq(GSE60143)/Homer\n",
      "\t\t4 of 994 (1e-13) MITF(bHLH)/MastCells-MITF-ChIP-Seq(GSE48085)/Homer\n",
      "\t\t5 of 994 (1e-12) Usf2(bHLH)/C2C12-Usf2-ChIP-Seq(GSE36030)/Homer\n",
      "\t\t6 of 994 (1e-11) CLOCK(bHLH)/Liver-Clock-ChIP-Seq(GSE39860)/Homer\n",
      "\t\t7 of 994 (1e-10) Cbf1(bHLH)/Yeast-Cbf1-ChIP-Seq(GSE29506)/Homer\n",
      "\t\t8 of 994 (1e-9) Atf4(bZIP)/MEF-Atf4-ChIP-Seq(GSE35681)/Homer\n",
      "\t\t9 of 994 (1e-7) USF1(bHLH)/GM12878-Usf1-ChIP-Seq(GSE32465)/Homer\n",
      "\t\t10 of 994 (1e-7) bHLH74(bHLH)/col-bHLH74-DAP-Seq(GSE60143)/Homer\n",
      "\t\t11 of 994 (1e-7) bHLHE40(bHLH)/HepG2-BHLHE40-ChIP-Seq(GSE31477)/Homer\n",
      "\t\t12 of 994 (1e-7) BIM3(bHLH)/col-BIM3-DAP-Seq(GSE60143)/Homer\n",
      "\t\t13 of 994 (1e-7) AARE(HLH)/mES-cMyc-ChIP-Seq/Homer\n",
      "\t\t14 of 994 (1e-6) BIM2(bHLH)/col-BIM2-DAP-Seq(GSE60143)/Homer\n",
      "\t\t15 of 994 (1e-6) SPCH(bHLH)/Seedling-SPCH-ChIP-Seq(GSE57497)/Homer\n",
      "\t\t16 of 994 (1e-6) TFE3(bHLH)/MEF-TFE3-ChIP-Seq(GSE75757)/Homer\n",
      "\t\t17 of 994 (1e-6) Chop(bZIP)/MEF-Chop-ChIP-Seq(GSE35681)/Homer\n",
      "\t\t18 of 994 (1e-6) FEA4(bZIP)/Corn-FEA4-ChIP-Seq(GSE61954)/Homer\n",
      "\t\t19 of 994 (1e-6) BMAL1(bHLH)/Liver-Bmal1-ChIP-Seq(GSE39860)/Homer\n",
      "\t\t20 of 994 (1e-6) Sp1(Zf)/Promoter/Homer\n",
      "\t\t21 of 994 (1e-6) NPAS(bHLH)/Liver-NPAS-ChIP-Seq(GSE39860)/Homer\n",
      "\t\t22 of 994 (1e-6) ETS(ETS)/Promoter/Homer\n",
      "\t\t23 of 994 (1e-6) Elk4(ETS)/Hela-Elk4-ChIP-Seq(GSE31477)/Homer\n",
      "\t\t24 of 994 (1e-6) Elk1(ETS)/Hela-Elk1-ChIP-Seq(GSE31477)/Homer\n",
      "\t\t25 of 994 (1e-6) NPAS2(bHLH)/Liver-NPAS2-ChIP-Seq(GSE39860)/Homer\n",
      "\t\t26 of 994 (1e-6) SeqBias: TA-repeat\n",
      "\t\t27 of 994 (1e-6) Fra1(bZIP)/BT549-Fra1-ChIP-Seq(GSE46166)/Homer\n",
      "\t\t28 of 994 (1e-5) Max(bHLH)/K562-Max-ChIP-Seq(GSE31477)/Homer\n",
      "\t\t29 of 994 (1e-5) Fra2(bZIP)/Striatum-Fra2-ChIP-Seq(GSE43429)/Homer\n",
      "\t\t30 of 994 (1e-5) c-Myc(bHLH)/LNCAP-cMyc-ChIP-Seq(Unpublished)/Homer\n",
      "\t\t31 of 994 (1e-5) ETV1(ETS)/GIST48-ETV1-ChIP-Seq(GSE22441)/Homer\n",
      "\t\t32 of 994 (1e-5) Atf3(bZIP)/GBM-ATF3-ChIP-Seq(GSE33912)/Homer\n",
      "\t\t33 of 994 (1e-5) HLF(bZIP)/HSC-HLF.Flag-ChIP-Seq(GSE69817)/Homer\n",
      "\t\t34 of 994 (1e-5) PIF5ox(bHLH)/Arabidopsis-PIF5ox-ChIP-Seq(GSE35062)/Homer\n",
      "\t\t35 of 994 (1e-5) PIF4(bHLH)/Seedling-PIF4-ChIP-Seq(GSE35315)/Homer\n",
      "\t\t36 of 994 (1e-5) Etv2(ETS)/ES-ER71-ChIP-Seq(GSE59402)/Homer\n",
      "\t\t37 of 994 (1e-5) bHLHE41(bHLH)/proB-Bhlhe41-ChIP-Seq(GSE93764)/Homer\n",
      "\t\t38 of 994 (1e-5) IBL1(bHLH)/Seedling-IBL1-ChIP-Seq(GSE51120)/Homer\n",
      "\t\t39 of 994 (1e-5) ELF1(ETS)/Jurkat-ELF1-ChIP-Seq(SRA014231)/Homer\n",
      "\t\t40 of 994 (1e-5) NFIL3(bZIP)/HepG2-NFIL3-ChIP-Seq(Encode)/Homer\n",
      "\t\t41 of 994 (1e-5) Sp2(Zf)/HEK293-Sp2.eGFP-ChIP-Seq(Encode)/Homer\n",
      "\t\t42 of 994 (1e-5) AP-1(bZIP)/ThioMac-PU.1-ChIP-Seq(GSE21512)/Homer\n",
      "\t\t43 of 994 (1e-5) BATF(bZIP)/Th17-BATF-ChIP-Seq(GSE39756)/Homer\n",
      "\t\t44 of 994 (1e-5) MNT(bHLH)/HepG2-MNT-ChIP-Seq(Encode)/Homer\n",
      "\t\t45 of 994 (1e-5) n-Myc(bHLH)/mES-nMyc-ChIP-Seq(GSE11431)/Homer\n",
      "\t\t46 of 994 (1e-5) Klf9(Zf)/GBM-Klf9-ChIP-Seq(GSE62211)/Homer\n",
      "\t\t47 of 994 (1e-4) GAGA-repeat/SacCer-Promoters/Homer\n",
      "\t\t48 of 994 (1e-4) BIM1(bHLH)/colamp-BIM1-DAP-Seq(GSE60143)/Homer\n",
      "\t\t49 of 994 (1e-4) Egr1(Zf)/K562-Egr1-ChIP-Seq(GSE32465)/Homer\n",
      "\t\t50 of 994 (1e-4) HuR(?)/HEK293-HuR-CLIP-Seq(GSE87887)/Homer\n",
      "\t\t51 of 994 (1e-4) Sp5(Zf)/mES-Sp5.Flag-ChIP-Seq(GSE72989)/Homer\n",
      "\t\t52 of 994 (1e-4) ETV4(ETS)/HepG2-ETV4-ChIP-Seq(ENCODE)/Homer\n",
      "\t\t53 of 994 (1e-4) Elf4(ETS)/BMDM-Elf4-ChIP-Seq(GSE88699)/Homer\n",
      "\t\t54 of 994 (1e-4) ATAF1(NAC)/col-ATAF1-DAP-Seq(GSE60143)/Homer\n",
      "\t\t55 of 994 (1e-4) Fli1(ETS)/CD8-FLI-ChIP-Seq(GSE20898)/Homer\n",
      "\t\t56 of 994 (1e-3) JunB(bZIP)/DendriticCells-Junb-ChIP-Seq(GSE36099)/Homer\n",
      "\t\t57 of 994 (1e-3) GABPA(ETS)/Jurkat-GABPa-ChIP-Seq(GSE17954)/Homer\n",
      "\t\t58 of 994 (1e-3) Bach2(bZIP)/OCILy7-Bach2-ChIP-Seq(GSE44420)/Homer\n",
      "\t\t59 of 994 (1e-3) At4g38000(C2C2dof)/col-At4g38000-DAP-Seq(GSE60143)/Homer\n",
      "\t\t60 of 994 (1e-3) SeqBias: CA-repeat\n",
      "\t\t61 of 994 (1e-3) E-box/Arabidopsis-Promoters/Homer\n",
      "\t\t62 of 994 (1e-3) ABF1(bZIP)/Arabidopsis-ABF1-ChIP-Seq(GSE80564)/Homer\n",
      "\t\t63 of 994 (1e-3) Ets1-distal(ETS)/CD4+-PolII-ChIP-Seq(Barski_et_al.)/Homer\n",
      "\t\t64 of 994 (1e-3) Klf4(Zf)/mES-Klf4-ChIP-Seq(GSE11431)/Homer\n",
      "\t\t65 of 994 (1e-3) EHF(ETS)/LoVo-EHF-ChIP-Seq(GSE49402)/Homer\n",
      "\t\t66 of 994 (1e-3) NFY(CCAAT)/Promoter/Homer\n",
      "\t\t67 of 994 (1e-3) KLF3(Zf)/MEF-Klf3-ChIP-Seq(GSE44748)/Homer\n",
      "\t\t68 of 994 (1e-3) Egr2(Zf)/Thymocytes-Egr2-ChIP-Seq(GSE34254)/Homer\n",
      "\t\t69 of 994 (1e-3) HY5(bZIP)/colamp-HY5-DAP-Seq(GSE60143)/Homer\n",
      "\t\t70 of 994 (1e-3) ABF2(bZIP)/col-ABF2-DAP-Seq(GSE60143)/Homer\n",
      "\t\t71 of 994 (1e-3) CRE(bZIP)/Promoter/Homer\n",
      "\t\t72 of 994 (1e-3) EWS:FLI1-fusion(ETS)/SK_N_MC-EWS:FLI1-ChIP-Seq(SRA014231)/Homer\n",
      "\t\t73 of 994 (1e-3) Unknown3/Arabidopsis-Promoters/Homer\n",
      "\t\t74 of 994 (1e-3) At4g36780(BZR)/col-At4g36780-DAP-Seq(GSE60143)/Homer\n",
      "\t\t75 of 994 (1e-3) RAP21(AP2EREBP)/colamp-RAP21-DAP-Seq(GSE60143)/Homer\n",
      "\t\t76 of 994 (1e-2) KLF14(Zf)/HEK293-KLF14.GFP-ChIP-Seq(GSE58341)/Homer\n",
      "\t\t77 of 994 (1e-2) KLF6(Zf)/PDAC-KLF6-ChIP-Seq(GSE64557)/Homer\n",
      "\t\t78 of 994 (1e-2) TGA6(bZIP)/colamp-TGA6-DAP-Seq(GSE60143)/Homer\n",
      "\t\t79 of 994 (1e-2) Fosl2(bZIP)/3T3L1-Fosl2-ChIP-Seq(GSE56872)/Homer\n",
      "\t\t80 of 994 (1e-2) bZIP3(bZIP)/col-bZIP3-DAP-Seq(GSE60143)/Homer\n",
      "\t\t81 of 994 (1e-2) KLF5(Zf)/LoVo-KLF5-ChIP-Seq(GSE49402)/Homer\n",
      "\t\t82 of 994 (1e-2) Atf1(bZIP)/K562-ATF1-ChIP-Seq(GSE31477)/Homer\n",
      "\t\t83 of 994 (1e-2) GFY-Staf(?,Zf)/Promoter/Homer\n",
      "\t\t84 of 994 (1e-2) bZIP53(bZIP)/colamp-bZIP53-DAP-Seq(GSE60143)/Homer\n",
      "\t\t85 of 994 (1e-2) GBF3(bZIP)/Arabidopsis-GBF3-ChIP-Seq(GSE80564)/Homer\n",
      "\t\t86 of 994 (1e-2) Zfp281(Zf)/ES-Zfp281-ChIP-Seq(GSE81042)/Homer\n",
      "\t\t87 of 994 (1e-2) AZF1(C2H2)/colamp-AZF1-DAP-Seq(GSE60143)/Homer\n",
      "\t\t88 of 994 (1e-2) bZIP48(bZIP)/colamp-bZIP48-DAP-Seq(GSE60143)/Homer\n",
      "\t\t89 of 994 (1e-2) CEBP:AP1(bZIP)/ThioMac-CEBPb-ChIP-Seq(GSE21512)/Homer\n",
      "\t\t90 of 994 (1e-2) bZIP16(bZIP)/colamp-bZIP16-DAP-Seq(GSE60143)/Homer\n",
      "\t\t91 of 994 (1e-2) ETS1(ETS)/Jurkat-ETS1-ChIP-Seq(GSE17954)/Homer\n",
      "\t\t92 of 994 (1e-2) bHLH34(bHLH)/colamp-bHLH34-DAP-Seq(GSE60143)/Homer\n",
      "\t\t93 of 994 (1e-2) WT1(Zf)/Kidney-WT1-ChIP-Seq(GSE90016)/Homer\n",
      "\t\t94 of 994 (1e-2) GBF6(bZIP)/colamp-GBF6-DAP-Seq(GSE60143)/Homer\n",
      "\t\t95 of 994 (1e-2) Replumless(BLH)/Arabidopsis-RPL.GFP-ChIP-Seq(GSE78727)/Homer\n",
      "\t\t96 of 994 (1e-2) TGA9(bZIP)/colamp-TGA9-DAP-Seq(GSE60143)/Homer\n",
      "\t\t97 of 994 (1e-2) Ronin(THAP)/ES-Thap11-ChIP-Seq(GSE51522)/Homer\n",
      "\t\t98 of 994 (1e-2) AREB3(bZIP)/col-AREB3-DAP-Seq(GSE60143)/Homer\n",
      "\t\t99 of 994 (1e-2) Bach1(bZIP)/K562-Bach1-ChIP-Seq(GSE31477)/Homer\n",
      "\t\t100 of 994 (1e-2) Knotted(Homeobox)/Corn-KN1-ChIP-Seq(GSE39161)/Homer\n",
      "\t\t101 of 994 (1e-2) At1g78700(BZR)/col-At1g78700-DAP-Seq(GSE60143)/Homer\n",
      "\t\t102 of 994 (1e-2) Nkx2.2(Homeobox)/NPC-Nkx2.2-ChIP-Seq(GSE61673)/Homer\n",
      "\t\t103 of 994 (1e-2) bHLH28(bHLH)/col-bHLH28-DAP-Seq(GSE60143)/Homer\n",
      "\t\t104 of 994 (1e-2) Jun-AP1(bZIP)/K562-cJun-ChIP-Seq(GSE31477)/Homer\n",
      "\t\t105 of 994 (1e-2) TGA1(bZIP)/colamp-TGA1-DAP-Seq(GSE60143)/Homer\n",
      "\t\t106 of 994 (1e-2) Nkx2.1(Homeobox)/LungAC-Nkx2.1-ChIP-Seq(GSE43252)/Homer\n",
      "\t\t107 of 994 (1e-2) Six1(Homeobox)/Myoblast-Six1-ChIP-Chip(GSE20150)/Homer\n",
      "\t\t108 of 994 (1e-2) KLF10(Zf)/HEK293-KLF10.GFP-ChIP-Seq(GSE58341)/Homer\n",
      "\t\t109 of 994 (1e-2) ERG(ETS)/VCaP-ERG-ChIP-Seq(GSE14097)/Homer\n",
      "\t\t110 of 994 (1e-2) bZIP68(bZIP)/col-bZIP68-DAP-Seq(GSE60143)/Homer\n",
      "\t\t111 of 994 (1e-2) AT1G10720(BSD)/col-AT1G10720-DAP-Seq(GSE60143)/Homer\n",
      "\t\t112 of 994 (1e-2) Pho2(bHLH)/Yeast-Pho2-ChIP-Seq(GSE29506)/Homer\n",
      "\n",
      "\tProgress: Step8 - De novo motif finding (HOMER)\n",
      "\tSkipping...\n",
      "\tJob finished\n",
      "\n"
     ]
    }
   ],
   "source": [
    "cd /staging/leuven/stg_00002/lcb/lcb_projects/CSE/SOX10-KD_library/analysis/HOMER/data\n",
    "module load HOMER\n",
    "\n",
    "findMotifs.pl \\\n",
    "active-human-names.noNC.fa \\\n",
    "fasta \\\n",
    "results/ \\\n",
    "-mask \\\n",
    "-nomotif \\\n",
    "-len 6,8,10,11,12,17,18 \\\n",
    "-fastaBg inactive-human-names.noNC.fa\n",
    " "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash 4.2.46",
   "language": "bash",
   "name": "bash4246"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
