---
title: "ATAC_CHEQseq_500bp_activity_correlation"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Load data H3K27Ac quantification
```{r}
Ac.500 <- new.env()
Ac.500[["MM001"]]$H3K27Ac <- read.table(file = "../../CH1/H3K27Ac-ATAC_quantification/H3K27Ac_500bp_MM001.bed", header = F, sep = "\t")
Ac.500[["MM001"]]$Copy.Number <- read.table(file = "../../CH1/H3K27Ac-ATAC_quantification/Copy_number_500bp_MM001.bed", header = F, sep = "\t")
Ac.500[["MM029"]]$H3K27Ac <- read.table(file = "../../CH1/H3K27Ac-ATAC_quantification/H3K27Ac_500bp_MM029.bed", header = F, sep = "\t")
Ac.500[["MM029"]]$Copy.Number <- read.table(file = "../../CH1/H3K27Ac-ATAC_quantification/Copy_number_500bp_MM029.bed", header = F, sep = "\t")
Ac.500[["MM047"]]$H3K27Ac <- read.table(file = "../../CH1/H3K27Ac-ATAC_quantification/H3K27Ac_500bp_MM047.bed", header = F, sep = "\t")
Ac.500[["MM047"]]$Copy.Number <- read.table(file = "../../CH1/H3K27Ac-ATAC_quantification/Copy_number_500bp_MM047.bed", header = F, sep = "\t")
Ac.500[["MM057"]]$H3K27Ac <- read.table(file = "../../CH1/H3K27Ac-ATAC_quantification/H3K27Ac_500bp_MM057.bed", header = F, sep = "\t")
Ac.500[["MM057"]]$Copy.Number <- read.table(file = "../../CH1/H3K27Ac-ATAC_quantification/Copy_number_500bp_MM057.bed", header = F, sep = "\t")
Ac.500[["MM074"]]$H3K27Ac <- read.table(file = "../../CH1/H3K27Ac-ATAC_quantification/H3K27Ac_500bp_MM074.bed", header = F, sep = "\t")
Ac.500[["MM074"]]$Copy.Number <- read.table(file = "../../CH1/H3K27Ac-ATAC_quantification/Copy_number_500bp_MM074.bed", header = F, sep = "\t")
Ac.500[["MM087"]]$H3K27Ac <- read.table(file = "../../CH1/H3K27Ac-ATAC_quantification/H3K27Ac_500bp_MM087.bed", header = F, sep = "\t")
Ac.500[["MM087"]]$Copy.Number <- read.table(file = "../../CH1/H3K27Ac-ATAC_quantification/Copy_number_500bp_MM087.bed", header = F, sep = "\t")
Ac.500[["MM099"]]$H3K27Ac <- read.table(file = "../../CH1/H3K27Ac-ATAC_quantification/H3K27Ac_500bp_MM099.bed", header = F, sep = "\t")
Ac.500[["MM099"]]$Copy.Number <- read.table(file = "../../CH1/H3K27Ac-ATAC_quantification/Copy_number_500bp_MM099.bed", header = F, sep = "\t")
# Correct name FOSL2_1A
for (i in c("MM001", "MM029", "MM047", "MM057", "MM074", "MM087", "MM099")){
  levels(Ac.500[[i]]$H3K27Ac$V1)[match("FOLS2_1A", levels(Ac.500[[i]]$H3K27Ac$V1))] <- "FOSL2_1A"
  levels(Ac.500[[i]]$Copy.Number$V4)[match("FOLS2_1A", levels(Ac.500[[i]]$Copy.Number$V4))] <- "FOSL2_1A"
}
# Normalize for copy number variation
for (i in c("MM001", "MM029", "MM047", "MM057", "MM074", "MM087", "MM099")){
Ac.500[[i]]$H3K27Ac$CNNorm.score <- Ac.500[[i]]$H3K27Ac[,4]/Ac.500[[i]]$Copy.Number[,5]*2
Ac.500[[i]]$H3K27Ac$CNNorm.mean.score <- Ac.500[[i]]$H3K27Ac[,5]/Ac.500[[i]]$Copy.Number[,5]*2
}

Ac.500$combined <- cbind(Ac.500[["MM001"]]$H3K27Ac[,c(1,2,5,8)],
                          Ac.500[["MM057"]]$H3K27Ac[,c(5,8)],
                          Ac.500[["MM074"]]$H3K27Ac[,c(5,8)],
                          Ac.500[["MM087"]]$H3K27Ac[,c(5,8)],
                          Ac.500[["MM029"]]$H3K27Ac[,c(5,8)],
                          Ac.500[["MM047"]]$H3K27Ac[,c(5,8)],
                          Ac.500[["MM099"]]$H3K27Ac[,c(5,8)])
colnames(Ac.500$combined) <- c("enhancer", "length", "MM001.mean.score", "MM001.CNNorm.mean.score", "MM057.mean.score", "MM057.CNNorm.mean.score", "MM074.mean.score", "MM074.CNNorm.mean.score", "MM087.mean.score", "MM087.CNNorm.mean.score", "MM029.mean.score", "MM029.CNNorm.mean.score", "MM047.mean.score", "MM047.CNNorm.mean.score", "MM099.mean.score", "MM099.CNNorm.mean.score")
proliferative.enhancers <- c("CDH1_1", "CDH1_2", "SOX10_1A", "SOX10_1B", "SOX10_3A", "SOX10_3B", "SOX10_5A", "SOX10_5B", "IRF4_1", "KIT_1A", "KIT_1B", "MLANA_1", "SGCD_2A", "SGCD_2B", "SGCD_3", "MITF_1", "TYR_1", "TYR_2")
Ac.500$combined$phenotype<-"Invasive"
Ac.500$combined$phenotype[Ac.500$combined$enhancer %in% proliferative.enhancers]<-"Proliferative"
rownames(Ac.500$combined) <- Ac.500$combined[,1]
```

# Load data ATAC quantification
```{r}
ATAC.500 <- new.env()
ATAC.500[["MM001"]]$ATAC <- read.table(file = "../../CH1/H3K27Ac-ATAC_quantification/ATAC_MM001.bed", header = F, sep = "\t")
ATAC.500[["MM001"]]$Copy.Number <- read.table(file = "../../CH1/H3K27Ac-ATAC_quantification/Copy_number_500bp_MM001.bed", header = F, sep = "\t")
ATAC.500[["MM029"]]$ATAC <- read.table(file = "../../CH1/H3K27Ac-ATAC_quantification/ATAC_MM029.bed", header = F, sep = "\t")
ATAC.500[["MM029"]]$Copy.Number <- read.table(file = "../../CH1/H3K27Ac-ATAC_quantification/Copy_number_500bp_MM029.bed", header = F, sep = "\t")
ATAC.500[["MM047"]]$ATAC <- read.table(file = "../../CH1/H3K27Ac-ATAC_quantification/ATAC_MM047.bed", header = F, sep = "\t")
ATAC.500[["MM047"]]$Copy.Number <- read.table(file = "../../CH1/H3K27Ac-ATAC_quantification/Copy_number_500bp_MM047.bed", header = F, sep = "\t")
ATAC.500[["MM057"]]$ATAC <- read.table(file = "../../CH1/H3K27Ac-ATAC_quantification/ATAC_MM057.bed", header = F, sep = "\t")
ATAC.500[["MM057"]]$Copy.Number <- read.table(file = "../../CH1/H3K27Ac-ATAC_quantification/Copy_number_500bp_MM057.bed", header = F, sep = "\t")
ATAC.500[["MM074"]]$ATAC <- read.table(file = "../../CH1/H3K27Ac-ATAC_quantification/ATAC_MM074.bed", header = F, sep = "\t")
ATAC.500[["MM074"]]$Copy.Number <- read.table(file = "../../CH1/H3K27Ac-ATAC_quantification/Copy_number_500bp_MM074.bed", header = F, sep = "\t")
ATAC.500[["MM087"]]$ATAC <- read.table(file = "../../CH1/H3K27Ac-ATAC_quantification/ATAC_MM087.bed", header = F, sep = "\t")
ATAC.500[["MM087"]]$Copy.Number <- read.table(file = "../../CH1/H3K27Ac-ATAC_quantification/Copy_number_500bp_MM087.bed", header = F, sep = "\t")
ATAC.500[["MM099"]]$ATAC <- read.table(file = "../../CH1/H3K27Ac-ATAC_quantification/ATAC_MM099.bed", header = F, sep = "\t")
ATAC.500[["MM099"]]$Copy.Number <- read.table(file = "../../CH1/H3K27Ac-ATAC_quantification/Copy_number_500bp_MM099.bed", header = F, sep = "\t")
# Correct name FOSL2_1A
for (i in c("MM001", "MM029", "MM047", "MM057", "MM074", "MM087", "MM099")){
  levels(ATAC.500[[i]]$ATAC$V1)[match("FOLS2_1A", levels(ATAC.500[[i]]$ATAC$V1))] <- "FOSL2_1A"
  levels(ATAC.500[[i]]$Copy.Number$V4)[match("FOLS2_1A", levels(ATAC.500[[i]]$Copy.Number$V4))] <- "FOSL2_1A"
}
# Normalize for copy number variation
for (i in c("MM001", "MM029", "MM047", "MM057", "MM074", "MM087", "MM099")){
ATAC.500[[i]]$ATAC$CNNorm.score <- ATAC.500[[i]]$ATAC[,4]/ATAC.500[[i]]$Copy.Number[,5]*2
ATAC.500[[i]]$ATAC$CNNorm.mean.score <- ATAC.500[[i]]$ATAC[,5]/ATAC.500[[i]]$Copy.Number[,5]*2
}

ATAC.500$combined <- cbind(ATAC.500[["MM001"]]$ATAC[,c(1,2,5,8)],
                           ATAC.500[["MM057"]]$ATAC[,c(5,8)],
                           ATAC.500[["MM074"]]$ATAC[,c(5,8)],
                           ATAC.500[["MM087"]]$ATAC[,c(5,8)],
                           ATAC.500[["MM029"]]$ATAC[,c(5,8)],
                           ATAC.500[["MM047"]]$ATAC[,c(5,8)],
                           ATAC.500[["MM099"]]$ATAC[,c(5,8)])
colnames(ATAC.500$combined) <- c("enhancer", "length", "MM001.mean.score", "MM001.CNNorm.mean.score", "MM057.mean.score", "MM057.CNNorm.mean.score", "MM074.mean.score", "MM074.CNNorm.mean.score", "MM087.mean.score", "MM087.CNNorm.mean.score", "MM029.mean.score", "MM029.CNNorm.mean.score", "MM047.mean.score", "MM047.CNNorm.mean.score", "MM099.mean.score", "MM099.CNNorm.mean.score")
proliferative.enhancers <- c("CDH1_1", "CDH1_2", "SOX10_1A", "SOX10_1B", "SOX10_3A", "SOX10_3B", "SOX10_5A", "SOX10_5B", "IRF4_1", "KIT_1A", "KIT_1B", "MLANA_1", "SGCD_2A", "SGCD_2B", "SGCD_3", "MITF_1", "TYR_1", "TYR_2")
ATAC.500$combined$phenotype<-"Invasive"
ATAC.500$combined$phenotype[ATAC.500$combined$enhancer %in% proliferative.enhancers]<-"Proliferative"
rownames(ATAC.500$combined) <- ATAC.500$combined[,1]
```

# Heatmap profile
```{r}
# library(gplots)
# library(viridis)
# x <- log1p(x = Ac.500$combined[, c(3,5,7,9,11,13,15)])
# colnames(x) <- c("MM001", "MM057", "MM074", "MM087", "MM029", "MM047", "MM099")
# setEPS()
# postscript(file = "analysis/CHEQseq/Plots/ATAC_correlation/H3K27Ac_heatmap_500bp_enhancers.eps", 
#            horizontal = FALSE, onefile = FALSE, paper = "special", height = 8.27, width = 10)
# heat1 <-heatmap.2(x = as.matrix(x), col = viridis(n=50), trace = "none", cexRow = 0.6, srtCol=45, main = "H3K27Ac ChIP-seq ATAC enhancers - Mean score")
# dev.off()
# 
# x <- log1p(x = ATAC.500$combined[, c(3,5,7,9,11,13,15)])
# colnames(x) <- c("MM001", "MM057", "MM074", "MM087", "MM029", "MM047", "MM099")
# setEPS()
# postscript(file = "analysis/CHEQseq/Plots/ATAC_correlation/ATAC_heatmap_500bp_enhancers.eps", 
#            horizontal = FALSE, onefile = FALSE, paper = "special", height = 8.27, width = 10)
# heatmap.2(x = as.matrix(x[rev(heat1$rowInd),heat1$colInd]), Rowv = F, Colv = F, dendrogram = "none", col = inferno(n=50), trace = "none", cexRow = 0.6, srtCol=45, main = "ATAC-seq ATAC enhancers - Mean score")
# dev.off()
```

# Heatmap profile Copy number normalized
```{r}
library(gplots)
library(viridis)
x <- log1p(x = Ac.500$combined[, c(4,6,8,10,12,14,16)])
colnames(x) <- c("MM001", "MM057", "MM074", "MM087", "MM029", "MM047", "MM099")
setEPS()
postscript(file = "analysis/CHEQseq/Plots/ATAC_correlation/H3K27Ac_heatmap_CNNorm_500bp_enhancers.eps", 
           horizontal = FALSE, onefile = FALSE, paper = "special", height = 8.27, width = 10)
heatmap.2(x = as.matrix(x[rev(heat1$rowInd),heat1$colInd]), Rowv = F, Colv = F, dendrogram = "none", col = viridis(n=50), trace = "none", cexRow = 0.6, srtCol=45, main = "H3K27Ac ChIP-seq ATAC enhancers - Mean score CNNorm")
dev.off()

x <- log1p(x = ATAC.500$combined[, c(4,6,8,10,12,14,16)])
colnames(x) <- c("MM001", "MM057", "MM074", "MM087", "MM029", "MM047", "MM099")
setEPS()
postscript(file = "analysis/CHEQseq/Plots/ATAC_correlation/ATAC_heatmap_CNNorm_500bp_enhancers.eps", 
           horizontal = FALSE, onefile = FALSE, paper = "special", height = 8.27, width = 10)
heatmap.2(x = as.matrix(x[rev(heat1$rowInd),heat1$colInd]), Rowv = F, Colv = F, dendrogram = "none", col = inferno(n=50), trace = "none", cexRow = 0.6, srtCol=45, main = "ATAC-seq ATAC enhancers - Mean score CNNorm")
dev.off()
```

# Correlation tables Ac and ATAC intersamples
```{r}
library(corrplot)

setEPS()
postscript(file = "analysis/CHEQseq/Plots/ATAC_correlation/H3K27Ac_correlation_CNNorm_500bp_enhancers.eps", 
           horizontal = FALSE, onefile = FALSE, paper = "special", height = 7, width = 7)
M <- cor(x = Ac.500$combined[, c(4,6,8,10,12,14,16)], method = "pearson")
colnames(M) <- c("MM001", "MM057", "MM074", "MM087", "MM029", "MM047", "MM099")
row.names(M) <- c("MM001", "MM057", "MM074", "MM087", "MM029", "MM047", "MM099")
corrplot(M, method="color", addCoef.col = "grey", tl.col="black", title = "H3K27Ac Correlation CNNorm", order = "hclust", addrect = 2, mar = c(0, 0, 1.5, 0))
dev.off()

setEPS()
postscript(file = "analysis/CHEQseq/Plots/ATAC_correlation/ATAC_correlation_CNNorm_500bp_enhancers.eps", 
           horizontal = FALSE, onefile = FALSE, paper = "special", height = 7, width = 7)
M <- cor(x = ATAC.500$combined[, c(4,6,8,10,12,14,16)], method = "pearson")
colnames(M) <- c("MM001", "MM057", "MM074", "MM087", "MM029", "MM047", "MM099")
row.names(M) <- c("MM001", "MM057", "MM074", "MM087", "MM029", "MM047", "MM099")
corrplot(M, method="color", addCoef.col = "grey", tl.col="black", title = "ATACseq Correlation CNNorm", order = "hclust", addrect = 2, mar = c(0, 0, 1.5, 0))
dev.off()
```

# Correlation Ac vs CHEQseq 500 bp activity per cell line
```{r}
library(dplyr)
library(ggplot2)
Ac.500$Ac.CHEQ <- inner_join(x = Ac.500$combined[, c(1:3,5,7,9,11,13,15)],
                             y = CSE.500bp.CHEQ.df[,c(1:9)],
                             by = "enhancer")
Ac.500$Ac.CHEQ.CNNorm <- inner_join(x = Ac.500$combined[, c(1,2,4,6,8,10,12,14,16)],
                                    y = CSE.500bp.CHEQ.df[,c(1:9)],
                                    by = "enhancer")

for (i in c("MM001", "MM057", "MM074", "MM087", "MM029", "MM047", "MM099")) {
p<-ggplot(Ac.500$Ac.CHEQ.CNNorm, aes(x = Ac.500$Ac.CHEQ.CNNorm[,paste0(i,".CNNorm.mean.score")], 
                                      y = log2(Ac.500$Ac.CHEQ.CNNorm[,paste0(i,".CPMNorm.FC")]), color = phenotype)) +
  geom_smooth(method = "lm", alpha =0.1) +
  geom_point() +
  theme(panel.background = element_rect(fill = "white", colour = "white", size = 0.5),
        panel.grid = element_blank(),
        axis.text = element_text(color = "black"),
        legend.key = element_rect(fill = "white")) +
  scale_color_manual(values = c("#E41A1C", "#377EB8"), labels = c("MES", "MEL")) +
  annotate("text", x = 0, y = 4, label = paste0("r = ", round(cor(Ac.500$Ac.CHEQ.CNNorm[,c(paste0(i,".CNNorm.mean.score"),paste0(i,".CPMNorm.FC"))], method="pearson")[1,2],digits = 4)), hjust=0) +
  ggtitle(label = paste0("Correlation ",i," - H3K27Ac ChIP-seq vs CHEQ-seq ATAC Activity")) +
  labs(x = "H3K27Ac ChIP-seq mean score", y = "CHEQ-seq - Log2 FC", color = "Enhancer\nphenotype")
ggsave(filename = paste0("analysis/CHEQseq/Plots/ATAC_correlation/H3K27Ac_CCNorm_CHEQseq_correlation_scatter_",i,".pdf"),device = "pdf",width = 5.5,height = 4, useDingbats = F)
print(p)
}
```

# Correlation ATAC vs CHEQseq 500 bp activity per cell line
```{r}
library(dplyr)
library(ggplot2)
ATAC.500$ATAC.CHEQ <- inner_join(x = ATAC.500$combined[, c(1:3,5,7,9,11,13,15)],
                                 y = CSE.500bp.CHEQ.df[,c(1:9)],
                                 by = "enhancer")
ATAC.500$ATAC.CHEQ.CNNorm <- inner_join(x = ATAC.500$combined[, c(1,2,4,6,8,10,12,14,16)],
                                        y = CSE.500bp.CHEQ.df[,c(1:9)],
                                        by = "enhancer")

for (i in c("MM001", "MM057", "MM074", "MM087","MM029", "MM047", "MM099")) {
p<-ggplot(ATAC.500$ATAC.CHEQ.CNNorm, aes(x = ATAC.500$ATAC.CHEQ.CNNorm[,paste0(i,".CNNorm.mean.score")], 
                                         y = log2(ATAC.500$ATAC.CHEQ.CNNorm[,paste0(i,".CPMNorm.FC")]), color = phenotype)) +
  geom_smooth(method = "lm", alpha =0.1) +
  geom_point() +
  theme(panel.background = element_rect(fill = "white", colour = "white", size = 0.5),
        panel.grid = element_blank(),
        axis.text = element_text(color = "black"),
        legend.key = element_rect(fill = "white")) +
  scale_color_manual(values = c("#E41A1C", "#377EB8"), labels = c("MES", "MEL")) +
  annotate("text", x = 0, y = 4, label = paste0("r = ", round(cor(ATAC.500$ATAC.CHEQ.CNNorm[,c(paste0(i,".CNNorm.mean.score"),paste0(i,".CPMNorm.FC"))], method="pearson")[1,2],digits = 4)), hjust=0) +
  ggtitle(label = paste0("Correlation ",i," - ATAC-seq vs CHEQ-seq ATAC Activity")) +
  labs(x = "ATAC-seq mean score", y = "CHEQ-seq - Log2 FC", color = "Enhancer\nphenotype")
ggsave(filename = paste0("analysis/CHEQseq/Plots/ATAC_correlation/ATAC_CCNorm_CHEQseq_correlation_scatter_",i,".pdf"),device = "pdf",width = 5.5,height = 4, useDingbats = F)
print(p)
}
```

```{r}
for (i in c("MM001", "MM057", "MM074", "MM087","MM029", "MM047", "MM099")) {
  print(i)
  print(paste0("All: ",cor(ATAC.500$ATAC.CHEQ.CNNorm[,c(paste0(i,".CNNorm.mean.score"),paste0(i,".CPMNorm.FC"))], method="pearson")[1,2]))
  print(paste0("Invasive: ",cor(ATAC.500$ATAC.CHEQ.CNNorm[ATAC.500$ATAC.CHEQ.CNNorm$phenotype == "Invasive", c(paste0(i,".CNNorm.mean.score"),paste0(i,".CPMNorm.FC"))], method="pearson")[1,2]))
  print(paste0("Proliferative: ",cor(ATAC.500$ATAC.CHEQ.CNNorm[ATAC.500$ATAC.CHEQ.CNNorm$phenotype == "Proliferative", c(paste0(i,".CNNorm.mean.score"),paste0(i,".CPMNorm.FC"))], method="pearson")[1,2]))
}
```


# Correlation Ac vs STARRseq 500 bp activity per cell line
```{r}
library(dplyr)
library(ggplot2)
Ac.500$Ac.STARR <- inner_join(x = Ac.500$combined[, c(1:3,5,7,9,11,13,15)],
                              y = CSE.500bp.STARR.df[,c(1:9)],
                              by = "enhancer")
Ac.500$Ac.STARR.CNNorm <- inner_join(x = Ac.500$combined[, c(1,2,4,6,8,10,12,14,16)],
                                     y = CSE.500bp.STARR.df[,c(1:9)],
                                     by = "enhancer")

for (i in c("MM001", "MM057", "MM074", "MM087", "MM029", "MM047", "MM099")) {
p<-ggplot(Ac.500$Ac.STARR.CNNorm, aes(x = Ac.500$Ac.STARR.CNNorm[,paste0(i,".CNNorm.mean.score")], 
                                      y = log2(Ac.500$Ac.STARR.CNNorm[,paste0(i,".CPMNorm.FC")]), color = phenotype)) +
  geom_smooth(method = "lm", alpha =0.1) +
  geom_point() +
  theme(panel.background = element_rect(fill = "white", colour = "white", size = 0.5),
        panel.grid = element_blank(),
        axis.text = element_text(color = "black"),
        legend.key = element_rect(fill = "white")) +
  scale_color_manual(values = c("#E41A1C", "#377EB8"), labels = c("MES", "MEL")) +
  annotate("text", x = 0, y = 4, label = paste0("r = ", round(cor(Ac.500$Ac.STARR.CNNorm[,c(paste0(i,".CNNorm.mean.score"),paste0(i,".CPMNorm.FC"))], method="pearson")[1,2], digits = 4)), hjust=0) +
  ggtitle(label = paste0("Correlation ",i," - H3K27Ac ChIP-seq vs STARR-seq ATAC Activity")) +
  labs(x = "H3K27Ac ChIP-seq mean score", y = "STARR-seq Log2 FC", color = "Enhancer\nphenotype")
ggsave(filename = paste0("analysis/CHEQseq/Plots/ATAC_correlation/H3K27Ac_CCNorm_STARRseq_correlation_scatter_",i,".pdf"),device = "pdf",width = 5.5,height = 4, useDingbats = F)
print(p)
}
```

# Correlation ATAC vs STARRseq 500 bp activity per cell line
```{r}
library(dplyr)
library(ggplot2)
ATAC.500$ATAC.STARR <- inner_join(x = ATAC.500$combined[, c(1:3,5,7,9,11,13,15)],
                                  y = CSE.500bp.STARR.df[,c(1:9)], by = "enhancer")
ATAC.500$ATAC.STARR.CNNorm <- inner_join(x = ATAC.500$combined[, c(1,2,4,6,8,10,12,14,16)],
                                         y = CSE.500bp.STARR.df[,c(1:9)], by = "enhancer")

for (i in c("MM001", "MM057", "MM074", "MM087","MM029", "MM047", "MM099")) {
p<-ggplot(ATAC.500$ATAC.STARR.CNNorm, aes(x = ATAC.500$ATAC.STARR.CNNorm[,paste0(i,".CNNorm.mean.score")], 
                                      y = log2(ATAC.500$ATAC.STARR.CNNorm[,paste0(i,".CPMNorm.FC")]), color = phenotype)) +
  geom_smooth(method = "lm", alpha =0.1) +
  geom_point() +
  theme(panel.background = element_rect(fill = "white", colour = "white", size = 0.5),
        panel.grid = element_blank(),
        axis.text = element_text(color = "black"),
        legend.key = element_rect(fill = "white")) +
  scale_color_manual(values = c("#E41A1C", "#377EB8"), labels = c("MES", "MEL")) +
  annotate("text", x = 0, y = 4, label = paste0("r = ", round(cor(ATAC.500$ATAC.STARR.CNNorm[,c(paste0(i,".CNNorm.mean.score"),paste0(i,".CPMNorm.FC"))], method="pearson")[1,2], digits = 4)), hjust=0) +
  ggtitle(label = paste0("Correlation ",i," - ATAC-seq vs STARR-seq ATAC Activity")) +
  labs(x = "ATAC-seq mean score", y = "STARR-seq Log2 FC", color = "Enhancer\nphenotype")
ggsave(filename = paste0("analysis/CHEQseq/Plots/ATAC_correlation/ATAC_CCNorm_STARRseq_correlation_scatter_",i,".pdf"),device = "pdf",width = 5.5,height = 4, useDingbats = F)
print(p)
}
```

# Correlation Ac vs ATAC
```{r}
for (i in c("MM001", "MM057", "MM074", "MM087","MM029", "MM047", "MM099")) {
  print(i)
  print(paste0("All: ",cor(ATAC.500$ATAC.STARR.CNNorm[,paste0(i,".CNNorm.mean.score")], Ac.500$Ac.STARR.CNNorm[,paste0(i,".CNNorm.mean.score")], method="pearson")))
  print(paste0("Invasive: ",cor(ATAC.500$ATAC.STARR.CNNorm[ATAC.500$ATAC.STARR.CNNorm$phenotype == "Invasive",paste0(i,".CNNorm.mean.score")], Ac.500$Ac.STARR.CNNorm[Ac.500$Ac.STARR.CNNorm$phenotype == "Invasive", paste0(i,".CNNorm.mean.score")], method="pearson")))
  print(paste0("Proliferative: ",cor(ATAC.500$ATAC.STARR.CNNorm[ATAC.500$ATAC.STARR.CNNorm$phenotype == "Proliferative",paste0(i,".CNNorm.mean.score")], Ac.500$Ac.STARR.CNNorm[Ac.500$Ac.STARR.CNNorm$phenotype == "Proliferative", paste0(i,".CNNorm.mean.score")], method="pearson")))
}
```
```{r}
library(dplyr)
library(ggplot2)
tmp <- inner_join(x = ATAC.500$combined[, c(1,4,6,8,10,12,14,16)],
                  y = Ac.500$combined[, c(1,4,6,8,10,12,14,16,17)], by = "enhancer", suffix = c(".ATAC", ".Ac"))

for (i in c("MM001", "MM057", "MM074", "MM087","MM029", "MM047", "MM099")) {
p<-ggplot(tmp, aes(x = tmp[,paste0(i,".CNNorm.mean.score.ATAC")], 
                   y = tmp[,paste0(i,".CNNorm.mean.score.Ac")], color = phenotype)) +
  geom_smooth(method = "lm", alpha =0.1) +
  geom_point() +
  theme(panel.background = element_rect(fill = "white", colour = "white", size = 0.5),
        panel.grid = element_blank(),
        axis.text = element_text(color = "black"),
        legend.key = element_rect(fill = "white")) +
  scale_color_manual(values = c("#E41A1C", "#377EB8"), labels = c("MES", "MEL")) +
  annotate("text", x = 0, y = 2, label = paste0("r = ", round(cor(tmp[,c(paste0(i,".CNNorm.mean.score.ATAC"),paste0(i,".CNNorm.mean.score.Ac"))], method="pearson")[1,2], digits = 4)), hjust=0) +
  ggtitle(label = paste0("Correlation ",i," - ATAC-seq vs H3K27ac ChIP-seq ATAC Activity")) +
  labs(x = "ATAC-seq mean score", y = "Ac mean score", color = "Enhancer\nphenotype")
# ggsave(filename = paste0("analysis/CHEQseq/Plots/ATAC_correlation/ATAC_CCNorm_STARRseq_correlation_scatter_",i,".pdf"),device = "pdf",width = 5.5,height = 4, useDingbats = F)
print(p)
}
```

```{r}
# Function to generate scale with image()
image.scale <- function(z, zlim, col = heat.colors(12),
breaks, horiz=TRUE, ylim=NULL, xlim=NULL, ...){
 if(!missing(breaks)){
  if(length(breaks) != (length(col)+1)){stop("must have one more break than colour")}
 }
 if(missing(breaks) & !missing(zlim)){
  breaks <- seq(zlim[1], zlim[2], length.out=(length(col)+1)) 
 }
 if(missing(breaks) & missing(zlim)){
  zlim <- range(z, na.rm=TRUE)
  zlim[2] <- zlim[2]+c(zlim[2]-zlim[1])*(1E-3)#adds a bit to the range in both directions
  zlim[1] <- zlim[1]-c(zlim[2]-zlim[1])*(1E-3)
  breaks <- seq(zlim[1], zlim[2], length.out=(length(col)+1))
 }
 poly <- vector(mode="list", length(col))
 for(i in seq(poly)){
  poly[[i]] <- c(breaks[i], breaks[i+1], breaks[i+1], breaks[i])
 }
 xaxt <- ifelse(horiz, "s", "n")
 yaxt <- ifelse(horiz, "n", "s")
 if(horiz){YLIM<-c(0,1); XLIM<-range(breaks)}
 if(!horiz){YLIM<-range(breaks); XLIM<-c(0,1)}
 if(missing(xlim)) xlim=XLIM
 if(missing(ylim)) ylim=YLIM
 plot(1,1,t="n",ylim=ylim, xlim=xlim, xaxt=xaxt, yaxt=yaxt, xaxs="i", yaxs="i", ...)  
 for(i in seq(poly)){
  if(horiz){
   polygon(poly[[i]], c(0,0,1,1), col=col[i], border=NA)
  }
  if(!horiz){
   polygon(c(0,0,1,1), poly[[i]], col=col[i], border=NA)
  }
 }
}
```

# Heatmap H3K27Ac and ATAC CNN and activity next to each other (sorted by ATAC signal)
```{r}
for (i in c("MM001", "MM057", "MM074", "MM087", "MM029", "MM047", "MM099")){
  library(dplyr)
  library(RColorBrewer)
  pdf(file = paste0("analysis/CHEQseq/Plots/ATAC_correlation/",i,"_CNNorm_profile_comparison_with_Ac.pdf"), 
             height = 7, width = 6.8)
  # Set layout
  par(mar=c(0,1,0.5,1))
  layout(matrix(c(1,2,2,2,2,1,3,3,3,3,1,4,4,4,4,1,5,5,5,5,1,6,7,8,9), nrow=5, ncol=5), widths=c(4.5,3,3,3,1.6), heights=c(0.8,3.5,3.5,3.5,3.5))
  # layout.show(9)
  plot.new()
  text(0.5, 0.5, paste0(i), cex=1.5, font=2)
  
  # Prepare matrix
  heatx <- ATAC.500$ATAC.CHEQ.CNNorm[,c("enhancer", paste0(i, ".CNNorm.mean.score"), paste0(i, ".CPMNorm.FC"))] %>% 
    inner_join(y = Ac.500$Ac.STARR.CNNorm[,c("enhancer", paste0(i, ".CNNorm.mean.score"), paste0(i, ".CPMNorm.FC"))], 
               by = "enhancer",
               suffix = c(".CHEQ", ".STARR"))
  heatx <-  arrange(heatx, heatx[,2])
  rownames(heatx) <- heatx$enhancer
  heatx <- as.matrix(heatx[,-1])

  # H3K27Ac
  breaks1 <- seq(min(heatx[, paste0(i, ".CNNorm.mean.score.STARR")]), max(heatx[, paste0(i, ".CNNorm.mean.score.STARR")]),length.out=100)
  colfunc1 <- colorRampPalette(brewer.pal(9, "Reds"))
  par(mar =  c( 0.5, 6, 2, 0.5 ))
  image(x = t(heatx[, paste0(i, ".CNNorm.mean.score.STARR")]),
        col=colfunc1(n = length(breaks1)-1),
        breaks=breaks1,
        xaxt="n", yaxt="n",
        xlab="", ylab="",
        main = "H3K27Ac ChIP-seq",
        cex.main = 1.3)
  axis(side = 2,
       at=seq(0,1,length.out=nrow(heatx)),
       labels = rownames(heatx),
       las= 2, tick = F, cex.axis = 0.9, font=2)
  
  # ATAC
  breaks2 <- seq(min(heatx[, paste0(i, ".CNNorm.mean.score.CHEQ")]), max(heatx[, paste0(i, ".CNNorm.mean.score.CHEQ")]),length.out=100)
  colfunc2 <- colorRampPalette(brewer.pal(9, "Blues"))
  par(mar =  c( 0.5, 0.5, 2, 0.5 ))
  image(x = t(heatx[, paste0(i, ".CNNorm.mean.score.CHEQ")]),
        col=colfunc2(n = length(breaks2)-1),
        breaks=breaks2,
        xaxt="n", yaxt="n",
        xlab="", ylab="",
        main = "ATAC-seq",
        cex.main = 1.3)

  # Activity CHEQseq
  breaks3 <- c(seq(min(log2(heatx[, paste0(i, ".CPMNorm.FC.CHEQ")])), max(log2(CSE.500bp.CHEQ[[i]]$merged.cpm.input.norm[CSE.500bp.CHEQ[[i]]$merged.cpm.input.norm$padj >= 0.05, "CPMNorm.FC"])),length.out=75), 
               seq(max(log2(CSE.500bp.CHEQ[[i]]$merged.cpm.input.norm[CSE.500bp.CHEQ[[i]]$merged.cpm.input.norm$padj >= 0.05, "CPMNorm.FC"])),max(log2(heatx[, paste0(i, ".CPMNorm.FC.CHEQ")])),length.out=75)[-1])
  colfunc3 <- colorRampPalette(c("#FCFBFD","#FCFBFD","#3F007D"))
  par(mar =  c( 0.5, 0.5, 2, 0.5 ))
  image(x = t(log2(heatx[, paste0(i, ".CPMNorm.FC.CHEQ")])),
        col=colfunc3(n = length(breaks3)-1),
        breaks=breaks3,
        xaxt="n", yaxt="n",
        xlab="", ylab="",
        main = "CHEQ-seq",
        cex.main = 1.3)

  # Activity STARRseq
  breaks4 <- c(seq(min(log2(heatx[, paste0(i, ".CPMNorm.FC.STARR")])), max(log2(CSE.500bp.STARR[[i]]$merged.input.norm[CSE.500bp.STARR[[i]]$merged.input.norm$padj >= 0.05, "CPMNorm.FC"])),length.out=75), 
               seq(max(log2(CSE.500bp.STARR[[i]]$merged.input.norm[CSE.500bp.STARR[[i]]$merged.input.norm$padj >= 0.05, "CPMNorm.FC"])), max(log2(heatx[, paste0(i, ".CPMNorm.FC.STARR")])),length.out=75)[-1])
  colfunc4 <- colorRampPalette(c("#FCFBFD", "#FCFBFD", "#FCFBFD", "#66C2A4", "#00441B"))
  par(mar =  c( 0.5, 0.5, 2, 0.5 ))
  image(x = t(log2(heatx[, paste0(i, ".CPMNorm.FC.STARR")])),
        col=colfunc4(n = length(breaks4)-1),
        breaks=breaks4,
        xaxt="n", yaxt="n",
        xlab="", ylab="",
        main = "STARR-seq",
        cex.main = 1.3)
  
  # Scale H3K27Ac
  par(mar = c(2.5,0.5,3.25,3.8))
  image.scale(heatx[, paste0(i, ".CNNorm.mean.score.STARR")], col=colfunc1(length(breaks1)-1), breaks=breaks1, horiz=FALSE, yaxt="n")
  mtext(text = "Mean score", side = 3, cex = 0.55, font = 2, line = 0.5, adj = 0)
  axis(side=4, at=seq(0, max(heatx[, paste0(i, ".CNNorm.mean.score.STARR")]), by=1), las=2)
  
  # Scale ATAC
  par(mar = c(2.5,0.5,3.25,3.8))
  image.scale(heatx[, paste0(i, ".CNNorm.mean.score.CHEQ")], col=colfunc2(length(breaks2)-1), breaks=breaks2, horiz=FALSE, yaxt="n")
  mtext(text = "Mean score", side = 3, cex = 0.55, font = 2, line = 0.5, adj = 0)
  axis(side=4, at=seq(0, max(heatx[, paste0(i, ".CNNorm.mean.score.CHEQ")]), by=1), las=2)

  #Scale Activity CHEQseq
  par(mar = c(2.5,0.5,3.25,3.8))
  image.scale(log2(heatx[, paste0(i, ".CPMNorm.FC.CHEQ")]), col=colfunc3(length(breaks3)-1), breaks=breaks3, horiz=FALSE, yaxt="n")
  mtext(text = "Log2 FC", side = 3, cex = 0.55, font = 2, line = 0.5, adj = 0)
  axis(side=4, at=seq(-10, max(log2(heatx[, paste0(i, ".CPMNorm.FC.CHEQ")])), by=1), las=2)
  
  #Scale Activity STARRseq
  par(mar = c(2.5,0.5,3.25,3.8))
  image.scale(log2(heatx[, paste0(i, ".CPMNorm.FC.STARR")]), col=colfunc4(length(breaks4)-1), breaks=breaks4, horiz=FALSE, yaxt="n")
  mtext(text = "Log2 FC", side = 3, cex = 0.55, font = 2, line = 0.5, adj = 0)
  axis(side=4, at=seq(-10, max(log2(heatx[, paste0(i, ".CPMNorm.FC.STARR")])), by=1), las=2)
  
  box()
  dev.off()
}
```

# Heatmap H3K27Ac and ATAC CNN and activity next to each other (sorted by CHEQseq value)
```{r}
for (i in c("MM001", "MM057", "MM074", "MM087", "MM029", "MM047", "MM099")){
  library(dplyr)
  library(RColorBrewer)
  pdf(file = paste0("analysis/CHEQseq/Plots/ATAC_correlation/",i,"_CNNorm_profile_comparison_with_Ac_CHEQseq_sorted.pdf"), 
             height = 7, width = 6.8)
  # Set layout
  par(mar=c(0,1,0.5,1))
  layout(matrix(c(1,2,2,2,2,1,3,3,3,3,1,4,4,4,4,1,5,5,5,5,1,6,7,8,9), nrow=5, ncol=5), widths=c(4.5,3,3,3,1.6), heights=c(0.8,3.5,3.5,3.5,3.5))
  # layout.show(9)
  plot.new()
  text(0.5, 0.5, paste0(i), cex=1.5, font=2)
  
  # Prepare matrix
  heatx <- ATAC.500$ATAC.CHEQ.CNNorm[,c("enhancer", paste0(i, ".CNNorm.mean.score"), paste0(i, ".CPMNorm.FC"))] %>% 
    inner_join(y = Ac.500$Ac.STARR.CNNorm[,c("enhancer", paste0(i, ".CNNorm.mean.score"), paste0(i, ".CPMNorm.FC"))], 
               by = "enhancer",
               suffix = c(".CHEQ", ".STARR"))
  heatx <-  arrange(heatx, heatx[,3])
  rownames(heatx) <- heatx$enhancer
  heatx <- as.matrix(heatx[,-1])

  # H3K27Ac
  breaks1 <- seq(min(heatx[, paste0(i, ".CNNorm.mean.score.STARR")]), max(heatx[, paste0(i, ".CNNorm.mean.score.STARR")]),length.out=100)
  colfunc1 <- colorRampPalette(brewer.pal(9, "Reds"))
  par(mar =  c( 0.5, 6, 2, 0.5 ))
  image(x = t(heatx[, paste0(i, ".CNNorm.mean.score.STARR")]),
        col=colfunc1(n = length(breaks1)-1),
        breaks=breaks1,
        xaxt="n", yaxt="n",
        xlab="", ylab="",
        main = "H3K27Ac ChIP-seq",
        cex.main = 1.3)
  axis(side = 2,
       at=seq(0,1,length.out=nrow(heatx)),
       labels = rownames(heatx),
       las= 2, tick = F, cex.axis = 0.9, font=2)
  
  # ATAC
  breaks2 <- seq(min(heatx[, paste0(i, ".CNNorm.mean.score.CHEQ")]), max(heatx[, paste0(i, ".CNNorm.mean.score.CHEQ")]),length.out=100)
  colfunc2 <- colorRampPalette(brewer.pal(9, "Blues"))
  par(mar =  c( 0.5, 0.5, 2, 0.5 ))
  image(x = t(heatx[, paste0(i, ".CNNorm.mean.score.CHEQ")]),
        col=colfunc2(n = length(breaks2)-1),
        breaks=breaks2,
        xaxt="n", yaxt="n",
        xlab="", ylab="",
        main = "ATAC-seq",
        cex.main = 1.3)

  # Activity CHEQseq
  breaks3 <- c(seq(min(log2(heatx[, paste0(i, ".CPMNorm.FC.CHEQ")])), max(log2(CSE.500bp.CHEQ[[i]]$merged.cpm.input.norm[CSE.500bp.CHEQ[[i]]$merged.cpm.input.norm$padj >= 0.05, "CPMNorm.FC"])),length.out=75), 
               seq(max(log2(CSE.500bp.CHEQ[[i]]$merged.cpm.input.norm[CSE.500bp.CHEQ[[i]]$merged.cpm.input.norm$padj >= 0.05, "CPMNorm.FC"])),max(log2(heatx[, paste0(i, ".CPMNorm.FC.CHEQ")])),length.out=75)[-1])
  colfunc3 <- colorRampPalette(c("#FCFBFD","#FCFBFD","#3F007D"))
  par(mar =  c( 0.5, 0.5, 2, 0.5 ))
  image(x = t(log2(heatx[, paste0(i, ".CPMNorm.FC.CHEQ")])),
        col=colfunc3(n = length(breaks3)-1),
        breaks=breaks3,
        xaxt="n", yaxt="n",
        xlab="", ylab="",
        main = "CHEQ-seq",
        cex.main = 1.3)

  # Activity STARRseq
  breaks4 <- c(seq(min(log2(heatx[, paste0(i, ".CPMNorm.FC.STARR")])), max(log2(CSE.500bp.STARR[[i]]$merged.input.norm[CSE.500bp.STARR[[i]]$merged.input.norm$padj >= 0.05, "CPMNorm.FC"])),length.out=75),
               seq(max(log2(CSE.500bp.STARR[[i]]$merged.input.norm[CSE.500bp.STARR[[i]]$merged.input.norm$padj >= 0.05, "CPMNorm.FC"])), max(log2(heatx[, paste0(i, ".CPMNorm.FC.STARR")])),length.out=75)[-1])
  
  colfunc4 <- colorRampPalette(c("#FCFBFD", "#FCFBFD", "#FCFBFD", "#66C2A4", "#00441B"))
  par(mar =  c( 0.5, 0.5, 2, 0.5 ))
  image(x = t(log2(heatx[, paste0(i, ".CPMNorm.FC.STARR")])),
        col=colfunc4(n = length(breaks4)-1),
        breaks=breaks4,
        xaxt="n", yaxt="n",
        xlab="", ylab="",
        main = "STARR-seq",
        cex.main = 1.3)
  
  # Scale H3K27Ac
  par(mar = c(2.5,0.5,3.25,3.8))
  image.scale(heatx[, paste0(i, ".CNNorm.mean.score.STARR")], col=colfunc1(length(breaks1)-1), breaks=breaks1, horiz=FALSE, yaxt="n")
  mtext(text = "Mean score", side = 3, cex = 0.55, font = 2, line = 0.5, adj = 0)
  axis(side=4, at=seq(0, max(heatx[, paste0(i, ".CNNorm.mean.score.STARR")]), by=1), las=2)
  
  # Scale ATAC
  par(mar = c(2.5,0.5,3.25,3.8))
  image.scale(heatx[, paste0(i, ".CNNorm.mean.score.CHEQ")], col=colfunc2(length(breaks2)-1), breaks=breaks2, horiz=FALSE, yaxt="n")
  mtext(text = "Mean score", side = 3, cex = 0.55, font = 2, line = 0.5, adj = 0)
  axis(side=4, at=seq(0, max(heatx[, paste0(i, ".CNNorm.mean.score.CHEQ")]), by=1), las=2)

  #Scale Activity CHEQseq
  par(mar = c(2.5,0.5,3.25,3.8))
  image.scale(log2(heatx[, paste0(i, ".CPMNorm.FC.CHEQ")]), col=colfunc3(length(breaks3)-1), breaks=breaks3, horiz=FALSE, yaxt="n")
  mtext(text = "Log2 FC", side = 3, cex = 0.55, font = 2, line = 0.5, adj = 0)
  axis(side=4, at=seq(-10, max(log2(heatx[, paste0(i, ".CPMNorm.FC.CHEQ")])), by=1), las=2)
  
  #Scale Activity STARRseq
  par(mar = c(2.5,0.5,3.25,3.8))
  image.scale(log2(heatx[, paste0(i, ".CPMNorm.FC.STARR")]), col=colfunc4(length(breaks4)-1), breaks=breaks4, horiz=FALSE, yaxt="n")
  mtext(text = "Log2 FC", side = 3, cex = 0.55, font = 2, line = 0.5, adj = 0)
  axis(side=4, at=seq(-10, max(log2(heatx[, paste0(i, ".CPMNorm.FC.STARR")])), by=1), las=2)
  
  box()
  dev.off()
}
```

# Correlation table CHEQseq
```{r}
library(RColorBrewer)
# Generate correlation table
corratac <- matrix(NA, nrow = 7, ncol = 3)
colnames(corratac) <- c("Acvs.CHEQseq", "ATAC.vs.CHEQseq", "RNA.vs.CHEQseq")
rownames(corratac) <- c("MM001", "MM057", "MM074", "MM087", "MM029", "MM047", "MM099")
for (i in c(0:6)){
  p1 <- cor(Ac.500$Ac.CHEQ.CNNorm[,-c(1,2,17)], method="pearson")[8+i,1+i]
  p2 <- cor(ATAC.500$ATAC.CHEQ.CNNorm[,-c(1,2,17)], method="pearson")[8+i,1+i]
  p3 <- cor(CSE.500.rna[,-c(8,16:18)], method="pearson")[8+i,1+i]
  corratac[1+i,1] <- p1
  print(p1)
  corratac[1+i,2] <- p2
  print(p2)
  corratac[1+i,3] <- p3
  print(p3)
}

setEPS()
postscript(file = paste0("analysis/CHEQseq/Plots/ATAC_correlation/Correlation_table_CHEQ.eps"), 
           horizontal = FALSE, onefile = FALSE, paper = "special", height = 8.27, width = 8)
  # Set layout
  par(mar=c(0,1,0.5,1))
  layout(matrix(c(1,1,2,3), nrow=2, ncol=2), widths=c(5.5,0.7), heights=c(2,5))
  # layout.show(7)
  
  # Plot data
  breaks2 <- seq(-1, 1,length.out=100)
  colfunc2 <- colorRampPalette(brewer.pal(11, "RdBu"))
  par(mar =  c( 0.5, 5, 4, 0.5 ) )
  image(x = t(corratac[nrow(corratac):1,]),
        col=colfunc2(n = length(breaks2)-1),
        breaks=breaks2,
        xaxt="n", yaxt="n",
        xlab="", ylab="")
  title("Correlation score", line = 3, cex =1)
  axis(side = 2,
       at=seq(0,1,length.out=nrow(corratac)),
       labels = rownames(corratac[nrow(corratac):1,]),
       las= 2, tick = F, cex.axis = 1.2, font=1)
  axis(side = 3,
       at=seq(0,1,length.out=ncol(corratac)),
       labels = c("H3K27Ac ChIP-seq vs CHEQ-seq", "ATAC-seq vs CHEQ-seq", "RNA-seq vs CHEQ-seq"),
       las= 1, tick = F, cex.axis = 1, font=1)
    
  # Add scale
  par(mar=c(2.5,0.5,4,3))
  image.scale(corratac, col=colfunc2(length(breaks2)-1), breaks=breaks2, horiz=FALSE, yaxt="n")
  #mtext(text = "Mean score", side = 3, cex = 0.55, font = 2, line = 0.5, adj = 0)
  axis(side=4, at=seq(-1, 1, by=1), las=2)
  box()
dev.off()
```

# Correlation table STARRseq
```{r}
library(RColorBrewer)
rnaseq.mean.full.ATAC <- transform(rnaseq.mean.full.ATAC, 
                                   MM001 = as.numeric(MM001),
                                   MM057 = as.numeric(MM057),
                                   MM074 = as.numeric(MM074),
                                   MM087 = as.numeric(MM087),
                                   MM029 = as.numeric(MM029),
                                   MM047 = as.numeric(MM047),
                                   MM099 = as.numeric(MM099))
CSE.500.STARR.rna <- inner_join(x = rnaseq.mean.full.ATAC, y = CSE.500bp.STARR.df, by = "enhancer")
# Generate correlation table
corrAc <- matrix(NA, nrow = 7, ncol = 3)
colnames(corrAc) <- c("Ac.vs.STARRseq", "ATAC.vs.STARRseq", "RNA.vs.STARRseq")
rownames(corrAc) <- c("MM001", "MM057", "MM074", "MM087", "MM029", "MM047", "MM099")
for (i in c(0:6)){
  p1 <- cor(Ac.500$Ac.STARR.CNNorm[,-c(1,2,17)], method="pearson")[8+i,1+i]
  p2 <- cor(ATAC.500$ATAC.STARR.CNNorm[,-c(1,2,17)], method="pearson")[8+i,1+i]
  p3 <- cor(CSE.500.STARR.rna[,-c(8,16:18)], method="pearson")[8+i,1+i]
  corrAc[1+i,1] <- p1
  corrAc[1+i,2] <- p2
  corrAc[1+i,3] <- p3
}

setEPS()
postscript(file = paste0("analysis/CHEQseq/Plots/ATAC_correlation/Correlation_table_STARR.eps"), 
           horizontal = FALSE, onefile = FALSE, paper = "special", height = 8.27, width = 8)
  # Set layout
  par(mar=c(0,1,0.5,1))
  layout(matrix(c(1,1,2,3), nrow=2, ncol=2), widths=c(5.5,0.7), heights=c(2,5))
  # layout.show(7)
  
  # Plot data
  breaks2 <- seq(-1, 1,length.out=100)
  colfunc2 <- colorRampPalette(brewer.pal(11, "RdBu"))
  par(mar =  c( 0.5, 5, 4, 0.5 ) )
  image(x = t(corrAc[nrow(corrAc):1,]),
        col=colfunc2(n = length(breaks2)-1),
        breaks=breaks2,
        xaxt="n", yaxt="n",
        xlab="", ylab="")
  title("Correlation score", line = 3, cex =1)
  axis(side = 2,
       at=seq(0,1,length.out=nrow(corrAc)),
       labels = rownames(corrAc[nrow(corrAc):1,]),
       las= 2, tick = F, cex.axis = 1.2, font=1)
  axis(side = 3,
       at=seq(0,1,length.out=ncol(corrAc)),
       labels = c("H3K27Ac ChIP-seq vs STARR-seq", "ATAC-seq vs STARR-seq", "RNA-seq vs STARR-seq"),
       las= 1, tick = F, cex.axis = 1, font=1)
    
  # Add scale
  par(mar=c(2.5,0.5,4,3))
  image.scale(corrAc, col=colfunc2(length(breaks2)-1), breaks=breaks2, horiz=FALSE, yaxt="n")
  #mtext(text = "Mean score", side = 3, cex = 0.55, font = 2, line = 0.5, adj = 0)
  axis(side=4, at=seq(-1, 1, by=1), las=2)
  box()
dev.off()
```

