{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Process synthetic enhancers sequencing data\n",
    "# = BC quantification\n",
    "\n",
    "Run on one of the Genius nodes to match the modules."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "cd /staging/leuven/stg_00002/lcb/lminn/Enhancer_design/MPRA/CHEQ-seq_cloning2/BC_quantification_20201006"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 00. Combine raw reads and 01. Fastqc\n",
    "\n",
    "- Create within your folder the 00.raw_reads folder where you create a text file CHEQseq_synth_names.txt.\n",
    "- Within the text file put the names of the samples, e.g. CSE__8636ca__CheqSeq_cDNA_MM001_2000_enh_pool_Liesbeth. Make sure to put a line symbol at the end of the text file so that also the last sample is processed.\n",
    "- Provide the sequencing run paths \n",
    "- The script will then combine the same samples from different runs within 1 fastq\n",
    "- Make sure to check if you want to use R1 or R2, in this case R2 is the RC of R1, but we will use R1 because the script is made for this (the rBCs will be not be RC for instance in R1)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "__FYI: We have now only 1 run so the script will just put those fastqs within the 00.raw_reads folder; except for the MM029 where we have two runs__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘01.fastqc’: File exists\n",
      "mkdir: cannot create directory ‘02.cleaned_reads’: File exists\n",
      "mkdir: cannot create directory ‘03.sequencing_saturation’: File exists\n",
      "mkdir: cannot create directory ‘10.bc_count’: File exists\n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "mkdir 01.fastqc\n",
    "mkdir 02.cleaned_reads\n",
    "mkdir 03.sequencing_saturation\n",
    "mkdir 10.bc_count"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cat: /staging/leuven/stg_00002/lcb/ngs_runs/NovaSeq6000_20201013/Demultiplexed/CSE__6f8b76__*R1*: No such file or directory\n",
      "Started analysis of CSE__6f8b76__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_cDNA_R1.fastq.gz\n",
      "Approx 5% complete for CSE__6f8b76__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_cDNA_R1.fastq.gz\n",
      "Approx 10% complete for CSE__6f8b76__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_cDNA_R1.fastq.gz\n",
      "Approx 15% complete for CSE__6f8b76__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_cDNA_R1.fastq.gz\n",
      "Approx 20% complete for CSE__6f8b76__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_cDNA_R1.fastq.gz\n",
      "Approx 25% complete for CSE__6f8b76__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_cDNA_R1.fastq.gz\n",
      "Approx 30% complete for CSE__6f8b76__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_cDNA_R1.fastq.gz\n",
      "Approx 35% complete for CSE__6f8b76__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_cDNA_R1.fastq.gz\n",
      "Approx 40% complete for CSE__6f8b76__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_cDNA_R1.fastq.gz\n",
      "Approx 45% complete for CSE__6f8b76__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_cDNA_R1.fastq.gz\n",
      "Approx 50% complete for CSE__6f8b76__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_cDNA_R1.fastq.gz\n",
      "Approx 55% complete for CSE__6f8b76__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_cDNA_R1.fastq.gz\n",
      "Approx 60% complete for CSE__6f8b76__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_cDNA_R1.fastq.gz\n",
      "Approx 65% complete for CSE__6f8b76__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_cDNA_R1.fastq.gz\n",
      "Approx 70% complete for CSE__6f8b76__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_cDNA_R1.fastq.gz\n",
      "Approx 75% complete for CSE__6f8b76__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_cDNA_R1.fastq.gz\n",
      "Approx 80% complete for CSE__6f8b76__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_cDNA_R1.fastq.gz\n",
      "Approx 85% complete for CSE__6f8b76__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_cDNA_R1.fastq.gz\n",
      "Approx 90% complete for CSE__6f8b76__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_cDNA_R1.fastq.gz\n",
      "Approx 95% complete for CSE__6f8b76__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_cDNA_R1.fastq.gz\n",
      "Analysis complete for CSE__6f8b76__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_cDNA_R1.fastq.gz\n",
      "cat: /staging/leuven/stg_00002/lcb/ngs_runs/NovaSeq6000_20201013/Demultiplexed/CSE__81ba64__*R1*: No such file or directory\n",
      "Started analysis of CSE__81ba64__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_cDNA_R1.fastq.gz\n",
      "Approx 5% complete for CSE__81ba64__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_cDNA_R1.fastq.gz\n",
      "Approx 10% complete for CSE__81ba64__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_cDNA_R1.fastq.gz\n",
      "Approx 15% complete for CSE__81ba64__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_cDNA_R1.fastq.gz\n",
      "Approx 20% complete for CSE__81ba64__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_cDNA_R1.fastq.gz\n",
      "Approx 25% complete for CSE__81ba64__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_cDNA_R1.fastq.gz\n",
      "Approx 30% complete for CSE__81ba64__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_cDNA_R1.fastq.gz\n",
      "Approx 35% complete for CSE__81ba64__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_cDNA_R1.fastq.gz\n",
      "Approx 40% complete for CSE__81ba64__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_cDNA_R1.fastq.gz\n",
      "Approx 45% complete for CSE__81ba64__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_cDNA_R1.fastq.gz\n",
      "Approx 50% complete for CSE__81ba64__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_cDNA_R1.fastq.gz\n",
      "Approx 55% complete for CSE__81ba64__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_cDNA_R1.fastq.gz\n",
      "Approx 60% complete for CSE__81ba64__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_cDNA_R1.fastq.gz\n",
      "Approx 65% complete for CSE__81ba64__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_cDNA_R1.fastq.gz\n",
      "Approx 70% complete for CSE__81ba64__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_cDNA_R1.fastq.gz\n",
      "Approx 75% complete for CSE__81ba64__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_cDNA_R1.fastq.gz\n",
      "Approx 80% complete for CSE__81ba64__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_cDNA_R1.fastq.gz\n",
      "Approx 85% complete for CSE__81ba64__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_cDNA_R1.fastq.gz\n",
      "Approx 90% complete for CSE__81ba64__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_cDNA_R1.fastq.gz\n",
      "Approx 95% complete for CSE__81ba64__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_cDNA_R1.fastq.gz\n",
      "Analysis complete for CSE__81ba64__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_cDNA_R1.fastq.gz\n",
      "cat: /staging/leuven/stg_00002/lcb/ngs_runs/NovaSeq6000_20201013/Demultiplexed/CSE__839e9e__*R1*: No such file or directory\n",
      "Started analysis of CSE__839e9e__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_cDNA_R1.fastq.gz\n",
      "Approx 5% complete for CSE__839e9e__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_cDNA_R1.fastq.gz\n",
      "Approx 10% complete for CSE__839e9e__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_cDNA_R1.fastq.gz\n",
      "Approx 15% complete for CSE__839e9e__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_cDNA_R1.fastq.gz\n",
      "Approx 20% complete for CSE__839e9e__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_cDNA_R1.fastq.gz\n",
      "Approx 25% complete for CSE__839e9e__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_cDNA_R1.fastq.gz\n",
      "Approx 30% complete for CSE__839e9e__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_cDNA_R1.fastq.gz\n",
      "Approx 35% complete for CSE__839e9e__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_cDNA_R1.fastq.gz\n",
      "Approx 40% complete for CSE__839e9e__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_cDNA_R1.fastq.gz\n",
      "Approx 45% complete for CSE__839e9e__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_cDNA_R1.fastq.gz\n",
      "Approx 50% complete for CSE__839e9e__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_cDNA_R1.fastq.gz\n",
      "Approx 55% complete for CSE__839e9e__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_cDNA_R1.fastq.gz\n",
      "Approx 60% complete for CSE__839e9e__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_cDNA_R1.fastq.gz\n",
      "Approx 65% complete for CSE__839e9e__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_cDNA_R1.fastq.gz\n",
      "Approx 70% complete for CSE__839e9e__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_cDNA_R1.fastq.gz\n",
      "Approx 75% complete for CSE__839e9e__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_cDNA_R1.fastq.gz\n",
      "Approx 80% complete for CSE__839e9e__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_cDNA_R1.fastq.gz\n",
      "Approx 85% complete for CSE__839e9e__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_cDNA_R1.fastq.gz\n",
      "Approx 90% complete for CSE__839e9e__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_cDNA_R1.fastq.gz\n",
      "Approx 95% complete for CSE__839e9e__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_cDNA_R1.fastq.gz\n",
      "Analysis complete for CSE__839e9e__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_cDNA_R1.fastq.gz\n",
      "Started analysis of CSE__7cb57c__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_cDNA_R1.fastq.gz\n",
      "Approx 5% complete for CSE__7cb57c__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_cDNA_R1.fastq.gz\n",
      "Approx 10% complete for CSE__7cb57c__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_cDNA_R1.fastq.gz\n",
      "Approx 15% complete for CSE__7cb57c__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_cDNA_R1.fastq.gz\n",
      "Approx 20% complete for CSE__7cb57c__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_cDNA_R1.fastq.gz\n",
      "Approx 25% complete for CSE__7cb57c__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_cDNA_R1.fastq.gz\n",
      "Approx 30% complete for CSE__7cb57c__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_cDNA_R1.fastq.gz\n",
      "Approx 35% complete for CSE__7cb57c__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_cDNA_R1.fastq.gz\n",
      "Approx 40% complete for CSE__7cb57c__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_cDNA_R1.fastq.gz\n",
      "Approx 45% complete for CSE__7cb57c__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_cDNA_R1.fastq.gz\n",
      "Approx 50% complete for CSE__7cb57c__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_cDNA_R1.fastq.gz\n",
      "Approx 55% complete for CSE__7cb57c__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_cDNA_R1.fastq.gz\n",
      "Approx 60% complete for CSE__7cb57c__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_cDNA_R1.fastq.gz\n",
      "Approx 65% complete for CSE__7cb57c__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_cDNA_R1.fastq.gz\n",
      "Approx 70% complete for CSE__7cb57c__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_cDNA_R1.fastq.gz\n",
      "Approx 75% complete for CSE__7cb57c__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_cDNA_R1.fastq.gz\n",
      "Approx 80% complete for CSE__7cb57c__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_cDNA_R1.fastq.gz\n",
      "Approx 85% complete for CSE__7cb57c__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_cDNA_R1.fastq.gz\n",
      "Approx 90% complete for CSE__7cb57c__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_cDNA_R1.fastq.gz\n",
      "Approx 95% complete for CSE__7cb57c__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_cDNA_R1.fastq.gz\n",
      "Analysis complete for CSE__7cb57c__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_cDNA_R1.fastq.gz\n",
      "cat: /staging/leuven/stg_00002/lcb/ngs_runs/NovaSeq6000_20201013/Demultiplexed/CSE__8f7a3e__*R1*: No such file or directory\n",
      "Started analysis of CSE__8f7a3e__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_cDNA_R1.fastq.gz\n",
      "Approx 5% complete for CSE__8f7a3e__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_cDNA_R1.fastq.gz\n",
      "Approx 10% complete for CSE__8f7a3e__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_cDNA_R1.fastq.gz\n",
      "Approx 15% complete for CSE__8f7a3e__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_cDNA_R1.fastq.gz\n",
      "Approx 20% complete for CSE__8f7a3e__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_cDNA_R1.fastq.gz\n",
      "Approx 25% complete for CSE__8f7a3e__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_cDNA_R1.fastq.gz\n",
      "Approx 30% complete for CSE__8f7a3e__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_cDNA_R1.fastq.gz\n",
      "Approx 35% complete for CSE__8f7a3e__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_cDNA_R1.fastq.gz\n",
      "Approx 40% complete for CSE__8f7a3e__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_cDNA_R1.fastq.gz\n",
      "Approx 45% complete for CSE__8f7a3e__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_cDNA_R1.fastq.gz\n",
      "Approx 50% complete for CSE__8f7a3e__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_cDNA_R1.fastq.gz\n",
      "Approx 55% complete for CSE__8f7a3e__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_cDNA_R1.fastq.gz\n",
      "Approx 60% complete for CSE__8f7a3e__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_cDNA_R1.fastq.gz\n",
      "Approx 65% complete for CSE__8f7a3e__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_cDNA_R1.fastq.gz\n",
      "Approx 70% complete for CSE__8f7a3e__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_cDNA_R1.fastq.gz\n",
      "Approx 75% complete for CSE__8f7a3e__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_cDNA_R1.fastq.gz\n",
      "Approx 80% complete for CSE__8f7a3e__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_cDNA_R1.fastq.gz\n",
      "Approx 85% complete for CSE__8f7a3e__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_cDNA_R1.fastq.gz\n",
      "Approx 90% complete for CSE__8f7a3e__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_cDNA_R1.fastq.gz\n",
      "Approx 95% complete for CSE__8f7a3e__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_cDNA_R1.fastq.gz\n",
      "Analysis complete for CSE__8f7a3e__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_cDNA_R1.fastq.gz\n",
      "cat: /staging/leuven/stg_00002/lcb/ngs_runs/NovaSeq6000_20201013/Demultiplexed/CSE__8b9ec1__*R1*: No such file or directory\n",
      "Started analysis of CSE__8b9ec1__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_plasmid_R1.fastq.gz\n",
      "Approx 5% complete for CSE__8b9ec1__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_plasmid_R1.fastq.gz\n",
      "Approx 10% complete for CSE__8b9ec1__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_plasmid_R1.fastq.gz\n",
      "Approx 15% complete for CSE__8b9ec1__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_plasmid_R1.fastq.gz\n",
      "Approx 20% complete for CSE__8b9ec1__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_plasmid_R1.fastq.gz\n",
      "Approx 25% complete for CSE__8b9ec1__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_plasmid_R1.fastq.gz\n",
      "Approx 30% complete for CSE__8b9ec1__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_plasmid_R1.fastq.gz\n",
      "Approx 35% complete for CSE__8b9ec1__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_plasmid_R1.fastq.gz\n",
      "Approx 40% complete for CSE__8b9ec1__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_plasmid_R1.fastq.gz\n",
      "Approx 45% complete for CSE__8b9ec1__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_plasmid_R1.fastq.gz\n",
      "Approx 50% complete for CSE__8b9ec1__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_plasmid_R1.fastq.gz\n",
      "Approx 55% complete for CSE__8b9ec1__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_plasmid_R1.fastq.gz\n",
      "Approx 60% complete for CSE__8b9ec1__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_plasmid_R1.fastq.gz\n",
      "Approx 65% complete for CSE__8b9ec1__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_plasmid_R1.fastq.gz\n",
      "Approx 70% complete for CSE__8b9ec1__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_plasmid_R1.fastq.gz\n",
      "Approx 75% complete for CSE__8b9ec1__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_plasmid_R1.fastq.gz\n",
      "Approx 80% complete for CSE__8b9ec1__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_plasmid_R1.fastq.gz\n",
      "Approx 85% complete for CSE__8b9ec1__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_plasmid_R1.fastq.gz\n",
      "Approx 90% complete for CSE__8b9ec1__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_plasmid_R1.fastq.gz\n",
      "Approx 95% complete for CSE__8b9ec1__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_plasmid_R1.fastq.gz\n",
      "Analysis complete for CSE__8b9ec1__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_plasmid_R1.fastq.gz\n",
      "cat: /staging/leuven/stg_00002/lcb/ngs_runs/NovaSeq6000_20201013/Demultiplexed/CSE__cd9463__*R1*: No such file or directory\n",
      "Started analysis of CSE__cd9463__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_plasmid_R1.fastq.gz\n",
      "Approx 5% complete for CSE__cd9463__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_plasmid_R1.fastq.gz\n",
      "Approx 10% complete for CSE__cd9463__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_plasmid_R1.fastq.gz\n",
      "Approx 15% complete for CSE__cd9463__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_plasmid_R1.fastq.gz\n",
      "Approx 20% complete for CSE__cd9463__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_plasmid_R1.fastq.gz\n",
      "Approx 25% complete for CSE__cd9463__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_plasmid_R1.fastq.gz\n",
      "Approx 30% complete for CSE__cd9463__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_plasmid_R1.fastq.gz\n",
      "Approx 35% complete for CSE__cd9463__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_plasmid_R1.fastq.gz\n",
      "Approx 40% complete for CSE__cd9463__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_plasmid_R1.fastq.gz\n",
      "Approx 45% complete for CSE__cd9463__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_plasmid_R1.fastq.gz\n",
      "Approx 50% complete for CSE__cd9463__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_plasmid_R1.fastq.gz\n",
      "Approx 55% complete for CSE__cd9463__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_plasmid_R1.fastq.gz\n",
      "Approx 60% complete for CSE__cd9463__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_plasmid_R1.fastq.gz\n",
      "Approx 65% complete for CSE__cd9463__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_plasmid_R1.fastq.gz\n",
      "Approx 70% complete for CSE__cd9463__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_plasmid_R1.fastq.gz\n",
      "Approx 75% complete for CSE__cd9463__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_plasmid_R1.fastq.gz\n",
      "Approx 80% complete for CSE__cd9463__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_plasmid_R1.fastq.gz\n",
      "Approx 85% complete for CSE__cd9463__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_plasmid_R1.fastq.gz\n",
      "Approx 90% complete for CSE__cd9463__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_plasmid_R1.fastq.gz\n",
      "Approx 95% complete for CSE__cd9463__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_plasmid_R1.fastq.gz\n",
      "Analysis complete for CSE__cd9463__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_plasmid_R1.fastq.gz\n",
      "cat: /staging/leuven/stg_00002/lcb/ngs_runs/NovaSeq6000_20201013/Demultiplexed/CSE__0bf4ec__*R1*: No such file or directory\n",
      "Started analysis of CSE__0bf4ec__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_plasmid_R1.fastq.gz\n",
      "Approx 5% complete for CSE__0bf4ec__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_plasmid_R1.fastq.gz\n",
      "Approx 10% complete for CSE__0bf4ec__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_plasmid_R1.fastq.gz\n",
      "Approx 15% complete for CSE__0bf4ec__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_plasmid_R1.fastq.gz\n",
      "Approx 20% complete for CSE__0bf4ec__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_plasmid_R1.fastq.gz\n",
      "Approx 25% complete for CSE__0bf4ec__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_plasmid_R1.fastq.gz\n",
      "Approx 30% complete for CSE__0bf4ec__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_plasmid_R1.fastq.gz\n",
      "Approx 35% complete for CSE__0bf4ec__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_plasmid_R1.fastq.gz\n",
      "Approx 40% complete for CSE__0bf4ec__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_plasmid_R1.fastq.gz\n",
      "Approx 45% complete for CSE__0bf4ec__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_plasmid_R1.fastq.gz\n",
      "Approx 50% complete for CSE__0bf4ec__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_plasmid_R1.fastq.gz\n",
      "Approx 55% complete for CSE__0bf4ec__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_plasmid_R1.fastq.gz\n",
      "Approx 60% complete for CSE__0bf4ec__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_plasmid_R1.fastq.gz\n",
      "Approx 65% complete for CSE__0bf4ec__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_plasmid_R1.fastq.gz\n",
      "Approx 70% complete for CSE__0bf4ec__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_plasmid_R1.fastq.gz\n",
      "Approx 75% complete for CSE__0bf4ec__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_plasmid_R1.fastq.gz\n",
      "Approx 80% complete for CSE__0bf4ec__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_plasmid_R1.fastq.gz\n",
      "Approx 85% complete for CSE__0bf4ec__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_plasmid_R1.fastq.gz\n",
      "Approx 90% complete for CSE__0bf4ec__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_plasmid_R1.fastq.gz\n",
      "Approx 95% complete for CSE__0bf4ec__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_plasmid_R1.fastq.gz\n",
      "Analysis complete for CSE__0bf4ec__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_plasmid_R1.fastq.gz\n",
      "cat: /staging/leuven/stg_00002/lcb/ngs_runs/NovaSeq6000_20201013/Demultiplexed/CSE__f7b5af__*R1*: No such file or directory\n",
      "Started analysis of CSE__f7b5af__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_plasmid_R1.fastq.gz\n",
      "Approx 5% complete for CSE__f7b5af__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_plasmid_R1.fastq.gz\n",
      "Approx 10% complete for CSE__f7b5af__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_plasmid_R1.fastq.gz\n",
      "Approx 15% complete for CSE__f7b5af__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_plasmid_R1.fastq.gz\n",
      "Approx 20% complete for CSE__f7b5af__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_plasmid_R1.fastq.gz\n",
      "Approx 25% complete for CSE__f7b5af__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_plasmid_R1.fastq.gz\n",
      "Approx 30% complete for CSE__f7b5af__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_plasmid_R1.fastq.gz\n",
      "Approx 35% complete for CSE__f7b5af__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_plasmid_R1.fastq.gz\n",
      "Approx 40% complete for CSE__f7b5af__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_plasmid_R1.fastq.gz\n",
      "Approx 45% complete for CSE__f7b5af__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_plasmid_R1.fastq.gz\n",
      "Approx 50% complete for CSE__f7b5af__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_plasmid_R1.fastq.gz\n",
      "Approx 55% complete for CSE__f7b5af__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_plasmid_R1.fastq.gz\n",
      "Approx 60% complete for CSE__f7b5af__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_plasmid_R1.fastq.gz\n",
      "Approx 65% complete for CSE__f7b5af__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_plasmid_R1.fastq.gz\n",
      "Approx 70% complete for CSE__f7b5af__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_plasmid_R1.fastq.gz\n",
      "Approx 75% complete for CSE__f7b5af__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_plasmid_R1.fastq.gz\n",
      "Approx 80% complete for CSE__f7b5af__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_plasmid_R1.fastq.gz\n",
      "Approx 85% complete for CSE__f7b5af__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_plasmid_R1.fastq.gz\n",
      "Approx 90% complete for CSE__f7b5af__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_plasmid_R1.fastq.gz\n",
      "Approx 95% complete for CSE__f7b5af__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_plasmid_R1.fastq.gz\n",
      "Analysis complete for CSE__f7b5af__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_plasmid_R1.fastq.gz\n",
      "cat: /staging/leuven/stg_00002/lcb/ngs_runs/NovaSeq6000_20201013/Demultiplexed/CSE__0bd121__*R1*: No such file or directory\n",
      "Started analysis of CSE__0bd121__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_plasmid_R1.fastq.gz\n",
      "Approx 5% complete for CSE__0bd121__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_plasmid_R1.fastq.gz\n",
      "Approx 10% complete for CSE__0bd121__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_plasmid_R1.fastq.gz\n",
      "Approx 15% complete for CSE__0bd121__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_plasmid_R1.fastq.gz\n",
      "Approx 20% complete for CSE__0bd121__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_plasmid_R1.fastq.gz\n",
      "Approx 25% complete for CSE__0bd121__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_plasmid_R1.fastq.gz\n",
      "Approx 30% complete for CSE__0bd121__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_plasmid_R1.fastq.gz\n",
      "Approx 35% complete for CSE__0bd121__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_plasmid_R1.fastq.gz\n",
      "Approx 40% complete for CSE__0bd121__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_plasmid_R1.fastq.gz\n",
      "Approx 45% complete for CSE__0bd121__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_plasmid_R1.fastq.gz\n",
      "Approx 50% complete for CSE__0bd121__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_plasmid_R1.fastq.gz\n",
      "Approx 55% complete for CSE__0bd121__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_plasmid_R1.fastq.gz\n",
      "Approx 60% complete for CSE__0bd121__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_plasmid_R1.fastq.gz\n",
      "Approx 65% complete for CSE__0bd121__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_plasmid_R1.fastq.gz\n",
      "Approx 70% complete for CSE__0bd121__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_plasmid_R1.fastq.gz\n",
      "Approx 75% complete for CSE__0bd121__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_plasmid_R1.fastq.gz\n",
      "Approx 80% complete for CSE__0bd121__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_plasmid_R1.fastq.gz\n",
      "Approx 85% complete for CSE__0bd121__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_plasmid_R1.fastq.gz\n",
      "Approx 90% complete for CSE__0bd121__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_plasmid_R1.fastq.gz\n",
      "Approx 95% complete for CSE__0bd121__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_plasmid_R1.fastq.gz\n",
      "Analysis complete for CSE__0bd121__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_plasmid_R1.fastq.gz\n"
     ]
    }
   ],
   "source": [
    "module load FastQC/0.11.8-Java-1.8.0_162\n",
    "# For group of samples\n",
    "cat 00.raw_reads/CHEQseq_synth_names.txt | while read line\n",
    "do \n",
    "# Path to the different runs\n",
    "NovaSeq6000_20201006=\"/staging/leuven/stg_00002/lcb/ngs_runs/NovaSeq6000_20201006/Demultiplexed/${line%Cheq*}*R1*\"\n",
    "NovaSeq6000_20201013=\"/staging/leuven/stg_00002/lcb/ngs_runs/NovaSeq6000_20201013/Demultiplexed/${line%Cheq*}*R1*\"\n",
    "cat ${NovaSeq6000_20201006} ${NovaSeq6000_20201013} > 00.raw_reads/${line}_R1.fastq.gz\n",
    "fastqc -t 10 00.raw_reads/${line}_R1.fastq.gz -o 01.fastqc\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Combined number of reads for CSE__6f8b76__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_cDNA:  8457241\n",
      "Combined number of reads for CSE__81ba64__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_cDNA:  8626303\n",
      "Combined number of reads for CSE__839e9e__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_cDNA:  5335676\n",
      "Combined number of reads for CSE__7cb57c__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_cDNA:  1516983\n",
      "Combined number of reads for CSE__8f7a3e__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_cDNA:  4266283\n",
      "Combined number of reads for CSE__8b9ec1__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_plasmid:  14868109\n",
      "Combined number of reads for CSE__cd9463__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_plasmid:  16533113\n",
      "Combined number of reads for CSE__0bf4ec__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_plasmid:  13382837\n",
      "Combined number of reads for CSE__f7b5af__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_plasmid:  13441105\n",
      "Combined number of reads for CSE__0bd121__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_plasmid:  23372958\n"
     ]
    }
   ],
   "source": [
    "# Total number of reads for R1 per sample over the runs\n",
    "cat 00.raw_reads/CHEQseq_synth_names.txt | while read line\n",
    "do\n",
    "echo -e \"Combined number of reads for ${line}: \" $(( $(zcat 00.raw_reads/${line}_R1.fastq.gz | wc -l) / 4 ))\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 02. Generate clean reads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Python/3.7.0-foss-2018a(38):ERROR:150: Module 'Python/3.7.0-foss-2018a' conflicts with the currently loaded module(s) 'Python/3.7.4-foss-2018a'\n",
      "Python/3.7.0-foss-2018a(38):ERROR:102: Tcl command execution failed: conflict Python\n",
      "\n",
      "This is cutadapt 1.18 with Python 3.7.4\n",
      "Command line parameters: -g ACTCGAGCCGAGCAGGCGCGCCGATC...GGACCGG -e 0.2 -l 17 -m 17 -j 10 --discard-untrimmed -o 02.cleaned_reads/CSE__6f8b76__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_cDNA_R1.fastq.gz 00.raw_reads/CSE__6f8b76__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_cDNA_R1.fastq.gz\n",
      "Processing reads on 10 cores in single-end mode ...\n",
      "Finished in 38.85 s (5 us/read; 13.06 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               8,457,241\n",
      "Reads with adapters:                 8,422,072 (99.6%)\n",
      "Reads that were too short:              97,711 (1.2%)\n",
      "Reads written (passing filters):     8,324,361 (98.4%)\n",
      "\n",
      "Total basepairs processed:   414,404,809 bp\n",
      "Total written (filtered):    141,514,137 bp (34.1%)\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: ACTCGAGCCGAGCAGGCGCGCCGATC...GGACCGG; Type: linked; Length: 26+7; 5' trimmed: 8422072 times; 3' trimmed: 8422072 times\n",
      "\n",
      "No. of allowed errors:\n",
      "0-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-26 bp: 5\n",
      "\n",
      "No. of allowed errors:\n",
      "0-4 bp: 0; 5-7 bp: 1\n",
      "\n",
      "Overview of removed sequences at 5' end\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t20\t132144.4\t0\t20\n",
      "4\t5\t33036.1\t0\t2 3\n",
      "5\t2\t8259.0\t1\t0 2\n",
      "7\t2\t516.2\t1\t0 2\n",
      "10\t2\t8.1\t2\t0 0 2\n",
      "16\t1\t0.0\t3\t0 0 0 1\n",
      "18\t3\t0.0\t3\t3\n",
      "19\t51\t0.0\t3\t40 11\n",
      "20\t52\t0.0\t4\t19 32 0 0 0 1\n",
      "21\t191\t0.0\t4\t56 31 41 11 0 52\n",
      "22\t651\t0.0\t4\t35 55 120 14 337 90\n",
      "23\t1246\t0.0\t4\t218 68 97 710 102 51\n",
      "24\t5777\t0.0\t4\t319 859 4299 156 69 75\n",
      "25\t68038\t0.0\t5\t4940 60915 1624 330 170 59\n",
      "26\t8238082\t0.0\t5\t8109561 121657 5479 845 377 163\n",
      "27\t66576\t0.0\t5\t41946 23021 1003 374 165 67\n",
      "28\t20726\t0.0\t5\t11553 474 4028 517 455 3699\n",
      "29\t12303\t0.0\t5\t7261 263 72 3853 611 243\n",
      "30\t4462\t0.0\t5\t0 5 7 49 3856 545\n",
      "31\t3789\t0.0\t5\t1 0 8 13 42 3725\n",
      "32\t53\t0.0\t5\t0 0 0 4 10 39\n",
      "33\t18\t0.0\t5\t2 0 0 0 6 10\n",
      "34\t12\t0.0\t5\t0 0 0 3 0 9\n",
      "35\t1\t0.0\t5\t0 0 0 1\n",
      "36\t2\t0.0\t5\t1 0 1\n",
      "37\t1\t0.0\t5\t0 1\n",
      "38\t1\t0.0\t5\t1\n",
      "40\t1\t0.0\t5\t1\n",
      "41\t1\t0.0\t5\t1\n",
      "43\t2\t0.0\t5\t2\n",
      "44\t1\t0.0\t5\t1\n",
      "\n",
      "\n",
      "Overview of removed sequences at 3' end\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t12640\t132144.4\t0\t12640\n",
      "4\t15624\t33036.1\t0\t15596 28\n",
      "5\t97849\t8259.0\t1\t96078 1771\n",
      "6\t8138619\t2064.8\t1\t8083266 55353\n",
      "7\t113242\t516.2\t1\t109305 3937\n",
      "8\t9744\t516.2\t1\t9338 406\n",
      "9\t3121\t516.2\t1\t2505 616\n",
      "10\t2505\t516.2\t1\t1487 1018\n",
      "11\t4517\t516.2\t1\t3956 561\n",
      "12\t7405\t516.2\t1\t1173 6232\n",
      "13\t2598\t516.2\t1\t737 1861\n",
      "14\t1375\t516.2\t1\t557 818\n",
      "15\t1403\t516.2\t1\t695 708\n",
      "16\t1439\t516.2\t1\t505 934\n",
      "17\t1372\t516.2\t1\t596 776\n",
      "18\t1650\t516.2\t1\t775 875\n",
      "19\t1561\t516.2\t1\t640 921\n",
      "20\t1402\t516.2\t1\t481 921\n",
      "21\t1204\t516.2\t1\t500 704\n",
      "22\t1163\t516.2\t1\t351 812\n",
      "23\t1314\t516.2\t1\t450 864\n",
      "24\t258\t516.2\t1\t253 5\n",
      "25\t38\t516.2\t1\t37 1\n",
      "26\t6\t516.2\t1\t6\n",
      "27\t3\t516.2\t1\t3\n",
      "29\t1\t516.2\t1\t0 1\n",
      "30\t1\t516.2\t1\t1\n",
      "32\t1\t516.2\t1\t0 1\n",
      "35\t1\t516.2\t1\t0 1\n",
      "36\t12\t516.2\t1\t0 12\n",
      "37\t1\t516.2\t1\t0 1\n",
      "38\t2\t516.2\t1\t0 2\n",
      "39\t1\t516.2\t1\t0 1\n",
      "\n",
      "This is cutadapt 1.18 with Python 3.7.4\n",
      "Command line parameters: -g ACTCGAGCCGAGCAGGCGCGCCGATC...GGACCGG -e 0.2 -l 17 -m 17 -j 10 --discard-untrimmed -o 02.cleaned_reads/CSE__81ba64__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_cDNA_R1.fastq.gz 00.raw_reads/CSE__81ba64__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_cDNA_R1.fastq.gz\n",
      "Processing reads on 10 cores in single-end mode ...\n",
      "Finished in 38.67 s (4 us/read; 13.38 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               8,626,303\n",
      "Reads with adapters:                 8,542,027 (99.0%)\n",
      "Reads that were too short:             278,506 (3.2%)\n",
      "Reads written (passing filters):     8,263,521 (95.8%)\n",
      "\n",
      "Total basepairs processed:   422,688,847 bp\n",
      "Total written (filtered):    140,479,857 bp (33.2%)\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: ACTCGAGCCGAGCAGGCGCGCCGATC...GGACCGG; Type: linked; Length: 26+7; 5' trimmed: 8542027 times; 3' trimmed: 8542027 times\n",
      "\n",
      "No. of allowed errors:\n",
      "0-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-26 bp: 5\n",
      "\n",
      "No. of allowed errors:\n",
      "0-4 bp: 0; 5-7 bp: 1\n",
      "\n",
      "Overview of removed sequences at 5' end\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t4\t134786.0\t0\t4\n",
      "4\t4\t33696.5\t0\t2 2\n",
      "5\t1\t8424.1\t1\t0 1\n",
      "6\t1\t2106.0\t1\t0 1\n",
      "10\t1\t8.2\t2\t0 0 1\n",
      "19\t7\t0.0\t3\t7\n",
      "20\t14\t0.0\t4\t11 3\n",
      "21\t36\t0.0\t4\t17 5 14\n",
      "22\t156\t0.0\t4\t17 21 19 5 2 92\n",
      "23\t870\t0.0\t4\t73 23 47 59 556 112\n",
      "24\t1504\t0.0\t4\t341 54 53 608 140 308\n",
      "25\t5614\t0.0\t5\t470 986 3887 151 75 45\n",
      "26\t93045\t0.0\t5\t14554 75575 1831 414 198 473\n",
      "27\t8375582\t0.0\t5\t8242356 125427 6068 1025 492 214\n",
      "28\t38123\t0.0\t5\t17057 19693 828 362 139 44\n",
      "29\t14032\t0.0\t5\t7969 330 4170 596 649 318\n",
      "30\t4749\t0.0\t5\t1 7 17 3764 745 215\n",
      "31\t4614\t0.0\t5\t1 0 9 15 3980 609\n",
      "32\t3617\t0.0\t5\t0 0 0 11 14 3592\n",
      "33\t32\t0.0\t5\t0 0 0 0 10 22\n",
      "34\t10\t0.0\t5\t1 0 0 0 0 9\n",
      "35\t1\t0.0\t5\t1\n",
      "36\t2\t0.0\t5\t1 0 0 0 1\n",
      "37\t2\t0.0\t5\t1 0 1\n",
      "40\t1\t0.0\t5\t1\n",
      "41\t1\t0.0\t5\t0 0 1\n",
      "43\t1\t0.0\t5\t1\n",
      "44\t1\t0.0\t5\t1\n",
      "45\t1\t0.0\t5\t1\n",
      "46\t1\t0.0\t5\t1\n",
      "\n",
      "\n",
      "Overview of removed sequences at 3' end\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t12515\t134786.0\t0\t12515\n",
      "4\t69537\t33696.5\t0\t69437 100\n",
      "5\t8098307\t8424.1\t1\t8059543 38764\n",
      "6\t136866\t2106.0\t1\t132784 4082\n",
      "7\t9466\t526.5\t1\t9059 407\n",
      "8\t2847\t526.5\t1\t2375 472\n",
      "9\t31754\t526.5\t1\t1357 30397\n",
      "10\t65657\t526.5\t1\t3962 61695\n",
      "11\t17947\t526.5\t1\t1147 16800\n",
      "12\t8681\t526.5\t1\t731 7950\n",
      "13\t9601\t526.5\t1\t530 9071\n",
      "14\t8415\t526.5\t1\t745 7670\n",
      "15\t7560\t526.5\t1\t507 7053\n",
      "16\t7709\t526.5\t1\t668 7041\n",
      "17\t9299\t526.5\t1\t759 8540\n",
      "18\t9979\t526.5\t1\t703 9276\n",
      "19\t8572\t526.5\t1\t549 8023\n",
      "20\t8303\t526.5\t1\t477 7826\n",
      "21\t7323\t526.5\t1\t364 6959\n",
      "22\t11298\t526.5\t1\t481 10817\n",
      "23\t334\t526.5\t1\t320 14\n",
      "24\t36\t526.5\t1\t36\n",
      "25\t15\t526.5\t1\t14 1\n",
      "26\t1\t526.5\t1\t1\n",
      "30\t1\t526.5\t1\t1\n",
      "31\t1\t526.5\t1\t1\n",
      "37\t1\t526.5\t1\t0 1\n",
      "44\t1\t526.5\t1\t0 1\n",
      "45\t1\t526.5\t1\t0 1\n",
      "\n",
      "This is cutadapt 1.18 with Python 3.7.4\n",
      "Command line parameters: -g ACTCGAGCCGAGCAGGCGCGCCGATC...GGACCGG -e 0.2 -l 17 -m 17 -j 10 --discard-untrimmed -o 02.cleaned_reads/CSE__839e9e__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_cDNA_R1.fastq.gz 00.raw_reads/CSE__839e9e__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_cDNA_R1.fastq.gz\n",
      "Processing reads on 10 cores in single-end mode ...\n",
      "Finished in 24.15 s (5 us/read; 13.26 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               5,335,676\n",
      "Reads with adapters:                 5,273,888 (98.8%)\n",
      "Reads that were too short:             178,078 (3.3%)\n",
      "Reads written (passing filters):     5,095,810 (95.5%)\n",
      "\n",
      "Total basepairs processed:   261,448,124 bp\n",
      "Total written (filtered):     86,628,770 bp (33.1%)\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: ACTCGAGCCGAGCAGGCGCGCCGATC...GGACCGG; Type: linked; Length: 26+7; 5' trimmed: 5273888 times; 3' trimmed: 5273888 times\n",
      "\n",
      "No. of allowed errors:\n",
      "0-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-26 bp: 5\n",
      "\n",
      "No. of allowed errors:\n",
      "0-4 bp: 0; 5-7 bp: 1\n",
      "\n",
      "Overview of removed sequences at 5' end\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t1\t83369.9\t0\t1\n",
      "4\t2\t20842.5\t0\t1 1\n",
      "5\t4\t5210.6\t1\t0 4\n",
      "6\t7\t1302.7\t1\t0 7\n",
      "17\t1\t0.0\t3\t1\n",
      "19\t12\t0.0\t3\t12\n",
      "20\t17\t0.0\t4\t13 3 1\n",
      "21\t35\t0.0\t4\t16 14 5\n",
      "22\t48\t0.0\t4\t12 5 27 0 4\n",
      "23\t250\t0.0\t4\t44 41 18 48 24 75\n",
      "24\t644\t0.0\t4\t57 77 66 127 264 53\n",
      "25\t1356\t0.0\t5\t207 164 225 638 61 61\n",
      "26\t19315\t0.0\t5\t14087 1016 3975 138 53 46\n",
      "27\t78599\t0.0\t5\t21172 55396 1534 256 160 81\n",
      "28\t5130330\t0.0\t5\t5060748 66011 2946 403 151 71\n",
      "29\t30856\t0.0\t5\t12969 16658 531 220 446 32\n",
      "30\t3932\t0.0\t5\t4 10 2620 406 483 409\n",
      "31\t3045\t0.0\t5\t0 1 6 2473 400 165\n",
      "32\t3089\t0.0\t5\t0 0 0 4 2715 370\n",
      "33\t2324\t0.0\t5\t1 0 1 0 7 2315\n",
      "34\t13\t0.0\t5\t1 0 0 0 0 12\n",
      "35\t2\t0.0\t5\t1 0 0 1\n",
      "36\t3\t0.0\t5\t2 0 0 1\n",
      "37\t2\t0.0\t5\t0 2\n",
      "38\t1\t0.0\t5\t1\n",
      "\n",
      "\n",
      "Overview of removed sequences at 3' end\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t49931\t83369.9\t0\t49931\n",
      "4\t4958081\t20842.5\t0\t4957426 655\n",
      "5\t105232\t5210.6\t1\t101897 3335\n",
      "6\t21877\t1302.7\t1\t21313 564\n",
      "7\t2406\t325.7\t1\t2109 297\n",
      "8\t19791\t325.7\t1\t971 18820\n",
      "9\t40964\t325.7\t1\t2467 38497\n",
      "10\t11840\t325.7\t1\t739 11101\n",
      "11\t6031\t325.7\t1\t659 5372\n",
      "12\t5638\t325.7\t1\t585 5053\n",
      "13\t5339\t325.7\t1\t373 4966\n",
      "14\t5483\t325.7\t1\t263 5220\n",
      "15\t4733\t325.7\t1\t561 4172\n",
      "16\t5379\t325.7\t1\t327 5052\n",
      "17\t5718\t325.7\t1\t510 5208\n",
      "18\t5384\t325.7\t1\t343 5041\n",
      "19\t5988\t325.7\t1\t274 5714\n",
      "20\t5105\t325.7\t1\t195 4910\n",
      "21\t8644\t325.7\t1\t368 8276\n",
      "22\t287\t325.7\t1\t194 93\n",
      "23\t20\t325.7\t1\t17 3\n",
      "24\t12\t325.7\t1\t12\n",
      "25\t2\t325.7\t1\t2\n",
      "34\t3\t325.7\t1\t0 3\n",
      "\n",
      "This is cutadapt 1.18 with Python 3.7.4\n",
      "Command line parameters: -g ACTCGAGCCGAGCAGGCGCGCCGATC...GGACCGG -e 0.2 -l 17 -m 17 -j 10 --discard-untrimmed -o 02.cleaned_reads/CSE__7cb57c__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_cDNA_R1.fastq.gz 00.raw_reads/CSE__7cb57c__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_cDNA_R1.fastq.gz\n",
      "Processing reads on 10 cores in single-end mode ...\n",
      "Finished in 5.70 s (4 us/read; 15.97 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               1,516,983\n",
      "Reads with adapters:                 1,273,374 (83.9%)\n",
      "Reads that were too short:              48,557 (3.2%)\n",
      "Reads written (passing filters):     1,224,817 (80.7%)\n",
      "\n",
      "Total basepairs processed:    74,332,167 bp\n",
      "Total written (filtered):     20,821,889 bp (28.0%)\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: ACTCGAGCCGAGCAGGCGCGCCGATC...GGACCGG; Type: linked; Length: 26+7; 5' trimmed: 1273374 times; 3' trimmed: 1273374 times\n",
      "\n",
      "No. of allowed errors:\n",
      "0-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-26 bp: 5\n",
      "\n",
      "No. of allowed errors:\n",
      "0-4 bp: 0; 5-7 bp: 1\n",
      "\n",
      "Overview of removed sequences at 5' end\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t139\t23702.9\t0\t139\n",
      "4\t125\t5925.7\t0\t49 76\n",
      "5\t52\t1481.4\t1\t7 45\n",
      "6\t64\t370.4\t1\t2 62\n",
      "7\t10\t92.6\t1\t0 10\n",
      "8\t2\t23.1\t1\t0 2\n",
      "9\t5\t5.8\t1\t0 1 4\n",
      "20\t3\t0.0\t4\t1 1 0 1\n",
      "21\t5\t0.0\t4\t3 1 1\n",
      "22\t8\t0.0\t4\t1 3 2 1 0 1\n",
      "23\t18\t0.0\t4\t9 3 6\n",
      "24\t134\t0.0\t4\t6 21 8 38 17 44\n",
      "25\t306\t0.0\t5\t48 42 15 28 64 109\n",
      "26\t3851\t0.0\t5\t3045 210 45 190 20 341\n",
      "27\t9831\t0.0\t5\t8653 251 841 34 26 26\n",
      "28\t15364\t0.0\t5\t3968 10407 280 147 394 168\n",
      "29\t1237584\t0.0\t5\t1218020 18478 731 95 186 74\n",
      "30\t1001\t0.0\t5\t64 563 71 15 253 35\n",
      "31\t2688\t0.0\t5\t0 0 443 72 2104 69\n",
      "32\t913\t0.0\t5\t0 0 0 702 156 55\n",
      "33\t595\t0.0\t5\t0 0 0 1 514 80\n",
      "34\t673\t0.0\t5\t0 0 0 1 1 671\n",
      "35\t2\t0.0\t5\t0 0 0 0 2\n",
      "36\t1\t0.0\t5\t0 0 0 1\n",
      "\n",
      "\n",
      "Overview of removed sequences at 3' end\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t1198929\t23702.9\t0\t1198929\n",
      "4\t20473\t5925.7\t0\t20157 316\n",
      "5\t10447\t1481.4\t1\t10165 282\n",
      "6\t3870\t370.4\t1\t3700 170\n",
      "7\t4935\t92.6\t1\t288 4647\n",
      "8\t11514\t92.6\t1\t526 10988\n",
      "9\t3404\t92.6\t1\t192 3212\n",
      "10\t1488\t92.6\t1\t99 1389\n",
      "11\t1284\t92.6\t1\t129 1155\n",
      "12\t1377\t92.6\t1\t125 1252\n",
      "13\t3450\t92.6\t1\t83 3367\n",
      "14\t1090\t92.6\t1\t98 992\n",
      "15\t1619\t92.6\t1\t121 1498\n",
      "16\t1710\t92.6\t1\t102 1608\n",
      "17\t1115\t92.6\t1\t99 1016\n",
      "18\t1800\t92.6\t1\t125 1675\n",
      "19\t1588\t92.6\t1\t34 1554\n",
      "20\t2588\t92.6\t1\t120 2468\n",
      "21\t481\t92.6\t1\t449 32\n",
      "22\t24\t92.6\t1\t3 21\n",
      "23\t24\t92.6\t1\t5 19\n",
      "24\t6\t92.6\t1\t0 6\n",
      "25\t4\t92.6\t1\t0 4\n",
      "26\t3\t92.6\t1\t0 3\n",
      "27\t3\t92.6\t1\t0 3\n",
      "28\t4\t92.6\t1\t0 4\n",
      "29\t7\t92.6\t1\t1 6\n",
      "30\t3\t92.6\t1\t0 3\n",
      "31\t4\t92.6\t1\t0 4\n",
      "32\t3\t92.6\t1\t0 3\n",
      "33\t22\t92.6\t1\t1 21\n",
      "34\t20\t92.6\t1\t0 20\n",
      "35\t11\t92.6\t1\t0 11\n",
      "36\t6\t92.6\t1\t0 6\n",
      "37\t4\t92.6\t1\t0 4\n",
      "38\t1\t92.6\t1\t0 1\n",
      "39\t4\t92.6\t1\t0 4\n",
      "40\t9\t92.6\t1\t0 9\n",
      "41\t5\t92.6\t1\t0 5\n",
      "42\t3\t92.6\t1\t0 3\n",
      "43\t8\t92.6\t1\t0 8\n",
      "44\t27\t92.6\t1\t0 27\n",
      "45\t6\t92.6\t1\t0 6\n",
      "46\t1\t92.6\t1\t0 1\n",
      "\n",
      "This is cutadapt 1.18 with Python 3.7.4\n",
      "Command line parameters: -g ACTCGAGCCGAGCAGGCGCGCCGATC...GGACCGG -e 0.2 -l 17 -m 17 -j 10 --discard-untrimmed -o 02.cleaned_reads/CSE__8f7a3e__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_cDNA_R1.fastq.gz 00.raw_reads/CSE__8f7a3e__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_cDNA_R1.fastq.gz\n",
      "Processing reads on 10 cores in single-end mode ...\n",
      "Finished in 19.35 s (5 us/read; 13.23 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               4,266,283\n",
      "Reads with adapters:                 4,226,512 (99.1%)\n",
      "Reads that were too short:             152,128 (3.6%)\n",
      "Reads written (passing filters):     4,074,384 (95.5%)\n",
      "\n",
      "Total basepairs processed:   209,047,867 bp\n",
      "Total written (filtered):     69,264,528 bp (33.1%)\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: ACTCGAGCCGAGCAGGCGCGCCGATC...GGACCGG; Type: linked; Length: 26+7; 5' trimmed: 4226512 times; 3' trimmed: 4226512 times\n",
      "\n",
      "No. of allowed errors:\n",
      "0-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-26 bp: 5\n",
      "\n",
      "No. of allowed errors:\n",
      "0-4 bp: 0; 5-7 bp: 1\n",
      "\n",
      "Overview of removed sequences at 5' end\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t1\t66660.7\t0\t1\n",
      "16\t1\t0.0\t3\t1\n",
      "19\t6\t0.0\t3\t4 0 2\n",
      "20\t4\t0.0\t4\t1 3\n",
      "21\t13\t0.0\t4\t6 4 3\n",
      "22\t61\t0.0\t4\t5 17 10 8 2 19\n",
      "23\t1473\t0.0\t4\t35 20 19 27 1175 197\n",
      "24\t775\t0.0\t4\t167 31 27 299 82 169\n",
      "25\t2802\t0.0\t5\t230 480 1865 105 101 21\n",
      "26\t46229\t0.0\t5\t6859 37688 1029 294 157 202\n",
      "27\t4132360\t0.0\t5\t4059420 69000 3109 469 227 135\n",
      "28\t29325\t0.0\t5\t7112 10491 485 175 10850 212\n",
      "29\t6239\t0.0\t5\t3337 121 2029 291 402 59\n",
      "30\t2674\t0.0\t5\t0 7 6 2160 393 108\n",
      "31\t2505\t0.0\t5\t0 0 0 11 2184 310\n",
      "32\t2024\t0.0\t5\t0 0 0 4 10 2010\n",
      "33\t5\t0.0\t5\t1 0 0 0 3 1\n",
      "34\t7\t0.0\t5\t0 0 0 0 1 6\n",
      "35\t1\t0.0\t5\t0 0 0 1\n",
      "36\t2\t0.0\t5\t0 0 1 1\n",
      "37\t1\t0.0\t5\t1\n",
      "40\t2\t0.0\t5\t2\n",
      "41\t1\t0.0\t5\t0 0 1\n",
      "43\t1\t0.0\t5\t1\n",
      "\n",
      "\n",
      "Overview of removed sequences at 3' end\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t5402\t66660.7\t0\t5402\n",
      "4\t34861\t16665.2\t0\t34822 39\n",
      "5\t3993460\t4166.3\t1\t3972627 20833\n",
      "6\t68529\t1041.6\t1\t66490 2039\n",
      "7\t15634\t260.4\t1\t4491 11143\n",
      "8\t1466\t260.4\t1\t1188 278\n",
      "9\t16027\t260.4\t1\t627 15400\n",
      "10\t31437\t260.4\t1\t1881 29556\n",
      "11\t9007\t260.4\t1\t691 8316\n",
      "12\t4186\t260.4\t1\t289 3897\n",
      "13\t4641\t260.4\t1\t328 4313\n",
      "14\t4428\t260.4\t1\t451 3977\n",
      "15\t4232\t260.4\t1\t281 3951\n",
      "16\t3478\t260.4\t1\t283 3195\n",
      "17\t4754\t260.4\t1\t501 4253\n",
      "18\t5217\t260.4\t1\t334 4883\n",
      "19\t4809\t260.4\t1\t271 4538\n",
      "20\t4315\t260.4\t1\t238 4077\n",
      "21\t4027\t260.4\t1\t215 3812\n",
      "22\t6262\t260.4\t1\t282 5980\n",
      "23\t321\t260.4\t1\t310 11\n",
      "24\t8\t260.4\t1\t7 1\n",
      "25\t9\t260.4\t1\t7 2\n",
      "26\t2\t260.4\t1\t0 2\n",
      "\n",
      "This is cutadapt 1.18 with Python 3.7.4\n",
      "Command line parameters: -g ACTCGAGCCGAGCAGGCGCGCCGATC...GGACCGG -e 0.2 -l 17 -m 17 -j 10 --discard-untrimmed -o 02.cleaned_reads/CSE__8b9ec1__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_plasmid_R1.fastq.gz 00.raw_reads/CSE__8b9ec1__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_plasmid_R1.fastq.gz\n",
      "Processing reads on 10 cores in single-end mode ...\n",
      "Finished in 68.52 s (5 us/read; 13.02 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              14,868,109\n",
      "Reads with adapters:                14,835,517 (99.8%)\n",
      "Reads that were too short:             163,951 (1.1%)\n",
      "Reads written (passing filters):    14,671,566 (98.7%)\n",
      "\n",
      "Total basepairs processed:   728,537,341 bp\n",
      "Total written (filtered):    249,416,622 bp (34.2%)\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: ACTCGAGCCGAGCAGGCGCGCCGATC...GGACCGG; Type: linked; Length: 26+7; 5' trimmed: 14835517 times; 3' trimmed: 14835517 times\n",
      "\n",
      "No. of allowed errors:\n",
      "0-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-26 bp: 5\n",
      "\n",
      "No. of allowed errors:\n",
      "0-4 bp: 0; 5-7 bp: 1\n",
      "\n",
      "Overview of removed sequences at 5' end\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t3\t232314.2\t0\t3\n",
      "7\t1\t907.5\t1\t0 1\n",
      "11\t2\t3.5\t2\t0 0 2\n",
      "19\t20\t0.0\t3\t19 1\n",
      "20\t47\t0.0\t4\t30 16 1\n",
      "21\t216\t0.0\t4\t55 8 18 2 0 133\n",
      "22\t800\t0.0\t4\t46 81 64 7 530 72\n",
      "23\t1602\t0.0\t4\t220 128 82 975 110 87\n",
      "24\t8266\t0.0\t4\t441 1299 6060 280 123 63\n",
      "25\t100203\t0.0\t5\t8180 88629 2375 504 403 112\n",
      "26\t14586150\t0.0\t5\t14384269 190119 9351 1623 463 325\n",
      "27\t85917\t0.0\t5\t52689 30841 1553 529 215 90\n",
      "28\t20761\t0.0\t5\t11737 545 6904 934 435 206\n",
      "29\t17830\t0.0\t5\t8881 353 107 6910 1152 427\n",
      "30\t7027\t0.0\t5\t1 3 15 72 6030 906\n",
      "31\t6540\t0.0\t5\t1 0 16 20 52 6451\n",
      "32\t83\t0.0\t5\t0 0 0 7 15 61\n",
      "33\t26\t0.0\t5\t0 0 0 0 11 15\n",
      "34\t10\t0.0\t5\t0 0 1 2 0 7\n",
      "36\t5\t0.0\t5\t5\n",
      "37\t1\t0.0\t5\t1\n",
      "39\t1\t0.0\t5\t1\n",
      "41\t1\t0.0\t5\t1\n",
      "44\t1\t0.0\t5\t1\n",
      "46\t4\t0.0\t5\t4\n",
      "\n",
      "\n",
      "Overview of removed sequences at 3' end\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t12352\t232314.2\t0\t12352\n",
      "4\t18482\t58078.6\t0\t18430 52\n",
      "5\t136749\t14519.6\t1\t134488 2261\n",
      "6\t14419723\t3629.9\t1\t14343014 76709\n",
      "7\t177838\t907.5\t1\t171109 6729\n",
      "8\t14112\t907.5\t1\t13599 513\n",
      "9\t4837\t907.5\t1\t3914 923\n",
      "10\t3748\t907.5\t1\t2419 1329\n",
      "11\t7922\t907.5\t1\t7213 709\n",
      "12\t12725\t907.5\t1\t1855 10870\n",
      "13\t3899\t907.5\t1\t1196 2703\n",
      "14\t2096\t907.5\t1\t1046 1050\n",
      "15\t2148\t907.5\t1\t907 1241\n",
      "16\t2189\t907.5\t1\t911 1278\n",
      "17\t2055\t907.5\t1\t860 1195\n",
      "18\t2959\t907.5\t1\t1408 1551\n",
      "19\t2653\t907.5\t1\t1167 1486\n",
      "20\t2335\t907.5\t1\t991 1344\n",
      "21\t2220\t907.5\t1\t957 1263\n",
      "22\t1896\t907.5\t1\t729 1167\n",
      "23\t2104\t907.5\t1\t684 1420\n",
      "24\t409\t907.5\t1\t404 5\n",
      "25\t45\t907.5\t1\t45\n",
      "26\t15\t907.5\t1\t15\n",
      "27\t6\t907.5\t1\t6\n",
      "\n",
      "This is cutadapt 1.18 with Python 3.7.4\n",
      "Command line parameters: -g ACTCGAGCCGAGCAGGCGCGCCGATC...GGACCGG -e 0.2 -l 17 -m 17 -j 10 --discard-untrimmed -o 02.cleaned_reads/CSE__cd9463__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_plasmid_R1.fastq.gz 00.raw_reads/CSE__cd9463__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_plasmid_R1.fastq.gz\n",
      "Processing reads on 10 cores in single-end mode ...\n",
      "Finished in 74.78 s (5 us/read; 13.27 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              16,533,113\n",
      "Reads with adapters:                16,497,639 (99.8%)\n",
      "Reads that were too short:             540,267 (3.3%)\n",
      "Reads written (passing filters):    15,957,372 (96.5%)\n",
      "\n",
      "Total basepairs processed:   810,122,537 bp\n",
      "Total written (filtered):    271,275,324 bp (33.5%)\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: ACTCGAGCCGAGCAGGCGCGCCGATC...GGACCGG; Type: linked; Length: 26+7; 5' trimmed: 16497639 times; 3' trimmed: 16497639 times\n",
      "\n",
      "No. of allowed errors:\n",
      "0-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-26 bp: 5\n",
      "\n",
      "No. of allowed errors:\n",
      "0-4 bp: 0; 5-7 bp: 1\n",
      "\n",
      "Overview of removed sequences at 5' end\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "19\t14\t0.0\t3\t14\n",
      "20\t16\t0.0\t4\t16\n",
      "21\t23\t0.0\t4\t20 3\n",
      "22\t292\t0.0\t4\t15 28 13 7 1 228\n",
      "23\t1198\t0.0\t4\t61 37 63 35 866 136\n",
      "24\t1938\t0.0\t4\t561 99 70 879 173 156\n",
      "25\t10452\t0.0\t5\t797 1726 7315 328 147 139\n",
      "26\t155709\t0.0\t5\t23052 128323 2920 611 631 172\n",
      "27\t16227546\t0.0\t5\t16027325 186704 10458 2066 627 366\n",
      "28\t53944\t0.0\t5\t23735 27607 1556 602 257 187\n",
      "29\t20982\t0.0\t5\t10878 415 7498 1284 551 356\n",
      "30\t9893\t0.0\t5\t4 8 17 7662 1645 557\n",
      "31\t8179\t0.0\t5\t0 1 10 18 6829 1321\n",
      "32\t7373\t0.0\t5\t3 0 0 9 27 7334\n",
      "33\t46\t0.0\t5\t1 0 0 0 12 33\n",
      "34\t12\t0.0\t5\t0 0 0 0 0 12\n",
      "35\t9\t0.0\t5\t5 0 4\n",
      "36\t3\t0.0\t5\t0 1 2\n",
      "37\t2\t0.0\t5\t2\n",
      "38\t1\t0.0\t5\t0 0 1\n",
      "39\t1\t0.0\t5\t0 1\n",
      "42\t2\t0.0\t5\t2\n",
      "43\t3\t0.0\t5\t3\n",
      "45\t1\t0.0\t5\t1\n",
      "\n",
      "\n",
      "Overview of removed sequences at 3' end\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t17814\t258329.9\t0\t17814\n",
      "4\t109884\t64582.5\t0\t109688 196\n",
      "5\t15690358\t16145.6\t1\t15631265 59093\n",
      "6\t242925\t4036.4\t1\t234973 7952\n",
      "7\t17055\t1009.1\t1\t16526 529\n",
      "8\t5231\t1009.1\t1\t4438 793\n",
      "9\t64440\t1009.1\t1\t3033 61407\n",
      "10\t131661\t1009.1\t1\t8163 123498\n",
      "11\t34303\t1009.1\t1\t2116 32187\n",
      "12\t14987\t1009.1\t1\t1367 13620\n",
      "13\t17254\t1009.1\t1\t1314 15940\n",
      "14\t15398\t1009.1\t1\t1122 14276\n",
      "15\t14101\t1009.1\t1\t1173 12928\n",
      "16\t14536\t1009.1\t1\t1179 13357\n",
      "17\t18864\t1009.1\t1\t1763 17101\n",
      "18\t19255\t1009.1\t1\t1528 17727\n",
      "19\t16189\t1009.1\t1\t1281 14908\n",
      "20\t15578\t1009.1\t1\t1179 14399\n",
      "21\t14943\t1009.1\t1\t951 13992\n",
      "22\t22003\t1009.1\t1\t874 21129\n",
      "23\t711\t1009.1\t1\t691 20\n",
      "24\t122\t1009.1\t1\t121 1\n",
      "25\t13\t1009.1\t1\t13\n",
      "26\t14\t1009.1\t1\t14\n",
      "\n",
      "This is cutadapt 1.18 with Python 3.7.4\n",
      "Command line parameters: -g ACTCGAGCCGAGCAGGCGCGCCGATC...GGACCGG -e 0.2 -l 17 -m 17 -j 10 --discard-untrimmed -o 02.cleaned_reads/CSE__0bf4ec__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_plasmid_R1.fastq.gz 00.raw_reads/CSE__0bf4ec__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_plasmid_R1.fastq.gz\n",
      "Processing reads on 10 cores in single-end mode ...\n",
      "Finished in 59.14 s (4 us/read; 13.58 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              13,382,837\n",
      "Reads with adapters:                13,280,042 (99.2%)\n",
      "Reads that were too short:             438,508 (3.3%)\n",
      "Reads written (passing filters):    12,841,534 (96.0%)\n",
      "\n",
      "Total basepairs processed:   655,759,013 bp\n",
      "Total written (filtered):    218,306,078 bp (33.3%)\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: ACTCGAGCCGAGCAGGCGCGCCGATC...GGACCGG; Type: linked; Length: 26+7; 5' trimmed: 13280042 times; 3' trimmed: 13280042 times\n",
      "\n",
      "No. of allowed errors:\n",
      "0-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-26 bp: 5\n",
      "\n",
      "No. of allowed errors:\n",
      "0-4 bp: 0; 5-7 bp: 1\n",
      "\n",
      "Overview of removed sequences at 5' end\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "4\t5\t52276.7\t0\t0 5\n",
      "5\t1\t13069.2\t1\t0 1\n",
      "6\t6\t3267.3\t1\t0 6\n",
      "12\t2\t0.8\t2\t0 0 2\n",
      "19\t13\t0.0\t3\t13\n",
      "20\t37\t0.0\t4\t36 1\n",
      "21\t44\t0.0\t4\t14 28 2\n",
      "22\t69\t0.0\t4\t40 3 22 3 1\n",
      "23\t309\t0.0\t4\t55 74 11 38 8 123\n",
      "24\t1058\t0.0\t4\t102 127 123 99 523 84\n",
      "25\t2229\t0.0\t5\t377 260 331 1053 108 100\n",
      "26\t22053\t0.0\t5\t13286 1695 6572 320 106 74\n",
      "27\t163701\t0.0\t5\t40227 119657 2789 505 420 103\n",
      "28\t12998881\t0.0\t5\t12807358 181779 8092 1123 344 185\n",
      "29\t61233\t0.0\t5\t29846 29125 1466 517 180 99\n",
      "30\t8140\t0.0\t5\t10 18 6386 1051 436 239\n",
      "31\t8494\t0.0\t5\t2 2 29 6788 1228 445\n",
      "32\t7203\t0.0\t5\t1 2 0 22 6095 1083\n",
      "33\t6529\t0.0\t5\t2 0 0 0 24 6503\n",
      "34\t20\t0.0\t5\t1 0 0 0 0 19\n",
      "35\t1\t0.0\t5\t1\n",
      "37\t5\t0.0\t5\t1 0 3 1\n",
      "38\t4\t0.0\t5\t4\n",
      "42\t1\t0.0\t5\t0 0 1\n",
      "43\t3\t0.0\t5\t3\n",
      "45\t1\t0.0\t5\t1\n",
      "\n",
      "\n",
      "Overview of removed sequences at 3' end\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t105245\t209106.8\t0\t105245\n",
      "4\t12579337\t52276.7\t0\t12577468 1869\n",
      "5\t229894\t13069.2\t1\t221965 7929\n",
      "6\t27887\t3267.3\t1\t27033 854\n",
      "7\t5053\t816.8\t1\t4439 614\n",
      "8\t51268\t816.8\t1\t2736 48532\n",
      "9\t106179\t816.8\t1\t6616 99563\n",
      "10\t29222\t816.8\t1\t1846 27376\n",
      "11\t11996\t816.8\t1\t1166 10830\n",
      "12\t13924\t816.8\t1\t1050 12874\n",
      "13\t12093\t816.8\t1\t869 11224\n",
      "14\t10924\t816.8\t1\t815 10109\n",
      "15\t11525\t816.8\t1\t879 10646\n",
      "16\t14695\t816.8\t1\t1321 13374\n",
      "17\t15302\t816.8\t1\t1201 14101\n",
      "18\t12890\t816.8\t1\t1040 11850\n",
      "19\t12617\t816.8\t1\t902 11715\n",
      "20\t11929\t816.8\t1\t719 11210\n",
      "21\t17377\t816.8\t1\t686 16691\n",
      "22\t587\t816.8\t1\t434 153\n",
      "23\t73\t816.8\t1\t70 3\n",
      "24\t10\t816.8\t1\t10\n",
      "25\t12\t816.8\t1\t12\n",
      "30\t1\t816.8\t1\t0 1\n",
      "32\t1\t816.8\t1\t1\n",
      "38\t1\t816.8\t1\t0 1\n",
      "\n",
      "This is cutadapt 1.18 with Python 3.7.4\n",
      "Command line parameters: -g ACTCGAGCCGAGCAGGCGCGCCGATC...GGACCGG -e 0.2 -l 17 -m 17 -j 10 --discard-untrimmed -o 02.cleaned_reads/CSE__f7b5af__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_plasmid_R1.fastq.gz 00.raw_reads/CSE__f7b5af__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_plasmid_R1.fastq.gz\n",
      "Processing reads on 10 cores in single-end mode ...\n",
      "Finished in 60.13 s (4 us/read; 13.41 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              13,441,105\n",
      "Reads with adapters:                13,265,717 (98.7%)\n",
      "Reads that were too short:             427,093 (3.2%)\n",
      "Reads written (passing filters):    12,838,624 (95.5%)\n",
      "\n",
      "Total basepairs processed:   658,614,145 bp\n",
      "Total written (filtered):    218,256,608 bp (33.1%)\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: ACTCGAGCCGAGCAGGCGCGCCGATC...GGACCGG; Type: linked; Length: 26+7; 5' trimmed: 13265717 times; 3' trimmed: 13265717 times\n",
      "\n",
      "No. of allowed errors:\n",
      "0-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-26 bp: 5\n",
      "\n",
      "No. of allowed errors:\n",
      "0-4 bp: 0; 5-7 bp: 1\n",
      "\n",
      "Overview of removed sequences at 5' end\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "4\t1\t52504.3\t0\t0 1\n",
      "5\t1\t13126.1\t1\t0 1\n",
      "6\t6\t3281.5\t1\t0 6\n",
      "19\t11\t0.0\t3\t11\n",
      "20\t8\t0.0\t4\t3 5\n",
      "21\t28\t0.0\t4\t19 8 0 1\n",
      "22\t18\t0.0\t4\t6 3 7 2\n",
      "23\t80\t0.0\t4\t33 9 32 2 2 2\n",
      "24\t478\t0.0\t4\t50 154 40 104 9 121\n",
      "25\t1449\t0.0\t5\t286 267 83 147 594 72\n",
      "26\t19319\t0.0\t5\t16382 1266 312 1120 138 101\n",
      "27\t41858\t0.0\t5\t34293 1577 5551 271 94 72\n",
      "28\t124296\t0.0\t5\t27499 93074 2732 481 455 55\n",
      "29\t13038997\t0.0\t5\t12861279 164049 11289 1771 424 185\n",
      "30\t7882\t0.0\t5\t528 5482 1084 467 263 58\n",
      "31\t8501\t0.0\t5\t2 4 6602 1093 524 276\n",
      "32\t8577\t0.0\t5\t3 1 16 6750 1380 427\n",
      "33\t7533\t0.0\t5\t1 0 0 9 6365 1158\n",
      "34\t6647\t0.0\t5\t1 0 7 0 5 6634\n",
      "35\t4\t0.0\t5\t1 0 1 1 0 1\n",
      "36\t3\t0.0\t5\t1 0 0 2\n",
      "37\t5\t0.0\t5\t2 1 0 2\n",
      "38\t3\t0.0\t5\t1 1 1\n",
      "39\t6\t0.0\t5\t3 3\n",
      "40\t2\t0.0\t5\t1 0 0 1\n",
      "41\t2\t0.0\t5\t1 1\n",
      "43\t1\t0.0\t5\t1\n",
      "44\t1\t0.0\t5\t0 0 1\n",
      "\n",
      "\n",
      "Overview of removed sequences at 3' end\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t12683807\t210017.3\t0\t12683807\n",
      "4\t182833\t52504.3\t0\t182786 47\n",
      "5\t45802\t13126.1\t1\t44705 1097\n",
      "6\t21709\t3281.5\t1\t20761 948\n",
      "7\t51423\t820.4\t1\t2990 48433\n",
      "8\t106108\t820.4\t1\t6867 99241\n",
      "9\t28923\t820.4\t1\t1990 26933\n",
      "10\t12315\t820.4\t1\t1208 11107\n",
      "11\t13361\t820.4\t1\t1164 12197\n",
      "12\t11927\t820.4\t1\t844 11083\n",
      "13\t10939\t820.4\t1\t951 9988\n",
      "14\t12008\t820.4\t1\t899 11109\n",
      "15\t14591\t820.4\t1\t1424 13167\n",
      "16\t15394\t820.4\t1\t1310 14084\n",
      "17\t12745\t820.4\t1\t1100 11645\n",
      "18\t11979\t820.4\t1\t966 11013\n",
      "19\t11809\t820.4\t1\t747 11062\n",
      "20\t17271\t820.4\t1\t764 16507\n",
      "21\t612\t820.4\t1\t486 126\n",
      "22\t136\t820.4\t1\t94 42\n",
      "23\t18\t820.4\t1\t18\n",
      "24\t3\t820.4\t1\t3\n",
      "25\t1\t820.4\t1\t1\n",
      "26\t1\t820.4\t1\t1\n",
      "32\t2\t820.4\t1\t2\n",
      "\n",
      "This is cutadapt 1.18 with Python 3.7.4\n",
      "Command line parameters: -g ACTCGAGCCGAGCAGGCGCGCCGATC...GGACCGG -e 0.2 -l 17 -m 17 -j 10 --discard-untrimmed -o 02.cleaned_reads/CSE__0bd121__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_plasmid_R1.fastq.gz 00.raw_reads/CSE__0bd121__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_plasmid_R1.fastq.gz\n",
      "Processing reads on 10 cores in single-end mode ...\n",
      "Finished in 118.27 s (5 us/read; 11.86 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              23,372,958\n",
      "Reads with adapters:                23,332,603 (99.8%)\n",
      "Reads that were too short:             801,442 (3.4%)\n",
      "Reads written (passing filters):    22,531,161 (96.4%)\n",
      "\n",
      "Total basepairs processed: 1,145,274,942 bp\n",
      "Total written (filtered):    383,029,737 bp (33.4%)\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: ACTCGAGCCGAGCAGGCGCGCCGATC...GGACCGG; Type: linked; Length: 26+7; 5' trimmed: 23332603 times; 3' trimmed: 23332603 times\n",
      "\n",
      "No. of allowed errors:\n",
      "0-4 bp: 0; 5-9 bp: 1; 10-14 bp: 2; 15-19 bp: 3; 20-24 bp: 4; 25-26 bp: 5\n",
      "\n",
      "No. of allowed errors:\n",
      "0-4 bp: 0; 5-7 bp: 1\n",
      "\n",
      "Overview of removed sequences at 5' end\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t1\t365202.5\t0\t1\n",
      "11\t1\t5.6\t2\t1\n",
      "19\t13\t0.0\t3\t12 1\n",
      "20\t22\t0.0\t4\t14 6 2\n",
      "21\t32\t0.0\t4\t25 2 5\n",
      "22\t361\t0.0\t4\t37 33 23 5 4 259\n",
      "23\t1576\t0.0\t4\t123 44 131 69 1039 170\n",
      "24\t2765\t0.0\t4\t789 168 85 1345 208 170\n",
      "25\t15240\t0.0\t5\t1115 2390 10526 714 288 207\n",
      "26\t223883\t0.0\t5\t20049 193506 7720 1577 797 234\n",
      "27\t22928312\t0.0\t5\t22419568 460485 36504 8065 2622 1068\n",
      "28\t73596\t0.0\t5\t23578 44862 3327 1196 417 216\n",
      "29\t31399\t0.0\t5\t10242 820 15678 2879 1189 591\n",
      "30\t20825\t0.0\t5\t1 24 38 16571 3096 1095\n",
      "31\t18332\t0.0\t5\t2 0 21 39 15550 2720\n",
      "32\t16127\t0.0\t5\t1 1 2 22 50 16051\n",
      "33\t73\t0.0\t5\t0 0 0 0 33 40\n",
      "34\t34\t0.0\t5\t3 0 0 0 0 31\n",
      "37\t1\t0.0\t5\t1\n",
      "38\t1\t0.0\t5\t1\n",
      "39\t1\t0.0\t5\t1\n",
      "40\t1\t0.0\t5\t0 1\n",
      "41\t3\t0.0\t5\t0 0 1 0 0 2\n",
      "43\t1\t0.0\t5\t1\n",
      "44\t1\t0.0\t5\t1\n",
      "45\t2\t0.0\t5\t2\n",
      "\n",
      "\n",
      "Overview of removed sequences at 3' end\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t20706\t365202.5\t0\t20706\n",
      "4\t148959\t91300.6\t0\t148684 275\n",
      "5\t22197323\t22825.2\t1\t22055677 141646\n",
      "6\t339845\t5706.3\t1\t327131 12714\n",
      "7\t24465\t1426.6\t1\t23532 933\n",
      "8\t7311\t1426.6\t1\t6222 1089\n",
      "9\t90978\t1426.6\t1\t4297 86681\n",
      "10\t185334\t1426.6\t1\t11151 174183\n",
      "11\t49559\t1426.6\t1\t3074 46485\n",
      "12\t22116\t1426.6\t1\t1986 20130\n",
      "13\t25349\t1426.6\t1\t1805 23544\n",
      "14\t22495\t1426.6\t1\t1630 20865\n",
      "15\t20495\t1426.6\t1\t1525 18970\n",
      "16\t21597\t1426.6\t1\t1626 19971\n",
      "17\t27368\t1426.6\t1\t2454 24914\n",
      "18\t28149\t1426.6\t1\t2115 26034\n",
      "19\t23659\t1426.6\t1\t1854 21805\n",
      "20\t22975\t1426.6\t1\t1811 21164\n",
      "21\t21435\t1426.6\t1\t1250 20185\n",
      "22\t31618\t1426.6\t1\t1300 30318\n",
      "23\t682\t1426.6\t1\t649 33\n",
      "24\t146\t1426.6\t1\t143 3\n",
      "25\t18\t1426.6\t1\t18\n",
      "26\t21\t1426.6\t1\t21\n",
      "\n",
      "517\n"
     ]
    }
   ],
   "source": [
    "module use /data/leuven/software/biomed/skylake_centos7/2018a/modules/all/\n",
    "module load SeqKit/0.10.2\n",
    "module load cutadapt/1.18-foss-2018a-Python-3.7.0\n",
    "start=`date +%s`\n",
    "cat 00.raw_reads/CHEQseq_synth_names.txt | while read line\n",
    "do\n",
    "    # Do cutadapt trimming on R1 reads\n",
    "        # Infor form David: Adjust -e to 0.2 for the NovaSeq run because of low sequencing quality on 5 bases! \n",
    "        # In our case good quality except for the first 3bp of one of the samples -> keep the max error rate at 15%\n",
    "        # However, only bewteen 61% and 89% of the reads are written\n",
    "cutadapt -g ACTCGAGCCGAGCAGGCGCGCCGATC...GGACCGG -e 0.2 -l 17 -m 17 -j 10 --discard-untrimmed \\\n",
    "         -o 02.cleaned_reads/${line}_R1.fastq.gz \\\n",
    "         00.raw_reads/${line}_R1.fastq.gz\n",
    "done\n",
    "end=`date +%s`\n",
    "echo $((end-start))\n",
    "\n",
    "\n",
    "## This is if it is R2 (the also RC)\n",
    "            # Do cutadapt trimming on R2 reads and then reverse complement\n",
    "        #cutadapt -g GCGAATTAATTCGGGCCCCGGTCC...GATCGGCGCGCCTGC -e 0.15 -l 17 -m 17 -j 10 --discard-untrimmed \\\n",
    "        #         00.raw_reads/${line}_R2.fastq.gz | \\\n",
    "        #seqkit seq -r -p -o 02.cleaned_reads/${line}_R2.fastq.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Adapter auto-detection is disabled for STDIN mode\n",
      "Read1 before filtering:\n",
      "total reads: 8324361\n",
      "total bases: 141514137\n",
      "Q20 bases: 139248360(98.3989%)\n",
      "Q30 bases: 135548709(95.7846%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 8111342\n",
      "total bases: 137892814\n",
      "Q20 bases: 136812785(99.2168%)\n",
      "Q30 bases: 133755537(96.9996%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 8111342\n",
      "reads failed due to low quality: 213019\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 0\n",
      "reads with adapter trimmed: 0\n",
      "bases trimmed due to adapters: 0\n",
      "\n",
      "Duplication rate (may be overestimated since this is SE data): 0%\n",
      "\n",
      "JSON report: /dev/null/fastp.json\n",
      "HTML report: /dev/null/fastp.html\n",
      "\n",
      "fastp --stdin -e 30 -h /dev/null/fastp.html -j /dev/null/fastp.json -w 8 -o 02.cleaned_reads/CSE__6f8b76__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_cDNA_R1_q30.fastq.gz \n",
      "fastp v0.20.0, time used: 33 seconds\n",
      "Adapter auto-detection is disabled for STDIN mode\n",
      "Read1 before filtering:\n",
      "total reads: 8263521\n",
      "total bases: 140479857\n",
      "Q20 bases: 137685836(98.0111%)\n",
      "Q30 bases: 133612386(95.1114%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 7992966\n",
      "total bases: 135880422\n",
      "Q20 bases: 134738657(99.1597%)\n",
      "Q30 bases: 131462762(96.7489%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 7992966\n",
      "reads failed due to low quality: 270555\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 0\n",
      "reads with adapter trimmed: 0\n",
      "bases trimmed due to adapters: 0\n",
      "\n",
      "Duplication rate (may be overestimated since this is SE data): 0%\n",
      "\n",
      "JSON report: /dev/null/fastp.json\n",
      "HTML report: /dev/null/fastp.html\n",
      "\n",
      "fastp --stdin -e 30 -h /dev/null/fastp.html -j /dev/null/fastp.json -w 8 -o 02.cleaned_reads/CSE__81ba64__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_cDNA_R1_q30.fastq.gz \n",
      "fastp v0.20.0, time used: 31 seconds\n",
      "Adapter auto-detection is disabled for STDIN mode\n",
      "Read1 before filtering:\n",
      "total reads: 5095810\n",
      "total bases: 86628770\n",
      "Q20 bases: 85255501(98.4148%)\n",
      "Q30 bases: 83219285(96.0643%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 4971234\n",
      "total bases: 84510978\n",
      "Q20 bases: 83894323(99.2703%)\n",
      "Q30 bases: 82211449(97.279%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 4971234\n",
      "reads failed due to low quality: 124576\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 0\n",
      "reads with adapter trimmed: 0\n",
      "bases trimmed due to adapters: 0\n",
      "\n",
      "Duplication rate (may be overestimated since this is SE data): 0%\n",
      "\n",
      "JSON report: /dev/null/fastp.json\n",
      "HTML report: /dev/null/fastp.html\n",
      "\n",
      "fastp --stdin -e 30 -h /dev/null/fastp.html -j /dev/null/fastp.json -w 8 -o 02.cleaned_reads/CSE__839e9e__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_cDNA_R1_q30.fastq.gz \n",
      "fastp v0.20.0, time used: 20 seconds\n",
      "Adapter auto-detection is disabled for STDIN mode\n",
      "Read1 before filtering:\n",
      "total reads: 1224817\n",
      "total bases: 20821889\n",
      "Q20 bases: 20585256(98.8635%)\n",
      "Q30 bases: 20129150(96.673%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1203434\n",
      "total bases: 20458378\n",
      "Q20 bases: 20329440(99.3698%)\n",
      "Q30 bases: 19941982(97.4759%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1203434\n",
      "reads failed due to low quality: 21383\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 0\n",
      "reads with adapter trimmed: 0\n",
      "bases trimmed due to adapters: 0\n",
      "\n",
      "Duplication rate (may be overestimated since this is SE data): 0%\n",
      "\n",
      "JSON report: /dev/null/fastp.json\n",
      "HTML report: /dev/null/fastp.html\n",
      "\n",
      "fastp --stdin -e 30 -h /dev/null/fastp.html -j /dev/null/fastp.json -w 8 -o 02.cleaned_reads/CSE__7cb57c__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_cDNA_R1_q30.fastq.gz \n",
      "fastp v0.20.0, time used: 4 seconds\n",
      "Adapter auto-detection is disabled for STDIN mode\n",
      "Read1 before filtering:\n",
      "total reads: 4074384\n",
      "total bases: 69264528\n",
      "Q20 bases: 67976210(98.14%)\n",
      "Q30 bases: 65946095(95.209%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 3952322\n",
      "total bases: 67189474\n",
      "Q20 bases: 66618247(99.1498%)\n",
      "Q30 bases: 64957223(96.6777%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3952322\n",
      "reads failed due to low quality: 122062\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 0\n",
      "reads with adapter trimmed: 0\n",
      "bases trimmed due to adapters: 0\n",
      "\n",
      "Duplication rate (may be overestimated since this is SE data): 0%\n",
      "\n",
      "JSON report: /dev/null/fastp.json\n",
      "HTML report: /dev/null/fastp.html\n",
      "\n",
      "fastp --stdin -e 30 -h /dev/null/fastp.html -j /dev/null/fastp.json -w 8 -o 02.cleaned_reads/CSE__8f7a3e__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_cDNA_R1_q30.fastq.gz \n",
      "fastp v0.20.0, time used: 18 seconds\n",
      "Adapter auto-detection is disabled for STDIN mode\n",
      "Read1 before filtering:\n",
      "total reads: 14671566\n",
      "total bases: 249416622\n",
      "Q20 bases: 245160677(98.2936%)\n",
      "Q30 bases: 238511222(95.6276%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 14267169\n",
      "total bases: 242541873\n",
      "Q20 bases: 240611797(99.2042%)\n",
      "Q30 bases: 235161001(96.9569%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 14267169\n",
      "reads failed due to low quality: 404397\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 0\n",
      "reads with adapter trimmed: 0\n",
      "bases trimmed due to adapters: 0\n",
      "\n",
      "Duplication rate (may be overestimated since this is SE data): 0%\n",
      "\n",
      "JSON report: /dev/null/fastp.json\n",
      "HTML report: /dev/null/fastp.html\n",
      "\n",
      "fastp --stdin -e 30 -h /dev/null/fastp.html -j /dev/null/fastp.json -w 8 -o 02.cleaned_reads/CSE__8b9ec1__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_plasmid_R1_q30.fastq.gz \n",
      "fastp v0.20.0, time used: 56 seconds\n",
      "Adapter auto-detection is disabled for STDIN mode\n",
      "Read1 before filtering:\n",
      "total reads: 15957372\n",
      "total bases: 271275324\n",
      "Q20 bases: 265933755(98.0309%)\n",
      "Q30 bases: 258114391(95.1485%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 15434299\n",
      "total bases: 262383083\n",
      "Q20 bases: 260213676(99.1732%)\n",
      "Q30 bases: 253951950(96.7867%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 15434299\n",
      "reads failed due to low quality: 523073\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 0\n",
      "reads with adapter trimmed: 0\n",
      "bases trimmed due to adapters: 0\n",
      "\n",
      "Duplication rate (may be overestimated since this is SE data): 0%\n",
      "\n",
      "JSON report: /dev/null/fastp.json\n",
      "HTML report: /dev/null/fastp.html\n",
      "\n",
      "fastp --stdin -e 30 -h /dev/null/fastp.html -j /dev/null/fastp.json -w 8 -o 02.cleaned_reads/CSE__cd9463__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_plasmid_R1_q30.fastq.gz \n",
      "fastp v0.20.0, time used: 48 seconds\n",
      "Adapter auto-detection is disabled for STDIN mode\n",
      "Read1 before filtering:\n",
      "total reads: 12841534\n",
      "total bases: 218306078\n",
      "Q20 bases: 214944243(98.46%)\n",
      "Q30 bases: 209480989(95.9575%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 12540334\n",
      "total bases: 213185678\n",
      "Q20 bases: 211546658(99.2312%)\n",
      "Q30 bases: 206968938(97.0839%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 12540334\n",
      "reads failed due to low quality: 301200\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 0\n",
      "reads with adapter trimmed: 0\n",
      "bases trimmed due to adapters: 0\n",
      "\n",
      "Duplication rate (may be overestimated since this is SE data): 0%\n",
      "\n",
      "JSON report: /dev/null/fastp.json\n",
      "HTML report: /dev/null/fastp.html\n",
      "\n",
      "fastp --stdin -e 30 -h /dev/null/fastp.html -j /dev/null/fastp.json -w 8 -o 02.cleaned_reads/CSE__0bf4ec__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_plasmid_R1_q30.fastq.gz \n",
      "fastp v0.20.0, time used: 49 seconds\n",
      "Adapter auto-detection is disabled for STDIN mode\n",
      "Read1 before filtering:\n",
      "total reads: 12838624\n",
      "total bases: 218256608\n",
      "Q20 bases: 214394903(98.2307%)\n",
      "Q30 bases: 207705067(95.1655%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 12508271\n",
      "total bases: 212640607\n",
      "Q20 bases: 210583687(99.0327%)\n",
      "Q30 bases: 204869507(96.3454%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 12508271\n",
      "reads failed due to low quality: 330353\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 0\n",
      "reads with adapter trimmed: 0\n",
      "bases trimmed due to adapters: 0\n",
      "\n",
      "Duplication rate (may be overestimated since this is SE data): 0%\n",
      "\n",
      "JSON report: /dev/null/fastp.json\n",
      "HTML report: /dev/null/fastp.html\n",
      "\n",
      "fastp --stdin -e 30 -h /dev/null/fastp.html -j /dev/null/fastp.json -w 8 -o 02.cleaned_reads/CSE__f7b5af__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_plasmid_R1_q30.fastq.gz \n",
      "fastp v0.20.0, time used: 46 seconds\n",
      "Adapter auto-detection is disabled for STDIN mode\n",
      "Read1 before filtering:\n",
      "total reads: 22531161\n",
      "total bases: 383029737\n",
      "Q20 bases: 366615376(95.7146%)\n",
      "Q30 bases: 338954627(88.493%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 21383081\n",
      "total bases: 363512377\n",
      "Q20 bases: 353682884(97.296%)\n",
      "Q30 bases: 329489892(90.6406%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 21383081\n",
      "reads failed due to low quality: 1148080\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 0\n",
      "reads with adapter trimmed: 0\n",
      "bases trimmed due to adapters: 0\n",
      "\n",
      "Duplication rate (may be overestimated since this is SE data): 0%\n",
      "\n",
      "JSON report: /dev/null/fastp.json\n",
      "HTML report: /dev/null/fastp.html\n",
      "\n",
      "fastp --stdin -e 30 -h /dev/null/fastp.html -j /dev/null/fastp.json -w 8 -o 02.cleaned_reads/CSE__0bd121__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_plasmid_R1_q30.fastq.gz \n",
      "fastp v0.20.0, time used: 84 seconds\n"
     ]
    }
   ],
   "source": [
    "module load fastp/0.20.0-foss-2018a\n",
    "cat 00.raw_reads/CHEQseq_synth_names.txt | while read line\n",
    "do\n",
    "# Combine R1 and R2 \n",
    "#zcat 02.cleaned_reads/${line}_R*.fastq.gz | \\\n",
    "\n",
    "#  Filter for Q30 reads\n",
    "zcat 02.cleaned_reads/${line}_R1.fastq.gz | fastp --stdin -e 30 -h /dev/null/fastp.html -j /dev/null/fastp.json -w 8 \\\n",
    "      -o 02.cleaned_reads/${line}_R1_q30.fastq.gz\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of reads after Q30 filtering for CSE__6f8b76__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_cDNA:  8111342\n",
      "Number of reads after Q30 filtering for CSE__81ba64__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_cDNA:  7992966\n",
      "Number of reads after Q30 filtering for CSE__839e9e__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_cDNA:  4971234\n",
      "Number of reads after Q30 filtering for CSE__7cb57c__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_cDNA:  1203434\n",
      "Number of reads after Q30 filtering for CSE__8f7a3e__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_cDNA:  3952322\n",
      "Number of reads after Q30 filtering for CSE__8b9ec1__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_plasmid:  14267169\n",
      "Number of reads after Q30 filtering for CSE__cd9463__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_plasmid:  15434299\n",
      "Number of reads after Q30 filtering for CSE__0bf4ec__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_plasmid:  12540334\n",
      "Number of reads after Q30 filtering for CSE__f7b5af__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_plasmid:  12508271\n",
      "Number of reads after Q30 filtering for CSE__0bd121__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_plasmid:  21383081\n"
     ]
    }
   ],
   "source": [
    "# Total number of clean reads per sample\n",
    "cat 00.raw_reads/CHEQseq_synth_names.txt | while read line\n",
    "do\n",
    "echo -e \"Number of reads after Q30 filtering for ${line}: \" $(( $(zcat 02.cleaned_reads/${line}_R1_q30.fastq.gz | wc -l) / 4 ))\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 03. Sequencing saturarion\n",
    "\n",
    "- First extract the rBCs from the fastq and put it in a text file\n",
    "- To calculate the statureation, subsample this rBC text file (choose the subsampling frequency a bit depending on the data) and print the number of cleaned reads (i.e. rBCs) taken and the number of unqiue ones for that subsample.\n",
    "    - e.g. now we have in the 03.sequencing_saturation/${line}_R1_q30.txt files a max of 20,884,696 \n",
    "- You can then plot these numbers in excell in a line graph to assess the saturation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Extract DNA sequences from fastq\n",
    "cat 00.raw_reads/CHEQseq_synth_names.txt | while read line\n",
    "do\n",
    "zcat 02.cleaned_reads/${line}_R1_q30.fastq.gz | sed -n '2~4p' > 03.sequencing_saturation/${line}_R1_q30.txt\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "__Number of barcodes and unique barcodes__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "03.sequencing_saturation/CSE__6f8b76__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_cDNA_R1_q30.txt\n",
      "8111342\n",
      "03.sequencing_saturation/CSE__81ba64__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_cDNA_R1_q30.txt\n",
      "7992966\n",
      "03.sequencing_saturation/CSE__839e9e__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_cDNA_R1_q30.txt\n",
      "4971234\n",
      "03.sequencing_saturation/CSE__7cb57c__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_cDNA_R1_q30.txt\n",
      "1203434\n",
      "03.sequencing_saturation/CSE__8f7a3e__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_cDNA_R1_q30.txt\n",
      "3952322\n",
      "03.sequencing_saturation/CSE__8b9ec1__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_plasmid_R1_q30.txt\n",
      "14267169\n",
      "03.sequencing_saturation/CSE__cd9463__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_plasmid_R1_q30.txt\n",
      "15434299\n",
      "03.sequencing_saturation/CSE__0bf4ec__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_plasmid_R1_q30.txt\n",
      "12540334\n",
      "03.sequencing_saturation/CSE__f7b5af__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_plasmid_R1_q30.txt\n",
      "12508271\n",
      "03.sequencing_saturation/CSE__0bd121__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_plasmid_R1_q30.txt\n",
      "21383081\n",
      "03.sequencing_saturation/CSE__6f8b76__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_cDNA_R1_q30.txt\n",
      "1513297\n",
      "03.sequencing_saturation/CSE__81ba64__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_cDNA_R1_q30.txt\n",
      "1338883\n",
      "03.sequencing_saturation/CSE__839e9e__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_cDNA_R1_q30.txt\n",
      "967709\n",
      "03.sequencing_saturation/CSE__7cb57c__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_cDNA_R1_q30.txt\n",
      "103386\n",
      "03.sequencing_saturation/CSE__8f7a3e__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_cDNA_R1_q30.txt\n",
      "481129\n",
      "03.sequencing_saturation/CSE__8b9ec1__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_plasmid_R1_q30.txt\n",
      "2364178\n",
      "03.sequencing_saturation/CSE__cd9463__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_plasmid_R1_q30.txt\n",
      "2370333\n",
      "03.sequencing_saturation/CSE__0bf4ec__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_plasmid_R1_q30.txt\n",
      "2267394\n",
      "03.sequencing_saturation/CSE__f7b5af__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_plasmid_R1_q30.txt\n",
      "1489261\n",
      "03.sequencing_saturation/CSE__0bd121__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_plasmid_R1_q30.txt\n",
      "2849052\n"
     ]
    }
   ],
   "source": [
    "# Count total number of rBC in files\n",
    "cat 00.raw_reads/CHEQseq_synth_names.txt | while read line\n",
    "do\n",
    "echo 03.sequencing_saturation/${line}_R1_q30.txt && cat 03.sequencing_saturation/${line}_R1_q30.txt | wc -l\n",
    "done\n",
    "\n",
    "# Count total number of UNIQUE rBC in files\n",
    "cat 00.raw_reads/CHEQseq_synth_names.txt | while read line\n",
    "do\n",
    "echo 03.sequencing_saturation/${line}_R1_q30.txt && cat 03.sequencing_saturation/${line}_R1_q30.txt | sort | uniq | wc -l\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "559\n"
     ]
    }
   ],
   "source": [
    "# Subsample, count total and unique reads\n",
    "start=`date +%s`\n",
    "\n",
    "cat 00.raw_reads/CHEQseq_synth_names.txt | while read line\n",
    "do\n",
    "file=\"03.sequencing_saturation/${line}_R1_q30.txt\"\n",
    "nreads=$(cat $file | wc -l)\n",
    " # Start by removing parallel instructions file\n",
    "rm 03.sequencing_saturation/parallel.txt\n",
    " # Remove any previoulsy created sat values for that file\n",
    "rm ${file%_*}_sat.txt\n",
    "    for i in $(seq 0 200000 $nreads)\n",
    "    do\n",
    "     # Print number of total and unique reads for every subsamples\n",
    "    echo \"echo $i \\$(shuf -n $(($i)) $file | sort | uniq | wc -l) >> ${file%_*}_sat.txt\" \\\n",
    "    >> 03.sequencing_saturation/parallel.txt\n",
    "    done\n",
    "cat 03.sequencing_saturation/parallel.txt | parallel -j 16\n",
    " # Put values in numerical order\n",
    "cat ${file%_*}_sat.txt | sort -g > ${file%_*}_satsort.txt && mv ${file%_*}_satsort.txt ${file%_*}_sat.txt\n",
    "done\n",
    "\n",
    "end=`date +%s`\n",
    "echo $((end-start))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 10. Make rBC count matrix and assign to enhancers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10.bc_count/CSE__6f8b76__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_cDNA_count.txt\n",
      "1513297\n",
      "10.bc_count/CSE__81ba64__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_cDNA_count.txt\n",
      "1338883\n",
      "10.bc_count/CSE__839e9e__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_cDNA_count.txt\n",
      "967709\n",
      "10.bc_count/CSE__7cb57c__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_cDNA_count.txt\n",
      "103386\n",
      "10.bc_count/CSE__8f7a3e__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_cDNA_count.txt\n",
      "481129\n",
      "10.bc_count/CSE__8b9ec1__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_plasmid_count.txt\n",
      "2364178\n",
      "10.bc_count/CSE__cd9463__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_plasmid_count.txt\n",
      "2370333\n",
      "10.bc_count/CSE__0bf4ec__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_plasmid_count.txt\n",
      "2267394\n",
      "10.bc_count/CSE__f7b5af__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_plasmid_count.txt\n",
      "1489261\n",
      "10.bc_count/CSE__0bd121__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_plasmid_count.txt\n",
      "2849052\n"
     ]
    }
   ],
   "source": [
    "# Generate rBC count matrix for each samples\n",
    "parallel \\\n",
    "\"cat 03.sequencing_saturation/{}_R1_q30.txt | sort | uniq -c | awk -F \\\" \\\" '{print \\$2\\\"\\t\\\"\\$1}' \\\n",
    "   > 10.bc_count/{}_count.txt\" ::: $(cat 00.raw_reads/CHEQseq_synth_names.txt)\n",
    "\n",
    "# Count number of unique rBC per sample\n",
    "cat 00.raw_reads/CHEQseq_synth_names.txt | while read line\n",
    "do\n",
    "echo 10.bc_count/${line}_count.txt && cat 10.bc_count/${line}_count.txt | wc -l\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "__We used the 1x library for transfections__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Assign rBC to enhancers\n",
    "    # List of rBC-Enhancer files : I one have 1, I will use the high complexity\n",
    "    # The file had another order: rBC - enhancer -> adatp it to (1) readID (put .) (2) enhancer (3) rBC ; and \n",
    "\n",
    "# OLSA=\"/staging/leuven/stg_00002/lcb/dwmax/documents/aertslab/CH1/CHEQ-seq_OLS_DavidTileA/20.coupling/CH1__b87bb6__CheqSeq_OLS_David_Pool_A_enhancer-barcode_S2_enhancer_barcode_coupled_wo_empty_vectors_bc_rc.tsv.gz\"\n",
    "\n",
    "\n",
    "cat /user/leuven/317/vsc31703/stg_00002/Enhancer_design/MPRA/CHEQ-seq_cloning2/E-BC_coupling/03.E-BC_coupling/CSE__953401__CHEQseq-cl2_2000enh_1x_enhancerID_sBC_rBC_coupled.format_unique.sorted_1to1.matrix | \\\n",
    "awk '{print \".\" \"\\t\" $2 \"\\t\" $1 }' | gzip -c > /user/leuven/317/vsc31703/stg_00002/Enhancer_design/MPRA/CHEQ-seq_cloning2/E-BC_coupling/03.E-BC_coupling/CSE__953401__CHEQseq-cl2_2000enh_1x_enhancerID_sBC_rBC_coupled.format_unique.sorted_1to1.matrix.gz\n",
    "\n",
    "COUPLE=\"/user/leuven/317/vsc31703/stg_00002/Enhancer_design/MPRA/CHEQ-seq_cloning2/E-BC_coupling/03.E-BC_coupling/CSE__953401__CHEQseq-cl2_2000enh_1x_enhancerID_sBC_rBC_coupled.format_unique.sorted_1to1.matrix.gz\"\n",
    "   \n",
    "   # OLS A samples\n",
    "#parallel \\\n",
    "#\"awk -F '\\t' -v OFS='\\t' 'FNR==NR{a[\\$3]=\\$2 FS \\$3;next}{ print a[\\$1],\\$2}' \\\n",
    "#   <( zcat ${OLSA} ) \\\n",
    "#   <( cat 10.bc_count/{}_count.txt ) \\\n",
    "#   | awk -F '\\t' '{if (\\$1) print \\$0;}' \\\n",
    "#   > 10.bc_count/{}_count_final.txt\" ::: $(grep 'OLS_A' 00.raw_reads/CHEQseq_synth_names.txt)\n",
    "   \n",
    "   # For mine:\n",
    "parallel \\\n",
    "\"awk -F '\\t' -v OFS='\\t' 'FNR==NR{a[\\$3]=\\$2 FS \\$3;next}{ print a[\\$1],\\$2}' \\\n",
    "   <( zcat ${COUPLE} ) \\\n",
    "   <( cat 10.bc_count/{}_count.txt ) \\\n",
    "   | awk -F '\\t' '{if (\\$1) print \\$0;}' \\\n",
    "   > 10.bc_count/{}_count_final.txt\" ::: $(cat 00.raw_reads/CHEQseq_synth_names.txt)   \n",
    "   \n",
    " "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CSE__6f8b76__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_cDNA\t597159\n",
      "CSE__81ba64__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_cDNA\t522569\n",
      "CSE__839e9e__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_cDNA\t384723\n",
      "CSE__7cb57c__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_cDNA\t40027\n",
      "CSE__8f7a3e__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_cDNA\t183189\n",
      "CSE__8b9ec1__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_plasmid\t894810\n",
      "CSE__cd9463__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_plasmid\t887510\n",
      "CSE__0bf4ec__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_plasmid\t861629\n",
      "CSE__f7b5af__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_plasmid\t575865\n",
      "CSE__0bd121__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_plasmid\t907338\n"
     ]
    }
   ],
   "source": [
    "# Count number of rBC associated to an enhancer\n",
    "cat 00.raw_reads/CHEQseq_synth_names.txt | while read line\n",
    "do\n",
    "echo -e \"${line}\\t$(cat 10.bc_count/${line}_count_final.txt | wc -l)\"\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CSE__6f8b76__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_cDNA\n",
      "1996\n",
      "158\n",
      "CSE__81ba64__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_cDNA\n",
      "1997\n",
      "137\n",
      "CSE__839e9e__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_cDNA\n",
      "1996\n",
      "98\n",
      "CSE__7cb57c__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_cDNA\n",
      "1866\n",
      "9\n",
      "CSE__8f7a3e__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_cDNA\n",
      "1990\n",
      "41\n",
      "CSE__8b9ec1__CheqSeq_Liesbeth_specific_enhancer_pool_MM001_plasmid\n",
      "1998\n",
      "254\n",
      "CSE__cd9463__CheqSeq_Liesbeth_specific_enhancer_pool_MM074_plasmid\n",
      "1998\n",
      "253\n",
      "CSE__0bf4ec__CheqSeq_Liesbeth_specific_enhancer_pool_MM087_plasmid\n",
      "1998\n",
      "243\n",
      "CSE__f7b5af__CheqSeq_Liesbeth_specific_enhancer_pool_MM029_plasmid\n",
      "1998\n",
      "153\n",
      "CSE__0bd121__CheqSeq_Liesbeth_specific_enhancer_pool_MM099_plasmid\n",
      "1998\n",
      "260\n"
     ]
    }
   ],
   "source": [
    "# Coverage (i.e number of enhancer recovered) and Median number of rBC per enhancer\n",
    "cat 00.raw_reads/CHEQseq_synth_names.txt | while read line\n",
    "do\n",
    "file=\"10.bc_count/${line}_count_final.txt\"\n",
    "middle=$(($(awk '{print $1}' $file | sort | uniq -c | wc -l)/2))\n",
    "echo $line\n",
    "awk '{print $1}' $file | sort | uniq -c | wc -l\n",
    "awk '{print $1}' $file | sort | uniq -c | awk '{print $1}' | sort -g | sed -n ${middle}p\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash 4.2.46",
   "language": "bash",
   "name": "bash4246"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
