{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Cloning #2 - E-BC coupling CHEQ-seq library\n",
    "\n",
    "David re-did the cloning from scratch. the library was sequenced for E-BC coupling on 10100730. A 3x and 1x cloning condition were done."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Reminder of structure of cloned TWIST enhancer pool:\n",
    "\n",
    "     15bp adaptor – 11bp BC - 259bp long enhancer - 15 bp adaptors.\n",
    "\n",
    "- Note that for CHEQ-seq we will thus have 2 BCs:\n",
    "    - the synthetic designed 11bp BC next to the enhancer (1 mismatch allowed) = sBC\n",
    "    - the 17bp random BC within the CHEQ-seq vector = rBC\n",
    "- Adapters:\n",
    "    - 5' : GAGCATGCACCGGTG\n",
    "    - 3' : CGCTTCGAGCAGACA\n",
    "- Total of 2,000 sequences, 300 bp long \n",
    "- For amplification of the pool VC2536-2537 were used (a bit longer tail than the adapters above)\n",
    "- During the linearisation of the CHEQ-seq vector we used newly designed primers (VC2573 (F), VC2574 (R)), since the adapters in the TWIST clones were designed for the STAR-seq plasmid\n",
    "- For the E-BC sequencing, libraries were prepped with VC2544 (< David) and VC2588 / VC2589 (newly designed)\n",
    "- Raw data: \n",
    "    - */staging/leuven/stg_00002/lcb/ngs_runs/NovaSeq6000_20200730/Demultiplexed*\n",
    "    - Paired-end: R1 contains sBC + part of enhancer ; R2 contains the rBC\n",
    "    - We have 2 reads for the low and the high complexity library each"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "__Copy data__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "DIR=/user/leuven/317/vsc31703/stg_00002/Enhancer_design/MPRA/CHEQ-seq_cloning2\n",
    "\n",
    "cd $DIR && mkdir -p 00.RAW_FASTQ && cp /staging/leuven/stg_00002/lcb/ngs_runs/NovaSeq6000_20200730/Demultiplexed/*Liesbeth* 00.RAW_FASTQ/ 2> /dev/null \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "__Change names of fastqs__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "cd 00.RAW_FASTQ\n",
    "mv CSE__88dc35__amplified_enhancer-barcode_Twist_pool_Liesbeth_3x_S10_R1_001.fastq.gz CSE__88dc35__CHEQseq-cl2_2000enh_3x_R1.fastq.gz \n",
    "mv CSE__88dc35__amplified_enhancer-barcode_Twist_pool_Liesbeth_3x_S10_R2_001.fastq.gz CSE__88dc35__CHEQseq-cl2_2000enh_3x_R2.fastq.gz \n",
    "mv  CSE__953401__amplified_enhancer-barcode_Twist_pool_Liesbeth_1x_S9_R1_001.fastq.gz CSE__953401__CHEQseq-cl2_2000enh_1x_R1.fastq.gz \n",
    "mv  CSE__953401__amplified_enhancer-barcode_Twist_pool_Liesbeth_1x_S9_R2_001.fastq.gz CSE__953401__CHEQseq-cl2_2000enh_1x_R2.fastq.gz\n",
    "cd $DIR"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 1. Processing read 1"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "__Parameters__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "#mkdir E-BC_coupling\n",
    "DIR=/user/leuven/317/vsc31703/stg_00002/Enhancer_design/MPRA/CHEQ-seq_cloning2/E-BC_coupling\n",
    "#NAME=CSE__88dc35__CHEQseq-cl2_2000enh_3x\n",
    "NAME=CSE__953401__CHEQseq-cl2_2000enh_1x\n",
    "\n",
    "cd $DIR\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1.0. Fastqc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘01.FASTQC’: File exists\n",
      "Started analysis of CSE__953401__CHEQseq-cl2_2000enh_1x_R1.fastq.gz\n",
      "Approx 5% complete for CSE__953401__CHEQseq-cl2_2000enh_1x_R1.fastq.gz\n",
      "Approx 10% complete for CSE__953401__CHEQseq-cl2_2000enh_1x_R1.fastq.gz\n",
      "Approx 15% complete for CSE__953401__CHEQseq-cl2_2000enh_1x_R1.fastq.gz\n",
      "Approx 20% complete for CSE__953401__CHEQseq-cl2_2000enh_1x_R1.fastq.gz\n",
      "Approx 25% complete for CSE__953401__CHEQseq-cl2_2000enh_1x_R1.fastq.gz\n",
      "Approx 30% complete for CSE__953401__CHEQseq-cl2_2000enh_1x_R1.fastq.gz\n",
      "Approx 35% complete for CSE__953401__CHEQseq-cl2_2000enh_1x_R1.fastq.gz\n",
      "Approx 40% complete for CSE__953401__CHEQseq-cl2_2000enh_1x_R1.fastq.gz\n",
      "Approx 45% complete for CSE__953401__CHEQseq-cl2_2000enh_1x_R1.fastq.gz\n",
      "Approx 50% complete for CSE__953401__CHEQseq-cl2_2000enh_1x_R1.fastq.gz\n",
      "Approx 55% complete for CSE__953401__CHEQseq-cl2_2000enh_1x_R1.fastq.gz\n",
      "Approx 60% complete for CSE__953401__CHEQseq-cl2_2000enh_1x_R1.fastq.gz\n",
      "Approx 65% complete for CSE__953401__CHEQseq-cl2_2000enh_1x_R1.fastq.gz\n",
      "Approx 70% complete for CSE__953401__CHEQseq-cl2_2000enh_1x_R1.fastq.gz\n",
      "Approx 75% complete for CSE__953401__CHEQseq-cl2_2000enh_1x_R1.fastq.gz\n",
      "Approx 80% complete for CSE__953401__CHEQseq-cl2_2000enh_1x_R1.fastq.gz\n",
      "Approx 85% complete for CSE__953401__CHEQseq-cl2_2000enh_1x_R1.fastq.gz\n",
      "Approx 90% complete for CSE__953401__CHEQseq-cl2_2000enh_1x_R1.fastq.gz\n",
      "Approx 95% complete for CSE__953401__CHEQseq-cl2_2000enh_1x_R1.fastq.gz\n",
      "Analysis complete for CSE__953401__CHEQseq-cl2_2000enh_1x_R1.fastq.gz\n",
      "Started analysis of CSE__953401__CHEQseq-cl2_2000enh_1x_R2.fastq.gz\n",
      "Approx 5% complete for CSE__953401__CHEQseq-cl2_2000enh_1x_R2.fastq.gz\n",
      "Approx 10% complete for CSE__953401__CHEQseq-cl2_2000enh_1x_R2.fastq.gz\n",
      "Approx 15% complete for CSE__953401__CHEQseq-cl2_2000enh_1x_R2.fastq.gz\n",
      "Approx 20% complete for CSE__953401__CHEQseq-cl2_2000enh_1x_R2.fastq.gz\n",
      "Approx 25% complete for CSE__953401__CHEQseq-cl2_2000enh_1x_R2.fastq.gz\n",
      "Approx 30% complete for CSE__953401__CHEQseq-cl2_2000enh_1x_R2.fastq.gz\n",
      "Approx 35% complete for CSE__953401__CHEQseq-cl2_2000enh_1x_R2.fastq.gz\n",
      "Approx 40% complete for CSE__953401__CHEQseq-cl2_2000enh_1x_R2.fastq.gz\n",
      "Approx 45% complete for CSE__953401__CHEQseq-cl2_2000enh_1x_R2.fastq.gz\n",
      "Approx 50% complete for CSE__953401__CHEQseq-cl2_2000enh_1x_R2.fastq.gz\n",
      "Approx 55% complete for CSE__953401__CHEQseq-cl2_2000enh_1x_R2.fastq.gz\n",
      "Approx 60% complete for CSE__953401__CHEQseq-cl2_2000enh_1x_R2.fastq.gz\n",
      "Approx 65% complete for CSE__953401__CHEQseq-cl2_2000enh_1x_R2.fastq.gz\n",
      "Approx 70% complete for CSE__953401__CHEQseq-cl2_2000enh_1x_R2.fastq.gz\n",
      "Approx 75% complete for CSE__953401__CHEQseq-cl2_2000enh_1x_R2.fastq.gz\n",
      "Approx 80% complete for CSE__953401__CHEQseq-cl2_2000enh_1x_R2.fastq.gz\n",
      "Approx 85% complete for CSE__953401__CHEQseq-cl2_2000enh_1x_R2.fastq.gz\n",
      "Approx 90% complete for CSE__953401__CHEQseq-cl2_2000enh_1x_R2.fastq.gz\n",
      "Approx 95% complete for CSE__953401__CHEQseq-cl2_2000enh_1x_R2.fastq.gz\n",
      "Analysis complete for CSE__953401__CHEQseq-cl2_2000enh_1x_R2.fastq.gz\n"
     ]
    }
   ],
   "source": [
    "mkdir 01.FASTQC\n",
    "module load FastQC/0.11.8-Java-1.8.0\n",
    "\n",
    "fastqc -t 1 ../00.RAW_FASTQ/${NAME}_R1.fastq.gz -o 01.FASTQC\n",
    "fastqc -t 1 ../00.RAW_FASTQ/${NAME}_R2.fastq.gz -o 01.FASTQC"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1.1. Filter for sequencing qualtiy\n",
    "We only keep reads that have Q20 for 80% of the bp.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘02.TRIMMED_FQ’: File exists\n",
      "Read1 before filtering:\n",
      "total reads: 16815676\n",
      "total bases: 823968124\n",
      "Q20 bases: 812490745(98.6071%)\n",
      "Q30 bases: 789445512(95.8102%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 16544387\n",
      "total bases: 810575178\n",
      "Q20 bases: 802697832(99.0282%)\n",
      "Q30 bases: 781569218(96.4216%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 16544387\n",
      "reads failed due to low quality: 271285\n",
      "reads failed due to too many N: 0\n",
      "reads failed due to too short: 4\n",
      "\n",
      "Duplication rate (may be overestimated since this is SE data): 0.812216%\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp --disable_adapter_trimming -q 20 -u 20 -i ../00.RAW_FASTQ/CSE__953401__CHEQseq-cl2_2000enh_1x_R1.fastq.gz -o 02.TRIMMED_FQ/CSE__953401__CHEQseq-cl2_2000enh_1x_R1.Q20.fastq.gz \n",
      "fastp v0.20.0, time used: 55 seconds\n"
     ]
    }
   ],
   "source": [
    "# at least 80% of the NT (-u 20) must have a quality higher then 20 \n",
    "module load fastp/0.20.0-foss-2018a\n",
    "mkdir 02.TRIMMED_FQ\n",
    "fastp --disable_adapter_trimming -q 20 -u 20 -i ../00.RAW_FASTQ/${NAME}_R1.fastq.gz -o 02.TRIMMED_FQ/${NAME}_R1.Q20.fastq.gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1.2. Trim the 5' adapter\n",
    "Cut with *ATCTAGAGCATGCACCGGTG* (20bp).\n",
    "\n",
    "The stagger of 1bp should be tolerated normally (Cuadtapt is robust to errors of 10%).\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is cutadapt 2.4 with Python 3.7.4\n",
      "Command line parameters: -g ATCTAGAGCATGCACCGGTG --trimmed-only --output 02.TRIMMED_FQ/CSE__953401__CHEQseq-cl2_2000enh_1x_R1.Q20_trimmed5.fastq.gz 02.TRIMMED_FQ/CSE__953401__CHEQseq-cl2_2000enh_1x_R1.Q20.fastq.gz\n",
      "Processing reads on 1 core in single-end mode ...\n",
      "[    8<------] 00:03:19    16,544,387 reads  @     12.0 µs/read;   4.99 M reads/minute\n",
      "Finished in 199.10 s (12 us/read; 4.99 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              16,544,387\n",
      "Reads with adapters:                16,536,217 (100.0%)\n",
      "Reads written (passing filters):    16,536,217 (100.0%)\n",
      "\n",
      "Total basepairs processed:   810,575,178 bp\n",
      "Total written (filtered):    479,385,713 bp (59.1%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: ATCTAGAGCATGCACCGGTG; Type: regular 5'; Length: 20; Trimmed: 16536217 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t7\t258506.0\t0\t7\n",
      "9\t2\t63.1\t0\t2\n",
      "10\t5\t15.8\t1\t4 1\n",
      "12\t23\t1.0\t1\t18 5\n",
      "13\t15\t0.2\t1\t12 3\n",
      "14\t73\t0.1\t1\t58 15\n",
      "15\t56\t0.0\t1\t35 21\n",
      "16\t79\t0.0\t1\t64 15\n",
      "17\t161\t0.0\t1\t77 84\n",
      "18\t1672\t0.0\t1\t220 346 1106\n",
      "19\t41425\t0.0\t1\t2099 38933 393\n",
      "20\t16388338\t0.0\t2\t16251348 133915 3075\n",
      "21\t100688\t0.0\t2\t84249 16176 263\n",
      "22\t3204\t0.0\t2\t74 123 3007\n",
      "23\t139\t0.0\t2\t60 15 64\n",
      "24\t60\t0.0\t2\t51 6 3\n",
      "25\t44\t0.0\t2\t30 7 7\n",
      "26\t25\t0.0\t2\t20 4 1\n",
      "27\t10\t0.0\t2\t8 1 1\n",
      "28\t18\t0.0\t2\t13 4 1\n",
      "29\t9\t0.0\t2\t6 1 2\n",
      "30\t11\t0.0\t2\t8 2 1\n",
      "31\t9\t0.0\t2\t7 0 2\n",
      "32\t7\t0.0\t2\t6 1\n",
      "33\t7\t0.0\t2\t7\n",
      "34\t6\t0.0\t2\t5 0 1\n",
      "35\t8\t0.0\t2\t7 1\n",
      "36\t8\t0.0\t2\t8\n",
      "37\t8\t0.0\t2\t8\n",
      "38\t7\t0.0\t2\t7\n",
      "39\t12\t0.0\t2\t12\n",
      "40\t5\t0.0\t2\t5\n",
      "41\t16\t0.0\t2\t16\n",
      "42\t9\t0.0\t2\t9\n",
      "43\t13\t0.0\t2\t11 2\n",
      "44\t5\t0.0\t2\t5\n",
      "45\t6\t0.0\t2\t6\n",
      "46\t7\t0.0\t2\t7\n",
      "47\t6\t0.0\t2\t6\n",
      "48\t6\t0.0\t2\t6\n",
      "49\t8\t0.0\t2\t5 1 2\n"
     ]
    }
   ],
   "source": [
    "# write only the reads that are trimmed, so that contained the adaptor\n",
    "module load cutadapt/2.4-foss-2018a-Python-3.7.4\n",
    "\n",
    "cutadapt -g ATCTAGAGCATGCACCGGTG --trimmed-only \\\n",
    "--output 02.TRIMMED_FQ/${NAME}_R1.Q20_trimmed5.fastq.gz \\\n",
    "02.TRIMMED_FQ/${NAME}_R1.Q20.fastq.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "      8 0\n",
      "      6 1\n",
      "     17 10\n",
      "    976 11\n",
      "   3046 12\n",
      "    344 13\n",
      "    118 14\n",
      "    399 15\n",
      "    416 16\n",
      "    406 17\n",
      "    435 18\n",
      "    273 19\n",
      "      6 2\n",
      "     82 20\n",
      "     18 21\n",
      "     10 22\n",
      "     25 23\n",
      "     44 24\n",
      "     60 25\n",
      "    139 26\n",
      "   3200 27\n",
      " 100638 28\n",
      "16381982 29\n",
      "      7 3\n",
      "  41413 30\n",
      "   1672 31\n",
      "    161 32\n",
      "     79 33\n",
      "     56 34\n",
      "     73 35\n",
      "     15 36\n",
      "     23 37\n",
      "      5 39\n",
      "      6 4\n",
      "      2 40\n",
      "      7 46\n",
      "      5 5\n",
      "     13 6\n",
      "     10 7\n",
      "     16 8\n",
      "      6 9\n"
     ]
    }
   ],
   "source": [
    "# What is the length of the remainder?\n",
    "zcat  02.TRIMMED_FQ/${NAME}_R1.Q20_trimmed5.fastq.gz | sed -n '2~4p' | awk '{print length}'  | sort | uniq -c"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1.3. Remove empty plasmid reads\n",
    "\n",
    "cut adapt: with the next part of the vector as 3’ adapter: *CGCTTCGAGCAGACA* (15 bp).\n",
    "\n",
    "-> the reads that have that 3' adapter or empty plasmid reads and will be kept in a seperate file, the rest (untrimmed reads) will be written to a another file and with this file we will continue to get the enhancer out."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is cutadapt 2.4 with Python 3.7.4\n",
      "Command line parameters: -a CGCTTCGAGCAGACA --overlap 5 --output 02.TRIMMED_FQ/CSE__953401__CHEQseq-cl2_2000enh_1x_R1.Q20_trimmed5_emptyVector.fastq.gz --untrimmed-output 02.TRIMMED_FQ/CSE__953401__CHEQseq-cl2_2000enh_1x_R1.Q20_trimmed5_full.fastq.gz 02.TRIMMED_FQ/CSE__953401__CHEQseq-cl2_2000enh_1x_R1.Q20_trimmed5.fastq.gz\n",
      "Processing reads on 1 core in single-end mode ...\n",
      "[ 8=---------] 00:02:20    16,536,217 reads  @      8.5 µs/read;   7.04 M reads/minute\n",
      "Finished in 140.93 s (9 us/read; 7.04 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              16,536,217\n",
      "Reads with adapters:                     7,743 (0.0%)\n",
      "Reads written (passing filters):         7,743 (0.0%)\n",
      "\n",
      "Total basepairs processed:   479,385,713 bp\n",
      "Total written (filtered):         92,089 bp (0.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CGCTTCGAGCAGACA; Type: regular 3'; Length: 15; Trimmed: 7743 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-15 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 14.0%\n",
      "  C: 54.6%\n",
      "  G: 14.4%\n",
      "  T: 16.8%\n",
      "  none/other: 0.2%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "5\t762\t16148.6\t0\t762\n",
      "6\t432\t4037.2\t0\t432\n",
      "7\t436\t1009.3\t0\t436\n",
      "8\t324\t252.3\t0\t324\n",
      "9\t254\t63.1\t0\t254\n",
      "10\t186\t15.8\t1\t184 2\n",
      "11\t259\t3.9\t1\t257 2\n",
      "12\t255\t1.0\t1\t254 1\n",
      "13\t245\t0.2\t1\t244 1\n",
      "14\t146\t0.1\t1\t141 5\n",
      "15\t109\t0.0\t1\t109\n",
      "16\t108\t0.0\t1\t108\n",
      "17\t134\t0.0\t1\t132 2\n",
      "18\t57\t0.0\t1\t50 7\n",
      "19\t178\t0.0\t1\t176 2\n",
      "20\t114\t0.0\t1\t113 1\n",
      "21\t116\t0.0\t1\t115 1\n",
      "22\t98\t0.0\t1\t97 1\n",
      "23\t106\t0.0\t1\t104 2\n",
      "24\t110\t0.0\t1\t109 1\n",
      "25\t3134\t0.0\t1\t3110 24\n",
      "26\t115\t0.0\t1\t112 3\n",
      "27\t31\t0.0\t1\t31\n",
      "28\t21\t0.0\t1\t21\n",
      "29\t12\t0.0\t1\t12\n",
      "30\t1\t0.0\t1\t1\n"
     ]
    }
   ],
   "source": [
    "## Find empty plasmids:\n",
    "    # write the trimmed ones to an output ( = empty vector) ; the untrimmed also to an output (= vector with enhancer)\n",
    "    # Minimum overlap of 5 is required --overlap\n",
    "\n",
    "module load cutadapt/2.4-foss-2018a-Python-3.7.4\n",
    "\n",
    "cutadapt -a CGCTTCGAGCAGACA \\\n",
    "--overlap 5 \\\n",
    "--output 02.TRIMMED_FQ/${NAME}_R1.Q20_trimmed5_emptyVector.fastq.gz \\\n",
    "--untrimmed-output 02.TRIMMED_FQ/${NAME}_R1.Q20_trimmed5_full.fastq.gz  \\\n",
    "02.TRIMMED_FQ/${NAME}_R1.Q20_trimmed5.fastq.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16528474\n",
      "      8 0\n",
      "      6 1\n",
      "     17 10\n",
      "    976 11\n",
      "   3046 12\n",
      "    344 13\n",
      "    118 14\n",
      "    399 15\n",
      "    416 16\n",
      "    406 17\n",
      "    435 18\n",
      "    273 19\n",
      "      6 2\n",
      "     82 20\n",
      "     18 21\n",
      "     10 22\n",
      "     25 23\n",
      "     44 24\n",
      "     60 25\n",
      "    139 26\n",
      "   3193 27\n",
      " 100546 28\n",
      "16374449 29\n",
      "      7 3\n",
      "  41311 30\n",
      "   1663 31\n",
      "    161 32\n",
      "     79 33\n",
      "     56 34\n",
      "     73 35\n",
      "     15 36\n",
      "     23 37\n",
      "      5 39\n",
      "      6 4\n",
      "      2 40\n",
      "      7 46\n",
      "      5 5\n",
      "     13 6\n",
      "     10 7\n",
      "     16 8\n",
      "      6 9\n"
     ]
    }
   ],
   "source": [
    "zcat 02.TRIMMED_FQ/${NAME}_R1.Q20_trimmed5_full.fastq.gz | sed -n '{2~4p}' |wc -l\n",
    "zcat  02.TRIMMED_FQ/${NAME}_R1.Q20_trimmed5_full.fastq.gz | sed -n '2~4p' | awk '{print length}'  | sort | uniq -c"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1.4. Extract the sBCs\n",
    "\n",
    "First demand that the read should be at least 28bp long.\n",
    "\n",
    "Then make fasta only keeping the first 11bp = sBC."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is cutadapt 2.4 with Python 3.7.4\n",
      "Command line parameters: -l 11 -o 02.TRIMMED_FQ/CSE__953401__CHEQseq-cl2_2000enh_1x_R1.Q20_trimmed5_full_filt_sBC.fastq.gz 02.TRIMMED_FQ/CSE__953401__CHEQseq-cl2_2000enh_1x_R1.Q20_trimmed5_full_filt.fastq.gz\n",
      "Processing reads on 1 core in single-end mode ...\n",
      "[---->8      ] 00:01:20    16,518,390 reads  @      4.9 µs/read;  12.30 M reads/minute\n",
      "Finished in 80.62 s (5 us/read; 12.29 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              16,518,390\n",
      "Reads with adapters:                         0 (0.0%)\n",
      "Reads written (passing filters):    16,518,390 (100.0%)\n",
      "\n",
      "Total basepairs processed:   478,979,398 bp\n",
      "Total written (filtered):    181,702,290 bp (37.9%)\n"
     ]
    }
   ],
   "source": [
    "# Remove reads < 28bp\n",
    "zcat 02.TRIMMED_FQ/${NAME}_R1.Q20_trimmed5_full.fastq.gz| awk 'NR%4==1{a=$0} NR%4==2{b=$0} NR%4==3{c=$0} NR%4==0&&length(b)>=28{print a\"\\n\"b\"\\n\"c\"\\n\"$0;}' \\\n",
    "| gzip > 02.TRIMMED_FQ/${NAME}_R1.Q20_trimmed5_full_filt.fastq.gz\n",
    "\n",
    "# Keep only the first 11bp\n",
    "    # Positive values remove bases at the end while negative ones remove bases at the beginning. \n",
    "cutadapt -l 11 -o 02.TRIMMED_FQ/${NAME}_R1.Q20_trimmed5_full_filt_sBC.fastq.gz 02.TRIMMED_FQ/${NAME}_R1.Q20_trimmed5_full_filt.fastq.gz\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1.5. Q30 filtering on the sBCs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Detecting adapter sequence for read1...\n",
      "No adapter detected for read1\n",
      "\n",
      "Read1 before filtering:\n",
      "total reads: 16518390\n",
      "total bases: 181702290\n",
      "Q20 bases: 179989923(99.0576%)\n",
      "Q30 bases: 175306864(96.4803%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 16376089\n",
      "total bases: 180136569\n",
      "Q20 bases: 178825131(99.272%)\n",
      "Q30 bases: 174445565(96.8407%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 16376089\n",
      "reads failed due to low quality: 142301\n",
      "reads failed due to too many N: 0\n",
      "reads with adapter trimmed: 0\n",
      "bases trimmed due to adapters: 0\n",
      "\n",
      "Duplication rate (may be overestimated since this is SE data): 0%\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -e 30 --disable_length_filtering -w 8 -o 02.TRIMMED_FQ/CSE__953401__CHEQseq-cl2_2000enh_1x_R1.Q20_trimmed5_full_filt_sBCq30.fastq.gz -i 02.TRIMMED_FQ/CSE__953401__CHEQseq-cl2_2000enh_1x_R1.Q20_trimmed5_full_filt_sBC.fastq.gz \n",
      "fastp v0.20.0, time used: 33 seconds\n"
     ]
    }
   ],
   "source": [
    "# Q30 filtering\n",
    "module load fastp/0.20.0-foss-2018a\n",
    "fastp -e 30 --disable_length_filtering -w 8 \\\n",
    "      -o 02.TRIMMED_FQ/${NAME}_R1.Q20_trimmed5_full_filt_sBCq30.fastq.gz \\\n",
    "      -i 02.TRIMMED_FQ/${NAME}_R1.Q20_trimmed5_full_filt_sBC.fastq.gz\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16376089 21313 0.130147 CGAAGCCTCTG 130838 0.798958\n"
     ]
    }
   ],
   "source": [
    "## output one line with: \n",
    "# total number of reads, \n",
    "# number of unique reads, \n",
    "# percentage of unique reads, \n",
    "# most common sequence, \n",
    "# number of occurrence of the most common sequence and frequency of the most common sequence.\n",
    "\n",
    "zcat  02.TRIMMED_FQ/${NAME}_R1.Q20_trimmed5_full_filt_sBCq30.fastq.gz | awk '((NR-2)%4==0){read=$1;total++;count[read]++}END{for(read in count){if(!max||count[read]>max) {max=count[read];maxRead=read};if(count[read]==1){unique++}};print total,unique,unique*100/total,maxRead,count[maxRead],count[maxRead]*100/total}'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Remove reads < 28bp\n",
    "zcat  02.TRIMMED_FQ/${NAME}_R1.Q20_trimmed5_full_filt_sBCq30.fastq.gz| awk 'NR%4==1{a=$0} NR%4==2{b=$0} NR%4==3{c=$0} NR%4==0&&length(b)>=11{print a\"\\n\"b\"\\n\"c\"\\n\"$0;}' \\\n",
    "| gzip >  02.TRIMMED_FQ/${NAME}_R1.Q20_trimmed5_full_filt_sBCq30_filt.fastq.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16376048 21313 0.130147 CGAAGCCTCTG 130838 0.79896\n"
     ]
    }
   ],
   "source": [
    "## output one line with: \n",
    "# total number of reads, \n",
    "# number of unique reads, \n",
    "# percentage of unique reads, \n",
    "# most common sequence, \n",
    "# number of occurrence of the most common sequence and frequency of the most common sequence.\n",
    "\n",
    "zcat  02.TRIMMED_FQ/${NAME}_R1.Q20_trimmed5_full_filt_sBCq30_filt.fastq.gz | awk '((NR-2)%4==0){read=$1;total++;count[read]++}END{for(read in count){if(!max||count[read]>max) {max=count[read];maxRead=read};if(count[read]==1){unique++}};print total,unique,unique*100/total,maxRead,count[maxRead],count[maxRead]*100/total}'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1.6. Make matrix of sBC and read ID\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘03.E-BC_coupling’: File exists\n"
     ]
    }
   ],
   "source": [
    "# From the sBC reads, create table with following content:\n",
    "    # Read ID - sBC\n",
    "mkdir 03.E-BC_coupling\n",
    "paste \\\n",
    "   <( zcat 02.TRIMMED_FQ/${NAME}_R1.Q20_trimmed5_full_filt_sBCq30_filt.fastq.gz | sed -n '1~4p' | awk '{print $1}' | sed 's/^@//' ) \\\n",
    "   <( zcat 02.TRIMMED_FQ/${NAME}_R1.Q20_trimmed5_full_filt_sBCq30_filt.fastq.gz | sed -n '2~4p' ) \\\n",
    "   | sort -k1 \\\n",
    "   | gzip -c \\\n",
    "   > 03.E-BC_coupling/${NAME}_R1_readID-sBC.tsv.gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 2) Read 2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2.1) Filter for sequencing quality\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘02.TRIMMED_FQ’: File exists\n",
      "Read1 before filtering:\n",
      "total reads: 16815676\n",
      "total bases: 807152448\n",
      "Q20 bases: 782369279(96.9296%)\n",
      "Q30 bases: 742605151(92.0031%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 16023549\n",
      "total bases: 769050147\n",
      "Q20 bases: 755922381(98.293%)\n",
      "Q30 bases: 721751408(93.8497%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 16023549\n",
      "reads failed due to low quality: 707715\n",
      "reads failed due to too many N: 118\n",
      "reads failed due to too short: 84294\n",
      "\n",
      "Duplication rate (may be overestimated since this is SE data): 2.27793%\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp --disable_adapter_trimming -q 20 -u 20 -i ../00.RAW_FASTQ/CSE__953401__CHEQseq-cl2_2000enh_1x_R2.fastq.gz -o 02.TRIMMED_FQ/CSE__953401__CHEQseq-cl2_2000enh_1x_R2.Q20.fastq.gz \n",
      "fastp v0.20.0, time used: 54 seconds\n"
     ]
    }
   ],
   "source": [
    "# at least 80% of the NT (-u 20) must have a quality higher then 20 \n",
    "# (same filtering as R1)\n",
    "\n",
    "module load fastp/0.20.0-foss-2018a\n",
    "mkdir 02.TRIMMED_FQ\n",
    "fastp --disable_adapter_trimming -q 20 -u 20 -i ../00.RAW_FASTQ/${NAME}_R2.fastq.gz -o 02.TRIMMED_FQ/${NAME}_R2.Q20.fastq.gz\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2.2) cut 5' adapter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is cutadapt 2.4 with Python 3.7.4\n",
      "Command line parameters: -g AATTAATTCGGGCCCCGGTCC --trimmed-only --output 02.TRIMMED_FQ/CSE__953401__CHEQseq-cl2_2000enh_1x_R2.Q20_trimmed5.fastq.gz 02.TRIMMED_FQ/CSE__953401__CHEQseq-cl2_2000enh_1x_R2.Q20.fastq.gz\n",
      "Processing reads on 1 core in single-end mode ...\n",
      "[   8<-------] 00:03:16    16,023,549 reads  @     12.3 µs/read;   4.90 M reads/minute\n",
      "Finished in 196.40 s (12 us/read; 4.90 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              16,023,549\n",
      "Reads with adapters:                16,018,116 (100.0%)\n",
      "Reads written (passing filters):    16,018,116 (100.0%)\n",
      "\n",
      "Total basepairs processed:   769,050,147 bp\n",
      "Total written (filtered):    432,567,577 bp (56.2%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AATTAATTCGGGCCCCGGTCC; Type: regular 5'; Length: 21; Trimmed: 16018116 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-21 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "4\t2\t62592.0\t0\t2\n",
      "9\t1\t61.1\t0\t1\n",
      "11\t1\t3.8\t1\t1\n",
      "13\t5\t0.2\t1\t5\n",
      "14\t10\t0.1\t1\t4 6\n",
      "15\t20\t0.0\t1\t11 9\n",
      "16\t61\t0.0\t1\t32 29\n",
      "17\t185\t0.0\t1\t164 21\n",
      "18\t422\t0.0\t1\t88 305 29\n",
      "19\t5286\t0.0\t1\t432 2150 2704\n",
      "20\t153330\t0.0\t2\t61452 89611 2267\n",
      "21\t15828279\t0.0\t2\t15597988 216124 14167\n",
      "22\t18435\t0.0\t2\t3426 14322 687\n",
      "23\t4902\t0.0\t2\t34 253 4615\n",
      "24\t7035\t0.0\t2\t6806 144 85\n",
      "25\t33\t0.0\t2\t13 10 10\n",
      "26\t11\t0.0\t2\t3 2 6\n",
      "27\t5\t0.0\t2\t5\n",
      "28\t10\t0.0\t2\t9 1\n",
      "29\t5\t0.0\t2\t3 2\n",
      "30\t7\t0.0\t2\t7\n",
      "31\t3\t0.0\t2\t2 1\n",
      "32\t2\t0.0\t2\t2\n",
      "33\t9\t0.0\t2\t8 1\n",
      "34\t1\t0.0\t2\t0 1\n",
      "35\t3\t0.0\t2\t3\n",
      "36\t2\t0.0\t2\t0 1 1\n",
      "37\t1\t0.0\t2\t1\n",
      "38\t1\t0.0\t2\t1\n",
      "39\t4\t0.0\t2\t4\n",
      "40\t8\t0.0\t2\t8\n",
      "41\t1\t0.0\t2\t1\n",
      "42\t3\t0.0\t2\t2 1\n",
      "43\t4\t0.0\t2\t4\n",
      "44\t4\t0.0\t2\t4\n",
      "45\t4\t0.0\t2\t3 1\n",
      "46\t4\t0.0\t2\t4\n",
      "47\t3\t0.0\t2\t3\n",
      "48\t14\t0.0\t2\t3 5 6\n"
     ]
    }
   ],
   "source": [
    "# cut 5' adaptor\n",
    "# write only the reads that are trimmed, so that contained the adaptor\n",
    "module load cutadapt/2.4-foss-2018a-Python-3.7.4\n",
    "\n",
    "cutadapt -g AATTAATTCGGGCCCCGGTCC --trimmed-only \\\n",
    "--output 02.TRIMMED_FQ/${NAME}_R2.Q20_trimmed5.fastq.gz \\\n",
    "02.TRIMMED_FQ/${NAME}_R2.Q20.fastq.gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2.3) cut 3' adapter  + 17 bp long"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is cutadapt 2.4 with Python 3.7.4\n",
      "Command line parameters: -a GATCGGCGCGCC --trimmed-only -m 17 -M 17 --output 02.TRIMMED_FQ/CSE__953401__CHEQseq-cl2_2000enh_1x_R2.Q20_trimmed5-3_rBC.fastq.gz 02.TRIMMED_FQ/CSE__953401__CHEQseq-cl2_2000enh_1x_R2.Q20_trimmed5.fastq.gz\n",
      "Processing reads on 1 core in single-end mode ...\n",
      "[    8<------] 00:03:59    16,018,116 reads  @     15.0 µs/read;   4.01 M reads/minute\n",
      "Finished in 239.94 s (15 us/read; 4.01 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              16,018,116\n",
      "Reads with adapters:                15,961,896 (99.6%)\n",
      "Reads that were too short:             127,182 (0.8%)\n",
      "Reads that were too long:              139,142 (0.9%)\n",
      "Reads written (passing filters):    15,751,633 (98.3%)\n",
      "\n",
      "Total basepairs processed:   432,567,577 bp\n",
      "Total written (filtered):    267,777,761 bp (61.9%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: GATCGGCGCGCC; Type: regular 3'; Length: 12; Trimmed: 15961896 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-12 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 34.0%\n",
      "  C: 25.8%\n",
      "  G: 17.1%\n",
      "  T: 23.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t1452\t250283.1\t0\t1452\n",
      "4\t1594\t62570.8\t0\t1594\n",
      "5\t1603\t15642.7\t0\t1603\n",
      "6\t1715\t3910.7\t0\t1715\n",
      "7\t9793\t977.7\t0\t9793\n",
      "8\t5855\t244.4\t0\t5855\n",
      "9\t71572\t61.1\t0\t71020 552\n",
      "10\t15603994\t15.3\t1\t15406703 197291\n",
      "11\t239360\t3.8\t1\t208067 31293\n",
      "12\t11269\t1.0\t1\t10238 1031\n",
      "13\t3717\t1.0\t1\t3444 273\n",
      "14\t2270\t1.0\t1\t1792 478\n",
      "15\t1368\t1.0\t1\t973 395\n",
      "16\t1007\t1.0\t1\t773 234\n",
      "17\t902\t1.0\t1\t627 275\n",
      "18\t684\t1.0\t1\t417 267\n",
      "19\t669\t1.0\t1\t470 199\n",
      "20\t492\t1.0\t1\t346 146\n",
      "21\t568\t1.0\t1\t276 292\n",
      "22\t352\t1.0\t1\t202 150\n",
      "23\t380\t1.0\t1\t227 153\n",
      "24\t486\t1.0\t1\t238 248\n",
      "25\t393\t1.0\t1\t185 208\n",
      "26\t149\t1.0\t1\t45 104\n",
      "27\t231\t1.0\t1\t94 137\n",
      "28\t16\t1.0\t1\t16\n",
      "29\t5\t1.0\t1\t5\n"
     ]
    }
   ],
   "source": [
    "# cut 3' adaptor + demand it to be 17bp long\n",
    "\n",
    "cutadapt -a GATCGGCGCGCC --trimmed-only \\\n",
    "-m 17 -M 17 \\\n",
    "--output  02.TRIMMED_FQ/${NAME}_R2.Q20_trimmed5-3_rBC.fastq.gz \\\n",
    "02.TRIMMED_FQ/${NAME}_R2.Q20_trimmed5.fastq.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "@A00311:155:HNLWHDRXX:1:2101:1325:1000 2:N:0:TAATCGAT+GGTTCAAT\n",
      "ACCATCACCATTTCCCC\n",
      "+\n",
      "FFFFFFFFFFFFFFFFF\n",
      "@A00311:155:HNLWHDRXX:1:2101:1922:1000 2:N:0:TAATCGAT+GGTTCAAT\n",
      "CAGAAATAATGGACAGA\n",
      "+\n",
      "FFFFFFFF:FFFFFFFF\n",
      "@A00311:155:HNLWHDRXX:1:2101:3043:1000 2:N:0:TAATCGAT+GGTTCAAT\n",
      "GTGCTGTAAGGCACATT\n",
      "\n",
      "gzip: stdout: Broken pipe\n"
     ]
    }
   ],
   "source": [
    "zcat 02.TRIMMED_FQ/${NAME}_R2.Q20_trimmed5-3_rBC.fastq.gz| head"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2.4) Filter more for sequencing quality"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Detecting adapter sequence for read1...\n",
      "No adapter detected for read1\n",
      "\n",
      "Read1 before filtering:\n",
      "total reads: 15751633\n",
      "total bases: 267777761\n",
      "Q20 bases: 261917290(97.8114%)\n",
      "Q30 bases: 248596761(92.837%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 15470621\n",
      "total bases: 263000243\n",
      "Q20 bases: 258421608(98.2591%)\n",
      "Q30 bases: 246050852(93.5554%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 15470621\n",
      "reads failed due to low quality: 281012\n",
      "reads failed due to too many N: 0\n",
      "reads with adapter trimmed: 0\n",
      "bases trimmed due to adapters: 0\n",
      "\n",
      "Duplication rate (may be overestimated since this is SE data): 0%\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: fastp.html\n",
      "\n",
      "fastp -e 30 --disable_length_filtering -w 8 -o 02.TRIMMED_FQ/CSE__953401__CHEQseq-cl2_2000enh_1x_R2.Q20_trimmed5-3_rBCq30.fastq.gz -i 02.TRIMMED_FQ/CSE__953401__CHEQseq-cl2_2000enh_1x_R2.Q20_trimmed5-3_rBC.fastq.gz \n",
      "fastp v0.20.0, time used: 53 seconds\n"
     ]
    }
   ],
   "source": [
    "# Q30 filtering\n",
    "module load fastp/0.20.0-foss-2018a\n",
    "fastp -e 30 --disable_length_filtering -w 8 \\\n",
    "      -o  02.TRIMMED_FQ/${NAME}_R2.Q20_trimmed5-3_rBCq30.fastq.gz \\\n",
    "      -i  02.TRIMMED_FQ/${NAME}_R2.Q20_trimmed5-3_rBC.fastq.gz\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15470589 17\n",
      "      1 6\n",
      "     24 7\n",
      "      7 8\n"
     ]
    }
   ],
   "source": [
    "# checked if all reads are still 17 bp\n",
    "zcat  02.TRIMMED_FQ/${NAME}_R2.Q20_trimmed5-3_rBCq30.fastq.gz| sed -n '2~4p' | awk '{print length($0);}' | sort | uniq -c"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "# remove reads that are not 17 bp\n",
    "zcat 02.TRIMMED_FQ/${NAME}_R2.Q20_trimmed5-3_rBCq30.fastq.gz| awk 'NR%4==1{a=$0} NR%4==2{b=$0} NR%4==3{c=$0} NR%4==0&&length(b)==17{print a\"\\n\"b\"\\n\"c\"\\n\"$0;}' \\\n",
    "| gzip > 02.TRIMMED_FQ/${NAME}_R2.Q20_trimmed5-3_rBCq30_filt.fastq.gz\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15470589 17\n"
     ]
    }
   ],
   "source": [
    "# checked if all reads are still 17 bp\n",
    "zcat  02.TRIMMED_FQ/${NAME}_R2.Q20_trimmed5-3_rBCq30_filt.fastq.gz| sed -n '2~4p' | awk '{print length($0);}' | sort | uniq -c"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15470589\n"
     ]
    }
   ],
   "source": [
    "zcat 02.TRIMMED_FQ/${NAME}_R2.Q20_trimmed5-3_rBCq30_filt.fastq.gz |  sed -n '2~4p' |wc -l"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2.5) Make matrix of rBC and read ID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "# From the barcode reads,\n",
    "# Create table with following content:\n",
    "# Read ID<TAB> rBC\n",
    "paste \\\n",
    "   <( zcat 02.TRIMMED_FQ/${NAME}_R2.Q20_trimmed5-3_rBCq30_filt.fastq.gz | sed -n '1~4p' | awk '{print $1}' | sed 's/^@//' ) \\\n",
    "   <( zcat 02.TRIMMED_FQ/${NAME}_R2.Q20_trimmed5-3_rBCq30_filt.fastq.gz | sed -n '2~4p' ) \\\n",
    "   | sort -k1 \\\n",
    "   | gzip -c \\\n",
    "   > 03.E-BC_coupling/${NAME}_R2_readID-rBC.tsv.gz\n",
    "   \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "A00311:155:HNLWHDRXX:1:2101:10004:12054\tTGGTTAAACGCAGACCA\n",
      "A00311:155:HNLWHDRXX:1:2101:10004:12774\tAAACATTGTCCAACGAT\n",
      "A00311:155:HNLWHDRXX:1:2101:10004:13996\tACACCTCCGCCTCACCT\n",
      "A00311:155:HNLWHDRXX:1:2101:10004:16532\tGAGTCTTACCCAAACAT\n",
      "A00311:155:HNLWHDRXX:1:2101:10004:19163\tACCACTAATCATTTAGT\n",
      "A00311:155:HNLWHDRXX:1:2101:10004:20729\tTGCCCAACTCACTCTCC\n",
      "A00311:155:HNLWHDRXX:1:2101:10004:23359\tCAGGCCTTACCAACACG\n",
      "A00311:155:HNLWHDRXX:1:2101:10004:23672\tAGGAAAAGAGGGGCTAA\n",
      "A00311:155:HNLWHDRXX:1:2101:10004:24048\tTGATATTGTTATCACAA\n",
      "A00311:155:HNLWHDRXX:1:2101:10004:24612\tCCACCAGGACTGCCTCC\n",
      "\n",
      "gzip: stdout: Broken pipe\n"
     ]
    }
   ],
   "source": [
    "zcat  03.E-BC_coupling/${NAME}_R2_readID-rBC.tsv.gz| head"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 3) Linking rBC with sBC"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- (needed further bellow) Matrix with Enhancer ID and the sBC: /user/leuven/317/vsc31703/stg_00002/Enhancer_design/Cloning/259bp_fastas/All_sequences_ID-sBC.tsv\n",
    "- Matrix with read ID and sBC from R1: 03.E-BC_coupling/${NAME}_R1_readID-sBC.tsv.gz\n",
    "\n",
    "- Matrix with read ID and rBC from R2: 03.E-BC_coupling/${NAME}_R2_readID-rBC.tsv.gz\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Merge the sBC and rBC using the read IDs\n",
    "    # Importantly: sort on the read IDs\n",
    "    # ID - BC - coord - real seq\n",
    "    # Join together based on read ID (-j 1 because in both files the ID is in the first column)\n",
    "    # Specify the order of the columns you want to be printed\n",
    "join -j 1 -o 1.1,1.2,2.2 \\\n",
    "<(sort -k1 <(zcat 03.E-BC_coupling/${NAME}_R1_readID-sBC.tsv.gz)) \\\n",
    "<(sort -k1 <(zcat 03.E-BC_coupling/${NAME}_R2_readID-rBC.tsv.gz)) \\\n",
    "| tr \" \" \"\\t\" > 03.E-BC_coupling/${NAME}_readID_sBC_rBC_coupled.matrix\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "cat 03.E-BC_coupling/${NAME}_readID_sBC_rBC_coupled.matrix | sort -k2 > 03.E-BC_coupling/${NAME}_readID_sBC_rBC_coupled.sorted.matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15298304\n"
     ]
    }
   ],
   "source": [
    "# number of rBC-sBC links\n",
    "cat 03.E-BC_coupling/${NAME}_readID_sBC_rBC_coupled.sorted.matrix | wc -l"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2489241\n"
     ]
    }
   ],
   "source": [
    "# Number of unique rBCs (matrix has readID - sBC - rBC)\n",
    "cat 03.E-BC_coupling/${NAME}_readID_sBC_rBC_coupled.sorted.matrix| cut -f3 | sort | uniq | wc -l"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 4) Remove from the empty-rBCs lost that are also the ones linked to non-empty plasmid reads -> soleley empty-plasmid\n",
    "\n",
    "For the reads linked to an empty plasmid, we just want to couple these read IDs from R1 to R2. Then we want to remove within these the rBCs that are also linked to non-empty plasmid reads."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 4.1. Make matrix of sBC and read ID for empty plasmids\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘03.E-BC_coupling’: File exists\n"
     ]
    }
   ],
   "source": [
    "# From the sBC reads, create table with following content:\n",
    "    # Read ID - sBC\n",
    "mkdir 03.E-BC_coupling\n",
    "paste \\\n",
    "   <( zcat 02.TRIMMED_FQ/${NAME}_R1.Q20_trimmed5_emptyVector.fastq.gz  | sed -n '1~4p' | awk '{print $1}' | sed 's/^@//' ) \\\n",
    "   <( zcat 02.TRIMMED_FQ/${NAME}_R1.Q20_trimmed5_emptyVector.fastq.gz  | sed -n '2~4p' ) \\\n",
    "   | sort -k1 \\\n",
    "   | gzip -c \\\n",
    "   > 03.E-BC_coupling/${NAME}_R1_empty_readID_rest.tsv.gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 4.2. Linking rBC with sBC - read ID for empty plasmids"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Merge the sBC and rBC using the read IDs\n",
    "     # we will only print the readID and the rBC\n",
    "join -j 1 -o 1.1,2.2 \\\n",
    "<(sort -k1 <(zcat 03.E-BC_coupling/${NAME}_R1_empty_readID_rest.tsv.gz)) \\\n",
    "<(sort -k1 <(zcat 03.E-BC_coupling/${NAME}_R2_readID-rBC.tsv.gz)) \\\n",
    "| tr \" \" \"\\t\" > 03.E-BC_coupling/${NAME}_empty_readID_rest_rBC_coupled.matrix\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 4.3. Remove the normal rBCs (linked to non-empty) from previous matrix from these empty-coupled rBCs to get solely-empty-coupled rBCs\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Read ID both in first column; the empty plasmid file only contains the coupled rBC, the other file readID_sBC_rBC\n",
    "join -1 2 -2 3 -o 1.1,1.2 -v1 \\\n",
    "<(sort -k2 <(cat 03.E-BC_coupling/${NAME}_empty_readID_rest_rBC_coupled.matrix)) \\\n",
    "<(sort -k3 <(cat 03.E-BC_coupling/${NAME}_readID_sBC_rBC_coupled.sorted.matrix)) \\\n",
    "| tr \" \" \"\\t\" > 03.E-BC_coupling/${NAME}_solely-empty_readID_rest_rBC_coupled.matrix\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1937 03.E-BC_coupling/CSE__953401__CHEQseq-cl2_2000enh_1x_solely-empty_readID_rest_rBC_coupled.matrix\n"
     ]
    }
   ],
   "source": [
    "wc -l 03.E-BC_coupling/${NAME}_solely-empty_readID_rest_rBC_coupled.matrix"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 5) Create final rBC - enhancer file\n",
    "\n",
    "Now we have a matrix with readID_sBC_rBC in which the rBCs also linked to empty plasmid are removed.\n",
    "\n",
    "We will now do the final merge:\n",
    "- merge the readID_sBC_rBC  matrix with the enhancerID_sBC matrix\n",
    "- then we keep only the unique couples, so PCR duplicates are removed\n",
    "- then we will remove any rBC that is linked to multiple enhancers"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 5.1. Link back sBC to enhancer"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import csv\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load data\n",
    "enh = pd.read_csv('/user/leuven/317/vsc31703/stg_00002/Enhancer_design/Cloning/259bp_fastas/All_sequences_ID-sBC.tsv', sep='\\t', header=None,names=('enhancerID','sBC'))\n",
    "#bcs = pd.read_csv('/user/leuven/317/vsc31703/stg_00002/Enhancer_design/MPRA/CHEQ-seq_cloning2/E-BC_coupling/03.E-BC_coupling/CSE__88dc35__CHEQseq-cl2_2000enh_3x_readID_sBC_rBC_coupled.sorted.matrix', sep='\\t', header=None,names=('readID','sBC','rBC'))\n",
    "bcs = pd.read_csv('/user/leuven/317/vsc31703/stg_00002/Enhancer_design/MPRA/CHEQ-seq_cloning2/E-BC_coupling/03.E-BC_coupling/CSE__953401__CHEQseq-cl2_2000enh_1x_readID_sBC_rBC_coupled.sorted.matrix', sep='\\t', header=None,names=('readID','sBC','rBC'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Merge: link every sBC instance in the R1 (because there are many intances of the same sBC linked to different rBCs) back to its correct enhancer via the sBC\n",
    "result = pd.merge(enh, bcs, on='sBC')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Reverse complement the rBCs\n",
    "result['rBC_RC'] = result['rBC'].apply(lambda x: x.replace('A', 'temp').replace('T', 'A').replace('temp', 'T').replace('G', 'temp').replace('C','G').replace('temp','C')[::-1])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>enhancerID</th>\n",
       "      <th>sBC</th>\n",
       "      <th>readID</th>\n",
       "      <th>rBC</th>\n",
       "      <th>rBC_RC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>A_group1_seq1_chr1:17027607-17028107</td>\n",
       "      <td>AACACACGCAA</td>\n",
       "      <td>A00311:155:HNLWHDRXX:2:2176:12382:36839</td>\n",
       "      <td>AAAAAAACCGGCATGAA</td>\n",
       "      <td>TTCATGCCGGTTTTTTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>A_group1_seq1_chr1:17027607-17028107</td>\n",
       "      <td>AACACACGCAA</td>\n",
       "      <td>A00311:155:HNLWHDRXX:1:2213:27688:7420</td>\n",
       "      <td>AAAAAAACGACCCAACG</td>\n",
       "      <td>CGTTGGGTCGTTTTTTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>A_group1_seq1_chr1:17027607-17028107</td>\n",
       "      <td>AACACACGCAA</td>\n",
       "      <td>A00311:155:HNLWHDRXX:2:2178:28031:13651</td>\n",
       "      <td>AAAAAAACGGGTACATT</td>\n",
       "      <td>AATGTACCCGTTTTTTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>A_group1_seq1_chr1:17027607-17028107</td>\n",
       "      <td>AACACACGCAA</td>\n",
       "      <td>A00311:155:HNLWHDRXX:2:2178:29288:9314</td>\n",
       "      <td>AAAAAAACGGGTACATT</td>\n",
       "      <td>AATGTACCCGTTTTTTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>A_group1_seq1_chr1:17027607-17028107</td>\n",
       "      <td>AACACACGCAA</td>\n",
       "      <td>A00311:155:HNLWHDRXX:2:2243:32289:31234</td>\n",
       "      <td>AAAAAAAGTATAAGACA</td>\n",
       "      <td>TGTCTTATACTTTTTTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14827270</td>\n",
       "      <td>S5_MES_TEAD</td>\n",
       "      <td>CTGTTAATGGA</td>\n",
       "      <td>A00311:155:HNLWHDRXX:1:2207:13431:1172</td>\n",
       "      <td>TTTTGCTCGGTTTCGAC</td>\n",
       "      <td>GTCGAAACCGAGCAAAA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14827271</td>\n",
       "      <td>S5_MES_TEAD</td>\n",
       "      <td>CTGTTAATGGA</td>\n",
       "      <td>A00311:155:HNLWHDRXX:2:2136:9489:32236</td>\n",
       "      <td>TTTTGCTCGGTTTCGAC</td>\n",
       "      <td>GTCGAAACCGAGCAAAA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14827272</td>\n",
       "      <td>S5_MES_TEAD</td>\n",
       "      <td>CTGTTAATGGA</td>\n",
       "      <td>A00311:155:HNLWHDRXX:2:2218:7238:13933</td>\n",
       "      <td>TTTTGCTCGGTTTCGAC</td>\n",
       "      <td>GTCGAAACCGAGCAAAA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14827273</td>\n",
       "      <td>S5_MES_TEAD</td>\n",
       "      <td>CTGTTAATGGA</td>\n",
       "      <td>A00311:155:HNLWHDRXX:1:2174:9661:27524</td>\n",
       "      <td>TTTTTCGAATGTCAATC</td>\n",
       "      <td>GATTGACATTCGAAAAA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14827274</td>\n",
       "      <td>S5_MES_TEAD</td>\n",
       "      <td>CTGTTAATGGA</td>\n",
       "      <td>A00311:155:HNLWHDRXX:1:2271:22498:23077</td>\n",
       "      <td>TTTTTTTGTAGAATAAC</td>\n",
       "      <td>GTTATTCTACAAAAAAA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>14827275 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                    enhancerID          sBC  \\\n",
       "0         A_group1_seq1_chr1:17027607-17028107  AACACACGCAA   \n",
       "1         A_group1_seq1_chr1:17027607-17028107  AACACACGCAA   \n",
       "2         A_group1_seq1_chr1:17027607-17028107  AACACACGCAA   \n",
       "3         A_group1_seq1_chr1:17027607-17028107  AACACACGCAA   \n",
       "4         A_group1_seq1_chr1:17027607-17028107  AACACACGCAA   \n",
       "...                                        ...          ...   \n",
       "14827270                           S5_MES_TEAD  CTGTTAATGGA   \n",
       "14827271                           S5_MES_TEAD  CTGTTAATGGA   \n",
       "14827272                           S5_MES_TEAD  CTGTTAATGGA   \n",
       "14827273                           S5_MES_TEAD  CTGTTAATGGA   \n",
       "14827274                           S5_MES_TEAD  CTGTTAATGGA   \n",
       "\n",
       "                                           readID                rBC  \\\n",
       "0         A00311:155:HNLWHDRXX:2:2176:12382:36839  AAAAAAACCGGCATGAA   \n",
       "1          A00311:155:HNLWHDRXX:1:2213:27688:7420  AAAAAAACGACCCAACG   \n",
       "2         A00311:155:HNLWHDRXX:2:2178:28031:13651  AAAAAAACGGGTACATT   \n",
       "3          A00311:155:HNLWHDRXX:2:2178:29288:9314  AAAAAAACGGGTACATT   \n",
       "4         A00311:155:HNLWHDRXX:2:2243:32289:31234  AAAAAAAGTATAAGACA   \n",
       "...                                           ...                ...   \n",
       "14827270   A00311:155:HNLWHDRXX:1:2207:13431:1172  TTTTGCTCGGTTTCGAC   \n",
       "14827271   A00311:155:HNLWHDRXX:2:2136:9489:32236  TTTTGCTCGGTTTCGAC   \n",
       "14827272   A00311:155:HNLWHDRXX:2:2218:7238:13933  TTTTGCTCGGTTTCGAC   \n",
       "14827273   A00311:155:HNLWHDRXX:1:2174:9661:27524  TTTTTCGAATGTCAATC   \n",
       "14827274  A00311:155:HNLWHDRXX:1:2271:22498:23077  TTTTTTTGTAGAATAAC   \n",
       "\n",
       "                     rBC_RC  \n",
       "0         TTCATGCCGGTTTTTTT  \n",
       "1         CGTTGGGTCGTTTTTTT  \n",
       "2         AATGTACCCGTTTTTTT  \n",
       "3         AATGTACCCGTTTTTTT  \n",
       "4         TGTCTTATACTTTTTTT  \n",
       "...                     ...  \n",
       "14827270  GTCGAAACCGAGCAAAA  \n",
       "14827271  GTCGAAACCGAGCAAAA  \n",
       "14827272  GTCGAAACCGAGCAAAA  \n",
       "14827273  GATTGACATTCGAAAAA  \n",
       "14827274  GTTATTCTACAAAAAAA  \n",
       "\n",
       "[14827275 rows x 5 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1998"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Number of unique enhancers found back in library\n",
    "len(result.enhancerID.unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2456513"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Number of unique rBCs found back in library\n",
    "len(result.rBC.unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "#result.to_csv(\"/user/leuven/317/vsc31703/stg_00002/Enhancer_design/MPRA/CHEQ-seq_cloning2/E-BC_coupling/03.E-BC_coupling/CSE__88dc35__CHEQseq-cl2_2000enh_3x_enhancerID_sBC_rBC_coupled.matrix\", sep=\"\\t\", quoting=csv.QUOTE_NONE, header=False, index=False)\n",
    "result.to_csv(\"/user/leuven/317/vsc31703/stg_00002/Enhancer_design/MPRA/CHEQ-seq_cloning2/E-BC_coupling/03.E-BC_coupling/CSE__953401__CHEQseq-cl2_2000enh_1x_enhancerID_sBC_rBC_coupled.matrix\", sep=\"\\t\", quoting=csv.QUOTE_NONE, header=False, index=False)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 5.2. Keep only unique rBC-enhancer couples (so remove PCR duplicates)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# keep only unique couples (so remove PCR duplicates)\n",
    "cat 03.E-BC_coupling/${NAME}_enhancerID_sBC_rBC_coupled.matrix |cut -f1,5 > 03.E-BC_coupling/${NAME}_enhancerID_sBC_rBC_coupled.format.matrix\n",
    "\n",
    "cat 03.E-BC_coupling/${NAME}_enhancerID_sBC_rBC_coupled.format.matrix \\\n",
    "   | sort -k1 \\\n",
    "   | uniq \\\n",
    "   > 03.E-BC_coupling/${NAME}_enhancerID_sBC_rBC_coupled.format_unique.matrix\n",
    "\n",
    "cat 03.E-BC_coupling/${NAME}_enhancerID_sBC_rBC_coupled.format_unique.matrix| awk '{print $2 \".\" $1}' | sort |tr \".\" \"\\t\"  > 03.E-BC_coupling/${NAME}_enhancerID_sBC_rBC_coupled.format_unique.sorted.matrix\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5380674 03.E-BC_coupling/CSE__953401__CHEQseq-cl2_2000enh_1x_enhancerID_sBC_rBC_coupled.format_unique.sorted.matrix\n"
     ]
    }
   ],
   "source": [
    "wc -l 03.E-BC_coupling/${NAME}_enhancerID_sBC_rBC_coupled.format_unique.sorted.matrix"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 5.3. Keep only rBCs linked to only one enhancer\n",
    "\n",
    "Because in the previous step, we already removed the PCR duplicates, now every time a rBCs occurs more than once it is linked with a different enhancer. And since the matrix is ordered according to the rBCs, we will just print here only the rows for which no two following lines are the same."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BC</th>\n",
       "      <th>Enhancer</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>AAAAAAAAAAATGGATA</td>\n",
       "      <td>S5_MES_AP1_TEAD-5prime-6bp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>AAAAAAAAAAGCCAGTT</td>\n",
       "      <td>B1_group1_seq29_chr17:80513676-80514177</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>AAAAAAAAAAGCCAGTT</td>\n",
       "      <td>C_group1_seq86_chr4:130350980-130351481</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>AAAAAAAAAATATCCTT</td>\n",
       "      <td>A_group2_seq164_chr15:84852643-84853143</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>AAAAAAAAAATATCCTT</td>\n",
       "      <td>B2_group1_seq85_chr15:89084235-89084736</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5380669</td>\n",
       "      <td>TTTTTTTTTTTTTGAGT</td>\n",
       "      <td>C_group1_seq18_chr2:150384868-150385369</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5380670</td>\n",
       "      <td>TTTTTTTTTTTTTTACC</td>\n",
       "      <td>A_group1_seq18_chr11:49166081-49166581</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5380671</td>\n",
       "      <td>TTTTTTTTTTTTTTACC</td>\n",
       "      <td>A_group1_seq1_chr1:17027607-17028107</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5380672</td>\n",
       "      <td>TTTTTTTTTTTTTTACC</td>\n",
       "      <td>A_group3_seq15_chr10:47749356-47749856</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5380673</td>\n",
       "      <td>TTTTTTTTTTTTTTAGC</td>\n",
       "      <td>Random_sequence_trial3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5380674 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                        BC                                 Enhancer\n",
       "0        AAAAAAAAAAATGGATA               S5_MES_AP1_TEAD-5prime-6bp\n",
       "1        AAAAAAAAAAGCCAGTT  B1_group1_seq29_chr17:80513676-80514177\n",
       "2        AAAAAAAAAAGCCAGTT  C_group1_seq86_chr4:130350980-130351481\n",
       "3        AAAAAAAAAATATCCTT  A_group2_seq164_chr15:84852643-84853143\n",
       "4        AAAAAAAAAATATCCTT  B2_group1_seq85_chr15:89084235-89084736\n",
       "...                    ...                                      ...\n",
       "5380669  TTTTTTTTTTTTTGAGT  C_group1_seq18_chr2:150384868-150385369\n",
       "5380670  TTTTTTTTTTTTTTACC   A_group1_seq18_chr11:49166081-49166581\n",
       "5380671  TTTTTTTTTTTTTTACC     A_group1_seq1_chr1:17027607-17028107\n",
       "5380672  TTTTTTTTTTTTTTACC   A_group3_seq15_chr10:47749356-47749856\n",
       "5380673  TTTTTTTTTTTTTTAGC                   Random_sequence_trial3\n",
       "\n",
       "[5380674 rows x 2 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import csv\n",
    "#df = pd.read_csv('/user/leuven/317/vsc31703/stg_00002/Enhancer_design/MPRA/CHEQ-seq_cloning2/E-BC_coupling/03.E-BC_coupling/CSE__88dc35__CHEQseq-cl2_2000enh_3x_enhancerID_sBC_rBC_coupled.format_unique.sorted.matrix', sep='\\t', header=None,names=('BC','Enhancer'))\n",
    "df = pd.read_csv('/user/leuven/317/vsc31703/stg_00002/Enhancer_design/MPRA/CHEQ-seq_cloning2/E-BC_coupling/03.E-BC_coupling/CSE__953401__CHEQseq-cl2_2000enh_1x_enhancerID_sBC_rBC_coupled.format_unique.sorted.matrix', sep='\\t', header=None,names=('BC','Enhancer'))\n",
    "\n",
    "df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "5380674"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "list =[]\n",
    "list.append(\"a\"\n",
    "        )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "100000\n",
      "200000\n",
      "300000\n",
      "400000\n",
      "500000\n",
      "600000\n",
      "700000\n",
      "800000\n",
      "900000\n",
      "1000000\n",
      "1100000\n",
      "1200000\n",
      "1300000\n",
      "1400000\n",
      "1500000\n",
      "1600000\n",
      "1700000\n",
      "1800000\n",
      "1900000\n",
      "2000000\n",
      "2100000\n",
      "2200000\n",
      "2300000\n",
      "2400000\n",
      "2500000\n",
      "2600000\n",
      "2700000\n",
      "2800000\n",
      "2900000\n",
      "3000000\n",
      "3100000\n",
      "3200000\n",
      "3300000\n",
      "3400000\n",
      "3500000\n",
      "3600000\n",
      "3700000\n",
      "3800000\n",
      "3900000\n",
      "4000000\n",
      "4100000\n",
      "4200000\n",
      "4300000\n",
      "4400000\n",
      "4500000\n",
      "4600000\n",
      "4700000\n",
      "4800000\n",
      "4900000\n",
      "5000000\n",
      "5100000\n",
      "5200000\n",
      "5300000\n"
     ]
    }
   ],
   "source": [
    "# Print a list where only the rBCs that are coupled with only 1 enhancer are  written\n",
    "lastBC=\"\"\n",
    "tempList = []\n",
    "ToDeletPrevious=False\n",
    "i = 0\n",
    "initalLength = df.shape[0]\n",
    "while i < initalLength:\n",
    "    if df.loc[i]['BC'] == lastBC:\n",
    "        #if ToDeletPrevious==True:\n",
    "            #df.drop(i-1)\n",
    "            #i = i - 1\n",
    "        lastBC=df.loc[i]['BC']\n",
    "        #df.drop(i)\n",
    "        ToDeletPrevious=False\n",
    "        #i = i -1\n",
    "    else:\n",
    "        if ToDeletPrevious ==True:\n",
    "            tempList.append(df.loc[i])\n",
    "        ToDeletPrevious=True\n",
    "        lastBC=df.loc[i]['BC']\n",
    "    i = i +1\n",
    "    if i % 100000 == 0:\n",
    "        print(i)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1094441"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(tempList)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "myarray = np.array(tempList)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([['AAAAAAAAAAGCCAGTT', 'B1_group1_seq29_chr17:80513676-80514177'],\n",
       "       ['AAAAAAAAAGATCCTGA', 'A_group4_seq26_chr7:44008156-44008656'],\n",
       "       ['AAAAAAAAATCGATTAT', 'A_group1_seq45_chr20:24227971-24228471'],\n",
       "       ...,\n",
       "       ['TTTTTTTTTTTTTACCG', 'B1_group1_seq106_chr11:67363550-67364051'],\n",
       "       ['TTTTTTTTTTTTTAGTA', 'A_group2_seq137_chr11:118597666-118598166'],\n",
       "       ['TTTTTTTTTTTTTATGA', 'B1_group5_seq149_chr16:21401190-21401539']],\n",
       "      dtype=object)"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "myarray"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# This is the one-to-one matrix\n",
    "\n",
    "#np.savetxt('/user/leuven/317/vsc31703/stg_00002/Enhancer_design/MPRA/CHEQ-seq_cloning2/E-BC_coupling/03.E-BC_coupling/CSE__88dc35__CHEQseq-cl2_2000enh_3x_enhancerID_sBC_rBC_coupled.format_unique.sorted_1to1.matrix', myarray, fmt=\"%s\",delimiter='\\t')\n",
    "  \n",
    "np.savetxt('/user/leuven/317/vsc31703/stg_00002/Enhancer_design/MPRA/CHEQ-seq_cloning2/E-BC_coupling/03.E-BC_coupling/CSE__953401__CHEQseq-cl2_2000enh_1x_enhancerID_sBC_rBC_coupled.format_unique.sorted_1to1.matrix', myarray, fmt=\"%s\",delimiter='\\t')\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# = Final one-to-one matrix of rBC - enhancer coupled without empty plasmdids"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Check the distriution of rBC per enhancer"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "#m <- read.table(\"/user/leuven/317/vsc31703/stg_00002/Enhancer_design/MPRA/CHEQ-seq_cloning2/E-BC_coupling/03.E-BC_coupling/CSE__953401__CHEQseq-cl2_2000enh_1x_enhancerID_sBC_rBC_coupled.format_unique.sorted_1to1.matrix\", header =F)\n",
    "m <- read.table(\"/user/leuven/317/vsc31703/stg_00002/Enhancer_design/MPRA/CHEQ-seq_cloning2/E-BC_coupling/03.E-BC_coupling/CSE__88dc35__CHEQseq-cl2_2000enh_3x_enhancerID_sBC_rBC_coupled.format_unique.sorted_1to1.matrix\", header =F)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "colnames(m) <- c('rBC','Enhancer')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 6 x 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>rBC</th><th scope=col>Enhancer</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>AAAAAAAAAACAGATAA</td><td>B1_group3_seq12_chr1:185226215-185226716</td></tr>\n",
       "\t<tr><td>AAAAAAAAAACCAATAA</td><td>A_group2_seq153_chr14:46052209-46052709 </td></tr>\n",
       "\t<tr><td>AAAAAAAAAGAGTAGCC</td><td>B2_group1_seq211_chr12:31330453-31330954</td></tr>\n",
       "\t<tr><td>AAAAAAAAAGCAGATGG</td><td>A_group2_seq149_chr13:53495666-53496166 </td></tr>\n",
       "\t<tr><td>AAAAAAAAAGGGCTGAT</td><td>B1_group5_seq12_chr6:41537231-41538539  </td></tr>\n",
       "\t<tr><td>AAAAAAAAAGTGGCTGT</td><td>A_group1_seq1_chr1:17027607-17028107    </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 x 2\n",
       "\\begin{tabular}{r|ll}\n",
       " rBC & Enhancer\\\\\n",
       " <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\t AAAAAAAAAACAGATAA & B1\\_group3\\_seq12\\_chr1:185226215-185226716\\\\\n",
       "\t AAAAAAAAAACCAATAA & A\\_group2\\_seq153\\_chr14:46052209-46052709 \\\\\n",
       "\t AAAAAAAAAGAGTAGCC & B2\\_group1\\_seq211\\_chr12:31330453-31330954\\\\\n",
       "\t AAAAAAAAAGCAGATGG & A\\_group2\\_seq149\\_chr13:53495666-53496166 \\\\\n",
       "\t AAAAAAAAAGGGCTGAT & B1\\_group5\\_seq12\\_chr6:41537231-41538539  \\\\\n",
       "\t AAAAAAAAAGTGGCTGT & A\\_group1\\_seq1\\_chr1:17027607-17028107    \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 x 2\n",
       "\n",
       "| rBC &lt;fct&gt; | Enhancer &lt;fct&gt; |\n",
       "|---|---|\n",
       "| AAAAAAAAAACAGATAA | B1_group3_seq12_chr1:185226215-185226716 |\n",
       "| AAAAAAAAAACCAATAA | A_group2_seq153_chr14:46052209-46052709  |\n",
       "| AAAAAAAAAGAGTAGCC | B2_group1_seq211_chr12:31330453-31330954 |\n",
       "| AAAAAAAAAGCAGATGG | A_group2_seq149_chr13:53495666-53496166  |\n",
       "| AAAAAAAAAGGGCTGAT | B1_group5_seq12_chr6:41537231-41538539   |\n",
       "| AAAAAAAAAGTGGCTGT | A_group1_seq1_chr1:17027607-17028107     |\n",
       "\n"
      ],
      "text/plain": [
       "  rBC               Enhancer                                \n",
       "1 AAAAAAAAAACAGATAA B1_group3_seq12_chr1:185226215-185226716\n",
       "2 AAAAAAAAAACCAATAA A_group2_seq153_chr14:46052209-46052709 \n",
       "3 AAAAAAAAAGAGTAGCC B2_group1_seq211_chr12:31330453-31330954\n",
       "4 AAAAAAAAAGCAGATGG A_group2_seq149_chr13:53495666-53496166 \n",
       "5 AAAAAAAAAGGGCTGAT B1_group5_seq12_chr6:41537231-41538539  \n",
       "6 AAAAAAAAAGTGGCTGT A_group1_seq1_chr1:17027607-17028107    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>3427605</li>\n",
       "\t<li>2</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 3427605\n",
       "\\item 2\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 3427605\n",
       "2. 2\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 3427605       2"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(m)\n",
    "dim(m)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Calculate number of rBCs per enhancer (as each rBC only appears once as it can only be coupled to 1 enhancer, we can just count the frequencies each enhancer occurs)\n",
    "x <- as.data.frame(table(m$Enhancer))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 6 x 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Var1</th><th scope=col>Freq</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>A_group1_seq100_chr9:96385969-96386469  </td><td>4639</td></tr>\n",
       "\t<tr><td>A_group1_seq101_chr9:140681098-140681598</td><td> 647</td></tr>\n",
       "\t<tr><td>A_group1_seq102_chrX:3845338-3845838    </td><td>1006</td></tr>\n",
       "\t<tr><td>A_group1_seq103_chrX:9827376-9827876    </td><td>4136</td></tr>\n",
       "\t<tr><td>A_group1_seq104_chrX:14320175-14320675  </td><td>2371</td></tr>\n",
       "\t<tr><td>A_group1_seq105_chrX:81125209-81125709  </td><td>2372</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 x 2\n",
       "\\begin{tabular}{r|ll}\n",
       " Var1 & Freq\\\\\n",
       " <fct> & <int>\\\\\n",
       "\\hline\n",
       "\t A\\_group1\\_seq100\\_chr9:96385969-96386469   & 4639\\\\\n",
       "\t A\\_group1\\_seq101\\_chr9:140681098-140681598 &  647\\\\\n",
       "\t A\\_group1\\_seq102\\_chrX:3845338-3845838     & 1006\\\\\n",
       "\t A\\_group1\\_seq103\\_chrX:9827376-9827876     & 4136\\\\\n",
       "\t A\\_group1\\_seq104\\_chrX:14320175-14320675   & 2371\\\\\n",
       "\t A\\_group1\\_seq105\\_chrX:81125209-81125709   & 2372\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 x 2\n",
       "\n",
       "| Var1 &lt;fct&gt; | Freq &lt;int&gt; |\n",
       "|---|---|\n",
       "| A_group1_seq100_chr9:96385969-96386469   | 4639 |\n",
       "| A_group1_seq101_chr9:140681098-140681598 |  647 |\n",
       "| A_group1_seq102_chrX:3845338-3845838     | 1006 |\n",
       "| A_group1_seq103_chrX:9827376-9827876     | 4136 |\n",
       "| A_group1_seq104_chrX:14320175-14320675   | 2371 |\n",
       "| A_group1_seq105_chrX:81125209-81125709   | 2372 |\n",
       "\n"
      ],
      "text/plain": [
       "  Var1                                     Freq\n",
       "1 A_group1_seq100_chr9:96385969-96386469   4639\n",
       "2 A_group1_seq101_chr9:140681098-140681598  647\n",
       "3 A_group1_seq102_chrX:3845338-3845838     1006\n",
       "4 A_group1_seq103_chrX:9827376-9827876     4136\n",
       "5 A_group1_seq104_chrX:14320175-14320675   2371\n",
       "6 A_group1_seq105_chrX:81125209-81125709   2372"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>1998</li>\n",
       "\t<li>2</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 1998\n",
       "\\item 2\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 1998\n",
       "2. 2\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 1998    2"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(x)\n",
    "dim(x)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. \n",
       "      6     647    1156    1716    2165   27276 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "summary(x$Freq)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 20 x 2</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Var1</th><th scope=col>Freq</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1384</th><td>B2_group3_seq131_chr9:43376634-43376970             </td><td> 6</td></tr>\n",
       "\t<tr><th scope=row>546</th><td>B1_group1_seq212_chr1:232468422-232468923           </td><td>14</td></tr>\n",
       "\t<tr><th scope=row>1002</th><td>B1_group5_seq51_chr2:133025518-133026407            </td><td>16</td></tr>\n",
       "\t<tr><th scope=row>1371</th><td>B2_group3_seq116_chr2:133025518-133026407           </td><td>20</td></tr>\n",
       "\t<tr><th scope=row>1203</th><td>B2_group1_seq245_chr1:159017399-159017900           </td><td>30</td></tr>\n",
       "\t<tr><th scope=row>1752</th><td>Neg-Control_B2_group3_seq80_chr11:175385-175963     </td><td>31</td></tr>\n",
       "\t<tr><th scope=row>898</th><td>B1_group5_seq189_chr13:28033564-28034043            </td><td>36</td></tr>\n",
       "\t<tr><th scope=row>1282</th><td>B2_group1_seq66_chr13:25528701-25529202             </td><td>37</td></tr>\n",
       "\t<tr><th scope=row>697</th><td>B1_group2_seq17_chr13:25528701-25529202             </td><td>38</td></tr>\n",
       "\t<tr><th scope=row>604</th><td>B1_group1_seq53_chr16:58539224-58539725             </td><td>39</td></tr>\n",
       "\t<tr><th scope=row>662</th><td>B1_group2_seq108_chr1:162698935-162699436           </td><td>44</td></tr>\n",
       "\t<tr><th scope=row>116</th><td>A_group2_seq109_chr9:47216611-47217111              </td><td>61</td></tr>\n",
       "\t<tr><th scope=row>573</th><td>B1_group1_seq25_chr1:39887435-39887936              </td><td>68</td></tr>\n",
       "\t<tr><th scope=row>1346</th><td>B2_group2_seq34_chr3:10394211-10395104              </td><td>78</td></tr>\n",
       "\t<tr><th scope=row>1158</th><td>B2_group1_seq204_chr22:48100065-48100566            </td><td>82</td></tr>\n",
       "\t<tr><th scope=row>767</th><td>B1_group2_seq80_chr14:20544265-20544766             </td><td>84</td></tr>\n",
       "\t<tr><th scope=row>1581</th><td>Neg-Control_A_group1_seq84_chr7:1056602-1057102     </td><td>88</td></tr>\n",
       "\t<tr><th scope=row>1344</th><td>B2_group2_seq32_chr7:143489706-143490207            </td><td>91</td></tr>\n",
       "\t<tr><th scope=row>1653</th><td>Neg-Control_B1_group1_seq189_chr20:60750743-60751244</td><td>91</td></tr>\n",
       "\t<tr><th scope=row>1783</th><td>S4_MEL_SOX_MITF-3prime-50bp                         </td><td>92</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20 x 2\n",
       "\\begin{tabular}{r|ll}\n",
       "  & Var1 & Freq\\\\\n",
       "  & <fct> & <int>\\\\\n",
       "\\hline\n",
       "\t1384 & B2\\_group3\\_seq131\\_chr9:43376634-43376970              &  6\\\\\n",
       "\t546 & B1\\_group1\\_seq212\\_chr1:232468422-232468923            & 14\\\\\n",
       "\t1002 & B1\\_group5\\_seq51\\_chr2:133025518-133026407             & 16\\\\\n",
       "\t1371 & B2\\_group3\\_seq116\\_chr2:133025518-133026407            & 20\\\\\n",
       "\t1203 & B2\\_group1\\_seq245\\_chr1:159017399-159017900            & 30\\\\\n",
       "\t1752 & Neg-Control\\_B2\\_group3\\_seq80\\_chr11:175385-175963      & 31\\\\\n",
       "\t898 & B1\\_group5\\_seq189\\_chr13:28033564-28034043             & 36\\\\\n",
       "\t1282 & B2\\_group1\\_seq66\\_chr13:25528701-25529202              & 37\\\\\n",
       "\t697 & B1\\_group2\\_seq17\\_chr13:25528701-25529202              & 38\\\\\n",
       "\t604 & B1\\_group1\\_seq53\\_chr16:58539224-58539725              & 39\\\\\n",
       "\t662 & B1\\_group2\\_seq108\\_chr1:162698935-162699436            & 44\\\\\n",
       "\t116 & A\\_group2\\_seq109\\_chr9:47216611-47217111               & 61\\\\\n",
       "\t573 & B1\\_group1\\_seq25\\_chr1:39887435-39887936               & 68\\\\\n",
       "\t1346 & B2\\_group2\\_seq34\\_chr3:10394211-10395104               & 78\\\\\n",
       "\t1158 & B2\\_group1\\_seq204\\_chr22:48100065-48100566             & 82\\\\\n",
       "\t767 & B1\\_group2\\_seq80\\_chr14:20544265-20544766              & 84\\\\\n",
       "\t1581 & Neg-Control\\_A\\_group1\\_seq84\\_chr7:1056602-1057102      & 88\\\\\n",
       "\t1344 & B2\\_group2\\_seq32\\_chr7:143489706-143490207             & 91\\\\\n",
       "\t1653 & Neg-Control\\_B1\\_group1\\_seq189\\_chr20:60750743-60751244 & 91\\\\\n",
       "\t1783 & S4\\_MEL\\_SOX\\_MITF-3prime-50bp                          & 92\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20 x 2\n",
       "\n",
       "| <!--/--> | Var1 &lt;fct&gt; | Freq &lt;int&gt; |\n",
       "|---|---|---|\n",
       "| 1384 | B2_group3_seq131_chr9:43376634-43376970              |  6 |\n",
       "| 546 | B1_group1_seq212_chr1:232468422-232468923            | 14 |\n",
       "| 1002 | B1_group5_seq51_chr2:133025518-133026407             | 16 |\n",
       "| 1371 | B2_group3_seq116_chr2:133025518-133026407            | 20 |\n",
       "| 1203 | B2_group1_seq245_chr1:159017399-159017900            | 30 |\n",
       "| 1752 | Neg-Control_B2_group3_seq80_chr11:175385-175963      | 31 |\n",
       "| 898 | B1_group5_seq189_chr13:28033564-28034043             | 36 |\n",
       "| 1282 | B2_group1_seq66_chr13:25528701-25529202              | 37 |\n",
       "| 697 | B1_group2_seq17_chr13:25528701-25529202              | 38 |\n",
       "| 604 | B1_group1_seq53_chr16:58539224-58539725              | 39 |\n",
       "| 662 | B1_group2_seq108_chr1:162698935-162699436            | 44 |\n",
       "| 116 | A_group2_seq109_chr9:47216611-47217111               | 61 |\n",
       "| 573 | B1_group1_seq25_chr1:39887435-39887936               | 68 |\n",
       "| 1346 | B2_group2_seq34_chr3:10394211-10395104               | 78 |\n",
       "| 1158 | B2_group1_seq204_chr22:48100065-48100566             | 82 |\n",
       "| 767 | B1_group2_seq80_chr14:20544265-20544766              | 84 |\n",
       "| 1581 | Neg-Control_A_group1_seq84_chr7:1056602-1057102      | 88 |\n",
       "| 1344 | B2_group2_seq32_chr7:143489706-143490207             | 91 |\n",
       "| 1653 | Neg-Control_B1_group1_seq189_chr20:60750743-60751244 | 91 |\n",
       "| 1783 | S4_MEL_SOX_MITF-3prime-50bp                          | 92 |\n",
       "\n"
      ],
      "text/plain": [
       "     Var1                                                 Freq\n",
       "1384 B2_group3_seq131_chr9:43376634-43376970               6  \n",
       "546  B1_group1_seq212_chr1:232468422-232468923            14  \n",
       "1002 B1_group5_seq51_chr2:133025518-133026407             16  \n",
       "1371 B2_group3_seq116_chr2:133025518-133026407            20  \n",
       "1203 B2_group1_seq245_chr1:159017399-159017900            30  \n",
       "1752 Neg-Control_B2_group3_seq80_chr11:175385-175963      31  \n",
       "898  B1_group5_seq189_chr13:28033564-28034043             36  \n",
       "1282 B2_group1_seq66_chr13:25528701-25529202              37  \n",
       "697  B1_group2_seq17_chr13:25528701-25529202              38  \n",
       "604  B1_group1_seq53_chr16:58539224-58539725              39  \n",
       "662  B1_group2_seq108_chr1:162698935-162699436            44  \n",
       "116  A_group2_seq109_chr9:47216611-47217111               61  \n",
       "573  B1_group1_seq25_chr1:39887435-39887936               68  \n",
       "1346 B2_group2_seq34_chr3:10394211-10395104               78  \n",
       "1158 B2_group1_seq204_chr22:48100065-48100566             82  \n",
       "767  B1_group2_seq80_chr14:20544265-20544766              84  \n",
       "1581 Neg-Control_A_group1_seq84_chr7:1056602-1057102      88  \n",
       "1344 B2_group2_seq32_chr7:143489706-143490207             91  \n",
       "1653 Neg-Control_B1_group1_seq189_chr20:60750743-60751244 91  \n",
       "1783 S4_MEL_SOX_MITF-3prime-50bp                          92  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA0gAAANICAIAAAByhViMAAAABmJLR0QA/wD/AP+gvaeTAAAg\nAElEQVR4nOzdV1hU1xrG8TUMSpeiICCKUoIaOyj2ikSNvRyMkkSCvcUWNbH3mhijYuy9IfaE\nYAV7F7sijho0okREBARpcy5IjFEYB2VmmJ3/7+I8Ya/F8Hqu3mfP3uuTKZVKAQAAAP1noOsA\nAAAAKBgUOwAAAImg2AEAAEgExQ4AAEAiKHYAAAASQbEDAACQCIodAACARFDsAAAAJIJiBwAA\nIBEUOwAAAImg2AEAAEgExQ4AAEAiKHYAAAASQbEDAACQCIodAACARFDsAAAAJIJiBwAAIBEU\nOwAAAImg2AEAAEgExQ4AAEAiKHYAAAASQbEDAACQCIodAACARFDsAAAAJIJiBwAAIBEUOwAA\nAImg2AEAAEgExQ4AAEAiKHYAAAASQbEDAACQCIodAACARFDsAAAAJIJiBwAAIBEUOwAAAImg\n2AEAAEgExQ4AAEAiKHYAAAASQbEDAACQCIodAACARFDsAAAAJIJiBwAAIBEUOwAAAImg2AEA\nAEgExQ4AAEAiKHYAAAASQbEDAACQCIodAACARFDsAAAAJIJiBwAAIBEUOwAAAImg2AEAAEgE\nxQ4AAEAiKHYAAAASQbEDAACQCIodAACARFDsAAAAJIJiBwAAIBEUOwAAAImg2AEAAEgExQ4A\nAEAiKHYAAAASQbEDAACQCIodAACARFDsAAAAJIJiBwAAIBEUOwAAAImg2AEAAEgExQ4AAEAi\nKHYAAAASQbEDAACQCIodAACARFDsAAAAJIJiBwAAIBEUOwAAAImg2AEAAEgExQ4AAEAiKHYA\nAAASQbEDAACQCIodAACARFDsAAAAJIJiBwAAIBEUOwAAAImg2AEAAEgExQ4AAEAiKHYAAAAS\nQbEDAACQCIodAACARFDsAAAAJIJiBwAAIBEUOwAAAImg2AEAAEgExQ4AAEAiKHYAAAASQbED\nAACQCIodAACARFDsAAAAJMJQ1wH0QGJi4po1a1JTU3UdBAAAFAomJiZffvmlpaWlroO8iWL3\nbhs2bPj66691nQIAABQihoaG/fv313WKN1Hs3i0jI0MIsWLFiqpVq+o6CwAA0LFLly4FBgbm\n1IPChmKnLg8PD09PT12nAAAAOpaWlqbrCHni5QkAAACJoNgBAABIBMUOAABAIih2AAAAEkGx\nAwAAkAiKHQAAgERQ7AAAACSCYgcAACARFDsAAACJoNgBAABIBMUOAABAIih2AAAAEkGxAwAA\nkAiKHQAAgEQY6joAAACABr148eLAgQNRUVFCCA8PDx8fH1NTU12H0hSKHQAAkKwlS5aMGTMm\nPj7+1ZXixYtPnz69d+/eOkylOXwVCwAApGnatGl9+/a1sLCYP3/+6dOnT58+PX/+fHNz8z59\n+kyfPl3X6TSCO3YAAECCrl27NnHiRE9Pz4MHD1paWuZcrFWr1pdfftm0adMJEya0a9fu448/\n1m3IAscdOwAAIEErV67MzMwMCgp61epyWFpaBgUFZWZmrlq1SlfZNIdiBwAAJOj8+fO2tra1\natV6e8nb27tEiRLnzp3TfipNo9gBAAAJev78+Rv36l5nZWWVlJSkzTzaQbEDAAAS5ODg8PDh\nw5cvX769lJaW9scffzg4OGg/laZR7AAAgAQ1b978xYsXGzdufHtp48aNqampPj4+2k+laRQ7\nAAAgQYGBgQ4ODoMHD967d+/r1/fu3fv11187ODgEBgbqKpvmcNwJAACQIAsLix07drRo0aJF\nixZ16tSpU6eOTCY7ceLEyZMnraysdu7caWFhoeuMBY9iBwAApMnb2/vixYuTJk0KCQk5efKk\nEKJYsWIBAQETJ04sU6aMrtNpBMUOAABIlrOz88qVK5cuXfrw4UMhhKOjo6GhlMuPlP9tAAAA\nQghDQ0Op3qJ7Ay9PAAAASATFDgAAQCIodgAAABJBsQMAAJAIih0AAIBEUOwAAAAkgmIHAAAg\nERQ7AAAAiaDYAQAASATFDgAAQCIodgAAABJBsQMAAJAIih0AAIBEGOo6wPvJTIq9ExWl+ONp\nUkpqlomNg4Ojg6OjUylbM7mukwEAAOiKvhW7lzF7F0yZvXzb4aiErDfXjErV7hjQq2+f7g2d\njHSRDQAAQKf0qthlRs1vWXtI+DMhL+ZSr23dCk4lbGysLYqkP09ISHgaG33m8IlNU09tWr5z\n6dFtvdyK6DotAACAdulTsYuc3HF4eIqr3+ItiwI9i+dS3LKe39o+7rOvfurbZrLXjSnVtZ8Q\nAABAh/To5Ym7Yb9ez3IbFrK5b66tTgghL/ZRl/lhMxuLm3v3x2g5HQAAgK7pUbFTKBRCXqly\nxXdss61WzVFER0drJRMAAEDhoUfFroanpyzr2J7QZyp3pUXu2vdA5uVVQ0upAAAACgs9KnY2\n7QI7lHiypau335wdp2OSs99YVqbGRv72c98GTeZct27ZraW1TjICAADojj69POHYbXXItZi2\n04NHdgweWdSqtLNjcWsba4uimckJTxPiY2PuP01TCmFZf2pYcEAZXYcFAADQNn0qdkJYNJp2\nKrrdpqUrg38JP3tVcf3+X2fZGRhb2tq71Pdt6//VV1193IvJdBsTAABAF/Sr2Akh5Ha1/MfW\n8h8rhDIjJTExKeWlMCtua2XMyAkAAPBfp3fFLgcjxQAAAN6kb8WOkWIAAAB50Ktix0gxAACA\nvOlTsWOkGAAAgAp6dI4dI8UAAABU0aNix0gxAAAAVfSo2DFSDAAAQBU9KnaMFAMAAFBFn16e\nYKQYAACACvpU7BgpBgAAoIJ+FTvBSDEAAIC86F2xy8FIMQAAgDfpW7FjpBgAAEAe9KrYMVIM\nAAAgb/pU7BgpBgAAoIIeFbuckWKjQjb3rZbHjpyRYnGX7Qfu3R8zpbo6R55kZWWFhoampaWp\n2BMZGSmEyMjIyH9mAAAA7dGjYqdQKIS8iXojxVZHRwuhTrELDw9v27atOn9+48aNjRs3Vmcn\nAACATuhRsavh6SlbcWxP6LPu7a3y3pXPkWJNmjTZvXu36jt2QUFBERERTk5O+YoLAACgZXpU\n7GzaBXYY331LV2/llJnD/JrXLGP+r3loytTYixG7lowfveS6dasR6o4Uk8vlbdq0Ub0nNDRU\nCGFgoEfj1wAAwH+RHhU7RooBAACook/FjpFiAAAAKuhXsROMFAMAAMiL3hW7f8iKmFmVMLN6\n8eD88V2Xr955XsylUmXPWp7OFtyvAwAA/0l6VOyOTGwy4bBH780/f1byryvZ9/dO6tV75t6Y\n9L/3yEvWHzR/2VS/8mY6CgkAAKAzelTs/rwWERGR3OLVwSRZN3/o0n7y6ZeWVbsM7FK/gkOR\np9Endyzf8GPXZvHmV9d+quZrsQAAAFKhR8XuDWkh48adTrPz2xS5savjXyeR9PtmeJeAam3X\n9J7Q88FPDflKFgAA/Kfo79ls186dSxOeQ2e+anVCCCEr0Wba0Nri4ZEjCt0lAwAA0An9LXbx\n8fGiiJub85vXS7m7mwiFgmIHAAD+a/S32H1cubJBxs2bbxW4R3fupIoKFSroIhMAAIAO6dsz\ndrc3DOkTV8XNzc3NrcKnTU3HLPhmrf+2L8r+3U8zfw8e9sNxYT/Qm9ETAADgv0aPil1xlyou\ndoobO5de2fna1Z2Dpx76YrmPECL7/A+fdp+0NyqptP/uCY10lRIAAEBX9KjYNZ51STFLZCY9\n+l2huK1Q/PU/ilgnk5x15e2jYdGySn5Ba5e0LqHbqAAAADqgR8Uuh6GFvWs1e9dq9d5aMWgy\n50acc/niRXSQCgAAQPf0rtiJzITbZ0+duahItnKt1ax5NbtX/wKZnVt5IcTLWwdCzj4u06B7\nAx6zAwAA/yn6VezSbm0Z2b3vwnPPlDk/m5RtM2XL+uG1ir22J3n/NP+BEZ1DKHYAAOA/Rp+K\nXfrpCa0/WxBt5Npq0Je+HsXiz29dvm7PiKa+2aeOflOJL2ABAMB/nR4Vu2frJ82LNqgw8tDZ\nWXXMhBBCDAr06VrXf+sY/6m+5yZV1aN/CgAAgAbo0QHFNy9fzpD5DBvzV6sTQhg4d1v/y7jq\nykuzBi++q8toAAAAhYAeFbsXL14IIysrk39dLFr9u0UDXNOPTBwe/ERHuQAAAAoHPSp2NTw9\nRVp4aHjavy8b1Zm8uHfppzsG9tn4SDfBAAAACgU9KnZWDZt5Gsat6tZyyqEHqa8vFGs+d1V/\nlyfbA1sM2HHvpa7iAQAA6JgeFTtRYeS25R0c4iPGNytjbV262fyoVyvmzebuXtimxJWgjh5O\ntWdH6jAjAACAzuhTsRMGzl9uPB42b2Abb1erjITE12/OmXzcf8epvXN717eIjUnUWUAAAAAd\n0rczQozLfTJkwSdDcluSl/IZvsRn2E8JD+4q7qQ7aTsZAACAjulbsXsnmZF16fJepXUdAwAA\nQOv06qtYAAAA5I1iBwAAIBEUOwAAAImg2AEAAEgExQ4AAEAiKHYAAAASQbEDAACQCIodAACA\nRFDsAAAAJIJiBwAAIBEUOwAAAImg2AEAAEgExQ4AAEAiKHYAAAASQbEDAACQCIodAACARFDs\nAAAAJIJiBwAAIBEUOwAAAImg2AEAAEgExQ4AAEAiKHYAAAASQbEDAACQCIodAACARFDsAAAA\nJIJiBwAAIBEUOwAAAImg2AEAAEgExQ4AAEAiKHYAAAASQbEDAACQCIodAACARFDsAAAAJIJi\nBwAAIBEUOwAAAImg2AEAAEgExQ4AAEAiKHYAAAASQbEDAACQCIodAACARFDsAAAAJIJiBwAA\nIBEUOwAAAImg2AEAAEgExQ4AAEAiKHYAAAASQbEDAACQCIodAACARFDsAAAAJMJQ1wHeT2ZS\n7J2oKMUfT5NSUrNMbBwcHB0cHZ1K2ZrJdZ0MAABAV/St2L2M2btgyuzl2w5HJWS9uWZUqnbH\ngF59+3Rv6GSki2wAAAA6pVfFLjNqfsvaQ8KfCXkxl3pt61ZwKmFjY21RJP15QkLC09joM4dP\nbJp6atPynUuPbuvlVkTXaQEAALRLn4pd5OSOw8NTXP0Wb1kU6Fk8l+KW9fzW9nGfffVT3zaT\nvW5Mqa79hAAAADqkRy9P3A379XqW27CQzX1zbXVCCHmxj7rMD5vZWNzcuz9Gy+kAAAB0TY+K\nnUKhEPJKlSu+Y5tttWqOIjo6WiuZAAAACg89KnY1PD1lWcf2hD5TuSstcte+BzIvrxpaSgUA\nAFBY6FGxs2kX2KHEky1dvf3m7Dgdk5z9xrIyNTbyt5/7Nmgy57p1y24trXWSEQAAQHf06eUJ\nx26rQ67FtJ0ePLJj8MiiVqWdHYtb21hbFM1MTniaEB8bc/9pmlIIy/pTw4IDyug6LAAAgLbp\nU7ETwqLRtFPR7TYtXRn8S/jZq4rr9/86y87A2NLW3qW+b1v/r77q6uNeTKbbmAAAALqgX8VO\nCCG3q+U/tpb/WCGUGSmJiUkpL4VZcVsrY0ZOAACA/zq9K3Y5GCkGAADwJn0rdowUAwAAyINe\nFTtGigEAAORNn4odI8UAAABU0KNz7BgpBgAAoIoeFTtGigEAAKiiR8WOkWIAAACq6FGxY6QY\nAACAKvr08gQjxQAAAFTQp2LHSDEAAAAV9KvYCUaKAQAA5EXvil0ORooBAAC8Sd+KHSPFAAAA\n8qBXxY6RYgAAAHnTp2LHSDEAAAAV9KjY5YwUGxWyuW+1PHbkjBSLu2w/cO/+mCnV1TnyJCsr\nKzQ0NC0tTcWee/fuCSGys988OQ8AAKBQ0aNip1AohLyJeiPFVkdHC6FOsQsPD2/btq06f/7u\n3bvqbAMAANAVPSp2NTw9ZSuO7Ql91r29Vd678jlSrEmTJrt371Z9xy4oKCgiIqJcuXL5igsA\nAKBlelTsbNoFdhjffUtXb+WUmcP8mtcsY/6veWjK1NiLEbuWjB+95Lp1qxHqjhSTy+Vt2rRR\nvSc0NFQIYWCgR+PXAADAf5EeFTtGigEAAKiiT8WOkWIAAAAq6FexE4wUAwAAyIveFbscjBQD\nAAB4k74VO0aKAQAA5EGvih0jxQAAQD4plcobN248fPjQzMysatWqpqamuk6kQfpU7BgpBgAA\n1KdUKpcsWTJ9+vT79+/nXDEyMvr8889nzpxZvHhx3WbTED0qdhoZKQYAACRJqVT26tVrxYoV\npUqVGjlypKura2Ji4p49e5YvX37w4MFjx445OjrqOmPB06Nip5GRYgAAQJI2bNiwYsWKTz/9\ndPPmzebm5jkXv/nmm2XLlvXp0ycwMPC3337TbUJN0KNpCjU8PWVZx/aEPlO5K58jxQAAgCT9\n8MMPNjY2GzZseNXqcvTq1evzzz8PCwu7fv26rrJpjh4VO5t2gR1KPNnS1dtvzo7TMcnZbywr\nU2Mjf/u5b4Mmc65bt+ym7kgxAAAgPc+fP4+MjGzVqpWlpeXbq5999pkQ4siRI1rPpXF69FUs\nI8UAAIBa4uPjhRClSpXKdTXn+pMnT7SaSSv0qdgxUgwAAKjDxsZGCPH48eNcV3Ou5+yRGP0q\ndoKRYgAA4J0sLS0rVaoUGhqakpJiZmb2xmpISIgQom7durqIpll69IzdXzITbp/8bePihUs3\nH4hOt7IvVcr+363u5a0DGzZsOBqjs4AAAED3Bg8eHBcX17Nnz/T09Nevb926dfny5Y0aNapW\nLa/z0/SYft2xS7u1ZWT3vgvPPVPm/GxSts2ULeuH1yr22p7k/dP8B0Z0DunegOfsAAD4zwoM\nDNy3b9/mzZsjIyMDAgLc3d0TEhJ27969Z88ee3v7VatW6TqgRuhTsUs/PaH1ZwuijVxbDfrS\n16NY/Pmty9ftGdHUN/vU0W8qMUEMAAD8w8DAYPPmzbNnz54zZ87o0aNfXezYseP8+fPzeq9C\n3+lRsXu2ftK8aIMKIw+dnVUn58vyQYE+Xev6bx3jP9X33KSqevRPAQAAmieXy7/99tthw4ad\nOXPm4cOH5ubmXl5eJUuW1HUuDdKjNnTz8uUMmc+wMXVePQJp4Nxt/S9Rt2tNnjV4cY/Dg8rp\nMh0AACiUjIyMGjRooOsUWqJHL0+8ePFCGFlZmfzrYtHq3y0a4Jp+ZOLwYAkeRgMAAJAPelTs\nanh6irTw0PC0f182qjN5ce/ST3cM7LPxkW6CAQAAFAp6VOysGjbzNIxb1a3llEMPUl9fKNZ8\n7qr+Lk+2B7YYsOPeS13FAwAA0DE9Knaiwshtyzs4xEeMb1bG2rp0s/lRr1bMm83dvbBNiStB\nHT2cas+O1GFGAAAAndGnYicMnL/ceDxs3sA23q5WGQmJr9+cM/m4/45Te+f2rm8RG5Oos4AA\nAAA6pEdvxQohhDAu98mQBZ8MyW1JXspn+BKfYT8lPLiruJPupO1kAAAAOqZvxe6dZEbWpct7\nldZ1DAAAAK3Tq69iAQAAkDeKHQAAgERQ7AAAACSCYgcAACARFDsAAACJoNgBAABIBMUOAABA\nIih2AAAAEkGxAwAAkAg1J0+kPFI8SlH7M62cnIsbvW8iAAAAvBc1i13YILfOIep+ZvVZigsj\nXd43EQAAAN6LmsXOa8DqufIpI7cosg1sPOrXreZWyiItVnHxxNHrTzJtanX/vK6t7J/NFZsy\nqhUAAEDr1Cx2zhVNzxxUGFTouWHXvK7u5n9fTlGEDG/Xbf0tm4VLxtc001RGAAAAqEHNlycS\ndy3f+sT167U/v9bqhBBmrp0XrRvqeHby5BC1n8ADAACARqhZ7C6cO6c09PSuIX9rRV7N28so\n68yZCwUcDAAAAPmjZrErXry4yLx5U5HL0t2bN18KExOTAo0FAACA/FKz2H1cv76VuPz9gAU3\nXv7revrNRf3nRgrjmjUrayAcAAAA1KfmyxPyVjNW+u3tuGVwNfcdXwS09nJ3NEt5GH3+19Vr\nD8W8tGi4YFpnzq0DAADQLTWLnRAlO6w8tM115Ogfdy+fHL7876sGJTy/nDV/zsCP3n74DgAA\nAFqldrETwtSj47RdbYffjrx4PTr6Xrzcrpybe2XPGs4Wsnf/LgAAADQtn7Ni0548+P3unft/\nJr0oVa9r6xqORSh1AAAAhUQ+it3jg5N8P3Ku6uPXa+CQb77ddF2IK7Nr27t+Ov34U83lAwAA\ngJrULXbZ1+a2bTdxf1yplgOnzvr8r1dgnby8re6HjvHxnX1dqbGEAAAAUIuaxS5l57RpZ9LK\nfxN+JXTBmD6NSuVcde4efPno2JrK89OnbGfyBAAAgG6pWewuHz78TNZ06Jg6bwyEldnUHjuk\nVZHEU6duFnw2AAAA5IOaxS45OVmYOjhY5rJUxMLCSMTHxxdoLAAAAOSXmsWuatWqIuX0iUtZ\nb61kXzh8LFlUqlSpgIMBAAAgf9QsdnatOjcwiZoXMPSXmIzXLmf/GTb4i5/uFKne2tdRI/EA\nAACgLnUPKHYbtH7BPs9eC9qU31rZ0zZOCJMF3dr/eObgMUWyae2Z60dyww4AAEDH1D7HTlYm\ncNeNk0sHeJs/PHf1sRD3Dm/69WLqR11nhF4PH1UxHwMsAAAAoBH5aWTyEt69Fob3WpidFh8T\nk2Dk4OxgUURjwQAAAJA/6t2xSz6+cOjQiTvu/PU7xsXLfuRGqwMAAChU1Ct25i8vr/1x+ubz\n2RpOAwAAgPem5jN2jfoNqSH/bdnqB5pNAwAAgPem5jN28urj9oXJPw9s5HN1+IDWNVxLl7Qs\n+q9OaGrrbGuqiYAAAABQj5rFbrtfEb/tQpmdlTVvwMF5uWzoHKLc2qlAkwEAACBf1Cx25Zr1\n7GmjakN1l4JIAwAAgPemZrGr3nvxYs0GAQAAwIdR+4BiAAAAFG4q7tiFDXIfFOYz9+bidnLt\n5QEAACgwSqXy5MmTJ0+efP78uYODg4+Pj5ubm65DaZCKYpfy6Pbt25WSX7/0S/9yg35rGXQ3\nqKWmcwEAAHyYyMjIgICAS5cuvboik8n+97//BQUF2diofHdAb+VvyGvKn/fu3fvzhYayAAAA\nFJDIyMhGjRqlp6ePGDGiTZs2dnZ2t2/fXr58+ZYtW6Kioo4ePWpubq7rjAUvf8UOAACg8FMq\nlT179szIyDh48GC9evVyLpYvX75169bTpk0bO3bs9OnTp0+frtuQmsDLEwAAQGouXLhw4cKF\nvn37vmp1r3z77bcVKlRYuXJldrYER6VS7AAAgNRcuHBBCNGyZS4vBRgYGHzyySePHz/+448/\ntJ5L4yh2AABAapKSkoQQxYoVy3XV0tLy1R6JodgBAACpcXR0FELcuXMn11WFQiGTyRwcHLQb\nShve9fLEsWnNmiyX/f1T3DUhxOEJTZosfHNfo0nhExsWdDgAAID30LRpU0NDw2XLln322Wcy\nmez1pcePH+/evdvLy8va2lpX8TTnXcXuyY3DETf+fenPaxERb+0r8WeBRQIAAPggdnZ2vXv3\nDgoKGjBgwLx584yMjHKuP3jwoFOnTs+fPx83bpxuE2qIimLnO//mzanqfo6FY0GkUVtmUuyd\nqCjFH0+TUlKzTGwcHBwdHB2dStmaMSQDAAAIIebOnXvz5s3Fixfv2LHjk08+sbOzi46ODgsL\ne/ny5eTJk9u0aaPrgBqhothZOHp4aC+Iml7G7F0wZfbybYejErLeXDMqVbtjQK++fbo3dDLS\nRTYAAFBomJiY7Nu3b9GiRYsXL16zZo0QwtDQsEGDBqNHj/b19dV1Ok1R44Di7Bdxd248MatW\n0eHv22GZ98NXrttz/Ozd7LJe9ep+0r6jl71W7pRlRs1vWXtI+DMhL+ZSr23dCk4lbGysLYqk\nP09ISHgaG33m8IlNU09tWr5z6dFtvdyKaCMRAAAotORy+eDBgwcPHvzs2bNnz56VLFnSxMRE\n16E06x3F7nnk0oFfjFh3Nan16uQ9X5oJIURWzKbPG/pv/j1bVtS4aObO9T9OmN7u57DNPSsa\nazpr5OSOw8NTXP0Wb1kU6Fk8l+KW9fzW9nGfffVT3zaTvW5Mqa7pPAAAQC9YWVlZWVnpOoU2\nqDzuJHHv1237rH/40ZcTF33d8K/e9ufafl9t/t2y6fTTcS9SUxLv7p/tq9zTy3f4kQxNR70b\n9uv1LLdhIZv75trqhBDyYh91mR82s7G4uXd/jKbjAAAAFDKqit2TzXNXx1Yef+jE6gn9fcrl\nfNn6aNOqsDQD73Erv61VQi7k5mV9vglZ1cvhj/Xrw5UajqpQKIS8UuWK79hmW62ao4iOjtZw\nGgAAoBeys7MfPXoUGxublfXW8/mSk9dXsS9vR/yydttpUdLX8vaebbdfXT667ni2KOducm7b\ntnN/X8wwLW3w/PiWFduSrIu6NGhT3U4zUWt4espWHNsT+qx7exU3U9Mid+17IPPyqqGZEAAA\nQF88evRoypQpwcHBT548EUJYWVl17Nhx/Pjxzs7Ouo6mKXkVu+eHfxqy6FiSyNw/e8jpV5uU\nqfF/ZIsifx6YNiTitc1pT5UiI2TckH2G1v6bNFbsbNoFdhjffUtXb+WUmcP8mtcsY/6v243K\n1NiLEbuWjB+95Lp1qxEtJXjmIAAAUNvly5ebN28eFxdXrVo1Pz8/AwOD06dPr1y5cvv27b/+\n+mvdunV1HVAj8ip2toHb73uNq1Dtp5or7q5t9feuC995eM547L/p95Wtiv6z9+4cT5fvyi25\nE9JVs2+aOHZbHXItpu304JEdg0cWtSrt7Fjc2sbaomhmcsLThPjYmPtP05RCWNafGhYcUEaj\nSQAAQGGWlpbWvn3758+fb926tXPnzq+u79u3r0uXLu3bt79165YkX6dQ9YxdlZYtSz3fMmHc\n8QSlEEIkX5ozdMEtYe/3efPXWp2I2zl+4QVRvW4dLbw/bNFo2qno0+um9GnjXdb46Z3rF88c\nCz946OjpS4rHL4u51O86esm+WzFHxtQy03wUAABQaK1fv/7u3buzZs16vdUJIXx9fYOCgv78\n88+ff/5ZV9k0StVxJ7K644K+CG4/s1G5zVUr2iXfirwVn+3kt3xck5x3Uo/k71kAACAASURB\nVB8c+OHnPWcOhew4Ges6YFVvLX1bLber5T+2lv9YIZQZKYmJSSkvhVlxWytjRk4AAIC/7Nu3\nz9DQMCAg4O0lPz+/AQMG7N+/f/To0doPpmmqz7GzbrvywpEa42duPXH9D3n5T3p8Omj6KF+H\nvxYVu2ZMW/jExKX1mJB5Y5uaaz7raxgpBgAA8hQbG2tnZ2dhYfH2kqGhobOz88OHD7WfSgve\nNXlCblf/659/+Tq3Ja+xZx9MLGlf3ESbZYqRYgAA4F3Mzc2TkpKUSqVMJnt7NTEx0dbWVvup\ntECNkWJ5MStZVsvPsjFSDAAAqKFGjRphYWHHjh1r0KDBG0vXr1///fffW7RooZNgmqai2D2+\nuPfi4/f6UAv3+nVdCr70MVIMAACo48svv5wzZ86gQYMiIiJef/v1xYsXffv2lclkgYGBOoyn\nOSqK3bFpLTqHvNeHVpt6M3KMx3smylPOSLFRIZv7VstjR85IsbjL9gP37o+ZUp0jTwAA+G/6\n6KOPpk6dOmrUqGrVqn3zzTf16tWTy+WnTp36/vvvo6KiRo0aVbNmTV1n1Ih3fBVbqvXYb1vY\n5+cDI5cPXPEhgfKmUCiEvIl6I8VWR0cLQbEDAOA/a+TIkVZWVt9+++3AgQNfXbSwsJg7d+6w\nYcN0GEyj3lHsbGv7DxiQr3tv2yIGrrj97m3vgZFiAAAgH3r37t2tW7f9+/dfvXpVqVSWL1/e\n19dXkucSv6Ki2Hm0Hz06vq5NPj/w/X5LHYwUAwAA+WNubt6hQ4cOHTroOoiWqCh2lbrPmJH/\nD3y/31ILI8UAAABU+IDjTnTAotG0U9HtNi1dGfxL+Nmriuv3/zrLzsDY0tbepb5vW/+vvurq\n414slyNrAAAApE6/ip1gpBgAAEBe9K7Y5WCkGAAAwJv0rdgxUgwAACAPelXsGCkGAACQN30q\ndowUAwAAUEGPip1GRoplZWWFhoampaWp2HPv3j0hRHZ2dr4jAwAAaNEHF7v7QT41JlwSwsDI\nqtTHTfrN+L5XDYuCCPY2jYwUCw8Pb9u2rTp//u7du+psAwAAhcfZs2fDwsIePHhgampas2bN\ndu3amZmZ6TqUBn1wsctOTXzy5InovDVzfeNr8zp0nVbz+sy87qh9GI2MFGvSpMnu3btV37EL\nCgqKiIgoV65cvuICAAAdiouL69Gjx2+//fb6xRIlSixdulTCgyg+uNg5dFt1qn6KsHGXG9lU\nGbp5S4xJQcTKjUZGisnl8jZt2qjeExoaKoQwMDBQvQ0AABQSKSkpPj4+V65c+eKLL/r16+fh\n4fH8+fPQ0NBJkyZ16dJl27Zt7dq103VGjfjgYlfUoZK3w98/GJWq7P6hH5g3RooBAAB1zJ07\n98qVKzNmzBg9enTOFWtr6379+rVq1apWrVr9+/f39fU1MdHYzSjd0a+7UBaNpp2KPr1uSp82\n3mWNn965fvHMsfCDh46evqR4/LKYS/2uo5fsuxVzZEwtKX95DgAA3mXNmjUuLi7ffPPNG9ed\nnZ1Hjx798OHD/fv36ySYpuX7jl1WWuKTx49i45IMLO3s7UuWKGak3W7ISDEAAKBKcnLy3bt3\nAwIC5PJc6kHTpk2FEFeuXFHz7Un9ol6xUyZHH9i69df9Bw8eOn718ct/FmQmDpXrN23WrHnr\n/3VuWk67N8pkRcysSphZvXhw/viuy1fvPC/mUqmyZy1PZwuZVmMAAIBCJTU1VQhhamqa62rO\n9Zw90vOuYpfxx+Hl38+ZvyI06rnM1L6CV22/fuXsSxQvYW2mTHka/+RJ7N2r5w78/O2Ged8N\n+bhtz69HDg+oY6+hs/GOTGwy4bBH780/f1byryvZ9/dO6tV75t6Y9L/3yEvWHzR/2VS/8nwZ\nCwDAf1Tx4sVNTU1v3ryZ6+qNGzeEEKVLl9ZuKC1RUcKUT04vHtbr201/OPn6fbdp5f9a1Spn\nkfv2zETFqV+C169bO7Lh8sUBs5fN7ulpVfC3zf68FhERkdzi1cEkWTd/6NJ+8umXllW7DOxS\nv4JDkafRJ3cs3/Bj12bx5lfXfqrma7EAAEBaDAwMWrZsuXPnzosXL1ar9q9D2JRK5YIFCwwM\nDFq0aKGreBql4gG5nYPbrTLsEaL449qvQaP86ubV6oQQhpau9bt/+3PYjYdRG7u+CGr/TZgm\nor4hLWTcuNNpdn4br18I/n7M4J5f9Rs5Y+2JGzu/LPVwXe8JR5RaiAAAAAql8ePHy+Xytm3b\nnjhx4tXFxMTEwMDAAwcO9O7d29nZWYfxNEfFHbvmS6I7WORvioSJS8uR61v0S3rxganUce3c\nuTThOWFmV8fXyqmsRJtpQ2uvGXHkiEI0dNNCCgAAUPhUqVJl3bp1PXr0qFevXtmyZW1tbbOy\nsm7dupWcnNyqVat58+bpOqCmqLhjZ57PVvc3mYWFNp5wi4+PF0Xc3N7q26Xc3U2EQqHQQgQA\nAFBY1alTx9vbWwhx7969s2fPXrhwISUlpVmzZps3bzY2NtZ1Ok1Rccfu8cW9Fx+/14dauNev\n66Lpcvdx5coGGTdvKoT49525R3fupIoKFSpo+M8DAIBC6969e/Xq1YuNje3atWuHDh2sra0f\nPHiwatWqgwcP+vr6Hjx4MK93ZvWdimJ3bFqLziHv9aHVpt6MHOPxnolUu71hSJ+4Km5ubm5u\nFT5tajpmwTdr/bd9UfbvG4+ZvwcP++G4sB/ozegJAAD+s/r16/fo0aMtW7Z06dLl1cUePXqM\nGTNmxowZM2bMmDJlig7jac47jiYp1Xrsty3s8/OBkcsHrviQQHkr7lLFxU5xY+fSKztfu7pz\n8NRDXyz3EUJkn//h0+6T9kYllfbfPaGRZjIAAIDC7s6dO2FhYd27d3+91QkhZDLZlClTdu3a\ntXTp0okTJ+Z6fLG+e0exs63tP2BAvu69bYsYuOL2hyTKU+NZlxSzRGbSo98VitsKxV//o4h1\n+mvSm/L20bBoWSW/oLVLWpfQSAIAAFD4nT59WgjRrl27t5fkcnnr1q1nz559584dd3cNDrjX\nFRXFzqP96NHxdW3y+YHv91vqM7Swd61m71qt3lsrBk3m3IhzLl+8iMb+NgAAKPwSExOFECVK\n5H6bJ+d6zh7pUVHsKnWfMUMIkZlw++ypMxcVyVautZo1r2b35m+8vHUg5OzjMg26Nyjz2m/p\ngMzOrbxu/jIAACg87OzshBAxMTG5ruZcz9kjPSqOOxFCpN3aMriOy0d1W3XvP6hPt1bVy7q3\n/f7M83/vSd4/zd/f/6ezGgwJAACgrgYNGsjl8rVr1769lJKSsm3bNldX1zJlpPmapapil356\nQuvPFpxLc2k1aPKPC38cF1DP6sGeEU1951zN0Fo+AACAfLG1tQ0MDDx06NDo0aOzs7NfXU9K\nSurWrVtsbOyoUaN0GE+jVHwV+2z9pHnRBhVGHjo7q07OoXSDAn261vXfOsZ/qu+5SVXf8d5F\ngctKinv0XN1KaWJTysZEo3EAAEBh9f3331+6dGnWrFk7d+5s166dra3t77//HhwcHBcXFxAQ\n0LNnT10H1BQV7ezm5csZMp9hY+q8OmrYwLnb+l+ibteaPGvw4h6HB5XTRsB/3PvpE7exF9Xc\n3DlEubWTRuMAAIDCytzc/PDhwzNmzAgKCpo9e3bORXd39+nTp3/11VcymUy38TRHRbF78eKF\nMLKy+vd9r6LVv1s0YEO9+ROHB3+2/X9aPVWkbJ+1u4stmT056NgTpbCq0LS+q4ozoz3zdfge\nAACQGCMjo4kTJ44fPz46Ovr58+clS5aU6nN1r1NR7Gp4eopl4aHhaZ19X5+oZlRn8uLe232X\nDOyzseG2blrsT/ISldsMWtiqiZ1XlQkXyweu3DP8rTmxAAAArzEwMPDw0Mw0rEJJxcsTVg2b\neRrGrerWcsqhB6mvLxRrPndVf5cn2wNbDNhx76WmE75BXingi5pa/psAAAD6QNVbsRVGblve\nwSE+YnyzMtbWpZvNj3q1Yt5s7u6FbUpcCero4VR7dqTmY76udI06LiUsTVQf1AIAAPCfo7Ie\nGTh/ufF42LyBbbxdrTISEl+/OWfycf8dp/bO7V3fIjZG20c3N/pR8WdY/9Ja/qsAAACF3LvO\nLDEu98mQBZ8MyW1JXspn+BKfYT8lPLiruJPupIFwAAAAUN8HH0YnM7IuXd6L22cAAAC6xpNq\nAAAAEkGxAwAAkAiKHQAAgERQ7AAAACSCYgcAACARFDsAAACJoNgBAABIxAcXu/tBPra2tra2\nJZ3ca3zSe9mFpIJIBQAAgHz74GKXnZr45MmTJ40X/X513xyXtf7TLhZELAAAAOTXB0+ecOi2\n6lT9FGHjLjeyqTJ085YYk4KIBQAAgPz64GJX1KGSt8PfPxiVquz+oR8IAACA96Liq9jUp3/8\nEZeUmc8PfL/fAgAAwIdSUexC+zg5ffKTIp8f+H6/BQAAgA/FcScAAAAS8Y5n7C6OLS8bm+8P\nrfaeYQAAAPD+VBQ7e8/WrdPe60PLuZm9ZxwAAAC8LxXFrt7oPXu0FwQAAAAfRs1n7KKOHXmc\nnedq8rUNOzmXGAAAFDZZWVn379+/d+9eRkaGrrNog5rF7ur8xpUaDNt6+62vZjMe7J/SunIN\n/w28BwsAAAqPR48e9e3b19bWtkyZMuXKlbOysvLz87t586auc2mWmsWuyqedbSLn/a9q9c8W\nnXmqzLmmfHZ+Wc+aH/uO/zWhYkCnGprLCAAAkB/Xrl2rVq3akiVLXFxchg4dOmLECG9v761b\nt3p6eu7bt0/X6TRIzckT7j2CLzXYObn3gDkD60bsGL0syF+5akDfOYceFnVtP3PHouFNHT94\nhAUAAEABSE9Pb9++/bNnz7Zu3dq5c+dX10+cONGuXbsuXbpERUXZ29vrMKHmqH+OnbFr++kH\nr59bHuBwdlobjwptZx7JbjQy5PKVHaNodQAAoNDYsmXL7du3p06d+nqrE0LUrVt35cqVz58/\nX7hwoa6yaVr+DihOjVNERcemCFnRooYiOys9IzNbqaFgAAAA72P//v0ymeyrr756e6l169b2\n9vYHDhzQfirtULvYZcYemt2pcpVOc04Zfzr9wL3Yc8t6OF6a17Vq5XazDj1kNCwAACgkHj16\nZGlpaWNj8/aSTCYrW7ZsbGys9lNph5rF7sL0WhWbjdrxrEr/DReu7Pq2qYNN1Z4rzl75dWzd\n1NDRzSp69956T6MxAQAA1FOsWLGUlJTMzNzvOyUkJFhYWGg5ktaoWezuRt4o3nFW+LUTi7pV\nNP/7YhHnVlMOXDu5oHupm8uCz2sqIQAAQD54eXllZGTs3bv37aXbt2/funXLy8tL+6m0Q81i\nV2vqxcvbRjYqKX9zQWbtNXD9hUvbvvqooJMBAAC8B39/f1NT06FDhz5+/Pj16y9evOjZs6cQ\nonfv3jqKpnFqvs9a2sNDxaqRW8eWBZIGAADgAzk5Oc2bN69Pnz7VqlUbOnRow4YNixQpcvbs\n2Xnz5t26dWv48OF169bVdUZNUVHswsa0P15t6ojOlSxlan9c9tOLG2dPueO7bXzTAggHAADw\nHnr37m1paTl06NBRo0a9umhpafn9998PHTpUh8E0TUWxq97MMyiwlsvUlgG9enzetUXVEkVU\nfE76owu/blq7etnqg5nNxq8aUeA5AQAA8sHPz699+/bh4eHXrl3LzMz86KOPmjVrVqxYMV3n\n0iwVxa5k03G7r/4vZNrICWPbfz/M6iPvBvXr1PH2rFjOvkTx4tYmypSE+PgnsXeunjt16uSx\no2cUSZZV2g9ee2lUW1dj7eUHAADInZGRUYsWLVq0aKHrINrzjmfszDw6T9/V+TvF/jWLl2/5\nZdePu1ZmvLXHyLZCHZ/ei5f1829S1kRDMQEAAPAuar08Ye7afMDc5gPmZqc8uHTqXNT9R48e\n/ZkkK2Znb+9QpnzN2lUcTNR/Cg8AAEArMjMzz58/HxMTY2xs7Onp6ejoqOtEGpevKa8GZk7V\nmzlV11QWAACAgqBUKhctWjR16tRXJ57IZLIWLVosWLDA1dVVt9k0Sr1z7JKPLxw6dOKOOxoO\nAwAAUAD69u07aNAgY2PjSZMmBQcHr1y5skuXLvv27fPy8rpy5Yqu02mQesXO/OXltT9O33w+\nW8NpAAAAPtCOHTuWLl3asmXLc+fONWvWzMzMrGLFiuvXrw8NDU1LS/P398/KytJ1Rk1Rc/JE\no35Dash/W7b6gWbTAAAAfKCffvrJ1NTUycnJ2dm5fv36n376ae3atUuWLHnq1KkhQ4Zcvnw5\nIiJC1xk1Rc1n7OTVx+0Lk38e2Mjn6vABrWu4li5pWfRfndDU1tnWVBMBAQAA1KZUKk+cOGFs\nbLxs2bKaNWt27tzZzs7u/v37GzdunDBhQp06dYQQx48fb9asma6TaoSaxW67XxG/7UKZnZU1\nb8DBebls6Byi3NqpQJMBAADkV3Jycnp6enp6er9+/YoWLRoWFpaZmeni4jJ9+vSwsLClS5cK\nIZ4+farrmJqiZrEr16xnTxtVG6q7FEQaAACAD5Geni6EMDY2Xrx4sRDCysrK2Nj46NGja9as\nqVevnpOT04MHD2xsVJYafaZmsavee/FizQYBAAD4YCdPnhRCpKWlubm5WVlZPX36tGjRorVr\n1xZC7Nq1y9TUVAjh7Oys45Qak69z7IRIe3T5+JEzN2OfJZX+dHTHUrEPDewdzTmdGAAAFA4X\nL17M+Y/bt2+bm5t7eHikpqbu3r07Ozu7TJkyMTExQogyZcroNKMG5aPYPT446fOA6fvvpwsh\nhOhcbnTHp7Nrf7Kr8nfr1n1XT7K3NAEAgB45d+6cEMLAwCA7O9vExKRhw4Yff/xxTEzMkiVL\nclqdEMLW1lanGTVIzeNORPa1uW3bTdwfV6rlwKmzPq+cc9HJy9vqfugYH9/Z15UaSwgAAKCu\na9euCSEqVaq0fv36YsWKzZs3r2fPnpMnT46Pjy9ZsqQQwsDAoHz58rqOqSlqFruUndOmnUkr\n/034ldAFY/o0KpVz1bl78OWjY2sqz0+fsj1FcxkBAADU8+jRIyHE5cuXHz16dPv27Rs3buzd\nu/fIkSN//vln3bp1hRCGhoaGhvl8FE1/qFnsLh8+/EzWdOiYOmb/vi6zqT12SKsiiadO3Sz4\nbAAAAPmQnZ394sULIYSLi8uIESMaNWq0b9++xMTE06dPN27ceMeOHUIIa2trXcfUIDUba3Jy\nsjB1cLDMZamIhYWRiI+PL9BYAAAA+SWTyeRyeXZ2dsWKFX18fNasWXPkyJGcJQsLi/r16x87\ndszd3V23ITVKzTt2VatWFSmnT1x6e7Ra9oXDx5JFpUqVCjgYAABA/shkMg8PD1NT019++SU5\nOfnixYv79+8PDg4OCwsbMmTIiRMnhBCdOkl5ooKad+zsWnVuMPrreQFD6+38vvU/9+2y/wwb\n/MVPd4pUD/R11FBAAAAAtXXt2nXs2LGVK1feuHHjpk2b3NzcjIyMbt26lZ6ebmxsrFQqu3Tp\nouuMGqTuw4Nug9Yv2OfZa0Gb8lsre9rGCWGyoFv7H88cPKZINq09c/1ILd+wy0yKvRMVpfjj\naVJKapaJjYODo4Ojo1MpWzO5dnMAAIBCZciQIevWrbt27Vrnzp2zsrJu376dmZnp5eUVGxt7\n9+7dadOmlSpVStcZNUjtt0JkZQJ33ai0cuLo2ZtPXY0XQhze9MDcsUrXGVNnDmnprLWXS17G\n7F0wZfbybYejEt76WtioVO2OAb369une0MlIW3kAAEBhYmZmtm/fvi5duoSEhAghihcvnpqa\neuPGDblcPn78+G+//VbXATUrP41MXsK718LwXguz0+JjYhKMHJwdLIpoLFhuMqPmt6w9JPyZ\nkBdzqde2bgWnEjY21hZF0p8nJCQ8jY0+c/jEpqmnNi3fufTotl5u2o0GAAAKiTJlypw8eXL3\n7t2//PLLvXv3jI2Nq1ev/sUXX0j7tYkcKopdyiPFozxPp5PLXsTFKOJe/Wxm72pvltfmAhI5\nuePw8BRXv8VbFgV6Fs+luGU9v7V93Gdf/dS3zWSvG1OqazgOAAAopAwMDNq3b9++fXtdB9E2\nFcUubJBb5xB1P6dziHKrhl8yuRv26/Ust1Ehm/tWy2OHvNhHXeaHxV22H7h3f8yU6pKdAwcA\nAJAbFcXO7dMhQ5z++VH56OjqzecThZlzrfo1Pipja5jw+6UT4ZEPs8q0GzuuQ2MvjUdVKBRC\n3qRyxXdss61WzVGsjo4WgmIHAAD+U1QUu6o95s179UPy6QmNlyaV+2zZzoUBVWz+fvc07ffd\n37TtsmjVmb5Demo2pxCihqenbMWxPaHPure3yntXWuSufQ9kXl41NJ4HAACgcFF3Vuy2qdPO\n2w3auKbnP61OCGHs3Hb+mqFuMSsX7dL8rFibdoEdSjzZ0tXbb86O0zHJ2W8sK1NjI3/7uW+D\nJnOuW7fs1lLK40IAAAByo+ZbsedPn84ybFy35ttvLBhUreVllLXv5FnxZeOCjfYWx26rQ67F\ntJ0ePLJj8MiiVqWdHYtb21hbFM1MTniaEB8bc/9pmlIIy/pTw4ID+BoWAAD856hZ7MzNzUXm\nrVt3hXB7c+nuzZsvhY2NTUEny41Fo2mnotttWroy+Jfws1cV1+//dZadgbGlrb1Lfd+2/l99\n1dXHvZhMG2EAAAAKFzWLXeXatc3Epu8HLOy0e2CF107/Tb+5qP/cSGHu7/2udxoKityulv/Y\nWv5jhVBmpCQmJqW8FGbFba2MGTkBAAD+69QsdkU6zFzYcm/Ab4Oqu2//PKC1l7ujWcrD6PO/\nrl57KOZl8U/XTm+rtdkTQghGigEAABUSExPXrVt39OjRp0+flihRolGjRt27d7ewsNB1Lo1T\nt4/JyvTYeCB77NfjloQvnxy+/O/LRUs3GTT5+ymfl9bWl5+MFAMAACr9+uuvX375ZXx8vFwu\nL1as2PPnzzdv3jxx4sSNGzc2bdpU1+k0Kx832qyqf7XwyGffXTl/6eYtxcOXlqVdXctX9axY\nUnslipFiAABApePHj3fs2NHS0nLNmjWdOnUyMzNLSkravHnzqFGjWrdufezYsRo1pHwkmnrF\nLvn4wnEhTxoOmtjBxbFyfcfK9TWcKneMFAMAAKp9/fXXcrk8IiKiYsW/nv+3sLDo1auXp6dn\nnTp1hg0bFhERodOAmqXeOXbmLy+v/XH65vNvnh2nVTkjxYaFbO6ba6sTf48Um9lY3Ny7P0bL\n6QAAgK7dunXr/PnzPXr0eNXqXqlRo4afn9+RI0cePnyok2zaoeYBxY36Dakh/23Z6geaTaOS\nQqEQ8krqjRSLjo7WSiYAAFB4REVFCSHq1KmT62qdOnWUSmXOHqlS8xk7efVx+8Lknwc28rk6\nfEDrGq6lS1oW/VcnNLV1tjXVRMB/MFIMAACokpWVJYQwNMy93uRcz9kjVWresdvuV8TeZ8K+\nu3cPzhvQsVmdqh+5lP23/r9pNqdgpBgAAFCtXLlyQohLly7lunrx4kUhhIuLi1YzaZead+zK\nNevZU+Vsiepa+D+JkWIAAECFKlWquLq6Ll++fMiQIfb29q8v/f777+vWratSpQrFTghRvffi\nxZoNohZGigEAgDzJZLK5c+d27NixadOmK1asePWwXURERM+ePVNSUubOnavbhJqW34ER6fGK\na9dvKW7fTzRxdPvI4+NK7iWKaiRYXhgpBgAA8tS+ffsFCxYMGzasbt26zs7ODg4ODx48ePDg\ngbGx8YoVK5o3b67rgJqVj2KXdHXjlBFjFuy9l/bPNSNn38Gz5o33q2he8NFUYaQYAADI3YAB\nA3x8fIKCgg4fPvzw4UNbW9v//e9//fv3d3V11XU0jVO72D3e2qNZ9+1xxs5NA7o1q1rO0fzl\no7uXD21cv29O16b3ilwO7minyZivMFIMAAC8i4eHx/z583WdQgfULHZZYWP6bo8r2WnNyU1f\nlPvndODRE79b/1ntz7f2Hruv/VJfNd+wfX+MFAMAAMibmsUu+tSpp6LuDwteb3VCCGFY1n/h\nt0tChpw8GS18PTSQ73WMFAMAAFBBzWJ37949YdrJzSGXpZLu7hZi6717Qmi42OWMFBsVsrlv\ntTx25IwUi7tsP3Dv/pgp1dU58iQrKys0NDQtLU3Fnnv37gkhsrN1OlANAADgXdQsduXLlxcv\nzp+7oWxT4Y2jRJRRZ88niQoVKhR8tjcoFAohb6LeSLHV0dFCqFPswsPD27Ztq86fv3v3rjrb\nAAAAdEXNYle2STNX2ezZX4yqv3tac4dX34JmPDo47otZl2Vuo5po/kRgjYwUa9Kkye7du1Xf\nsQsKCoqIiMg5zBoAAKDQUvetWO+JG0b8Un/OHF+XzbV9G1ZxcTB9EXv38tG9p2LSDCuMWDeh\npkZTCiFyRoqN776lq7dyysxhfs1rljH/19saytTYixG7lowfveS6dasR6o4Uk8vlbdq0Ub0n\nNDRUCGFgoPGXQwAAAD6E2sedmHjPPHau8vSR44P2795wKueazKRs8+GTZo/xr2aiqXyvY6QY\nAACACvk4oNjAuurnc/Z+Pist/r7i9h/J5qXcXEsXN9bqbSxGigEAAORJRbFLffrH08xiJe0s\n/r3HwLi488fFnfP5WwWGkWIAAAC5U3HDLbSPk9MnPyny+YHv91v5lZkUe+v88SOHjxw5HBEe\nHn701KWomD9T3ppFAQAA8B+ioftqGsNIMQAAgDy8o9hdHFteNjbfH5rXAcIfipFiAABAPenp\n6dHR0RkZGc7OztbWap6WofdUFDt7z9atVR3vlrdybmbvGUclRooBAIB3io2NHTNmzJYtW168\neCGEMDAwaNCgwdSpU+vXr6/raBqnotjVG71nj/aCvJtGRooBAAApuXnzZuPGjR8/flyvXr2m\nTZsaGxtfunRp165djRs3XrZsWUBAgK4Dapaaz9hFHTtiVbdhyTxer68swQAAIABJREFUtUi+\ntuFARvf2mvoG9i8aGSkGAAAkIzMzs0uXLgkJCVu3bu3cufOr69HR0a1aterdu7e3t3fFiu+q\nEvpMzWPors5vXKnBsK233/pqNuPB/imtK9fw36Dp92BzRoplHdsT+kzlrnyOFAMAAJLx66+/\nXr169bvvvnu91Qkh3N3dt2zZkpWV9cMPP+gqm3aoWeyqfNrZJnLe/6pW/2zRmafKnGvKZ+eX\n9az5se/4XxMqBnTSfJGyaRfYocSTLV29/ebsOB2TnP3GsjI1NvK3n/s2aDLnunXLbuqOFAMA\nAJJx6NAhIcQXX3zx9lKNGjUqVaqUs0HC1Pwq1r1H8KUGOyf3HjBnYN2IHaOXBfkrVw3oO+fQ\nw6Ku7WfuWDS8qaMWzk1hpBgAAFAhLi5OJpM5Ojpu3Lhx7dq1MTExJiYm9erVCwgIqF69epky\nZcLDw3WdUbPU72PGru2nH2zit2JEwLBpbTymCWFo33RkyJKJndy0MihWCMFIMQAAoIKlpaVS\nqSxTpkxcXNyrixcuXFiwYEG/fv2ePHliZWWlw3hakL8bbalxiqjo2BQhK1pUnp6ZlZ6Rma3U\nULC8MFIMAADk7uOPPxZCxMXF2dnZ9ezZ093dPTY2dvXq1bdu3Vq8eLGBgUG7du10nVGz1HzG\nTojM2EOzO1Wu0mnOKeNPpx+4F3tuWQ/HS/O6Vq3cbtahh5majJgXWREzqxL2pUrZ0+oAAIAQ\nIjg4WAhRpEiRSZMmmZqa3rlzp2TJkgcPHhwyZIgQIjs7W/LFTs07dhem12o2JjKxeK3+G1bN\n6lbRXAjRc8XZ5p0mf9Vz5uhmFYN7Ldu2tEtZTQYFAABQIS0t7eTJkwYGBhkZGf369Xt1XS6X\nm5ub5/z3gwcPdJROS9S8Y3c38kbxjrPCr51YlNPqhBBCFHFuNeXAtZMLupe6uSz4vKYSAgAA\nvFtkZGRWVlZ2draxsXGRIv9MqMrKykpMTCxZsqQQ4uTJk7oLqA1q3rGrNfXiZQ8P07cXZNZe\nA9dfaNHxUGrB5spFVlLco+cZam42sSllo72XOgAAgK7FxMTk/EdaWppcLpfL5TKZLDs7Ozs7\n287O7vHjx0KI1FTNFxadUrPYlfbwULFq5NaxZYGkUeneT5+4jb2o5ubOIcqtnTQaBwAAFCbn\nzp3L+Q+ZTFa+fHknJ6e4uLhL/2/vvuOjqPY+jp/Z3TRSSSAQUiQQSkiAUAxBilJUlFAfITQL\ncGmKgqIIVrqoiN0rioIFaYpAqF6pEaSHGoiQGAiQUNKBtN2d54+FGHokyc7O7Of9x30NM2dn\nf9yzI9/XzJxzDhwQQpw/f16n05nN5uDgYEVrrHRWmH6uotQe8f1KjznvTfnij4uy8Art2Lbu\nLe4gXtOipvUKAwAAylu5cqVlY+LEidOnT7ds//XXX08//fSOHTvMZrMQYuTIkYrVZxUqCnb6\nao27Pf/Z4x18WzZ5e3/Dod/GjrtP6ZIAAICtSEq6ur7pzJkz4+LiwsLCWrdu3bNnzy+//LJl\ny5ZGo1EI0aJFC0VrrHQqCnYW+vDBT93/dlmfyAIAAHuQkpJiMpmEEJb36uLi4uLi4r788svS\nbZycnCRJ44sYlHkeO9sR2Lx1nWqeLiqsHAAAVI6TJ09aNmRZdnV1vSHAubq6CiE0v+yEUOEd\nOyHEgx8lXVC6BgAAYENSU1NLti9fvlyy7efnd+HCBcueyMhIBSqzLjUGOwAAgOvMnTvXsuHk\n5FRYWBgYGNi8efOzZ8/u3bvXMmxCCDF69GjlCrQSgh0AAFC9Xbt2WTYKCwv1en1qaqrlHp6D\ng0NJsHv44YcVq89aeFMNAACoW0FBQemZh0tGUQghiouLSzezfm1WRrADAADqduLECctGnTp1\nnJ2dLduyLFs2LAlPr9e7uGh/USqCHQAAULeSGeySk5MdHR1DQ0Nr1qyp1+uFEP7+/pZDfn5+\nitVnRQQ7AACgbocOHSrZnjhxYkJCQlpaWnZ29oQJE86cOWO5dTdixAjlCrQeBk8AAAB1++OP\nP4QQBoPBaDROnDjxo48+ioqKcnR03LdvX0mbwYMHK1eg9RDsAACAulkexRqNxjp16ly8ePHc\nuXMrVqywHLKkPSFE1apVlSzRWgh2AABA3a5cuSKE6Nq165o1axwcHNq0aSPL8qVLl5KSki5f\nvixJkrOzc5UqVZQu0xp4xw4AAKibl5eXXq+Pj49fvHhxx44d9+7du3379iNHjjRu3Lh///6y\nLJcModA8gh0AAFC38PBwWZbT0tJeeumlvn37Hj9+PC0tbfv27Y0aNVq0aJEQ4tFHH1W6Rivh\nUSwAAFC3AQMGLFmypF27docOHRoyZEjpQz4+PpmZmU899ZRStVkZd+wAAIC69ejRo1u3bnFx\ncYWFhSU7DQaDr69vRkbG8OHDIyMjFSzPmgh2AABA3cxms2XBidILixmNxvPnz3t7e0+ePFm5\n0qyNYAcAANTts88+W7p0qRCiatWqjRo1Cg4O9vX11el0Op0uMzNzzJgxShdoPQQ7AACgbm++\n+aYQYtSoUWlpaUeOHElOTj537lxCQkLDhg0lSVqyZElqaqrSNVoJwQ4AAKjYrl27cnNzg4KC\nvvjiCycnp5L9DRo0iI2NNRgMsixv3bpVwQqtiWAHAABUbM2aNUKIdu3a3XyoTp06lmET586d\ns3ZZCmG6EwAAoGJ5eXmWDZPJtGfPnsTERCFEaGhoixYtdDqdj4+PEEKns5c7WQQ7AACgYn5+\nfkKINWvW1KtX7++//y7ZHxISMnv27ISEBHGb+3maRLADAAAq1rx5cyFEVlZWfn7+xIkTW7du\nLcvyn3/++dVXX/Xo0UOWZRcXlxYtWihdppUQ7AAAgIr5+/tLkiSEKCgomDdv3pYtW2rWrFmj\nRo2IiIiNGzcKIZ588kmla7Qegh0AAFCxefPmybJsyXbp6enp6ek3NKhWrZoSdSnDXt4lBAAA\nmhQXF6fX64UQrq6uNxzS6XQuLi67d+9Woi5lEOwAAICKHT9+3GQy+fr6Xr58uWSnTqfr1q2b\nm5ub0WjMzMxUsDwrI9gBAAC1MpvN2dnZQohz587pdLratWuHhIQYDAaz2RwbG+vh4VFcXGww\n2NGLZwQ7AACgVunp6cXFxUIIf3//pk2bFhQUSJLUs2fPQYMGCSFOnz4thKhRo4bCVVqRHWVY\nAACgMV988YVl48yZMwUFBY0aNcrIyFi2bJnZbG7QoIFlsuImTZooWqNVEewAAIBaxcbGCiH0\ner1Op8vJyalevXr37t0zMzO/+eYbS6oT12YwthMEOwAAoFZJSUlCCEmSfvzxx1mzZi1fvnzZ\nsmVCCL1e7+npmZOTI4SIiIhQuEor4h07AACgSmazOT8/XwhhMpn69eu3e/dus9kshHB3d3/x\nxRfDwsIszUJDQ5Ws0rq4YwcAAFTp4sWLliQny3Lp/Xl5ebNmzbJsu7i4VK1aVYHiFMIdOwAA\noEplmcekZcuWVqjEdhDsAACAKk2bNq30H4OCggYNGhQVFWVZXszihRdesHpdSuJRLAAAUKXP\nPvvMshEVFbVjx45Tp079+OOPN7QpedPOThDsAACA+iQkJFimJvb09Ny5c6erq6tOpzMYDNWr\nVw8KCtq6dWtRUZEQwjK6wn4Q7AAAgPrs3LnTsmGZ06RkodisrKy//vqr5PU7Hx8fRcpTCu/Y\nAQAA9bEsEXs7VapUEULo9fqgoCBrVWQTCHYAAEBlZFmeOHHiHRrk5uYKIcLCwkoPpLAHBDsA\nAKAyU6dOLSwsvGGnJEk3xLjVq1dbsSibwDt2AABAZaZPn37zzhumKXZwcAgICLBWRbaCO3YA\nAEBNjh07ZhnxKklS586db9dMr9dbsShbQbADAABq8vXXX1s2ZFn+/fffSx9ycHDQ6a5mG7ta\nSawEwQ4AAKhJYmJiybYlxjk5OdWtW9fV1dVsNvv6+loO9erVS5n6FMU7dgAAQE0yMjJKts1m\nsxCisLAwOTlZluUqVaqkp6cLISRJmj17tmIlKoc7dgAAQE2SkpJu3mkZOXHlyhXLHxcvXuzk\n5GTVsmwDwQ4AAKhGcXHxhQsX7tzG3d29T58+1qnH1hDsAACAasyaNUsIcedph8PDw61Vjs0h\n2AEAANX4/vvvhRBubm5eXl6i1KTEpaOep6enUuUpjmAHAABUIy0tTQiRl5dnWStWlmXL23Wy\nLDdr1szSpmnTpgpWqCyCHQAAUI38/PySbZ1OVzJrnRAiPj7esjFs2DBrl2UzCHYAAEAd8vLy\nLGtOWJjNZst0J6UXmXB0dKxbt64CxdkGgh0AAFCH/v37l/5jyXt1JpOpZGeHDh2sWpONIdgB\nAAAVMBqNq1evLr3H8nbdDbp27WqtimwRwQ4AAKjA8OHDy9LM0dGxsiuxZQQ7AABg6/Lz8+fP\nn2/ZDgoKmjBhwu3mNGnSpIn1yrI9BDsAAGDrRowYUfLgNTU1NTIy8vDhw2vXrn311VdLD4zV\n6XSRkZEK1WgTDEoXcG+MeWnJiYlJZzLzLuebXLz9/Gr51aoV4F/dVX/3zwIAAJVZunRpybYs\ny71795YkKTw8/K233ho9evQnn3xiOdSxY8fSI2TtkNqCXeGp9Z9OfW/uL1sSs0w3HnPyj+o9\neNjIEQPbB9jjsr8AAGhTdnZ2QUHBDTtlWT506FCfPn1KrzmxZMkS65Zmc1QV7IyJHz8WNXZT\nttB71GnT/YHQgGre3lXdHYpys7KyMtOO79qyfeG0HQvnLv8q7pdhIQ5KVwsAACrCpUuX7nC0\n5BFt3bp1q1atapWKbJeagl38lN7jNl2uG/PfxZ8PbeFzi+Bmyv1r2Zv9h3wystuUlkenNrN+\nhQAAoMK1bdu2LM327NlT2ZXYPhUNnvh73eoEU8hLPy8aectUJ4TQe9Tv8/G6mQ+JY+v/d8rK\n1QEAgEoQHR198uTJuzYLDg728vKyQj02TkXBLikpSejDGze6S7PqERG1xPHjx61SEwAAqDxj\nxoy5YVLi25k7d25lF6MKKgp2zVu0kEx/xK7JvmOrgvgVv52WWrZsbqWqAABApViwYEHJcNe7\nqlOnTqUWoxYqCnbePYb2qnZxcb9WMe//uvPUJfMNh+X8tPi1X45s1+H9hKqPDXjM3l+eBABA\nzfLz8wcNGlTGxjqdLigoqFLrUQs1DZ6oNWD+z0dOdZ+xZHzvJeMdvQLvq+VT1buqu6PxUlZm\nVkbaqdTMAlkIz7bT1i0ZTO8CAKBi3t7eZW8cFhZWeppie6amYCeE+4PTdxzvsfCrb5es2rT7\ncFJC6tW57HTOntVr1mn7SPdBQ4b061zPQ7rzaQAAgA2LiYm5eeK6Oyjje3j2QF3BTgih940c\n9EbkoDeEkIsv5+TkXS4Urj7VvZztepppAAA04+LFi/9qnuFnnnkmMDCw8upRF9UFOwuWFAMA\nQJv+VUrz9fWdN29e5RWjOmoLdiwpBgCARhUXF7u7uxcWFpaxvcFgOHfuXKWWpDqqCnYsKQYA\ngHbVr1+/7KlOkqQLFy5Uaj1qpKZgx5JiAABo1W+//ZaSklL29qdPn2apiZupaGwwS4oBAKBN\nZrO5S5cuZW//yy+/1KpVq/LqUS8VBTuWFAMAQJsaNmwoy3IZGzdq1Kh3796VWo96qSjYsaQY\nAAAa9Mknn5T9howkSbt27arUelRNRcGOJcUAANCa3NzcMWPGlL19ly5dXF1dK68etVPT4AmW\nFAMAQGMaNGhQ9sbOzs5r1qypvGI0QE3BrjKWFDOZTGvWrLnzuiWWQTpm8433CAEAQHmsW7cu\nPT29jI0dHR0zMzMrtR4NUFewExW+pNimTZu6d+9elpZ///33vX0FAAC4pV69epWxpV6vL/sU\nd/ZMdcHOosKWFOvQocPKlSvvfMfuiy++2Lx5c3BwcHkqBgAApe3Zs+fO//7e0LhSi9EMtQW7\nil5STK/Xd+vW7c5tLI/zdToVDTQBAMDWde7cuYwtBw0aFBERUanFaIaqgh1LigEAoAkPP/xw\nTk5OWVp6eHj88MMPlV2PZqgp2LGkGAAAGjBp0qTff/+9jI0ZMPGvqOjxIkuKAQCgeiNHjpw8\neXIZG7/33nt6/T2Oj7RPKgp2LCkGAIC6jRs3bs6cOWVsbDAYXnnllUqtR3tUFOxYUgwAABVb\ntWrV7Nmzy97+9ddfr7xitEpFwY4lxQAAUKvt27ffdRqK0iRJmjhxYuXVo1VqGjzBkmIAAKjR\nrl272rRp868+MnjwYCenMs9ehmvUFOwqY0kxAABQeQoLC6Oiovbv3/+vPuXl5fX1119XUkna\npq5gJyp8STEAAFBJxo4d+/HHH//bT+n1+gsXLrAuwL1RXbD7h+Tg6lXN1evK6b3bVhw8nJzr\nUSe8cYvIFve5c78OAABFvfzyyx988ME9fFCSpIyMDINBxflEWSr6P27rpA5vb2kwfNGX/Wtc\n3WNOXT952PCZ608VXWujr9H2+Y+/nhbT0FWhIgEAsF+yLHfu3Hnjxo33fIa1a9d6enpWYEn2\nRkXB7sKRzZs3X+pSslyw6djsPj2n7Cz0bNpndJ+2oX4Omcf//HXugo/6dcpwO/x9V4bFAgBg\nRfn5+dWqVbty5co9n+Hw4cNhYWEVWJIdUlGwu0HBz2++ubPAN2Zh/E/9al19Dj/qlXF9Bkd0\n/2742/85/Ul7HskCAGA1Xl5eRUVFd293G6tWrSLVlZ9630w8smdPgWjx4sySVCeEEFK1btNf\njBJnt25NUq4yAADsyuXLlw0GQ3lS3cSJE7t27VqBJdkt9Qa7jIwM4RASct+N+/3r1XMRSUkE\nOwAArGDHjh3u7u4mk+mez/Dhhx/OmDGjAkuyZ+oNdmGNG+uKjx27KcClJyfni9DQUCVqAgDA\nfqxfv97BwaF169ayLN/zST799NOxY8dWYFV2Tm3v2J1YMHbE+SYhISEhIaFdO1Z5/dNXvh/0\ny1O1r+VT48klL83eJmqObsXSEwAAVIb09PROnTodPXq0PHnOYtmyZb169aqQqmChomDnU6dJ\nHd+ko8u/OrS81N7lL0zb+NTczkII897ZXQdOXp+YFzho5dsPKlUlAABalZubGxYWdvr06fKf\nqkqVKikpKdWrVy//qVCaioLdQ+8eSHpXGPPSTyYlnUhKuvo/SWkBLpbj8om4dcel8Jgvvp8T\nXU3ZUgEA0JouXbqsX7++/OeRJGnt2rWPPvpo+U+Fm6ko2FkY3GvWjahZN+LmpYR1Hd4/ev6+\nhj4OClQFAIBmTZ48edKkSRVyqgkTJrzzzjsVcirckuqC3e1JviENla4BAABtqVmz5rlz58p/\nnqFDh86dO7f858GdaSjYAQCACmIymfr377906dLyn8rf3z81NVWSWDfAGgh2AADgOunp6YGB\ngUajsfynqlev3l9//VX+86CM1DuPHQAAqHj79u3z8/OrkFQ3fPhwUp2VcccOAAAIIcTDDz+8\nYcOG8s9OJ4SQJGnTpk0PPsjsY9ZGsAMAwK7FxcV16tSpuLi4ok4YEBCQnJzs4MA0FQrgUSwA\nAHbq6NGjBoOhffv2FZXqHnroIVmWU1NTSXVKIdgBAGBfduzY4erqKklSo0aNTCZT+U+o0+nm\nzJkjy/KmTZvKfzaUB49iAQCwCz/99NOYMWMuXrxYgec0GAy7d++OiIiowHOiPAh2AABoU2pq\n6scff7x06dLU1NQKGRJRmiRJv//+e8eOHSv2tCgngh0AANpRVFQUFRW1f//+Ck9ypT300EM8\ndbVNBDsAADQiPDz8yJEjlfoVOp1u8+bN7dq1q9RvwT1j8AQAAKq3Y8cOnU5X2alu6NChJpOJ\nVGfLCHYAAKjVp59+6uHhIUlS69atK/XZqyRJa9eunTt3buV9BSoEj2IBAFCZFStWPPHEExWy\n6ldZ1K9f/9ixY5IkWefrUB7csQMAQB2Ki4ubNWsmSVLPnj2tkOokSWrfvn1OTk5iYiKpTi0I\ndgAAqMBzzz3n6Oi4f/9+K3xXaGio0Wg0m81btmzx8PCwwjeiovAoFgAAG5WVldW8efOTJ09W\n6vtzpTVr1mzbtm0uLi7W+TpUOO7YAQBgc2bMmKHT6by9vVNSUqyQ6tzd3WNjY2VZ3rdvH6lO\n1bhjBwCATdi9e3eXLl2ysrKsdn9Op9NNnz59woQJ1vk6WAHBDgAAJX333XcjR44sKCiwztdJ\nkhQYGBgXFxcUFGSdb4Q1EewAAFDA3r17o6Oj09PTK/uLJEkyGAwBAQHPPPPMyy+/XKVKlcr+\nRiiIYAcAgPWYTKamTZtW9hIRJaZPn/7aa69Z57tgCwh2AABYw4kTJ1q3bn3x4kXrfJ2Li8u6\ndevat29vna+DjWBULAAAlSU7O7tJkyY6nU6SpHr16lkh1el0ui5dupw/f/7KlSukOjvEHTsA\nACpMampqz549Dxw4YDKZrPm9derU2b17t7e3tzW/FDaIYAcAQHklJSUNGDBg9+7dVpupxEKS\npIMHD4aHh1vzS2HLeBQLAMC/9u2337q5uVmesUqSFBISsmvXLiunum7duhmNRlIdSiPYAQBQ\nJt9++62Li4slyQ0dOvTy5ctWTnJCCEmSgoODV69eLcvyypUrdTr+Hcd1eBQLAMCtXbx4sV+/\nflu2bDEajcpWIknSG2+8MWXKFGXLgO0j2AEAcJ0ffvhhyJAhioc5IYQkSeHh4atXrw4MDFS6\nFqgDwQ4AYL/MZvMnn3wya9astLQ0s9msdDlXeXh47Nq1q0GDBkoXAvUh2AEA7M6iRYueffbZ\nrKwspQu5joODw9atW6OiopQuBCrGS5cAAO0rKCiYPn26h4eHZRxr//79bSfV6fV6y2CIoqIi\nUh3KiTt2AAANysvLi4mJ+e2336w8UfC/4uTktGrVqs6dOytdCLSDYAcA0IhDhw49/fTTBw4c\nsJ235W4mSVL16tVjY2MjIyOVrgUaxKNYAIAqvfvuuyWPVi2aNGkSHx9vg6lOkiQfH5+lS5fK\nsmw2m8+dO0eqQyXhjh0AwHYZjcZvvvlm/vz5CQkJly5dkmXZ+nMC3xtJksaPHz9z5kylC4F9\nIdgBAGxCcXHxmDFjFixYoK4AV5okSR4eHv369Zs5c6aXl5fS5cAeEewAAMo4ePDg888/v3Pn\nzqKiIjXGOCGETqdr3br1xo0bHR0dla4FEIJgBwCwgvj4+KFDhx45cqS4uFilGa6ETqeLjIxc\ntmyZn5+f0rUAN2LwBACg4mVkZHTq1MlgMFiGNTRv3jw+Pl69d+bc3Nw+/PBDywNik8n0559/\nkupgm7hjBwCoAGvXru3bt++lS5eULqTC6PX68PDwBQsWhIWFKV0LUFbcsQMAlNWZM2f69etn\nmWSk9DwjkiQ9/vjjqk51kiSFhISsXbtWvsZoNO7fv59UB3Xhjh0A4EZFRUULFy789ddfDx48\nmJaWVlhYqNJHqHfl6em5cuXK9u3bK10IUDEIdgBgp0wmU3x8/NKlS1esWJGSkqLeF+DKTpIk\ng8Hw0EMPzZkzJzg4WOlygIpHsAMAbUpJSVm1atXy5cv37duXk5Oj0pnhys/FxWX8+PGTJk1S\nuhDAGgh2AKBW+/fvnzdvXmxs7JkzZzQwjUhF0el0Pj4+o0ePfu211wwG/pmDfeEXDwC2aOvW\nrTNnzoyPj8/Ozi4uLrasf0p0u4EkSY6Ojp07d168eLGrq6vS5QDKI9gBgDLOnDkzbty4zZs3\nW6Kb3T4qLQvLwFsvL69GjRqNHDnyiSeecHJyUroowBYR7ACgsuTk5MyePXvBggWnT5+2h6EJ\nFcVgMPTq1WvatGn169dXuhZAZQh2AFAB4uLi3n///W3btmVnZ1sem6KMJEny9vYeMmTIm2++\n6e7urnQ5gLoR7ADgtkwm07p16xYtWnTo0KHz58/n5eUVFBSYTCbuvd0bS4abMWNGz549fX19\nlS4H0CCCHQC7c+HChZ9++unXX389fvx4Zmam5SEpWa0y6HS6qlWrvvXWW88++ywDVAEr4DID\noBH79+9/6623du3aVTIWQTCM1Or0en1ERMTy5csDAgKUrgWwRwQ7ADbNbDafPHly586d27dv\n379///Hjx3NyciwrXBHalCJJkpubW48ePT777DNPT0+lywHwD4IdAAUkJydv2rRp1apVBw4c\nOH/+fEFBgdlsJqjZGsskIw4ODt7e3m3btn377bfDwsKULgrAnRDsAFQAo9G4bdu21atXb926\n9eTJk9nZ2by4phaSJAkhHBwcatas2aNHj9dff71GjRpKFwXgHhHsANxaVlbWunXrVq9enZCQ\nkJaWduXKlaKiIpZAUB1JknQ6naura0RExEsvvdSjRw+lKwJQiQh2gH2RZTkhISE5OTk+Pv7P\nP/88duxYRkZGfn6+5UkocU11LPfbHB0dAwMDu3fvPnr06ODgYKWLAqAYgh2gen/88ceGDRu2\nbNmSkJCQnZ1tNBq5qaYZOp3OycnJ19c3NDS0efPmjzzySJs2bZg3BMDt8F8HwBbl5+cfPnw4\nIyNjw4YNq1atSk1Nzc/PZ/4OTZIkydnZ2d/fPyIiIiYm5v/+7/8sN+EA4B4Q7AAruXjx4sKF\nC1euXJmQkMCkuHZFkiSDwRAQENCtW7dXXnmFCd4AVB6CHfDv5Obmbt++fdmyZQcOHDh79mxu\nbm5hYSFPPyGuTQ7i5ubWuHHjUaNGPfHEE05OTkoXBcC+EOxgp44ePbp3796UlJQjR47s27cv\nLS3thjVAiWgoYXk2qtfrHR0dnZ2dg4ODY2JiXnzxRd51A2Br+K8S1KqoqGj//v2bN28+cODA\niRMnzp07l5+fb7l5ZjQaTSZTyYS3RDTckmUeECcnJ09Pz9DQ0N69e/fs2dPf31/pugDg3hHs\nYCuSk5N37ty5bdu2+Pj4lJSUnJycoqKikltohDOUheVhqIuLS61atRo2bNi+ffuHH364adOm\nStcFAFZCsEPFk2V5x44dv/32244dOywDBW54ygncA8vzUJ10f9N1AAAThElEQVRO5+zs7O3t\n3bRp00GDBsXExChdFwDYEIIdyspsNm/YsGHv3r27du06dOhQeno6cQ3lVLIUqZeXV7169fr2\n7RsdHc38ugBwzwh2uConJyc2Nnby5MknT540Go3ENdyDkptqLi4u1apVa9GixdNPP921a1ed\nTqd0aQBgFwh2diE5Ofmzzz5bt25dSkpKYWEh06eh7Cw31XQ6nYODg2WcQUBAwP3339+hQ4fW\nrVv7+PgQ2gDAdhDstODy5cvffPPNggULjh49euXKlZLRoMAdWO6uSZLk6Ojo5ubm7+/fsGHD\nqKio6OjokJAQpasDANwLgp06XLlyZe7cuStWrDh27FhOTo5ldSnSG0qUHlhQs2bN2rVrN2rU\nKCoqqlOnTjVq1FC6OgCAlRDsbFFhYeGoUaMWLlxoeWyqdDlQkiWxGQyGatWqhYWFdezYMTo6\nOjw8nOVEAQA3I9gpr6io6Ndff507d+7u3bvz8vIsK1NBq0omxa1evXp4eHjnzp07d+4cHh6u\ndF0AAC0g2Cljy5YtPXr0yMnJUboQ3KOSR5+WIQVeXl61a9du1apVdHR0u3btlK4OAGCnCHZW\ntX379ujo6KysLKULgRDXwpllw2AwuLu716xZMygoKCAgIDg4uH79+i1atKhdu7aiNQIA8C+o\nNNgZ89KSExOTzmTmXc43uXj7+dXyq1UrwL+6q17pym5n27ZtHTp0KC4uVroQrSkZ2ilJkl6v\n1+l0lpXa3d3dq1WrFhAQEBoa2q5duyZNmgQGBipdLAAAlUttwa7w1PpPp74395ctiVmmG485\n+Uf1Hjxs5IiB7QOclKjt9pYuXRoTE8MwiLKQrnF0dPT09PTx8QkKCmrTpk2fPn3q16+vdHUA\nANg0VQU7Y+LHj0WN3ZQt9B512nR/IDSgmrd3VXeHotysrKzMtOO7tmxfOG3HwrnLv4r7ZViI\ng9LVXpObm9uvXz87THUlb6FZIpqfn19ERMSwYcNat26tdGkAAGiTmoJd/JTe4zZdrhvz38Wf\nD23hc4vgZsr9a9mb/Yd8MrLblJZHpzazfoW3NHXqVC0NdC159GlZ3zMoKKhly5aPP/54165d\nmYADAABlqSjY/b1udYIp5NWfF42MuE0LvUf9Ph+vO3+w5uj1/zs1tVlQGU5qMpnWrFlTUFBw\nhzYpKSlCiHsOZ7Gxsff2QUVYhhGEhITMmDGje/furBYFAICKqCjYJSUlCX2Hxo3u0qx6REQt\nMf/4cSHKEuw2bdrUvXv3snz96dOny9LsZpmZmff2wQpnuc1WvXr1+++//6mnnmrZsiXjCQAA\n0BIVBbvmLVpI3/wRuyZ7YE+v27cqiF/x22mpZcvmZTtphw4dVq5ceec7dqtXr/7uu+8GDBjw\nr8ot4e7ufuHChXv77L9imfnWw8OjadOm48aNi46OtsKXAgAA26GiYOfdY2ivtwYu7tdKnjrz\npZiH7w9yu+4poZyftn/zijlvTZiTUPXxlx+rWraT6vX6bt263bnN2bNnv/vuOweHexyO0a5d\nu+Tk5Hv77A0sw0WdnZ3d3d3r1avXp0+fUaNG3XNhAABAY1QU7EStAfN/PnKq+4wl43svGe/o\nFXhfLZ+q3lXdHY2XsjKzMtJOpWYWyEJ4tp22bsngsjyGtZLx48d///339zYq1mAwPP/887Nn\nz67wqgAAgPaoKdgJ4f7g9B3Heyz86tslqzbtPpyUkHp1Ljuds2f1mnXaPtJ90JAh/TrX87Cp\nwZmNGjUaP378u+++e+dmlrtx7u7urVq1mjVrVuPGja1THgAA0Ax1BTshhN43ctAbkYPeEEIu\nvpyTk3e5ULj6VPdyttklJ4QQYubMmb6+vq+++qrRaLzhkMFgmDhx4pQpUxQpDAAAaInqgt0/\nJAdXr2qudxhHYVNeeumlUaNGxcbG/vzzz2fPnq1Xr96MGTP8/PyUrgsAAGiHioOd6ri4uPTt\n27dv375KFwIAALSJ6WcBAAA0gmAHAACgEQQ7AAAAjSDYAQAAaATBDgAAQCMIdgAAABpBsAMA\nANAIgh0AAIBGEOwAAAA0gmAHAACgEQQ7AAAAjSDYAQAAaATBDgAAQCMIdgAAABphULoA1UhM\nTHR2di7nSYqLi+fPn3/ffffpdERqDTKbzSdOnAgJCaF/tYou1jb6V9vMZvPJkyefeeYZBweH\ncp4qMTGxQkqqDAS7u7P8AoYOHap0IQAAoFzmzJlTUacqf0CsDAS7uxs4cKDRaMzPzy//qQ4e\nPPjTTz+1bdv2vvvuK//ZYGtOnjz5xx9/0L8aRhdrG/2rbZb+HTBgQJMmTcp/NhcXl4EDB5b/\nPBVPhhUtWbJECLFkyRKlC0GloH81jy7WNvpX2+ykf3mNAAAAQCMIdgAAABpBsAMAANAIgh0A\nAIBGEOwAAAA0gmAHAACgEQQ7AAAAjSDYAQAAaATBDgAAQCMIdlbl4uJS8r/QHvpX8+hibaN/\ntc1O+leSZVnpGuyIyWTasGFDp06d9Hq90rWg4tG/mkcXaxv9q2120r8EOwAAAI3gUSwAAIBG\nEOwAAAA0gmAHAACgEQQ7AAAAjSDYAQAAaATBDgAAQCMIdgAAABpBsAMAANAIgh0AAIBGEOwA\nAAA0gmAHAACgEQQ7AAAAjSDYAQAAaATBDgAAQCMIdgAAABpBsAMAANAIgp3VFBz76dWYzhEB\nXt7BLbs8OWnlKaPSFaGs/prSRLqFDp9n/NPmrv3LD8AmHZrdpdlTP52/+UD5O5QetwG37l+u\naPUzn1k7c1TMI/fX9XHzCmzUqutzX+26aL6uhR1fwjKsIXvza/d7COHgE9qxZ/c2dT31QvLt\n+MH+AqXrQpmsecpVCK/gFi2vN2Jx9tUGd+1ffgC2KX/z6NpCRH10+ob95e9QetwW3K5/uaJV\nLj9hTq8gvRBO1cM69X2y7yPh1fRCSL59F6Vda2HXlzDBzhqKd71aVxKGeiN/z5RlWZbNZxf1\n9RfC0Or9vxSuDGWR9mkbIaoO/+12x+/av/wAbI8xL3XP0ildAyVx8z/85e9Qelxpd+pfrmi1\nS/mwjU6Iqg/NOphv2WHO2P76/c5CVP2/pZmybPeXMMHOCoz/G+ErRJX+v+T/s+/i1510QjR4\n47ByZaGs/hhbU4jI95Jvc/iu/csPwNb8+UodJ+mf5xY3/MNf/g6lx5V15/7lila707OihJCi\nPkotvdO0ebS/EG5D1pi4hHnHzgoSNm06Lxw7RT/i/M8+n65do4RI3LgxTbm6UDaXk5LShUNI\nSNBtjt+1f/kB2JqAR8ZNf3/WrFmz3okJvflo+TuUHlfWnfuXK1rt/v77byH8o6ICSu/UhYbW\nF+JSUtJ5LmGCXeUrPno0SYigsDC30nv9GjXyEiIxMVGpslBWSSdOCFFTHH13YNuG/p5uPnWa\nd+w7YUnilauH79q//ABsTkDnZ8eNGzdu3LgXH6t708Hydyg9rrA79i9XtOo1eeW3/ft/nxBx\n3c7CvXsPC+Fap44vlzDBrvLlZGaahPD29r5+t7e3txBZmZnKFIUyk5OS/hYi9afXp2++5Nfq\nsUebVT2/Y+m7MRERL6zPFqIM/csPQF3K36H0uE3jilY7j/uaNG3aoIbTP3uMyQv/M/qHC6L2\nf4Y/ouMSJthVvsLCQiGEu7v79bvd3d2FMOfnFypSFMosLTXN4ObZcNB3+5L3b1q26Jff9yYd\nXDgouPD4pyMmbr5Shv7lB6Au5e9QetymcUVrS+HJNZMee2Dgj8mOLSctnh5l4BIm2FmBj4+P\nEOLSpUvX787NzRWiire30y0/BJtR64WNeXnZR394qsG1rnIK6ff5zCc8xclFi/8sQ//yA1CX\n8ncoPW7TuKI1w3j6fzN6hIV2nfx7fuPB8/7c+HakqxBcwgQ7K3CuWdNTiKysrOt3Z2VlCVGr\nVi1likL5eLRt20SI7MOHz9y9f/kBqEv5O5QeVx+uaLUp+Gvh6AcaP/r6qsyGgz/ecmzvt880\nvnZ/ze4vYYKdFfj7+wtxKjExv/TOCydO5KjiJ2LvZLPJZDKZb9zt4OAghPDwcC9D//IDUJfy\ndyg9bsO4ojXAlDIv5sEBnx/w7jkr7q89377Q3s9Q+rC9X8IEOysI7d69nij439rNpZYjyV2/\ndpusi+retbpydaEsEmc0NRhco7+9/n3Z/B07DghxX+PGHmXoX34A6lL+DqXHbRhXtPqd+Kj/\n8JUXQkfF7lw27oFqN8cYu7+ElZ5Izz4kzmgihMv9k+KvrkaSs/GFUINwfOyb88rWhTI4Milc\nJ6TAgYtPFV/dU3QqdkRDnXBp82GSWZblMvQvPwAbVTA/WtxiAtvydyg9bhNu2b9c0Spn3jMu\nWIhqz6zLv30b+76ECXbWkfvn1LaeQvIOe/w/L74wsEOIm5BqRX+RoIZV53Bp36ePBxiEqFKn\nbc+BT/V7vHWwmxAGv0f/m2C81uSu/csPwDbdJthVQIfS47bg1v3LFa1up2ZHCiE8Qx546Bbe\n+F+RLMt2fgkT7Kzm0r5vxvbt1NTfs2rtll2efG3piSKlK0KZmc9v/fDZPp1b1PV28wpq0r7n\nc5/vyjBf3+Su/csPwAbdLtjJFdGh9Ljibtu/XNEqtnl0tds/hOz5Y0nyst9LWJJl2QoPfAEA\nAFDZGDwBAACgEQQ7AAAAjSDYAQAAaATBDgAAQCMIdgAAABpBsAMAANAIgh0AAIBGEOwAAAA0\ngmAHAACgEQQ7AAAAjSDYAQAAaATBDgAAQCMIdgAAABpBsAMAANAIgh0AAIBGEOwAAAA0gmAH\nAACgEQQ7AAAAjSDYAQAAaATBDgAAQCMIdgAAABpBsAMAANAIgh0AAIBGEOwAAAA0gmAHAACg\nEQQ7AAAAjSDYAQAAaATBDgAAQCMIdgAAABpBsAMAANAIgh0AAIBGEOwAAAA0gmAHwH6sH+wu\nSS5Pri7fWba9GChJzd5NqpiaAKACEewAAAA0gmAHAACgEQQ7ABBCCFNRoVFWuggAKB+CHQD7\nteV5P0nq83Pm9hm9GvpUcXZwcPOrF9VvyuqUon/aFJ9aPWnQoy3r+njWaNi2x8gvduXcGP/k\nzO2fPtfzoaZBXq5eAQ3bD5y09Eie5Yjp2AdRLpJTi6lHjCWtLy7u4yvp6o7derny/34A7A7B\nDoCdS/78/x57I86h3ZNjxo2KDszds/jtHo+9ucMS7YrjP+jYvNvkBVvOuEV0al+3+MBPozu0\nez2uVCYzH5vTremDL/x3S5r3/d2feCTUcHTx5L6t2r6yMUsIoW/44rzJUdK+aUNnHzULIYTI\nXDH2hZ8z6o+d/057VwX+rgA0TwYAe7HuGTchnAetuvbnzaNrCiFEQMzSU8WWPebUrzpVESLo\npV2yLMsnP27vLITv45/EX7IcLkr+qX+wJIQQETNPyLIsn/7qkSrCED56Xbrx6gkubnutpZMQ\ngSM358uyLMvGo+9EOguXBz46bpKz1zxVS+gbjPvzitX+ygDsC3fsANg5qcPEj54INFz9Q0C3\n6GZCnD9/Xghx6JvPtxY4d3j76+cjrt5ecwju/9nUrv/cazvw2Xu/Xakx+OMPHq2hv3oCnwem\nzBoWIFKXLNomCyGEvuEr8ya1NG9/fdi7X7088vtzoeO/nxblYs2/IAA7YlC6AABQVlBkZM1S\nf3RxuZa6io8dSxIiMjq6Vunm3t17tBWr1gshhCg4ciRZiLDiYz9883epJpdzPIQ4fezYedGp\nhhBC3+jleW//0uK110ZsNjR+ffukSOdK/fsAsGcEOwB2rmrVqrc+cP7sWZMQtWr5Xb/bPSjo\n2gdOpaSYhTg0/7n/zL/543l5eULUEEIIoQ8f8WyHSYPXF4XGDGrpWGGVA8CNeBQLALdWMzDQ\nQYizZ9Ou323Myro66FX4+fkJ4TRgmflWb7rsmRBy7SO5a195a73R0dFwaOZz/z1pxb8CAHtD\nsAOAW9M3aFBXiPg1a65LdgVxG7dfm7zEvWHDWqJw5/Z95tItig8veO3FF99Zd+1j2WvG/ufb\n0/XGrFn/UuiVjROGf33KGtUDsEsEOwC4jbD/PN/BuWDDpBH/PXTFssecvv6VsfPOlrSIGjm6\npSHp86EvrDp7Ldvl73wr5pl3PlqeUc3y5l7W6ueHzUur89w30zo9NOnr0XUv/fby8Pmnrfw3\nAWAvCHYAcDv+wz6Y1tbnfOyzreq3eDRmYO8HG4U9/nlm1+4trjWQQl/8/K1Wzgc+7xnWoG3P\np4cPfaJN/QdnJugbv/rDWy0lIUTW8ueH/Xg2aNTX77SrIoRLm+lfjQrKXf/iyPlpd/haALhX\nBDsAuC2HZuM27Yt9e2A7v5y9a9cfKgrpMW3N9nm9Sw2ncI5884+Da6YObFUtY+eyBct359Tu\n+caSvXHvtPUQQlz8dfSIBWmBw76a2dHN0tyt4zv/HRKQvfqlZxcQ7QBUPEmWWRwRAABAC7hj\nBwAAoBEEOwAAAI0g2AEAAGgEwQ4AAEAjCHYAAAAaQbADAADQCIIdAACARhDsAAAANIJgBwAA\noBEEOwAAAI0g2AEAAGgEwQ4AAEAjCHYAAAAaQbADAADQCIIdAACARhDsAAAANIJgBwAAoBEE\nOwAAAI0g2AEAAGgEwQ4AAEAjCHYAAAAaQbADAADQCIIdAACARhDsAAAANIJgBwAAoBEEOwAA\nAI0g2AEAAGgEwQ4AAEAj/h/Oc7xwP9eG1AAAAABJRU5ErkJggg==",
      "text/plain": [
       "plot without title"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "plot(x[order(x[,2]),]$Freq)\n",
    "x[order(x[,2]),][c(1:20),]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "  x$class <- \"unassigned\"\n",
    "  x[grep(pattern = \"A_group1\", x = x$Var1), \"class\"] <- \"A_group1\"\n",
    "  x[grep(pattern = \"A_group2\", x = x$Var1), \"class\"] <- \"A_group2\"\n",
    "  x[grep(pattern = \"A_group3\", x = x$Var1), \"class\"] <- \"A_group3\"\n",
    "  x[grep(pattern = \"A_group4\", x = x$Var1), \"class\"] <- \"A_group4\"\n",
    "  x[grep(pattern = \"B1_group1\", x = x$Var1), \"class\"] <- \"B1_group1\"\n",
    "  x[grep(pattern = \"B1_group2\", x = x$Var1), \"class\"] <- \"B1_group2\"\n",
    "  x[grep(pattern = \"B1_group3\", x = x$Var1), \"class\"] <- \"B1_group3\"\n",
    "  x[grep(pattern = \"B1_group4\", x = x$Var1), \"class\"] <- \"B1_group4\"\n",
    "  x[grep(pattern = \"B1_group5\", x = x$Var1), \"class\"] <- \"B1_group5\"\n",
    "  x[grep(pattern = \"B2_group1\", x = x$Var1), \"class\"] <- \"B2_group1\"\n",
    "  x[grep(pattern = \"B2_group2\", x = x$Var1), \"class\"] <- \"B2_group2\"\n",
    "  x[grep(pattern = \"B2_group3\", x = x$Var1), \"class\"] <- \"B2_group3\"\n",
    "  x[grep(pattern = \"C_group1\", x = x$Var1), \"class\"] <- \"C_group1\"\n",
    "  x[grep(pattern = \"C_group2\", x = x$Var1), \"class\"] <- \"C_group2\"\n",
    "  x[grep(pattern = \"C_group3\", x = x$Var1), \"class\"] <- \"C_group3\"\n",
    "  x[grep(pattern = \"Neg-Control\", x = x$Var1), \"class\"] <- \"Neg-Control\"\n",
    "  x[grep(pattern = \"Random_sequence_trial2\", x = x$Var1), \"class\"] <- \"Random_sequence_trial2\"\n",
    "  x[grep(pattern = \"Random_sequence_trial3\", x = x$Var1), \"class\"] <- \"Random_sequence_trial3\"\n",
    "  x[grep(pattern = \"S4_MEL\", x = x$Var1), \"class\"] <- \"S4_MEL\"\n",
    "  x[grep(pattern = \"S4_MES\", x = x$Var1), \"class\"] <- \"S4_MES\"\n",
    "  x[grep(pattern = \"S4_combi\", x = x$Var1), \"class\"] <- \"S4_combi\"\n",
    "  x[grep(pattern = \"S5_MEL\", x = x$Var1), \"class\"] <- \"S5_MEL\"\n",
    "  x[grep(pattern = \"S5_MES\", x = x$Var1), \"class\"] <- \"S5_MES\"\n",
    "  x[grep(pattern = \"S5_combi\", x = x$Var1), \"class\"] <- \"S5_combi\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'A_group1'</li>\n",
       "\t<li>'A_group2'</li>\n",
       "\t<li>'A_group3'</li>\n",
       "\t<li>'A_group4'</li>\n",
       "\t<li>'B1_group1'</li>\n",
       "\t<li>'B1_group2'</li>\n",
       "\t<li>'B1_group3'</li>\n",
       "\t<li>'B1_group4'</li>\n",
       "\t<li>'B1_group5'</li>\n",
       "\t<li>'B2_group1'</li>\n",
       "\t<li>'B2_group2'</li>\n",
       "\t<li>'B2_group3'</li>\n",
       "\t<li>'C_group1'</li>\n",
       "\t<li>'C_group2'</li>\n",
       "\t<li>'C_group3'</li>\n",
       "\t<li>'Neg-Control'</li>\n",
       "\t<li>'Random_sequence_trial2'</li>\n",
       "\t<li>'Random_sequence_trial3'</li>\n",
       "\t<li>'S4_MEL'</li>\n",
       "\t<li>'S4_MES'</li>\n",
       "\t<li>'S4_combi'</li>\n",
       "\t<li>'S5_MEL'</li>\n",
       "\t<li>'S5_MES'</li>\n",
       "\t<li>'S5_combi'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'A\\_group1'\n",
       "\\item 'A\\_group2'\n",
       "\\item 'A\\_group3'\n",
       "\\item 'A\\_group4'\n",
       "\\item 'B1\\_group1'\n",
       "\\item 'B1\\_group2'\n",
       "\\item 'B1\\_group3'\n",
       "\\item 'B1\\_group4'\n",
       "\\item 'B1\\_group5'\n",
       "\\item 'B2\\_group1'\n",
       "\\item 'B2\\_group2'\n",
       "\\item 'B2\\_group3'\n",
       "\\item 'C\\_group1'\n",
       "\\item 'C\\_group2'\n",
       "\\item 'C\\_group3'\n",
       "\\item 'Neg-Control'\n",
       "\\item 'Random\\_sequence\\_trial2'\n",
       "\\item 'Random\\_sequence\\_trial3'\n",
       "\\item 'S4\\_MEL'\n",
       "\\item 'S4\\_MES'\n",
       "\\item 'S4\\_combi'\n",
       "\\item 'S5\\_MEL'\n",
       "\\item 'S5\\_MES'\n",
       "\\item 'S5\\_combi'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'A_group1'\n",
       "2. 'A_group2'\n",
       "3. 'A_group3'\n",
       "4. 'A_group4'\n",
       "5. 'B1_group1'\n",
       "6. 'B1_group2'\n",
       "7. 'B1_group3'\n",
       "8. 'B1_group4'\n",
       "9. 'B1_group5'\n",
       "10. 'B2_group1'\n",
       "11. 'B2_group2'\n",
       "12. 'B2_group3'\n",
       "13. 'C_group1'\n",
       "14. 'C_group2'\n",
       "15. 'C_group3'\n",
       "16. 'Neg-Control'\n",
       "17. 'Random_sequence_trial2'\n",
       "18. 'Random_sequence_trial3'\n",
       "19. 'S4_MEL'\n",
       "20. 'S4_MES'\n",
       "21. 'S4_combi'\n",
       "22. 'S5_MEL'\n",
       "23. 'S5_MES'\n",
       "24. 'S5_combi'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"A_group1\"               \"A_group2\"               \"A_group3\"              \n",
       " [4] \"A_group4\"               \"B1_group1\"              \"B1_group2\"             \n",
       " [7] \"B1_group3\"              \"B1_group4\"              \"B1_group5\"             \n",
       "[10] \"B2_group1\"              \"B2_group2\"              \"B2_group3\"             \n",
       "[13] \"C_group1\"               \"C_group2\"               \"C_group3\"              \n",
       "[16] \"Neg-Control\"            \"Random_sequence_trial2\" \"Random_sequence_trial3\"\n",
       "[19] \"S4_MEL\"                 \"S4_MES\"                 \"S4_combi\"              \n",
       "[22] \"S5_MEL\"                 \"S5_MES\"                 \"S5_combi\"              "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "unique(x$class)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "library(ggplot2)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 6 x 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Var1</th><th scope=col>Freq</th><th scope=col>class</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>A_group1_seq100_chr9:96385969-96386469  </td><td>2227</td><td>A_group1</td></tr>\n",
       "\t<tr><td>A_group1_seq101_chr9:140681098-140681598</td><td> 270</td><td>A_group1</td></tr>\n",
       "\t<tr><td>A_group1_seq102_chrX:3845338-3845838    </td><td> 447</td><td>A_group1</td></tr>\n",
       "\t<tr><td>A_group1_seq103_chrX:9827376-9827876    </td><td>1824</td><td>A_group1</td></tr>\n",
       "\t<tr><td>A_group1_seq104_chrX:14320175-14320675  </td><td>1041</td><td>A_group1</td></tr>\n",
       "\t<tr><td>A_group1_seq105_chrX:81125209-81125709  </td><td>1176</td><td>A_group1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 x 3\n",
       "\\begin{tabular}{r|lll}\n",
       " Var1 & Freq & class\\\\\n",
       " <fct> & <int> & <chr>\\\\\n",
       "\\hline\n",
       "\t A\\_group1\\_seq100\\_chr9:96385969-96386469   & 2227 & A\\_group1\\\\\n",
       "\t A\\_group1\\_seq101\\_chr9:140681098-140681598 &  270 & A\\_group1\\\\\n",
       "\t A\\_group1\\_seq102\\_chrX:3845338-3845838     &  447 & A\\_group1\\\\\n",
       "\t A\\_group1\\_seq103\\_chrX:9827376-9827876     & 1824 & A\\_group1\\\\\n",
       "\t A\\_group1\\_seq104\\_chrX:14320175-14320675   & 1041 & A\\_group1\\\\\n",
       "\t A\\_group1\\_seq105\\_chrX:81125209-81125709   & 1176 & A\\_group1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 x 3\n",
       "\n",
       "| Var1 &lt;fct&gt; | Freq &lt;int&gt; | class &lt;chr&gt; |\n",
       "|---|---|---|\n",
       "| A_group1_seq100_chr9:96385969-96386469   | 2227 | A_group1 |\n",
       "| A_group1_seq101_chr9:140681098-140681598 |  270 | A_group1 |\n",
       "| A_group1_seq102_chrX:3845338-3845838     |  447 | A_group1 |\n",
       "| A_group1_seq103_chrX:9827376-9827876     | 1824 | A_group1 |\n",
       "| A_group1_seq104_chrX:14320175-14320675   | 1041 | A_group1 |\n",
       "| A_group1_seq105_chrX:81125209-81125709   | 1176 | A_group1 |\n",
       "\n"
      ],
      "text/plain": [
       "  Var1                                     Freq class   \n",
       "1 A_group1_seq100_chr9:96385969-96386469   2227 A_group1\n",
       "2 A_group1_seq101_chr9:140681098-140681598  270 A_group1\n",
       "3 A_group1_seq102_chrX:3845338-3845838      447 A_group1\n",
       "4 A_group1_seq103_chrX:9827376-9827876     1824 A_group1\n",
       "5 A_group1_seq104_chrX:14320175-14320675   1041 A_group1\n",
       "6 A_group1_seq105_chrX:81125209-81125709   1176 A_group1"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(x)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "ggplot(x, aes(x=class,y=Freq))+geom_violin() + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + geom_jitter(shape=16, position=position_jitter(0.2), cex = 0.2) "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Extra. Sequencing saturarion\n",
    "\n",
    "- First extract the rBCs from the fastq and put it in a text file\n",
    "- To calculate the statureation, subsample this rBC text file (choose the subsampling frequency a bit depending on the data) and print the number of cleaned reads (i.e. rBCs) taken and the number of unqiue ones for that subsample.\n",
    "    - e.g. now we have in the 03.sequencing_saturation/${line}_R1_q30.txt files a max of 20,884,696 \n",
    "- You can then plot these numbers in excell in a line graph to assess the saturation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Extract DNA sequences from fastq\n",
    "cd /user/leuven/317/vsc31703/stg_00002/Enhancer_design/MPRA/CHEQ-seq_cloning2/E-BC_coupling/\n",
    "zcat 02.TRIMMED_FQ/CSE__88dc35__CHEQseq-cl2_2000enh_3x_R2.Q20_trimmed5-3_rBCq30_filt.fastq.gz  | sed -n '2~4p' > 04.sequencing_saturation/CSE__88dc35__CHEQseq-cl2_2000enh_3x_R2.txt\n",
    "zcat 02.TRIMMED_FQ/CSE__953401__CHEQseq-cl2_2000enh_1x_R2.Q20_trimmed5-3_rBCq30_filt.fastq.gz  | sed -n '2~4p' > 04.sequencing_saturation/CSE__953401__CHEQseq-cl2_2000enh_1x_R2.txt\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15495989\n",
      "15470589\n"
     ]
    }
   ],
   "source": [
    "# Count total number of rBC in files\n",
    "cat 04.sequencing_saturation/CSE__88dc35__CHEQseq-cl2_2000enh_3x_R2.txt| wc -l\n",
    "cat 04.sequencing_saturation/CSE__953401__CHEQseq-cl2_2000enh_1x_R2.txt| wc -l"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "rm: cannot remove ‘04.sequencing_saturation/CSE__953401__CHEQseq-cl2_2000enh_1x_sat.txt’: No such file or directory\n",
      "bash: syntax error near unexpected token `done'\n",
      "270\n"
     ]
    }
   ],
   "source": [
    "# Subsample, count total and unique reads\n",
    "start=`date +%s`\n",
    "\n",
    "file=\"04.sequencing_saturation/CSE__953401__CHEQseq-cl2_2000enh_1x_R2.txt\"\n",
    "nreads=$(cat $file | wc -l)\n",
    " # Start by removing parallel instructions file\n",
    "rm 04.sequencing_saturation/parallel.txt\n",
    " # Remove any previoulsy created sat values for that file\n",
    "rm ${file%_*}_sat.txt\n",
    "    for i in $(seq 0 200000 $nreads)\n",
    "    do\n",
    "     # Print number of total and unique reads for every subsamples\n",
    "    echo \"echo $i \\$(shuf -n $(($i)) $file | sort | uniq | wc -l) >> ${file%_*}_sat.txt\" \\\n",
    "    >> 04.sequencing_saturation/parallel.txt\n",
    "    done\n",
    "cat 04.sequencing_saturation/parallel.txt | parallel -j 16\n",
    " # Put values in numerical order\n",
    "cat ${file%_*}_sat.txt | sort -g > ${file%_*}_satsort.txt && mv ${file%_*}_satsort.txt ${file%_*}_sat.txt\n",
    "done\n",
    "\n",
    "end=`date +%s`\n",
    "echo $((end-start))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash 4.2.46",
   "language": "bash",
   "name": "bash4246"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
