{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Processing of STARR-seq reads from 500 bp library"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "cd /staging/leuven/stg_00002/lcb/lcb_projects/CSE/500bp_ATAC_peak_library/counting"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 01.fastqc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Skipping '/staging/leuven/stg_00002/lcb/ngs_runs/NovaSeq6000_20200123/Demultiplexed/STA__073ce2__MM029_*R2*' which didn't exist, or couldn't be read\n",
      "Started analysis of STA__073ce2__MM029_STARRseq_500_bp_enh_plasmid_S16_R2_001.fastq.gz\n",
      "Approx 5% complete for STA__073ce2__MM029_STARRseq_500_bp_enh_plasmid_S16_R2_001.fastq.gz\n",
      "Approx 10% complete for STA__073ce2__MM029_STARRseq_500_bp_enh_plasmid_S16_R2_001.fastq.gz\n",
      "Approx 15% complete for STA__073ce2__MM029_STARRseq_500_bp_enh_plasmid_S16_R2_001.fastq.gz\n",
      "Approx 20% complete for STA__073ce2__MM029_STARRseq_500_bp_enh_plasmid_S16_R2_001.fastq.gz\n",
      "Approx 25% complete for STA__073ce2__MM029_STARRseq_500_bp_enh_plasmid_S16_R2_001.fastq.gz\n",
      "Approx 30% complete for STA__073ce2__MM029_STARRseq_500_bp_enh_plasmid_S16_R2_001.fastq.gz\n",
      "Approx 35% complete for STA__073ce2__MM029_STARRseq_500_bp_enh_plasmid_S16_R2_001.fastq.gz\n",
      "Approx 40% complete for STA__073ce2__MM029_STARRseq_500_bp_enh_plasmid_S16_R2_001.fastq.gz\n",
      "Approx 45% complete for STA__073ce2__MM029_STARRseq_500_bp_enh_plasmid_S16_R2_001.fastq.gz\n",
      "Approx 50% complete for STA__073ce2__MM029_STARRseq_500_bp_enh_plasmid_S16_R2_001.fastq.gz\n",
      "Approx 55% complete for STA__073ce2__MM029_STARRseq_500_bp_enh_plasmid_S16_R2_001.fastq.gz\n",
      "Approx 60% complete for STA__073ce2__MM029_STARRseq_500_bp_enh_plasmid_S16_R2_001.fastq.gz\n",
      "Approx 65% complete for STA__073ce2__MM029_STARRseq_500_bp_enh_plasmid_S16_R2_001.fastq.gz\n",
      "Approx 70% complete for STA__073ce2__MM029_STARRseq_500_bp_enh_plasmid_S16_R2_001.fastq.gz\n",
      "Approx 75% complete for STA__073ce2__MM029_STARRseq_500_bp_enh_plasmid_S16_R2_001.fastq.gz\n",
      "Approx 80% complete for STA__073ce2__MM029_STARRseq_500_bp_enh_plasmid_S16_R2_001.fastq.gz\n",
      "Approx 85% complete for STA__073ce2__MM029_STARRseq_500_bp_enh_plasmid_S16_R2_001.fastq.gz\n",
      "Approx 90% complete for STA__073ce2__MM029_STARRseq_500_bp_enh_plasmid_S16_R2_001.fastq.gz\n",
      "Approx 95% complete for STA__073ce2__MM029_STARRseq_500_bp_enh_plasmid_S16_R2_001.fastq.gz\n",
      "Analysis complete for STA__073ce2__MM029_STARRseq_500_bp_enh_plasmid_S16_R2_001.fastq.gz\n",
      "Skipping '/staging/leuven/stg_00002/lcb/ngs_runs/NovaSeq6000_20200123/Demultiplexed/STA__1a9f59__MM087_*R2*' which didn't exist, or couldn't be read\n",
      "Started analysis of STA__1a9f59__MM087_STARRseq_500_bp_enh_plasmid_S20_R2_001.fastq.gz\n",
      "Approx 5% complete for STA__1a9f59__MM087_STARRseq_500_bp_enh_plasmid_S20_R2_001.fastq.gz\n",
      "Approx 10% complete for STA__1a9f59__MM087_STARRseq_500_bp_enh_plasmid_S20_R2_001.fastq.gz\n",
      "Approx 15% complete for STA__1a9f59__MM087_STARRseq_500_bp_enh_plasmid_S20_R2_001.fastq.gz\n",
      "Approx 20% complete for STA__1a9f59__MM087_STARRseq_500_bp_enh_plasmid_S20_R2_001.fastq.gz\n",
      "Approx 25% complete for STA__1a9f59__MM087_STARRseq_500_bp_enh_plasmid_S20_R2_001.fastq.gz\n",
      "Approx 30% complete for STA__1a9f59__MM087_STARRseq_500_bp_enh_plasmid_S20_R2_001.fastq.gz\n",
      "Approx 35% complete for STA__1a9f59__MM087_STARRseq_500_bp_enh_plasmid_S20_R2_001.fastq.gz\n",
      "Approx 40% complete for STA__1a9f59__MM087_STARRseq_500_bp_enh_plasmid_S20_R2_001.fastq.gz\n",
      "Approx 45% complete for STA__1a9f59__MM087_STARRseq_500_bp_enh_plasmid_S20_R2_001.fastq.gz\n",
      "Approx 50% complete for STA__1a9f59__MM087_STARRseq_500_bp_enh_plasmid_S20_R2_001.fastq.gz\n",
      "Approx 55% complete for STA__1a9f59__MM087_STARRseq_500_bp_enh_plasmid_S20_R2_001.fastq.gz\n",
      "Approx 60% complete for STA__1a9f59__MM087_STARRseq_500_bp_enh_plasmid_S20_R2_001.fastq.gz\n",
      "Approx 65% complete for STA__1a9f59__MM087_STARRseq_500_bp_enh_plasmid_S20_R2_001.fastq.gz\n",
      "Approx 70% complete for STA__1a9f59__MM087_STARRseq_500_bp_enh_plasmid_S20_R2_001.fastq.gz\n",
      "Approx 75% complete for STA__1a9f59__MM087_STARRseq_500_bp_enh_plasmid_S20_R2_001.fastq.gz\n",
      "Approx 80% complete for STA__1a9f59__MM087_STARRseq_500_bp_enh_plasmid_S20_R2_001.fastq.gz\n",
      "Approx 85% complete for STA__1a9f59__MM087_STARRseq_500_bp_enh_plasmid_S20_R2_001.fastq.gz\n",
      "Approx 90% complete for STA__1a9f59__MM087_STARRseq_500_bp_enh_plasmid_S20_R2_001.fastq.gz\n",
      "Approx 95% complete for STA__1a9f59__MM087_STARRseq_500_bp_enh_plasmid_S20_R2_001.fastq.gz\n",
      "Analysis complete for STA__1a9f59__MM087_STARRseq_500_bp_enh_plasmid_S20_R2_001.fastq.gz\n",
      "Skipping '/staging/leuven/stg_00002/lcb/ngs_runs/NovaSeq6000_20200123/Demultiplexed/STA__34085e__MM047_*R2*' which didn't exist, or couldn't be read\n",
      "Started analysis of STA__34085e__MM047_STARRseq_500_bp_enh_cDNA_S11_R2_001.fastq.gz\n",
      "Approx 5% complete for STA__34085e__MM047_STARRseq_500_bp_enh_cDNA_S11_R2_001.fastq.gz\n",
      "Approx 10% complete for STA__34085e__MM047_STARRseq_500_bp_enh_cDNA_S11_R2_001.fastq.gz\n",
      "Approx 15% complete for STA__34085e__MM047_STARRseq_500_bp_enh_cDNA_S11_R2_001.fastq.gz\n",
      "Approx 20% complete for STA__34085e__MM047_STARRseq_500_bp_enh_cDNA_S11_R2_001.fastq.gz\n",
      "Approx 25% complete for STA__34085e__MM047_STARRseq_500_bp_enh_cDNA_S11_R2_001.fastq.gz\n",
      "Approx 30% complete for STA__34085e__MM047_STARRseq_500_bp_enh_cDNA_S11_R2_001.fastq.gz\n",
      "Approx 35% complete for STA__34085e__MM047_STARRseq_500_bp_enh_cDNA_S11_R2_001.fastq.gz\n",
      "Approx 40% complete for STA__34085e__MM047_STARRseq_500_bp_enh_cDNA_S11_R2_001.fastq.gz\n",
      "Approx 45% complete for STA__34085e__MM047_STARRseq_500_bp_enh_cDNA_S11_R2_001.fastq.gz\n",
      "Approx 50% complete for STA__34085e__MM047_STARRseq_500_bp_enh_cDNA_S11_R2_001.fastq.gz\n",
      "Approx 55% complete for STA__34085e__MM047_STARRseq_500_bp_enh_cDNA_S11_R2_001.fastq.gz\n",
      "Approx 60% complete for STA__34085e__MM047_STARRseq_500_bp_enh_cDNA_S11_R2_001.fastq.gz\n",
      "Approx 65% complete for STA__34085e__MM047_STARRseq_500_bp_enh_cDNA_S11_R2_001.fastq.gz\n",
      "Approx 70% complete for STA__34085e__MM047_STARRseq_500_bp_enh_cDNA_S11_R2_001.fastq.gz\n",
      "Approx 75% complete for STA__34085e__MM047_STARRseq_500_bp_enh_cDNA_S11_R2_001.fastq.gz\n",
      "Approx 80% complete for STA__34085e__MM047_STARRseq_500_bp_enh_cDNA_S11_R2_001.fastq.gz\n",
      "Approx 85% complete for STA__34085e__MM047_STARRseq_500_bp_enh_cDNA_S11_R2_001.fastq.gz\n",
      "Approx 90% complete for STA__34085e__MM047_STARRseq_500_bp_enh_cDNA_S11_R2_001.fastq.gz\n",
      "Approx 95% complete for STA__34085e__MM047_STARRseq_500_bp_enh_cDNA_S11_R2_001.fastq.gz\n",
      "Analysis complete for STA__34085e__MM047_STARRseq_500_bp_enh_cDNA_S11_R2_001.fastq.gz\n",
      "Skipping '/staging/leuven/stg_00002/lcb/ngs_runs/NovaSeq6000_20200123/Demultiplexed/STA__3f00ec__MM001_*R2*' which didn't exist, or couldn't be read\n",
      "Started analysis of STA__3f00ec__MM001_STARRseq_500_bp_enh_cDNA_S9_R2_001.fastq.gz\n",
      "Approx 5% complete for STA__3f00ec__MM001_STARRseq_500_bp_enh_cDNA_S9_R2_001.fastq.gz\n",
      "Approx 10% complete for STA__3f00ec__MM001_STARRseq_500_bp_enh_cDNA_S9_R2_001.fastq.gz\n",
      "Approx 15% complete for STA__3f00ec__MM001_STARRseq_500_bp_enh_cDNA_S9_R2_001.fastq.gz\n",
      "Approx 20% complete for STA__3f00ec__MM001_STARRseq_500_bp_enh_cDNA_S9_R2_001.fastq.gz\n",
      "Approx 25% complete for STA__3f00ec__MM001_STARRseq_500_bp_enh_cDNA_S9_R2_001.fastq.gz\n",
      "Approx 30% complete for STA__3f00ec__MM001_STARRseq_500_bp_enh_cDNA_S9_R2_001.fastq.gz\n",
      "Approx 35% complete for STA__3f00ec__MM001_STARRseq_500_bp_enh_cDNA_S9_R2_001.fastq.gz\n",
      "Approx 40% complete for STA__3f00ec__MM001_STARRseq_500_bp_enh_cDNA_S9_R2_001.fastq.gz\n",
      "Approx 45% complete for STA__3f00ec__MM001_STARRseq_500_bp_enh_cDNA_S9_R2_001.fastq.gz\n",
      "Approx 50% complete for STA__3f00ec__MM001_STARRseq_500_bp_enh_cDNA_S9_R2_001.fastq.gz\n",
      "Approx 55% complete for STA__3f00ec__MM001_STARRseq_500_bp_enh_cDNA_S9_R2_001.fastq.gz\n",
      "Approx 60% complete for STA__3f00ec__MM001_STARRseq_500_bp_enh_cDNA_S9_R2_001.fastq.gz\n",
      "Approx 65% complete for STA__3f00ec__MM001_STARRseq_500_bp_enh_cDNA_S9_R2_001.fastq.gz\n",
      "Approx 70% complete for STA__3f00ec__MM001_STARRseq_500_bp_enh_cDNA_S9_R2_001.fastq.gz\n",
      "Approx 75% complete for STA__3f00ec__MM001_STARRseq_500_bp_enh_cDNA_S9_R2_001.fastq.gz\n",
      "Approx 80% complete for STA__3f00ec__MM001_STARRseq_500_bp_enh_cDNA_S9_R2_001.fastq.gz\n",
      "Approx 85% complete for STA__3f00ec__MM001_STARRseq_500_bp_enh_cDNA_S9_R2_001.fastq.gz\n",
      "Approx 90% complete for STA__3f00ec__MM001_STARRseq_500_bp_enh_cDNA_S9_R2_001.fastq.gz\n",
      "Approx 95% complete for STA__3f00ec__MM001_STARRseq_500_bp_enh_cDNA_S9_R2_001.fastq.gz\n",
      "Analysis complete for STA__3f00ec__MM001_STARRseq_500_bp_enh_cDNA_S9_R2_001.fastq.gz\n",
      "Skipping '/staging/leuven/stg_00002/lcb/ngs_runs/NovaSeq6000_20200123/Demultiplexed/STA__4642d6__MM087_*R2*' which didn't exist, or couldn't be read\n",
      "Started analysis of STA__4642d6__MM087_STARRseq_500_bp_enh_cDNA_S14_R2_001.fastq.gz\n",
      "Approx 5% complete for STA__4642d6__MM087_STARRseq_500_bp_enh_cDNA_S14_R2_001.fastq.gz\n",
      "Approx 10% complete for STA__4642d6__MM087_STARRseq_500_bp_enh_cDNA_S14_R2_001.fastq.gz\n",
      "Approx 15% complete for STA__4642d6__MM087_STARRseq_500_bp_enh_cDNA_S14_R2_001.fastq.gz\n",
      "Approx 20% complete for STA__4642d6__MM087_STARRseq_500_bp_enh_cDNA_S14_R2_001.fastq.gz\n",
      "Approx 25% complete for STA__4642d6__MM087_STARRseq_500_bp_enh_cDNA_S14_R2_001.fastq.gz\n",
      "Approx 30% complete for STA__4642d6__MM087_STARRseq_500_bp_enh_cDNA_S14_R2_001.fastq.gz\n",
      "Approx 35% complete for STA__4642d6__MM087_STARRseq_500_bp_enh_cDNA_S14_R2_001.fastq.gz\n",
      "Approx 40% complete for STA__4642d6__MM087_STARRseq_500_bp_enh_cDNA_S14_R2_001.fastq.gz\n",
      "Approx 45% complete for STA__4642d6__MM087_STARRseq_500_bp_enh_cDNA_S14_R2_001.fastq.gz\n",
      "Approx 50% complete for STA__4642d6__MM087_STARRseq_500_bp_enh_cDNA_S14_R2_001.fastq.gz\n",
      "Approx 55% complete for STA__4642d6__MM087_STARRseq_500_bp_enh_cDNA_S14_R2_001.fastq.gz\n",
      "Approx 60% complete for STA__4642d6__MM087_STARRseq_500_bp_enh_cDNA_S14_R2_001.fastq.gz\n",
      "Approx 65% complete for STA__4642d6__MM087_STARRseq_500_bp_enh_cDNA_S14_R2_001.fastq.gz\n",
      "Approx 70% complete for STA__4642d6__MM087_STARRseq_500_bp_enh_cDNA_S14_R2_001.fastq.gz\n",
      "Approx 75% complete for STA__4642d6__MM087_STARRseq_500_bp_enh_cDNA_S14_R2_001.fastq.gz\n",
      "Approx 80% complete for STA__4642d6__MM087_STARRseq_500_bp_enh_cDNA_S14_R2_001.fastq.gz\n",
      "Approx 85% complete for STA__4642d6__MM087_STARRseq_500_bp_enh_cDNA_S14_R2_001.fastq.gz\n",
      "Approx 90% complete for STA__4642d6__MM087_STARRseq_500_bp_enh_cDNA_S14_R2_001.fastq.gz\n",
      "Approx 95% complete for STA__4642d6__MM087_STARRseq_500_bp_enh_cDNA_S14_R2_001.fastq.gz\n",
      "Analysis complete for STA__4642d6__MM087_STARRseq_500_bp_enh_cDNA_S14_R2_001.fastq.gz\n",
      "Skipping '/staging/leuven/stg_00002/lcb/ngs_runs/NovaSeq6000_20200123/Demultiplexed/STA__60a828__MM047_*R2*' which didn't exist, or couldn't be read\n",
      "Started analysis of STA__60a828__MM047_STARRseq_500_bp_enh_plasmid_S17_R2_001.fastq.gz\n",
      "Approx 5% complete for STA__60a828__MM047_STARRseq_500_bp_enh_plasmid_S17_R2_001.fastq.gz\n",
      "Approx 10% complete for STA__60a828__MM047_STARRseq_500_bp_enh_plasmid_S17_R2_001.fastq.gz\n",
      "Approx 15% complete for STA__60a828__MM047_STARRseq_500_bp_enh_plasmid_S17_R2_001.fastq.gz\n",
      "Approx 20% complete for STA__60a828__MM047_STARRseq_500_bp_enh_plasmid_S17_R2_001.fastq.gz\n",
      "Approx 25% complete for STA__60a828__MM047_STARRseq_500_bp_enh_plasmid_S17_R2_001.fastq.gz\n",
      "Approx 30% complete for STA__60a828__MM047_STARRseq_500_bp_enh_plasmid_S17_R2_001.fastq.gz\n",
      "Approx 35% complete for STA__60a828__MM047_STARRseq_500_bp_enh_plasmid_S17_R2_001.fastq.gz\n",
      "Approx 40% complete for STA__60a828__MM047_STARRseq_500_bp_enh_plasmid_S17_R2_001.fastq.gz\n",
      "Approx 45% complete for STA__60a828__MM047_STARRseq_500_bp_enh_plasmid_S17_R2_001.fastq.gz\n",
      "Approx 50% complete for STA__60a828__MM047_STARRseq_500_bp_enh_plasmid_S17_R2_001.fastq.gz\n",
      "Approx 55% complete for STA__60a828__MM047_STARRseq_500_bp_enh_plasmid_S17_R2_001.fastq.gz\n",
      "Approx 60% complete for STA__60a828__MM047_STARRseq_500_bp_enh_plasmid_S17_R2_001.fastq.gz\n",
      "Approx 65% complete for STA__60a828__MM047_STARRseq_500_bp_enh_plasmid_S17_R2_001.fastq.gz\n",
      "Approx 70% complete for STA__60a828__MM047_STARRseq_500_bp_enh_plasmid_S17_R2_001.fastq.gz\n",
      "Approx 75% complete for STA__60a828__MM047_STARRseq_500_bp_enh_plasmid_S17_R2_001.fastq.gz\n",
      "Approx 80% complete for STA__60a828__MM047_STARRseq_500_bp_enh_plasmid_S17_R2_001.fastq.gz\n",
      "Approx 85% complete for STA__60a828__MM047_STARRseq_500_bp_enh_plasmid_S17_R2_001.fastq.gz\n",
      "Approx 90% complete for STA__60a828__MM047_STARRseq_500_bp_enh_plasmid_S17_R2_001.fastq.gz\n",
      "Approx 95% complete for STA__60a828__MM047_STARRseq_500_bp_enh_plasmid_S17_R2_001.fastq.gz\n",
      "Analysis complete for STA__60a828__MM047_STARRseq_500_bp_enh_plasmid_S17_R2_001.fastq.gz\n",
      "Skipping '/staging/leuven/stg_00002/lcb/ngs_runs/NovaSeq6000_20200123/Demultiplexed/STA__616867__MM029_*R2*' which didn't exist, or couldn't be read\n",
      "Started analysis of STA__616867__MM029_STARRseq_500_bp_enh_cDNA_S10_R2_001.fastq.gz\n",
      "Approx 5% complete for STA__616867__MM029_STARRseq_500_bp_enh_cDNA_S10_R2_001.fastq.gz\n",
      "Approx 10% complete for STA__616867__MM029_STARRseq_500_bp_enh_cDNA_S10_R2_001.fastq.gz\n",
      "Approx 15% complete for STA__616867__MM029_STARRseq_500_bp_enh_cDNA_S10_R2_001.fastq.gz\n",
      "Approx 20% complete for STA__616867__MM029_STARRseq_500_bp_enh_cDNA_S10_R2_001.fastq.gz\n",
      "Approx 25% complete for STA__616867__MM029_STARRseq_500_bp_enh_cDNA_S10_R2_001.fastq.gz\n",
      "Approx 30% complete for STA__616867__MM029_STARRseq_500_bp_enh_cDNA_S10_R2_001.fastq.gz\n",
      "Approx 35% complete for STA__616867__MM029_STARRseq_500_bp_enh_cDNA_S10_R2_001.fastq.gz\n",
      "Approx 40% complete for STA__616867__MM029_STARRseq_500_bp_enh_cDNA_S10_R2_001.fastq.gz\n",
      "Approx 45% complete for STA__616867__MM029_STARRseq_500_bp_enh_cDNA_S10_R2_001.fastq.gz\n",
      "Approx 50% complete for STA__616867__MM029_STARRseq_500_bp_enh_cDNA_S10_R2_001.fastq.gz\n",
      "Approx 55% complete for STA__616867__MM029_STARRseq_500_bp_enh_cDNA_S10_R2_001.fastq.gz\n",
      "Approx 60% complete for STA__616867__MM029_STARRseq_500_bp_enh_cDNA_S10_R2_001.fastq.gz\n",
      "Approx 65% complete for STA__616867__MM029_STARRseq_500_bp_enh_cDNA_S10_R2_001.fastq.gz\n",
      "Approx 70% complete for STA__616867__MM029_STARRseq_500_bp_enh_cDNA_S10_R2_001.fastq.gz\n",
      "Approx 75% complete for STA__616867__MM029_STARRseq_500_bp_enh_cDNA_S10_R2_001.fastq.gz\n",
      "Approx 80% complete for STA__616867__MM029_STARRseq_500_bp_enh_cDNA_S10_R2_001.fastq.gz\n",
      "Approx 85% complete for STA__616867__MM029_STARRseq_500_bp_enh_cDNA_S10_R2_001.fastq.gz\n",
      "Approx 90% complete for STA__616867__MM029_STARRseq_500_bp_enh_cDNA_S10_R2_001.fastq.gz\n",
      "Approx 95% complete for STA__616867__MM029_STARRseq_500_bp_enh_cDNA_S10_R2_001.fastq.gz\n",
      "Analysis complete for STA__616867__MM029_STARRseq_500_bp_enh_cDNA_S10_R2_001.fastq.gz\n",
      "Skipping '/staging/leuven/stg_00002/lcb/ngs_runs/NovaSeq6000_20200123/Demultiplexed/STA__a6d28a__MM001_*R2*' which didn't exist, or couldn't be read\n",
      "Started analysis of STA__a6d28a__MM001_STARRseq_500_bp_enh_plasmid_S15_R2_001.fastq.gz\n",
      "Approx 5% complete for STA__a6d28a__MM001_STARRseq_500_bp_enh_plasmid_S15_R2_001.fastq.gz\n",
      "Approx 10% complete for STA__a6d28a__MM001_STARRseq_500_bp_enh_plasmid_S15_R2_001.fastq.gz\n",
      "Approx 15% complete for STA__a6d28a__MM001_STARRseq_500_bp_enh_plasmid_S15_R2_001.fastq.gz\n",
      "Approx 20% complete for STA__a6d28a__MM001_STARRseq_500_bp_enh_plasmid_S15_R2_001.fastq.gz\n",
      "Approx 25% complete for STA__a6d28a__MM001_STARRseq_500_bp_enh_plasmid_S15_R2_001.fastq.gz\n",
      "Approx 30% complete for STA__a6d28a__MM001_STARRseq_500_bp_enh_plasmid_S15_R2_001.fastq.gz\n",
      "Approx 35% complete for STA__a6d28a__MM001_STARRseq_500_bp_enh_plasmid_S15_R2_001.fastq.gz\n",
      "Approx 40% complete for STA__a6d28a__MM001_STARRseq_500_bp_enh_plasmid_S15_R2_001.fastq.gz\n",
      "Approx 45% complete for STA__a6d28a__MM001_STARRseq_500_bp_enh_plasmid_S15_R2_001.fastq.gz\n",
      "Approx 50% complete for STA__a6d28a__MM001_STARRseq_500_bp_enh_plasmid_S15_R2_001.fastq.gz\n",
      "Approx 55% complete for STA__a6d28a__MM001_STARRseq_500_bp_enh_plasmid_S15_R2_001.fastq.gz\n",
      "Approx 60% complete for STA__a6d28a__MM001_STARRseq_500_bp_enh_plasmid_S15_R2_001.fastq.gz\n",
      "Approx 65% complete for STA__a6d28a__MM001_STARRseq_500_bp_enh_plasmid_S15_R2_001.fastq.gz\n",
      "Approx 70% complete for STA__a6d28a__MM001_STARRseq_500_bp_enh_plasmid_S15_R2_001.fastq.gz\n",
      "Approx 75% complete for STA__a6d28a__MM001_STARRseq_500_bp_enh_plasmid_S15_R2_001.fastq.gz\n",
      "Approx 80% complete for STA__a6d28a__MM001_STARRseq_500_bp_enh_plasmid_S15_R2_001.fastq.gz\n",
      "Approx 85% complete for STA__a6d28a__MM001_STARRseq_500_bp_enh_plasmid_S15_R2_001.fastq.gz\n",
      "Approx 90% complete for STA__a6d28a__MM001_STARRseq_500_bp_enh_plasmid_S15_R2_001.fastq.gz\n",
      "Approx 95% complete for STA__a6d28a__MM001_STARRseq_500_bp_enh_plasmid_S15_R2_001.fastq.gz\n",
      "Analysis complete for STA__a6d28a__MM001_STARRseq_500_bp_enh_plasmid_S15_R2_001.fastq.gz\n",
      "Skipping '/staging/leuven/stg_00002/lcb/ngs_runs/NovaSeq6000_20200123/Demultiplexed/STA__af2595__MM074_*R2*' which didn't exist, or couldn't be read\n",
      "Started analysis of STA__af2595__MM074_STARRseq_500_bp_enh_cDNA_S13_R2_001.fastq.gz\n",
      "Approx 5% complete for STA__af2595__MM074_STARRseq_500_bp_enh_cDNA_S13_R2_001.fastq.gz\n",
      "Approx 10% complete for STA__af2595__MM074_STARRseq_500_bp_enh_cDNA_S13_R2_001.fastq.gz\n",
      "Approx 15% complete for STA__af2595__MM074_STARRseq_500_bp_enh_cDNA_S13_R2_001.fastq.gz\n",
      "Approx 20% complete for STA__af2595__MM074_STARRseq_500_bp_enh_cDNA_S13_R2_001.fastq.gz\n",
      "Approx 25% complete for STA__af2595__MM074_STARRseq_500_bp_enh_cDNA_S13_R2_001.fastq.gz\n",
      "Approx 30% complete for STA__af2595__MM074_STARRseq_500_bp_enh_cDNA_S13_R2_001.fastq.gz\n",
      "Approx 35% complete for STA__af2595__MM074_STARRseq_500_bp_enh_cDNA_S13_R2_001.fastq.gz\n",
      "Approx 40% complete for STA__af2595__MM074_STARRseq_500_bp_enh_cDNA_S13_R2_001.fastq.gz\n",
      "Approx 45% complete for STA__af2595__MM074_STARRseq_500_bp_enh_cDNA_S13_R2_001.fastq.gz\n",
      "Approx 50% complete for STA__af2595__MM074_STARRseq_500_bp_enh_cDNA_S13_R2_001.fastq.gz\n",
      "Approx 55% complete for STA__af2595__MM074_STARRseq_500_bp_enh_cDNA_S13_R2_001.fastq.gz\n",
      "Approx 60% complete for STA__af2595__MM074_STARRseq_500_bp_enh_cDNA_S13_R2_001.fastq.gz\n",
      "Approx 65% complete for STA__af2595__MM074_STARRseq_500_bp_enh_cDNA_S13_R2_001.fastq.gz\n",
      "Approx 70% complete for STA__af2595__MM074_STARRseq_500_bp_enh_cDNA_S13_R2_001.fastq.gz\n",
      "Approx 75% complete for STA__af2595__MM074_STARRseq_500_bp_enh_cDNA_S13_R2_001.fastq.gz\n",
      "Approx 80% complete for STA__af2595__MM074_STARRseq_500_bp_enh_cDNA_S13_R2_001.fastq.gz\n",
      "Approx 85% complete for STA__af2595__MM074_STARRseq_500_bp_enh_cDNA_S13_R2_001.fastq.gz\n",
      "Approx 90% complete for STA__af2595__MM074_STARRseq_500_bp_enh_cDNA_S13_R2_001.fastq.gz\n",
      "Approx 95% complete for STA__af2595__MM074_STARRseq_500_bp_enh_cDNA_S13_R2_001.fastq.gz\n",
      "Analysis complete for STA__af2595__MM074_STARRseq_500_bp_enh_cDNA_S13_R2_001.fastq.gz\n",
      "Skipping '/staging/leuven/stg_00002/lcb/ngs_runs/NovaSeq6000_20200123/Demultiplexed/STA__c67b3a__MM057_*R2*' which didn't exist, or couldn't be read\n",
      "Started analysis of STA__c67b3a__MM057_STARRseq_500_bp_enh_plasmid_S18_R2_001.fastq.gz\n",
      "Approx 5% complete for STA__c67b3a__MM057_STARRseq_500_bp_enh_plasmid_S18_R2_001.fastq.gz\n",
      "Approx 10% complete for STA__c67b3a__MM057_STARRseq_500_bp_enh_plasmid_S18_R2_001.fastq.gz\n",
      "Approx 15% complete for STA__c67b3a__MM057_STARRseq_500_bp_enh_plasmid_S18_R2_001.fastq.gz\n",
      "Approx 20% complete for STA__c67b3a__MM057_STARRseq_500_bp_enh_plasmid_S18_R2_001.fastq.gz\n",
      "Approx 25% complete for STA__c67b3a__MM057_STARRseq_500_bp_enh_plasmid_S18_R2_001.fastq.gz\n",
      "Approx 30% complete for STA__c67b3a__MM057_STARRseq_500_bp_enh_plasmid_S18_R2_001.fastq.gz\n",
      "Approx 35% complete for STA__c67b3a__MM057_STARRseq_500_bp_enh_plasmid_S18_R2_001.fastq.gz\n",
      "Approx 40% complete for STA__c67b3a__MM057_STARRseq_500_bp_enh_plasmid_S18_R2_001.fastq.gz\n",
      "Approx 45% complete for STA__c67b3a__MM057_STARRseq_500_bp_enh_plasmid_S18_R2_001.fastq.gz\n",
      "Approx 50% complete for STA__c67b3a__MM057_STARRseq_500_bp_enh_plasmid_S18_R2_001.fastq.gz\n",
      "Approx 55% complete for STA__c67b3a__MM057_STARRseq_500_bp_enh_plasmid_S18_R2_001.fastq.gz\n",
      "Approx 60% complete for STA__c67b3a__MM057_STARRseq_500_bp_enh_plasmid_S18_R2_001.fastq.gz\n",
      "Approx 65% complete for STA__c67b3a__MM057_STARRseq_500_bp_enh_plasmid_S18_R2_001.fastq.gz\n",
      "Approx 70% complete for STA__c67b3a__MM057_STARRseq_500_bp_enh_plasmid_S18_R2_001.fastq.gz\n",
      "Approx 75% complete for STA__c67b3a__MM057_STARRseq_500_bp_enh_plasmid_S18_R2_001.fastq.gz\n",
      "Approx 80% complete for STA__c67b3a__MM057_STARRseq_500_bp_enh_plasmid_S18_R2_001.fastq.gz\n",
      "Approx 85% complete for STA__c67b3a__MM057_STARRseq_500_bp_enh_plasmid_S18_R2_001.fastq.gz\n",
      "Approx 90% complete for STA__c67b3a__MM057_STARRseq_500_bp_enh_plasmid_S18_R2_001.fastq.gz\n",
      "Approx 95% complete for STA__c67b3a__MM057_STARRseq_500_bp_enh_plasmid_S18_R2_001.fastq.gz\n",
      "Analysis complete for STA__c67b3a__MM057_STARRseq_500_bp_enh_plasmid_S18_R2_001.fastq.gz\n",
      "Skipping '/staging/leuven/stg_00002/lcb/ngs_runs/NovaSeq6000_20200123/Demultiplexed/STA__e17937__MM074_*R2*' which didn't exist, or couldn't be read\n",
      "Started analysis of STA__e17937__MM074_STARRseq_500_bp_enh_plasmid_S19_R2_001.fastq.gz\n",
      "Approx 5% complete for STA__e17937__MM074_STARRseq_500_bp_enh_plasmid_S19_R2_001.fastq.gz\n",
      "Approx 10% complete for STA__e17937__MM074_STARRseq_500_bp_enh_plasmid_S19_R2_001.fastq.gz\n",
      "Approx 15% complete for STA__e17937__MM074_STARRseq_500_bp_enh_plasmid_S19_R2_001.fastq.gz\n",
      "Approx 20% complete for STA__e17937__MM074_STARRseq_500_bp_enh_plasmid_S19_R2_001.fastq.gz\n",
      "Approx 25% complete for STA__e17937__MM074_STARRseq_500_bp_enh_plasmid_S19_R2_001.fastq.gz\n",
      "Approx 30% complete for STA__e17937__MM074_STARRseq_500_bp_enh_plasmid_S19_R2_001.fastq.gz\n",
      "Approx 35% complete for STA__e17937__MM074_STARRseq_500_bp_enh_plasmid_S19_R2_001.fastq.gz\n",
      "Approx 40% complete for STA__e17937__MM074_STARRseq_500_bp_enh_plasmid_S19_R2_001.fastq.gz\n",
      "Approx 45% complete for STA__e17937__MM074_STARRseq_500_bp_enh_plasmid_S19_R2_001.fastq.gz\n",
      "Approx 50% complete for STA__e17937__MM074_STARRseq_500_bp_enh_plasmid_S19_R2_001.fastq.gz\n",
      "Approx 55% complete for STA__e17937__MM074_STARRseq_500_bp_enh_plasmid_S19_R2_001.fastq.gz\n",
      "Approx 60% complete for STA__e17937__MM074_STARRseq_500_bp_enh_plasmid_S19_R2_001.fastq.gz\n",
      "Approx 65% complete for STA__e17937__MM074_STARRseq_500_bp_enh_plasmid_S19_R2_001.fastq.gz\n",
      "Approx 70% complete for STA__e17937__MM074_STARRseq_500_bp_enh_plasmid_S19_R2_001.fastq.gz\n",
      "Approx 75% complete for STA__e17937__MM074_STARRseq_500_bp_enh_plasmid_S19_R2_001.fastq.gz\n",
      "Approx 80% complete for STA__e17937__MM074_STARRseq_500_bp_enh_plasmid_S19_R2_001.fastq.gz\n",
      "Approx 85% complete for STA__e17937__MM074_STARRseq_500_bp_enh_plasmid_S19_R2_001.fastq.gz\n",
      "Approx 90% complete for STA__e17937__MM074_STARRseq_500_bp_enh_plasmid_S19_R2_001.fastq.gz\n",
      "Approx 95% complete for STA__e17937__MM074_STARRseq_500_bp_enh_plasmid_S19_R2_001.fastq.gz\n",
      "Analysis complete for STA__e17937__MM074_STARRseq_500_bp_enh_plasmid_S19_R2_001.fastq.gz\n",
      "Skipping '/staging/leuven/stg_00002/lcb/ngs_runs/NovaSeq6000_20200123/Demultiplexed/STA__ed22cd__MM057_*R2*' which didn't exist, or couldn't be read\n",
      "Started analysis of STA__ed22cd__MM057_STARRseq_500_bp_enh_cDNA_S12_R2_001.fastq.gz\n",
      "Approx 5% complete for STA__ed22cd__MM057_STARRseq_500_bp_enh_cDNA_S12_R2_001.fastq.gz\n",
      "Approx 10% complete for STA__ed22cd__MM057_STARRseq_500_bp_enh_cDNA_S12_R2_001.fastq.gz\n",
      "Approx 15% complete for STA__ed22cd__MM057_STARRseq_500_bp_enh_cDNA_S12_R2_001.fastq.gz\n",
      "Approx 20% complete for STA__ed22cd__MM057_STARRseq_500_bp_enh_cDNA_S12_R2_001.fastq.gz\n",
      "Approx 25% complete for STA__ed22cd__MM057_STARRseq_500_bp_enh_cDNA_S12_R2_001.fastq.gz\n",
      "Approx 30% complete for STA__ed22cd__MM057_STARRseq_500_bp_enh_cDNA_S12_R2_001.fastq.gz\n",
      "Approx 35% complete for STA__ed22cd__MM057_STARRseq_500_bp_enh_cDNA_S12_R2_001.fastq.gz\n",
      "Approx 40% complete for STA__ed22cd__MM057_STARRseq_500_bp_enh_cDNA_S12_R2_001.fastq.gz\n",
      "Approx 45% complete for STA__ed22cd__MM057_STARRseq_500_bp_enh_cDNA_S12_R2_001.fastq.gz\n",
      "Approx 50% complete for STA__ed22cd__MM057_STARRseq_500_bp_enh_cDNA_S12_R2_001.fastq.gz\n",
      "Approx 55% complete for STA__ed22cd__MM057_STARRseq_500_bp_enh_cDNA_S12_R2_001.fastq.gz\n",
      "Approx 60% complete for STA__ed22cd__MM057_STARRseq_500_bp_enh_cDNA_S12_R2_001.fastq.gz\n",
      "Approx 65% complete for STA__ed22cd__MM057_STARRseq_500_bp_enh_cDNA_S12_R2_001.fastq.gz\n",
      "Approx 70% complete for STA__ed22cd__MM057_STARRseq_500_bp_enh_cDNA_S12_R2_001.fastq.gz\n",
      "Approx 75% complete for STA__ed22cd__MM057_STARRseq_500_bp_enh_cDNA_S12_R2_001.fastq.gz\n",
      "Approx 80% complete for STA__ed22cd__MM057_STARRseq_500_bp_enh_cDNA_S12_R2_001.fastq.gz\n",
      "Approx 85% complete for STA__ed22cd__MM057_STARRseq_500_bp_enh_cDNA_S12_R2_001.fastq.gz\n",
      "Approx 90% complete for STA__ed22cd__MM057_STARRseq_500_bp_enh_cDNA_S12_R2_001.fastq.gz\n",
      "Approx 95% complete for STA__ed22cd__MM057_STARRseq_500_bp_enh_cDNA_S12_R2_001.fastq.gz\n",
      "Analysis complete for STA__ed22cd__MM057_STARRseq_500_bp_enh_cDNA_S12_R2_001.fastq.gz\n"
     ]
    }
   ],
   "source": [
    "module load FastQC/0.11.8-Java-1.8.0_162\n",
    "cat STARRseq_names.txt | while read line\n",
    "do\n",
    "NovaSeq6000_200123=\"/staging/leuven/stg_00002/lcb/ngs_runs/NovaSeq6000_20200123/Demultiplexed/${line%STARR*}*R2*\"\n",
    "NovaSeq6000_200416=\"/staging/leuven/stg_00002/lcb/ngs_runs/NovaSeq6000_20200416/Demultiplexed/${line%STARR*}*R2*\"\n",
    "fastqc -t 10 ${NovaSeq6000_200123} ${NovaSeq6000_200416} -o 01.fastqc\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 02.clean_reads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is cutadapt 1.18 with Python 3.7.0\n",
      "Command line parameters: -g CTTATCATGTCTGCTCGAAGCGCACTGAGCCTCCACCTAGCCT -e 0.15 -m 7 -M 7 -j 10 --discard-untrimmed /staging/leuven/stg_00002/lcb/ngs_runs/NovaSeq6000_20200416/Demultiplexed/STA__073ce2__MM029_STARRseq_500_bp_enh_plasmid_S16_R2_001.fastq.gz\n",
      "Processing reads on 10 cores in single-end mode ...\n",
      "Finished in 43.34 s (4 us/read; 16.28 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              11,762,241\n",
      "Reads with adapters:                10,665,592 (90.7%)\n",
      "Reads that were too short:             366,781 (3.1%)\n",
      "Reads that were too long:            1,377,004 (11.7%)\n",
      "Reads written (passing filters):    10,018,456 (85.2%)\n",
      "\n",
      "Total basepairs processed:   588,112,050 bp\n",
      "Total written (filtered):     70,129,192 bp (11.9%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTATCATGTCTGCTCGAAGCGCACTGAGCCTCCACCTAGCCT; Type: regular 5'; Length: 43; Trimmed: 10665592 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20-25 bp: 3; 26-32 bp: 4; 33-39 bp: 5; 40-43 bp: 6\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t541\t183785.0\t0\t541\n",
      "4\t14\t45946.3\t0\t14\n",
      "5\t2\t11486.6\t0\t2\n",
      "6\t10\t2871.6\t0\t0 10\n",
      "7\t136\t717.9\t1\t12 124\n",
      "8\t1088\t179.5\t1\t1 1087\n",
      "33\t1\t0.0\t4\t0 1\n",
      "34\t1\t0.0\t5\t0 1\n",
      "35\t1\t0.0\t5\t0 0 0 1\n",
      "36\t7\t0.0\t5\t1 0 1 1 2 0 2\n",
      "37\t19\t0.0\t5\t3 2 0 6 0 1 7\n",
      "38\t93\t0.0\t5\t17 6 8 3 3 17 39\n",
      "39\t263\t0.0\t5\t9 29 22 21 32 51 99\n",
      "40\t1549\t0.0\t6\t83 115 101 261 262 306 421\n",
      "41\t13042\t0.0\t6\t534 1024 3415 2481 1863 1713 2012\n",
      "42\t263588\t0.0\t6\t37777 110898 45618 26543 17681 13003 12068\n",
      "43\t10018456\t0.0\t6\t7659298 1505981 480504 188341 95167 54906 34259\n",
      "44\t103463\t0.0\t6\t1064 50723 23173 12679 7427 4924 3473\n",
      "45\t73752\t0.0\t6\t5 13 30686 17879 11514 7831 5824\n",
      "46\t53578\t0.0\t6\t9 1 11 22172 17846 8313 5226\n",
      "47\t60742\t0.0\t6\t2 1 2 8 29680 21268 9781\n",
      "48\t54037\t0.0\t6\t5 2 4 1 6 26082 27937\n",
      "49\t21149\t0.0\t6\t0 3 0 4 2 10 21130\n",
      "50\t60\t0.0\t6\t6 6 5 5 7 18 13\n",
      "\n",
      "This is cutadapt 1.18 with Python 3.7.0\n",
      "Command line parameters: -g CTTATCATGTCTGCTCGAAGCGCACTGAGCCTCCACCTAGCCT -e 0.15 -m 7 -M 7 -j 10 --discard-untrimmed /staging/leuven/stg_00002/lcb/ngs_runs/NovaSeq6000_20200416/Demultiplexed/STA__1a9f59__MM087_STARRseq_500_bp_enh_plasmid_S20_R2_001.fastq.gz\n",
      "Processing reads on 10 cores in single-end mode ...\n",
      "Finished in 39.81 s (3 us/read; 17.79 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              11,801,088\n",
      "Reads with adapters:                 9,864,696 (83.6%)\n",
      "Reads that were too short:             390,305 (3.3%)\n",
      "Reads that were too long:            2,189,505 (18.6%)\n",
      "Reads written (passing filters):     9,221,278 (78.1%)\n",
      "\n",
      "Total basepairs processed:   590,054,400 bp\n",
      "Total written (filtered):     64,548,946 bp (10.9%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTATCATGTCTGCTCGAAGCGCACTGAGCCTCCACCTAGCCT; Type: regular 5'; Length: 43; Trimmed: 9864696 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20-25 bp: 3; 26-32 bp: 4; 33-39 bp: 5; 40-43 bp: 6\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t9739\t184392.0\t0\t9739\n",
      "4\t3430\t46098.0\t0\t3430\n",
      "5\t603\t11524.5\t0\t603\n",
      "6\t194\t2881.1\t0\t5 189\n",
      "7\t250\t720.3\t1\t17 233\n",
      "8\t2043\t180.1\t1\t3 2040\n",
      "9\t81\t45.0\t1\t0 81\n",
      "10\t117\t11.3\t1\t2 115\n",
      "11\t12\t2.8\t1\t0 12\n",
      "12\t5\t0.7\t1\t0 1 4\n",
      "13\t4\t0.2\t1\t0 0 4\n",
      "14\t3\t0.0\t2\t0 0 3\n",
      "15\t13\t0.0\t2\t0 0 13\n",
      "16\t9\t0.0\t2\t0 0 9\n",
      "21\t1\t0.0\t3\t0 1\n",
      "31\t1\t0.0\t4\t1\n",
      "34\t3\t0.0\t5\t1 0 1 0 1\n",
      "35\t3\t0.0\t5\t0 1 1 0 1\n",
      "36\t2\t0.0\t5\t0 0 1 0 1\n",
      "37\t17\t0.0\t5\t0 3 4 3 0 1 6\n",
      "38\t158\t0.0\t5\t8 5 10 7 2 31 95\n",
      "39\t282\t0.0\t5\t2 16 10 15 33 58 148\n",
      "40\t1791\t0.0\t6\t31 64 48 233 320 398 697\n",
      "41\t12266\t0.0\t6\t234 457 2397 2299 1925 2219 2735\n",
      "42\t222086\t0.0\t6\t17956 80625 40688 28096 20868 16864 16989\n",
      "43\t9221278\t0.0\t6\t6653320 1529914 540056 235917 128819 80228 53024\n",
      "44\t108057\t0.0\t6\t875 48685 24422 14136 8969 6284 4686\n",
      "45\t84071\t0.0\t6\t5 13 31330 20369 14098 10347 7909\n",
      "46\t58200\t0.0\t6\t2 2 8 22623 18778 10020 6767\n",
      "47\t63251\t0.0\t6\t5 5 3 13 28914 22680 11631\n",
      "48\t55359\t0.0\t6\t0 1 2 0 6 26562 28788\n",
      "49\t21337\t0.0\t6\t2 1 1 0 1 5 21327\n",
      "50\t30\t0.0\t6\t0 2 2 3 2 3 18\n",
      "\n",
      "This is cutadapt 1.18 with Python 3.7.0\n",
      "Command line parameters: -g CTTATCATGTCTGCTCGAAGCGCACTGAGCCTCCACCTAGCCT -e 0.15 -m 7 -M 7 -j 10 --discard-untrimmed /staging/leuven/stg_00002/lcb/ngs_runs/NovaSeq6000_20200416/Demultiplexed/STA__34085e__MM047_STARRseq_500_bp_enh_cDNA_S11_R2_001.fastq.gz\n",
      "Processing reads on 10 cores in single-end mode ...\n",
      "Finished in 8.61 s (3 us/read; 21.60 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               3,099,592\n",
      "Reads with adapters:                 1,794,121 (57.9%)\n",
      "Reads that were too short:              71,493 (2.3%)\n",
      "Reads that were too long:            1,371,970 (44.3%)\n",
      "Reads written (passing filters):     1,656,129 (53.4%)\n",
      "\n",
      "Total basepairs processed:   154,979,600 bp\n",
      "Total written (filtered):     11,592,903 bp (7.5%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTATCATGTCTGCTCGAAGCGCACTGAGCCTCCACCTAGCCT; Type: regular 5'; Length: 43; Trimmed: 1794121 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20-25 bp: 3; 26-32 bp: 4; 33-39 bp: 5; 40-43 bp: 6\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t10871\t48431.1\t0\t10871\n",
      "4\t4285\t12107.8\t0\t4285\n",
      "5\t752\t3026.9\t0\t752\n",
      "6\t203\t756.7\t0\t9 194\n",
      "7\t169\t189.2\t1\t6 163\n",
      "8\t437\t47.3\t1\t1 436\n",
      "9\t102\t11.8\t1\t2 100\n",
      "10\t176\t3.0\t1\t1 175\n",
      "11\t11\t0.7\t1\t0 11\n",
      "12\t7\t0.2\t1\t0 2 5\n",
      "13\t9\t0.0\t1\t0 0 9\n",
      "14\t4\t0.0\t2\t0 0 4\n",
      "15\t9\t0.0\t2\t0 0 9\n",
      "16\t12\t0.0\t2\t0 0 12\n",
      "19\t1\t0.0\t2\t0 0 0 1\n",
      "37\t8\t0.0\t5\t0 1 2 1 0 1 3\n",
      "38\t27\t0.0\t5\t1 2 1 0 0 3 20\n",
      "39\t60\t0.0\t5\t0 3 4 4 3 14 32\n",
      "40\t418\t0.0\t6\t6 5 12 48 64 118 165\n",
      "41\t2299\t0.0\t6\t35 102 374 385 408 451 544\n",
      "42\t46639\t0.0\t6\t3354 18121 7681 5930 4382 3598 3573\n",
      "43\t1656129\t0.0\t6\t1200094 269079 93375 43061 24235 15468 10817\n",
      "44\t13509\t0.0\t6\t163 6154 2935 1720 1057 853 627\n",
      "45\t16247\t0.0\t6\t2 2 5736 3962 2803 2088 1654\n",
      "46\t8378\t0.0\t6\t3 0 1 3643 2311 1429 991\n",
      "47\t13885\t0.0\t6\t0 0 0 3 7072 4505 2305\n",
      "48\t18168\t0.0\t6\t1 0 1 0 0 8514 9652\n",
      "49\t1302\t0.0\t6\t1 0 0 0 0 3 1298\n",
      "50\t4\t0.0\t6\t0 0 1 0 0 0 3\n",
      "\n",
      "This is cutadapt 1.18 with Python 3.7.0\n",
      "Command line parameters: -g CTTATCATGTCTGCTCGAAGCGCACTGAGCCTCCACCTAGCCT -e 0.15 -m 7 -M 7 -j 10 --discard-untrimmed /staging/leuven/stg_00002/lcb/ngs_runs/NovaSeq6000_20200416/Demultiplexed/STA__3f00ec__MM001_STARRseq_500_bp_enh_cDNA_S9_R2_001.fastq.gz\n",
      "Processing reads on 10 cores in single-end mode ...\n",
      "Finished in 3.36 s (4 us/read; 14.58 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                 815,364\n",
      "Reads with adapters:                   752,282 (92.3%)\n",
      "Reads that were too short:              25,160 (3.1%)\n",
      "Reads that were too long:               81,122 (9.9%)\n",
      "Reads written (passing filters):       709,082 (87.0%)\n",
      "\n",
      "Total basepairs processed:    40,768,200 bp\n",
      "Total written (filtered):      4,963,574 bp (12.2%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTATCATGTCTGCTCGAAGCGCACTGAGCCTCCACCTAGCCT; Type: regular 5'; Length: 43; Trimmed: 752282 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20-25 bp: 3; 26-32 bp: 4; 33-39 bp: 5; 40-43 bp: 6\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t53\t12740.1\t0\t53\n",
      "4\t4\t3185.0\t0\t4\n",
      "5\t1\t796.3\t0\t1\n",
      "7\t4\t49.8\t1\t1 3\n",
      "8\t44\t12.4\t1\t0 44\n",
      "36\t1\t0.0\t5\t0 0 0 1\n",
      "37\t7\t0.0\t5\t1 2 0 2 1 0 1\n",
      "38\t6\t0.0\t5\t2 1 0 0 0 0 3\n",
      "39\t17\t0.0\t5\t1 1 1 0 0 4 10\n",
      "40\t96\t0.0\t6\t2 10 9 12 25 17 21\n",
      "41\t800\t0.0\t6\t43 80 177 156 112 99 133\n",
      "42\t17007\t0.0\t6\t3037 7128 2876 1524 923 835 684\n",
      "43\t709082\t0.0\t6\t565281 97597 27505 9802 4646 2691 1560\n",
      "44\t3884\t0.0\t6\t81 2104 812 406 221 146 114\n",
      "45\t2447\t0.0\t6\t1 1 912 568 470 294 201\n",
      "46\t10183\t0.0\t6\t0 0 0 4253 3684 1463 783\n",
      "47\t3412\t0.0\t6\t0 0 0 1 1843 1089 479\n",
      "48\t3113\t0.0\t6\t1 0 0 0 0 1517 1595\n",
      "49\t2117\t0.0\t6\t0 0 0 0 0 1 2116\n",
      "50\t4\t0.0\t6\t0 0 2 0 0 1 1\n",
      "\n",
      "This is cutadapt 1.18 with Python 3.7.0\n",
      "Command line parameters: -g CTTATCATGTCTGCTCGAAGCGCACTGAGCCTCCACCTAGCCT -e 0.15 -m 7 -M 7 -j 10 --discard-untrimmed /staging/leuven/stg_00002/lcb/ngs_runs/NovaSeq6000_20200416/Demultiplexed/STA__4642d6__MM087_STARRseq_500_bp_enh_cDNA_S14_R2_001.fastq.gz\n",
      "Processing reads on 10 cores in single-end mode ...\n",
      "Finished in 14.02 s (4 us/read; 16.79 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               3,922,351\n",
      "Reads with adapters:                 3,529,379 (90.0%)\n",
      "Reads that were too short:             123,288 (3.1%)\n",
      "Reads that were too long:              477,730 (12.2%)\n",
      "Reads written (passing filters):     3,321,333 (84.7%)\n",
      "\n",
      "Total basepairs processed:   196,117,550 bp\n",
      "Total written (filtered):     23,249,331 bp (11.9%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTATCATGTCTGCTCGAAGCGCACTGAGCCTCCACCTAGCCT; Type: regular 5'; Length: 43; Trimmed: 3529379 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20-25 bp: 3; 26-32 bp: 4; 33-39 bp: 5; 40-43 bp: 6\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t440\t61286.7\t0\t440\n",
      "4\t5\t15321.7\t0\t5\n",
      "6\t3\t957.6\t0\t0 3\n",
      "7\t19\t239.4\t1\t0 19\n",
      "8\t387\t59.9\t1\t0 387\n",
      "9\t1\t15.0\t1\t0 1\n",
      "34\t1\t0.0\t5\t0 1\n",
      "35\t1\t0.0\t5\t0 0 0 1\n",
      "36\t1\t0.0\t5\t0 0 0 1\n",
      "37\t5\t0.0\t5\t0 0 2 1 0 0 2\n",
      "38\t26\t0.0\t5\t4 1 3 2 1 3 12\n",
      "39\t81\t0.0\t5\t3 6 1 2 3 14 52\n",
      "40\t600\t0.0\t6\t8 15 14 58 122 159 224\n",
      "41\t4320\t0.0\t6\t81 141 907 804 800 773 814\n",
      "42\t78868\t0.0\t6\t6187 32644 12821 9349 7046 5642 5179\n",
      "43\t3321333\t0.0\t6\t2534067 486968 157670 67807 36610 22794 15417\n",
      "44\t16720\t0.0\t6\t318 8410 3267 1882 1206 926 711\n",
      "45\t19563\t0.0\t6\t2 2 7063 4650 3356 2532 1958\n",
      "46\t17526\t0.0\t6\t0 0 3 6978 5787 2898 1860\n",
      "47\t30613\t0.0\t6\t2 0 1 2 18080 8429 4099\n",
      "48\t33618\t0.0\t6\t1 0 0 0 8 16064 17545\n",
      "49\t5240\t0.0\t6\t1 1 0 1 1 6 5230\n",
      "50\t8\t0.0\t6\t0 0 1 1 0 2 4\n",
      "\n",
      "This is cutadapt 1.18 with Python 3.7.0\n",
      "Command line parameters: -g CTTATCATGTCTGCTCGAAGCGCACTGAGCCTCCACCTAGCCT -e 0.15 -m 7 -M 7 -j 10 --discard-untrimmed /staging/leuven/stg_00002/lcb/ngs_runs/NovaSeq6000_20200416/Demultiplexed/STA__60a828__MM047_STARRseq_500_bp_enh_plasmid_S17_R2_001.fastq.gz\n",
      "Processing reads on 10 cores in single-end mode ...\n",
      "Finished in 34.91 s (3 us/read; 18.11 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              10,536,160\n",
      "Reads with adapters:                 8,844,352 (83.9%)\n",
      "Reads that were too short:             395,298 (3.8%)\n",
      "Reads that were too long:            1,935,066 (18.4%)\n",
      "Reads written (passing filters):     8,205,796 (77.9%)\n",
      "\n",
      "Total basepairs processed:   526,808,000 bp\n",
      "Total written (filtered):     57,440,572 bp (10.9%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTATCATGTCTGCTCGAAGCGCACTGAGCCTCCACCTAGCCT; Type: regular 5'; Length: 43; Trimmed: 8844352 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20-25 bp: 3; 26-32 bp: 4; 33-39 bp: 5; 40-43 bp: 6\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t7910\t164627.5\t0\t7910\n",
      "4\t2630\t41156.9\t0\t2630\n",
      "5\t459\t10289.2\t0\t459\n",
      "6\t142\t2572.3\t0\t2 140\n",
      "7\t228\t643.1\t1\t11 217\n",
      "8\t2286\t160.8\t1\t3 2283\n",
      "9\t64\t40.2\t1\t0 64\n",
      "10\t100\t10.0\t1\t2 98\n",
      "11\t5\t2.5\t1\t0 5\n",
      "12\t5\t0.6\t1\t0 1 4\n",
      "13\t2\t0.2\t1\t0 1 1\n",
      "14\t7\t0.0\t2\t0 0 7\n",
      "15\t7\t0.0\t2\t0 0 7\n",
      "16\t10\t0.0\t2\t0 0 10\n",
      "33\t1\t0.0\t4\t0 0 1\n",
      "34\t2\t0.0\t5\t1 0 1\n",
      "36\t2\t0.0\t5\t0 0 0 1 1\n",
      "37\t15\t0.0\t5\t0 1 1 1 0 2 10\n",
      "38\t164\t0.0\t5\t5 3 4 5 2 32 113\n",
      "39\t310\t0.0\t5\t5 14 7 15 22 64 183\n",
      "40\t2060\t0.0\t6\t30 35 37 219 338 533 868\n",
      "41\t13076\t0.0\t6\t159 343 1982 2359 2300 2459 3474\n",
      "42\t213773\t0.0\t6\t11391 67807 40640 30383 23660 19480 20412\n",
      "43\t8205796\t0.0\t6\t5574159 1493570 567116 263690 148521 94768 63972\n",
      "44\t105840\t0.0\t6\t640 42940 24339 14914 10079 7331 5597\n",
      "45\t89020\t0.0\t6\t5 13 30327 21753 15547 11737 9638\n",
      "46\t57358\t0.0\t6\t1 2 2 21593 17981 10359 7420\n",
      "47\t62811\t0.0\t6\t2 1 1 2 27909 22244 12652\n",
      "48\t59454\t0.0\t6\t1 1 1 1 11 28333 31106\n",
      "49\t20791\t0.0\t6\t0 1 2 1 1 7 20779\n",
      "50\t24\t0.0\t6\t1 3 1 3 2 7 7\n",
      "\n",
      "This is cutadapt 1.18 with Python 3.7.0\n",
      "Command line parameters: -g CTTATCATGTCTGCTCGAAGCGCACTGAGCCTCCACCTAGCCT -e 0.15 -m 7 -M 7 -j 10 --discard-untrimmed /staging/leuven/stg_00002/lcb/ngs_runs/NovaSeq6000_20200416/Demultiplexed/STA__616867__MM029_STARRseq_500_bp_enh_cDNA_S10_R2_001.fastq.gz\n",
      "Processing reads on 10 cores in single-end mode ...\n",
      "Finished in 16.48 s (3 us/read; 21.43 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               5,886,866\n",
      "Reads with adapters:                 5,676,879 (96.4%)\n",
      "Reads that were too short:             201,669 (3.4%)\n",
      "Reads that were too long:              386,660 (6.6%)\n",
      "Reads written (passing filters):     5,298,537 (90.0%)\n",
      "\n",
      "Total basepairs processed:   294,343,300 bp\n",
      "Total written (filtered):     37,089,759 bp (12.6%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTATCATGTCTGCTCGAAGCGCACTGAGCCTCCACCTAGCCT; Type: regular 5'; Length: 43; Trimmed: 5676879 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20-25 bp: 3; 26-32 bp: 4; 33-39 bp: 5; 40-43 bp: 6\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t258\t91982.3\t0\t258\n",
      "4\t4\t22995.6\t0\t4\n",
      "6\t4\t1437.2\t0\t0 4\n",
      "7\t66\t359.3\t1\t6 60\n",
      "8\t601\t89.8\t1\t0 601\n",
      "9\t1\t22.5\t1\t0 1\n",
      "32\t1\t0.0\t4\t1\n",
      "35\t2\t0.0\t5\t1 0 0 1\n",
      "36\t7\t0.0\t5\t0 0 2 3 1 1\n",
      "37\t10\t0.0\t5\t1 1 0 2 1 0 5\n",
      "38\t69\t0.0\t5\t5 7 5 10 2 6 34\n",
      "39\t157\t0.0\t5\t4 17 11 16 12 31 66\n",
      "40\t901\t0.0\t6\t34 61 43 149 161 195 258\n",
      "41\t8173\t0.0\t6\t298 532 2052 1667 1297 1148 1179\n",
      "42\t166419\t0.0\t6\t19266 69395 29355 19716 12529 8696 7462\n",
      "43\t5298537\t0.0\t6\t3943518 860838 281271 111602 54161 29485 17662\n",
      "44\t28724\t0.0\t6\t549 13770 6198 3511 2198 1485 1013\n",
      "45\t41047\t0.0\t6\t5 4 16233 10459 6577 4554 3215\n",
      "46\t25286\t0.0\t6\t2 0 1 10862 7323 4355 2743\n",
      "47\t43339\t0.0\t6\t1 0 1 3 24260 13287 5787\n",
      "48\t58475\t0.0\t6\t0 0 1 1 6 28290 30177\n",
      "49\t4768\t0.0\t6\t2 1 1 4 1 7 4752\n",
      "50\t30\t0.0\t6\t3 2 1 4 8 5 7\n",
      "\n",
      "This is cutadapt 1.18 with Python 3.7.0\n",
      "Command line parameters: -g CTTATCATGTCTGCTCGAAGCGCACTGAGCCTCCACCTAGCCT -e 0.15 -m 7 -M 7 -j 10 --discard-untrimmed /staging/leuven/stg_00002/lcb/ngs_runs/NovaSeq6000_20200416/Demultiplexed/STA__a6d28a__MM001_STARRseq_500_bp_enh_plasmid_S15_R2_001.fastq.gz\n",
      "Processing reads on 10 cores in single-end mode ...\n",
      "Finished in 22.20 s (2 us/read; 25.54 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               9,450,621\n",
      "Reads with adapters:                 8,507,427 (90.0%)\n",
      "Reads that were too short:             257,475 (2.7%)\n",
      "Reads that were too long:            1,150,937 (12.2%)\n",
      "Reads written (passing filters):     8,042,209 (85.1%)\n",
      "\n",
      "Total basepairs processed:   472,531,050 bp\n",
      "Total written (filtered):     56,295,463 bp (11.9%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTATCATGTCTGCTCGAAGCGCACTGAGCCTCCACCTAGCCT; Type: regular 5'; Length: 43; Trimmed: 8507427 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20-25 bp: 3; 26-32 bp: 4; 33-39 bp: 5; 40-43 bp: 6\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t335\t147666.0\t0\t335\n",
      "4\t9\t36916.5\t0\t9\n",
      "6\t4\t2307.3\t0\t0 4\n",
      "7\t79\t576.8\t1\t7 72\n",
      "8\t819\t144.2\t1\t1 818\n",
      "10\t1\t9.0\t1\t0 1\n",
      "33\t1\t0.0\t4\t1\n",
      "34\t1\t0.0\t5\t1\n",
      "35\t1\t0.0\t5\t0 0 0 0 0 1\n",
      "36\t1\t0.0\t5\t0 0 0 1\n",
      "37\t19\t0.0\t5\t0 5 2 2 3 2 5\n",
      "38\t79\t0.0\t5\t5 10 7 1 1 12 43\n",
      "39\t213\t0.0\t5\t5 22 17 26 22 43 78\n",
      "40\t1040\t0.0\t6\t52 82 60 185 172 228 261\n",
      "41\t9126\t0.0\t6\t448 802 2675 1799 1189 1061 1152\n",
      "42\t196015\t0.0\t6\t31808 87882 33385 17737 10883 7665 6655\n",
      "43\t8042209\t0.0\t6\t6370638 1122659 331818 116007 53877 29428 17782\n",
      "44\t75166\t0.0\t6\t900 40264 16323 8183 4686 2859 1951\n",
      "45\t47176\t0.0\t6\t5 6 21491 11433 6944 4366 2931\n",
      "46\t41834\t0.0\t6\t5 0 13 17989 14421 5882 3524\n",
      "47\t42637\t0.0\t6\t1 3 1 10 20921 15640 6061\n",
      "48\t33806\t0.0\t6\t1 4 2 2 4 16639 17154\n",
      "49\t16817\t0.0\t6\t3 2 1 2 1 6 16802\n",
      "50\t39\t0.0\t6\t2 3 3 4 6 10 11\n",
      "\n",
      "This is cutadapt 1.18 with Python 3.7.0\n",
      "Command line parameters: -g CTTATCATGTCTGCTCGAAGCGCACTGAGCCTCCACCTAGCCT -e 0.15 -m 7 -M 7 -j 10 --discard-untrimmed /staging/leuven/stg_00002/lcb/ngs_runs/NovaSeq6000_20200416/Demultiplexed/STA__af2595__MM074_STARRseq_500_bp_enh_cDNA_S13_R2_001.fastq.gz\n",
      "Processing reads on 10 cores in single-end mode ...\n",
      "Finished in 3.84 s (4 us/read; 14.78 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                 945,186\n",
      "Reads with adapters:                   824,553 (87.2%)\n",
      "Reads that were too short:              40,455 (4.3%)\n",
      "Reads that were too long:              144,003 (15.2%)\n",
      "Reads written (passing filters):       760,728 (80.5%)\n",
      "\n",
      "Total basepairs processed:    47,259,300 bp\n",
      "Total written (filtered):      5,325,096 bp (11.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTATCATGTCTGCTCGAAGCGCACTGAGCCTCCACCTAGCCT; Type: regular 5'; Length: 43; Trimmed: 824553 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20-25 bp: 3; 26-32 bp: 4; 33-39 bp: 5; 40-43 bp: 6\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t101\t14768.5\t0\t101\n",
      "4\t5\t3692.1\t0\t5\n",
      "7\t11\t57.7\t1\t0 11\n",
      "8\t116\t14.4\t1\t0 116\n",
      "10\t1\t0.9\t1\t0 1\n",
      "37\t1\t0.0\t5\t1\n",
      "38\t9\t0.0\t5\t1 2 1 0 0 1 4\n",
      "39\t18\t0.0\t5\t0 3 0 2 0 4 9\n",
      "40\t144\t0.0\t6\t6 8 6 19 30 31 44\n",
      "41\t1169\t0.0\t6\t45 75 271 202 195 172 209\n",
      "42\t21795\t0.0\t6\t2967 8774 3699 2263 1727 1145 1220\n",
      "43\t760728\t0.0\t6\t563139 124537 39885 16805 8543 4901 2918\n",
      "44\t7500\t0.0\t6\t86 3593 1656 885 611 411 258\n",
      "45\t9694\t0.0\t6\t1 0 3732 2531 1554 1147 729\n",
      "46\t5238\t0.0\t6\t2 0 0 1937 1876 916 507\n",
      "47\t8898\t0.0\t6\t1 0 0 1 4767 2821 1308\n",
      "48\t7891\t0.0\t6\t0 1 1 0 3 3712 4174\n",
      "49\t1232\t0.0\t6\t0 0 1 0 0 1 1230\n",
      "50\t2\t0.0\t6\t0 0 0 1 0 0 1\n",
      "\n",
      "This is cutadapt 1.18 with Python 3.7.0\n",
      "Command line parameters: -g CTTATCATGTCTGCTCGAAGCGCACTGAGCCTCCACCTAGCCT -e 0.15 -m 7 -M 7 -j 10 --discard-untrimmed /staging/leuven/stg_00002/lcb/ngs_runs/NovaSeq6000_20200416/Demultiplexed/STA__c67b3a__MM057_STARRseq_500_bp_enh_plasmid_S18_R2_001.fastq.gz\n",
      "Processing reads on 10 cores in single-end mode ...\n",
      "Finished in 35.24 s (3 us/read; 19.53 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              11,470,522\n",
      "Reads with adapters:                 9,622,887 (83.9%)\n",
      "Reads that were too short:             394,093 (3.4%)\n",
      "Reads that were too long:            2,096,505 (18.3%)\n",
      "Reads written (passing filters):     8,979,924 (78.3%)\n",
      "\n",
      "Total basepairs processed:   573,526,100 bp\n",
      "Total written (filtered):     62,859,468 bp (11.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTATCATGTCTGCTCGAAGCGCACTGAGCCTCCACCTAGCCT; Type: regular 5'; Length: 43; Trimmed: 9622887 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20-25 bp: 3; 26-32 bp: 4; 33-39 bp: 5; 40-43 bp: 6\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t9561\t179226.9\t0\t9561\n",
      "4\t3378\t44806.7\t0\t3378\n",
      "5\t621\t11201.7\t0\t621\n",
      "6\t185\t2800.4\t0\t8 177\n",
      "7\t227\t700.1\t1\t16 211\n",
      "8\t2098\t175.0\t1\t2 2096\n",
      "9\t81\t43.8\t1\t1 80\n",
      "10\t137\t10.9\t1\t1 136\n",
      "11\t5\t2.7\t1\t0 5\n",
      "12\t2\t0.7\t1\t0 0 2\n",
      "13\t5\t0.2\t1\t0 0 5\n",
      "14\t2\t0.0\t2\t0 0 2\n",
      "15\t9\t0.0\t2\t0 1 8\n",
      "16\t5\t0.0\t2\t0 0 5\n",
      "34\t3\t0.0\t5\t1 1 0 0 1\n",
      "36\t1\t0.0\t5\t0 0 0 1\n",
      "37\t13\t0.0\t5\t1 3 1 0 2 0 6\n",
      "38\t144\t0.0\t5\t6 4 7 0 4 38 85\n",
      "39\t280\t0.0\t5\t2 17 11 13 35 53 149\n",
      "40\t1879\t0.0\t6\t34 41 52 244 341 449 718\n",
      "41\t12726\t0.0\t6\t204 446 2324 2204 2231 2256 3061\n",
      "42\t217508\t0.0\t6\t15147 78070 39918 27954 21288 17529 17602\n",
      "43\t8979924\t0.0\t6\t6451052 1493434 533292 235012 130063 81624 55447\n",
      "44\t109094\t0.0\t6\t868 48664 24656 14255 9300 6405 4946\n",
      "45\t86588\t0.0\t6\t7 10 32320 21195 14487 10537 8032\n",
      "46\t57774\t0.0\t6\t1 4 10 22646 18239 10026 6848\n",
      "47\t62400\t0.0\t6\t5 0 2 6 28504 22288 11595\n",
      "48\t57333\t0.0\t6\t2 1 1 1 5 27616 29707\n",
      "49\t20881\t0.0\t6\t0 1 0 0 0 7 20873\n",
      "50\t23\t0.0\t6\t3 1 3 2 3 4 7\n",
      "\n",
      "This is cutadapt 1.18 with Python 3.7.0\n",
      "Command line parameters: -g CTTATCATGTCTGCTCGAAGCGCACTGAGCCTCCACCTAGCCT -e 0.15 -m 7 -M 7 -j 10 --discard-untrimmed /staging/leuven/stg_00002/lcb/ngs_runs/NovaSeq6000_20200416/Demultiplexed/STA__e17937__MM074_STARRseq_500_bp_enh_plasmid_S19_R2_001.fastq.gz\n",
      "Processing reads on 10 cores in single-end mode ...\n",
      "Finished in 22.54 s (2 us/read; 28.59 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              10,738,316\n",
      "Reads with adapters:                 9,687,558 (90.2%)\n",
      "Reads that were too short:             336,313 (3.1%)\n",
      "Reads that were too long:            1,310,393 (12.2%)\n",
      "Reads written (passing filters):     9,091,610 (84.7%)\n",
      "\n",
      "Total basepairs processed:   536,915,800 bp\n",
      "Total written (filtered):     63,641,270 bp (11.9%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTATCATGTCTGCTCGAAGCGCACTGAGCCTCCACCTAGCCT; Type: regular 5'; Length: 43; Trimmed: 9687558 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20-25 bp: 3; 26-32 bp: 4; 33-39 bp: 5; 40-43 bp: 6\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t1041\t167786.2\t0\t1041\n",
      "4\t15\t41946.5\t0\t15\n",
      "5\t4\t10486.6\t0\t4\n",
      "6\t7\t2621.7\t0\t0 7\n",
      "7\t130\t655.4\t1\t15 115\n",
      "8\t1349\t163.9\t1\t1 1348\n",
      "9\t2\t41.0\t1\t0 2\n",
      "30\t1\t0.0\t4\t0 1\n",
      "32\t1\t0.0\t4\t0 1\n",
      "34\t2\t0.0\t5\t1 0 0 0 1\n",
      "35\t2\t0.0\t5\t1 1\n",
      "36\t6\t0.0\t5\t0 1 3 2\n",
      "37\t22\t0.0\t5\t0 4 5 2 3 2 6\n",
      "38\t138\t0.0\t5\t16 6 10 4 2 15 85\n",
      "39\t259\t0.0\t5\t10 22 19 18 25 63 102\n",
      "40\t1655\t0.0\t6\t55 104 94 217 244 354 587\n",
      "41\t12828\t0.0\t6\t472 806 3113 2319 1841 1956 2321\n",
      "42\t242173\t0.0\t6\t33497 100092 39871 24515 17129 13723 13346\n",
      "43\t9091610\t0.0\t6\t6943753 1328489 435716 181655 99171 61475 41351\n",
      "44\t92641\t0.0\t6\t996 45655 19670 11079 6949 4722 3570\n",
      "45\t69639\t0.0\t6\t6 17 28618 16598 10813 7698 5889\n",
      "46\t48953\t0.0\t6\t2 1 5 19738 16094 7860 5253\n",
      "47\t56087\t0.0\t6\t3 1 3 10 26655 19934 9481\n",
      "48\t50308\t0.0\t6\t2 2 0 3 13 24104 26184\n",
      "49\t18650\t0.0\t6\t2 1 1 0 2 4 18640\n",
      "50\t35\t0.0\t6\t2 2 3 6 3 5 14\n",
      "\n",
      "This is cutadapt 1.18 with Python 3.7.0\n",
      "Command line parameters: -g CTTATCATGTCTGCTCGAAGCGCACTGAGCCTCCACCTAGCCT -e 0.15 -m 7 -M 7 -j 10 --discard-untrimmed /staging/leuven/stg_00002/lcb/ngs_runs/NovaSeq6000_20200416/Demultiplexed/STA__ed22cd__MM057_STARRseq_500_bp_enh_cDNA_S12_R2_001.fastq.gz\n",
      "Processing reads on 10 cores in single-end mode ...\n",
      "Finished in 8.13 s (2 us/read; 29.13 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:               3,947,921\n",
      "Reads with adapters:                 3,432,584 (86.9%)\n",
      "Reads that were too short:             129,330 (3.3%)\n",
      "Reads that were too long:              601,971 (15.2%)\n",
      "Reads written (passing filters):     3,216,620 (81.5%)\n",
      "\n",
      "Total basepairs processed:   197,396,050 bp\n",
      "Total written (filtered):     22,516,340 bp (11.4%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTATCATGTCTGCTCGAAGCGCACTGAGCCTCCACCTAGCCT; Type: regular 5'; Length: 43; Trimmed: 3432584 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20-25 bp: 3; 26-32 bp: 4; 33-39 bp: 5; 40-43 bp: 6\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t697\t61686.3\t0\t697\n",
      "4\t31\t15421.6\t0\t31\n",
      "5\t2\t3855.4\t0\t2\n",
      "6\t4\t963.8\t0\t0 4\n",
      "7\t50\t241.0\t1\t4 46\n",
      "8\t560\t60.2\t1\t0 560\n",
      "10\t1\t3.8\t1\t0 1\n",
      "36\t1\t0.0\t5\t0 0 0 0 1\n",
      "37\t6\t0.0\t5\t0 0 1 2 0 0 3\n",
      "38\t37\t0.0\t5\t2 1 2 0 0 8 24\n",
      "39\t122\t0.0\t5\t0 4 4 1 12 36 65\n",
      "40\t661\t0.0\t6\t11 17 15 69 141 158 250\n",
      "41\t4504\t0.0\t6\t66 125 774 787 858 866 1028\n",
      "42\t79958\t0.0\t6\t5642 29383 13489 10336 8102 6621 6385\n",
      "43\t3216620\t0.0\t6\t2346754 518777 179865 80350 44165 27839 18870\n",
      "44\t18471\t0.0\t6\t316 8495 3784 2276 1490 1131 979\n",
      "45\t21963\t0.0\t6\t1 2 7259 5324 3882 3069 2426\n",
      "46\t21260\t0.0\t6\t0 0 0 8133 6799 3793 2535\n",
      "47\t30624\t0.0\t6\t0 0 0 2 16577 9376 4669\n",
      "48\t31118\t0.0\t6\t0 1 1 0 7 14570 16539\n",
      "49\t5885\t0.0\t6\t1 0 0 0 1 5 5878\n",
      "50\t9\t0.0\t6\t0 1 0 2 5 0 1\n",
      "\n",
      "263\n"
     ]
    }
   ],
   "source": [
    "module use /data/leuven/software/biomed/skylake_centos7/2018a/modules/all/\n",
    "module load SeqKit/0.10.2\n",
    "module load cutadapt/1.18-foss-2018a-Python-3.7.0\n",
    "start=`date +%s`\n",
    "cat STARRseq_names.txt | while read line\n",
    "do\n",
    "NovaSeq6000_200123=\"/staging/leuven/stg_00002/lcb/ngs_runs/NovaSeq6000_20200123/Demultiplexed/${line%STARR*}*R2*\"\n",
    "NovaSeq6000_200416=\"/staging/leuven/stg_00002/lcb/ngs_runs/NovaSeq6000_20200416/Demultiplexed/${line%STARR*}*R2*\"\n",
    "    # Do cutadapt trimming on R2 reads and then reverse complement\n",
    "cutadapt -g CTTATCATGTCTGCTCGAAGCGCACTGAGCCTCCACCTAGCCT -e 0.15 -m 7 -M 7 -j 10 --discard-untrimmed \\\n",
    "         ${NovaSeq6000_200416} | \\\n",
    "seqkit seq -r -p -o 02.clean_reads/${line}_R2.fastq.gz\n",
    "\n",
    "done\n",
    "end=`date +%s`\n",
    "echo $((end-start))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Detecting adapter sequence for read1...\n",
      "No adapter detected for read1\n",
      "\n",
      "Read1 before filtering:\n",
      "total reads: 10018456\n",
      "total bases: 70129192\n",
      "Q20 bases: 60498494(86.2672%)\n",
      "Q30 bases: 49862678(71.1012%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 6491053\n",
      "total bases: 45437371\n",
      "Q20 bases: 43266174(95.2216%)\n",
      "Q30 bases: 37538162(82.6152%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 6491053\n",
      "reads failed due to low quality: 3527403\n",
      "reads failed due to too many N: 0\n",
      "reads with adapter trimmed: 0\n",
      "bases trimmed due to adapters: 0\n",
      "\n",
      "Duplication rate (may be overestimated since this is SE data): 0%\n",
      "\n",
      "JSON report: /dev/null/fastp.json\n",
      "HTML report: /dev/null/fastp.html\n",
      "\n",
      "fastp -i 02.clean_reads/STA__073ce2__MM029_STARRseq_500_bp_enh_plasmid_R2.fastq.gz -L -e 30 -h /dev/null/fastp.html -j /dev/null/fastp.json -w 8 -o 02.clean_reads/STA__073ce2__MM029_STARRseq_500_bp_enh_plasmid_q30.fastq.gz \n",
      "fastp v0.20.0, time used: 18 seconds\n",
      "Number of reads after Q30 filtering for STA__073ce2__MM029_STARRseq_500_bp_enh_plasmid:  6491053\n",
      "Detecting adapter sequence for read1...\n",
      "No adapter detected for read1\n",
      "\n",
      "Read1 before filtering:\n",
      "total reads: 9221278\n",
      "total bases: 64548946\n",
      "Q20 bases: 54048259(83.7322%)\n",
      "Q30 bases: 43614563(67.5682%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 5331091\n",
      "total bases: 37317637\n",
      "Q20 bases: 35397191(94.8538%)\n",
      "Q30 bases: 30407027(81.4817%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 5331091\n",
      "reads failed due to low quality: 3890187\n",
      "reads failed due to too many N: 0\n",
      "reads with adapter trimmed: 0\n",
      "bases trimmed due to adapters: 0\n",
      "\n",
      "Duplication rate (may be overestimated since this is SE data): 0%\n",
      "\n",
      "JSON report: /dev/null/fastp.json\n",
      "HTML report: /dev/null/fastp.html\n",
      "\n",
      "fastp -i 02.clean_reads/STA__1a9f59__MM087_STARRseq_500_bp_enh_plasmid_R2.fastq.gz -L -e 30 -h /dev/null/fastp.html -j /dev/null/fastp.json -w 8 -o 02.clean_reads/STA__1a9f59__MM087_STARRseq_500_bp_enh_plasmid_q30.fastq.gz \n",
      "fastp v0.20.0, time used: 17 seconds\n",
      "Number of reads after Q30 filtering for STA__1a9f59__MM087_STARRseq_500_bp_enh_plasmid:  5331091\n",
      "Detecting adapter sequence for read1...\n",
      "No adapter detected for read1\n",
      "\n",
      "Read1 before filtering:\n",
      "total reads: 1656129\n",
      "total bases: 11592903\n",
      "Q20 bases: 9835386(84.8397%)\n",
      "Q30 bases: 8094848(69.8259%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1021323\n",
      "total bases: 7149261\n",
      "Q20 bases: 6806411(95.2044%)\n",
      "Q30 bases: 5939886(83.0839%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1021323\n",
      "reads failed due to low quality: 634806\n",
      "reads failed due to too many N: 0\n",
      "reads with adapter trimmed: 0\n",
      "bases trimmed due to adapters: 0\n",
      "\n",
      "Duplication rate (may be overestimated since this is SE data): 0%\n",
      "\n",
      "JSON report: /dev/null/fastp.json\n",
      "HTML report: /dev/null/fastp.html\n",
      "\n",
      "fastp -i 02.clean_reads/STA__34085e__MM047_STARRseq_500_bp_enh_cDNA_R2.fastq.gz -L -e 30 -h /dev/null/fastp.html -j /dev/null/fastp.json -w 8 -o 02.clean_reads/STA__34085e__MM047_STARRseq_500_bp_enh_cDNA_q30.fastq.gz \n",
      "fastp v0.20.0, time used: 4 seconds\n",
      "Number of reads after Q30 filtering for STA__34085e__MM047_STARRseq_500_bp_enh_cDNA:  1021323\n",
      "Detecting adapter sequence for read1...\n",
      "No adapter detected for read1\n",
      "\n",
      "Read1 before filtering:\n",
      "total reads: 709082\n",
      "total bases: 4963574\n",
      "Q20 bases: 4332132(87.2785%)\n",
      "Q30 bases: 3606868(72.6668%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 479392\n",
      "total bases: 3355744\n",
      "Q20 bases: 3204600(95.496%)\n",
      "Q30 bases: 2798478(83.3937%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 479392\n",
      "reads failed due to low quality: 229690\n",
      "reads failed due to too many N: 0\n",
      "reads with adapter trimmed: 0\n",
      "bases trimmed due to adapters: 0\n",
      "\n",
      "Duplication rate (may be overestimated since this is SE data): 0%\n",
      "\n",
      "JSON report: /dev/null/fastp.json\n",
      "HTML report: /dev/null/fastp.html\n",
      "\n",
      "fastp -i 02.clean_reads/STA__3f00ec__MM001_STARRseq_500_bp_enh_cDNA_R2.fastq.gz -L -e 30 -h /dev/null/fastp.html -j /dev/null/fastp.json -w 8 -o 02.clean_reads/STA__3f00ec__MM001_STARRseq_500_bp_enh_cDNA_q30.fastq.gz \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Number of reads after Q30 filtering for STA__3f00ec__MM001_STARRseq_500_bp_enh_cDNA:  479392\n",
      "Detecting adapter sequence for read1...\n",
      "No adapter detected for read1\n",
      "\n",
      "Read1 before filtering:\n",
      "total reads: 3321333\n",
      "total bases: 23249331\n",
      "Q20 bases: 20082559(86.3791%)\n",
      "Q30 bases: 16708500(71.8666%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 2187680\n",
      "total bases: 15313760\n",
      "Q20 bases: 14610285(95.4063%)\n",
      "Q30 bases: 12787527(83.5035%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 2187680\n",
      "reads failed due to low quality: 1133653\n",
      "reads failed due to too many N: 0\n",
      "reads with adapter trimmed: 0\n",
      "bases trimmed due to adapters: 0\n",
      "\n",
      "Duplication rate (may be overestimated since this is SE data): 0%\n",
      "\n",
      "JSON report: /dev/null/fastp.json\n",
      "HTML report: /dev/null/fastp.html\n",
      "\n",
      "fastp -i 02.clean_reads/STA__4642d6__MM087_STARRseq_500_bp_enh_cDNA_R2.fastq.gz -L -e 30 -h /dev/null/fastp.html -j /dev/null/fastp.json -w 8 -o 02.clean_reads/STA__4642d6__MM087_STARRseq_500_bp_enh_cDNA_q30.fastq.gz \n",
      "fastp v0.20.0, time used: 7 seconds\n",
      "Number of reads after Q30 filtering for STA__4642d6__MM087_STARRseq_500_bp_enh_cDNA:  2187680\n",
      "Detecting adapter sequence for read1...\n",
      "No adapter detected for read1\n",
      "\n",
      "Read1 before filtering:\n",
      "total reads: 8205796\n",
      "total bases: 57440572\n",
      "Q20 bases: 46931640(81.7047%)\n",
      "Q30 bases: 37299346(64.9355%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 4296752\n",
      "total bases: 30077264\n",
      "Q20 bases: 28416706(94.479%)\n",
      "Q30 bases: 24246059(80.6126%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 4296752\n",
      "reads failed due to low quality: 3909044\n",
      "reads failed due to too many N: 0\n",
      "reads with adapter trimmed: 0\n",
      "bases trimmed due to adapters: 0\n",
      "\n",
      "Duplication rate (may be overestimated since this is SE data): 0%\n",
      "\n",
      "JSON report: /dev/null/fastp.json\n",
      "HTML report: /dev/null/fastp.html\n",
      "\n",
      "fastp -i 02.clean_reads/STA__60a828__MM047_STARRseq_500_bp_enh_plasmid_R2.fastq.gz -L -e 30 -h /dev/null/fastp.html -j /dev/null/fastp.json -w 8 -o 02.clean_reads/STA__60a828__MM047_STARRseq_500_bp_enh_plasmid_q30.fastq.gz \n",
      "fastp v0.20.0, time used: 16 seconds\n",
      "Number of reads after Q30 filtering for STA__60a828__MM047_STARRseq_500_bp_enh_plasmid:  4296752\n",
      "Detecting adapter sequence for read1...\n",
      "No adapter detected for read1\n",
      "\n",
      "Read1 before filtering:\n",
      "total reads: 5298537\n",
      "total bases: 37089759\n",
      "Q20 bases: 32047205(86.4045%)\n",
      "Q30 bases: 26686859(71.9521%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 3478133\n",
      "total bases: 24346931\n",
      "Q20 bases: 23216209(95.3558%)\n",
      "Q30 bases: 20357165(83.6129%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 3478133\n",
      "reads failed due to low quality: 1820404\n",
      "reads failed due to too many N: 0\n",
      "reads with adapter trimmed: 0\n",
      "bases trimmed due to adapters: 0\n",
      "\n",
      "Duplication rate (may be overestimated since this is SE data): 0%\n",
      "\n",
      "JSON report: /dev/null/fastp.json\n",
      "HTML report: /dev/null/fastp.html\n",
      "\n",
      "fastp -i 02.clean_reads/STA__616867__MM029_STARRseq_500_bp_enh_cDNA_R2.fastq.gz -L -e 30 -h /dev/null/fastp.html -j /dev/null/fastp.json -w 8 -o 02.clean_reads/STA__616867__MM029_STARRseq_500_bp_enh_cDNA_q30.fastq.gz \n",
      "fastp v0.20.0, time used: 10 seconds\n",
      "Number of reads after Q30 filtering for STA__616867__MM029_STARRseq_500_bp_enh_cDNA:  3478133\n",
      "Detecting adapter sequence for read1...\n",
      "No adapter detected for read1\n",
      "\n",
      "Read1 before filtering:\n",
      "total reads: 8042209\n",
      "total bases: 56295463\n",
      "Q20 bases: 48910922(86.8825%)\n",
      "Q30 bases: 40382230(71.7327%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 5327715\n",
      "total bases: 37294005\n",
      "Q20 bases: 35566021(95.3666%)\n",
      "Q30 bases: 30846485(82.7116%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 5327715\n",
      "reads failed due to low quality: 2714494\n",
      "reads failed due to too many N: 0\n",
      "reads with adapter trimmed: 0\n",
      "bases trimmed due to adapters: 0\n",
      "\n",
      "Duplication rate (may be overestimated since this is SE data): 0%\n",
      "\n",
      "JSON report: /dev/null/fastp.json\n",
      "HTML report: /dev/null/fastp.html\n",
      "\n",
      "fastp -i 02.clean_reads/STA__a6d28a__MM001_STARRseq_500_bp_enh_plasmid_R2.fastq.gz -L -e 30 -h /dev/null/fastp.html -j /dev/null/fastp.json -w 8 -o 02.clean_reads/STA__a6d28a__MM001_STARRseq_500_bp_enh_plasmid_q30.fastq.gz \n",
      "fastp v0.20.0, time used: 16 seconds\n",
      "Number of reads after Q30 filtering for STA__a6d28a__MM001_STARRseq_500_bp_enh_plasmid:  5327715\n",
      "Detecting adapter sequence for read1...\n",
      "No adapter detected for read1\n",
      "\n",
      "Read1 before filtering:\n",
      "total reads: 760728\n",
      "total bases: 5325096\n",
      "Q20 bases: 4527188(85.0161%)\n",
      "Q30 bases: 3716848(69.7987%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 469448\n",
      "total bases: 3286136\n",
      "Q20 bases: 3120328(94.9543%)\n",
      "Q30 bases: 2709187(82.4429%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 469448\n",
      "reads failed due to low quality: 291280\n",
      "reads failed due to too many N: 0\n",
      "reads with adapter trimmed: 0\n",
      "bases trimmed due to adapters: 0\n",
      "\n",
      "Duplication rate (may be overestimated since this is SE data): 0%\n",
      "\n",
      "JSON report: /dev/null/fastp.json\n",
      "HTML report: /dev/null/fastp.html\n",
      "\n",
      "fastp -i 02.clean_reads/STA__af2595__MM074_STARRseq_500_bp_enh_cDNA_R2.fastq.gz -L -e 30 -h /dev/null/fastp.html -j /dev/null/fastp.json -w 8 -o 02.clean_reads/STA__af2595__MM074_STARRseq_500_bp_enh_cDNA_q30.fastq.gz \n",
      "fastp v0.20.0, time used: 2 seconds\n",
      "Number of reads after Q30 filtering for STA__af2595__MM074_STARRseq_500_bp_enh_cDNA:  469448\n",
      "Detecting adapter sequence for read1...\n",
      "No adapter detected for read1\n",
      "\n",
      "Read1 before filtering:\n",
      "total reads: 8979924\n",
      "total bases: 62859468\n",
      "Q20 bases: 52577442(83.6428%)\n",
      "Q30 bases: 42439708(67.5152%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 5192368\n",
      "total bases: 36346576\n",
      "Q20 bases: 34493430(94.9015%)\n",
      "Q30 bases: 29660906(81.6058%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 5192368\n",
      "reads failed due to low quality: 3787556\n",
      "reads failed due to too many N: 0\n",
      "reads with adapter trimmed: 0\n",
      "bases trimmed due to adapters: 0\n",
      "\n",
      "Duplication rate (may be overestimated since this is SE data): 0%\n",
      "\n",
      "JSON report: /dev/null/fastp.json\n",
      "HTML report: /dev/null/fastp.html\n",
      "\n",
      "fastp -i 02.clean_reads/STA__c67b3a__MM057_STARRseq_500_bp_enh_plasmid_R2.fastq.gz -L -e 30 -h /dev/null/fastp.html -j /dev/null/fastp.json -w 8 -o 02.clean_reads/STA__c67b3a__MM057_STARRseq_500_bp_enh_plasmid_q30.fastq.gz \n",
      "fastp v0.20.0, time used: 16 seconds\n",
      "Number of reads after Q30 filtering for STA__c67b3a__MM057_STARRseq_500_bp_enh_plasmid:  5192368\n",
      "Detecting adapter sequence for read1...\n",
      "No adapter detected for read1\n",
      "\n",
      "Read1 before filtering:\n",
      "total reads: 9091610\n",
      "total bases: 63641270\n",
      "Q20 bases: 54433652(85.532%)\n",
      "Q30 bases: 44619160(70.1104%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 5724180\n",
      "total bases: 40069260\n",
      "Q20 bases: 38167627(95.2541%)\n",
      "Q30 bases: 33051804(82.4867%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 5724180\n",
      "reads failed due to low quality: 3367430\n",
      "reads failed due to too many N: 0\n",
      "reads with adapter trimmed: 0\n",
      "bases trimmed due to adapters: 0\n",
      "\n",
      "Duplication rate (may be overestimated since this is SE data): 0%\n",
      "\n",
      "JSON report: /dev/null/fastp.json\n",
      "HTML report: /dev/null/fastp.html\n",
      "\n",
      "fastp -i 02.clean_reads/STA__e17937__MM074_STARRseq_500_bp_enh_plasmid_R2.fastq.gz -L -e 30 -h /dev/null/fastp.html -j /dev/null/fastp.json -w 8 -o 02.clean_reads/STA__e17937__MM074_STARRseq_500_bp_enh_plasmid_q30.fastq.gz \n",
      "fastp v0.20.0, time used: 17 seconds\n",
      "Number of reads after Q30 filtering for STA__e17937__MM074_STARRseq_500_bp_enh_plasmid:  5724180\n",
      "Detecting adapter sequence for read1...\n",
      "No adapter detected for read1\n",
      "\n",
      "Read1 before filtering:\n",
      "total reads: 3216620\n",
      "total bases: 22516340\n",
      "Q20 bases: 19094914(84.8047%)\n",
      "Q30 bases: 15681740(69.646%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 1980945\n",
      "total bases: 13866615\n",
      "Q20 bases: 13184606(95.0816%)\n",
      "Q30 bases: 11465559(82.6846%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 1980945\n",
      "reads failed due to low quality: 1235675\n",
      "reads failed due to too many N: 0\n",
      "reads with adapter trimmed: 0\n",
      "bases trimmed due to adapters: 0\n",
      "\n",
      "Duplication rate (may be overestimated since this is SE data): 0%\n",
      "\n",
      "JSON report: /dev/null/fastp.json\n",
      "HTML report: /dev/null/fastp.html\n",
      "\n",
      "fastp -i 02.clean_reads/STA__ed22cd__MM057_STARRseq_500_bp_enh_cDNA_R2.fastq.gz -L -e 30 -h /dev/null/fastp.html -j /dev/null/fastp.json -w 8 -o 02.clean_reads/STA__ed22cd__MM057_STARRseq_500_bp_enh_cDNA_q30.fastq.gz \n",
      "fastp v0.20.0, time used: 6 seconds\n",
      "Number of reads after Q30 filtering for STA__ed22cd__MM057_STARRseq_500_bp_enh_cDNA:  1980945\n"
     ]
    }
   ],
   "source": [
    "module load fastp/0.20.0-foss-2018a\n",
    "cat STARRseq_names.txt | while read line\n",
    "do\n",
    "# Filter for Q30 reads\n",
    "fastp -i 02.clean_reads/${line}_R2.fastq.gz -L -e 30 -h /dev/null/fastp.html -j /dev/null/fastp.json -w 8 -o 02.clean_reads/${line}_q30.fastq.gz\n",
    "echo -e \"Number of reads after Q30 filtering for ${line}: \" $(( $(zcat 02.clean_reads/${line}_q30.fastq.gz | wc -l) / 4 ))\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 10. Make BC count matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Extract DNA sequences from fastq\n",
    "cat STARRseq_names.txt | while read line\n",
    "do\n",
    "zcat 02.clean_reads/${line}_q30.fastq.gz | sed -n '2~4p' > 03.sequencing_saturation/${line}_q30.txt\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10.bc_count/STA__073ce2__MM029_STARRseq_500_bp_enh_plasmid_count.txt\n",
      "3861\n",
      "10.bc_count/STA__1a9f59__MM087_STARRseq_500_bp_enh_plasmid_count.txt\n",
      "3882\n",
      "10.bc_count/STA__34085e__MM047_STARRseq_500_bp_enh_cDNA_count.txt\n",
      "1516\n",
      "10.bc_count/STA__3f00ec__MM001_STARRseq_500_bp_enh_cDNA_count.txt\n",
      "921\n",
      "10.bc_count/STA__4642d6__MM087_STARRseq_500_bp_enh_cDNA_count.txt\n",
      "1914\n",
      "10.bc_count/STA__60a828__MM047_STARRseq_500_bp_enh_plasmid_count.txt\n",
      "3884\n",
      "10.bc_count/STA__616867__MM029_STARRseq_500_bp_enh_cDNA_count.txt\n",
      "2306\n",
      "10.bc_count/STA__a6d28a__MM001_STARRseq_500_bp_enh_plasmid_count.txt\n",
      "3174\n",
      "10.bc_count/STA__af2595__MM074_STARRseq_500_bp_enh_cDNA_count.txt\n",
      "1176\n",
      "10.bc_count/STA__c67b3a__MM057_STARRseq_500_bp_enh_plasmid_count.txt\n",
      "3903\n",
      "10.bc_count/STA__e17937__MM074_STARRseq_500_bp_enh_plasmid_count.txt\n",
      "3711\n",
      "10.bc_count/STA__ed22cd__MM057_STARRseq_500_bp_enh_cDNA_count.txt\n",
      "1974\n"
     ]
    }
   ],
   "source": [
    "# Generate BC count matrix for each samples\n",
    "parallel \\\n",
    "\"cat 03.sequencing_saturation/{}_q30.txt | sort | uniq -c | awk -F \\\" \\\" '{print \\$2\\\"\\t\\\"\\$1}' \\\n",
    "   > 10.bc_count/{}_count.txt\" ::: $(cat STARRseq_names.txt) 2>/dev/null\n",
    "\n",
    "# Count number of unique BC per sample\n",
    "cat STARRseq_names.txt | while read line\n",
    "do\n",
    "echo 10.bc_count/${line}_count.txt && cat 10.bc_count/${line}_count.txt | wc -l\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Assign BC to enhancers\n",
    "parallel \\\n",
    "\"awk -F '\\t' -v OFS='\\t' 'FNR==NR{a[\\$2]=\\$1 FS \\$2;next}{ print a[\\$1],\\$2}' \\\n",
    "   <( cat Enh-freeBC.tsv |  sed 's/\\t./\\t/') \\\n",
    "   <( cat 10.bc_count/{}_count.txt ) \\\n",
    "   | awk -F '\\t' '{if (\\$1) print \\$0;}' \\\n",
    "   > 10.bc_count/{}_count_final.txt\" ::: $(cat STARRseq_names.txt) 2>/dev/null"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "45\n"
     ]
    }
   ],
   "source": [
    "cat 10.bc_count/STA__6541c2__MM099_STARRseq_500_bp_plasmid_count_final.txt | wc -l"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "45\n"
     ]
    }
   ],
   "source": [
    "cat 10.bc_count/STA__fc62b1__MM099_STARRseq_500_bp_cDNA_count_final.txt | wc -l"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GPR39_2\tAACAC\t32604\n",
      "SOX10_5A\tAACCT\t24319\n",
      "COL5A1_3\tAAGAC\t37687\n",
      "EDN1_1\tAAGGT\t31853\n",
      "SERPINE1_1B\tAATCG\t63405\n",
      "FOSL2_1A\tACAAT\t20783\n",
      "TYR_1\tACAGG\t38207\n",
      "SOX10_1B\tACAGT\t52004\n",
      "COL5A1_8\tACCAT\t36634\n",
      "BLK_5\tACCTA\t27501\n"
     ]
    }
   ],
   "source": [
    "head 10.bc_count/STA__6541c2__MM099_STARRseq_500_bp_plasmid_count_final.txt\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "FOSL2_1A\tCAACAAT\n",
      "FOSL2_1B\tCGTATGG\n",
      "SERPINE1_1A\tAGGACAA\n",
      "SERPINE1_1B\tAGAATCG\n",
      "SERPINE1_6\tATACTGT\n",
      "FGF2_1A\tCACGACT\n",
      "FGF2_1B\tCGTCGGT\n",
      "FGF2_2\tTACGTCC\n",
      "FLNC_1A\tGGTGTCT\n",
      "FLNC_1B\tTTAGCCA\n",
      "HEG1_1\tGACGCTT\n",
      "BLK_3\tGTGCCAA\n",
      "BLK_5\tACACCTA\n",
      "COL1A2_1\tCAGTGTG\n",
      "COL5A1_1\tAGAGGTT\n",
      "COL5A1_2\tTCAGTGG\n",
      "COL5A1_3\tCTAAGAC\n",
      "COL5A1_4\tGGTTATC\n",
      "COL5A1_5\tGGTGTGG\n",
      "COL5A1_8\tCGACCAT\n",
      "ALPK_2\tTTCGCTC\n",
      "AXL_2\tGCGGAAT\n",
      "ABCC3_1\tATGCCTT\n",
      "GPR39_1\tGAACTGG\n",
      "GPR39_2\tACAACAC\n",
      "EDN1_1\tACAAGGT\n",
      "EDN1_3\tGCCTATA\n",
      "CDH1_1\tGTAGCTT\n",
      "CDH1_2\tCATTCAT\n",
      "SOX10_1A\tCAACTTC\n",
      "SOX10_1B\tCCACAGT\n",
      "SOX10_3A\tTGTTCAC\n",
      "SOX10_3B\tCAAGCGT\n",
      "SOX10_5A\tAGAACCT\n",
      "SOX10_5B\tGTAGTAT\n",
      "IRF4_1\tTCCATGA\n",
      "KIT_1A\tTATCACG\n",
      "KIT_1B\tGATTGCT\n",
      "MLANA_1\tGAGTCTT\n",
      "SGCD_2A\tAGCTATT\n",
      "SGCD_2B\tACACTGA\n",
      "SGCD_3\tTCTATCC\n",
      "MITF_1\tCTACTCC\n",
      "TYR_1\tCAACAGG\n",
      "TYR_2\tAAGAAGA"
     ]
    }
   ],
   "source": [
    "cat Enh-freeBC.tsv |  sed 's/\\t./\\t/'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
